<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Fabio Bedin | MS-Unit at IEO" />


<title>LFQ_Analysis</title>

<script src="site_libs/jquery-3.5.1/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/pagedtable-1.1/css/pagedtable.css" rel="stylesheet" />
<script src="site_libs/pagedtable-1.1/js/pagedtable.js"></script>
<script src="site_libs/htmlwidgets-1.5.3/htmlwidgets.js"></script>
<link href="site_libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="site_libs/datatables-binding-0.18/datatables.js"></script>
<link href="site_libs/dt-core-1.10.20/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="site_libs/dt-core-1.10.20/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="site_libs/dt-core-1.10.20/js/jquery.dataTables.min.js"></script>
<link href="site_libs/crosstalk-1.1.1/css/crosstalk.css" rel="stylesheet" />
<script src="site_libs/crosstalk-1.1.1/js/crosstalk.min.js"></script>
<script src="site_libs/plotly-binding-4.9.4.1/plotly.js"></script>
<script src="site_libs/typedarray-0.1/typedarray.min.js"></script>
<link href="site_libs/plotly-htmlwidgets-css-1.57.1/plotly-htmlwidgets.css" rel="stylesheet" />
<script src="site_libs/plotly-main-1.57.1/plotly-latest.min.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Template_LFQ</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/FabioBedin/GN_MR_ZC3H4">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">LFQ_Analysis</h1>
<h4 class="author">Fabio Bedin | MS-Unit at IEO</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-02-16
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>GN_MR_ZC3H4/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20220215code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20220215)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20220215code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20220215)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomFabioBedinGNMRZC3H4tree3e779377b92dbb0fcbea3b5bb9238beb1772ab6dtargetblank3e77937a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/FabioBedin/GN_MR_ZC3H4/tree/3e779377b92dbb0fcbea3b5bb9238beb1772ab6d" target="_blank">3e77937</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomFabioBedinGNMRZC3H4tree3e779377b92dbb0fcbea3b5bb9238beb1772ab6dtargetblank3e77937a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/FabioBedin/GN_MR_ZC3H4/tree/3e779377b92dbb0fcbea3b5bb9238beb1772ab6d" target="_blank">3e77937</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/

Untracked files:
    Untracked:  code/custom_functions.R
    Untracked:  data/expdesign.tsv
    Untracked:  data/proteinGroups.txt
    Untracked:  output/results_table_TURBO._vs_WT..xlsx
    Untracked:  output/results_table_TURBO_BIO_vs_TURBO..xlsx
    Untracked:  output/results_table_TURBO_BIO_vs_WT_BIO.xlsx
    Untracked:  output/results_table_WT_BIO_vs_WT..xlsx

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/Template_LFQ.Rmd</code>) and HTML (<code>docs/Template_LFQ.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/FabioBedin/GN_MR_ZC3H4/blob/3e779377b92dbb0fcbea3b5bb9238beb1772ab6d/analysis/Template_LFQ.Rmd" target="_blank">3e77937</a>
</td>
<td>
FabioBedin
</td>
<td>
2022-02-16
</td>
<td>
Update imputation
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/FabioBedin/GN_MR_ZC3H4/3e779377b92dbb0fcbea3b5bb9238beb1772ab6d/docs/Template_LFQ.html" target="_blank">3e77937</a>
</td>
<td>
FabioBedin
</td>
<td>
2022-02-16
</td>
<td>
Update imputation
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/FabioBedin/GN_MR_ZC3H4/9ecde79132cfac88eaf9a4160cd1b06092b45f53/docs/Template_LFQ.html" target="_blank">9ecde79</a>
</td>
<td>
FabioBedin
</td>
<td>
2022-02-15
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/FabioBedin/GN_MR_ZC3H4/blob/24ebe3abcc4514b8925fc1a3fc0b6a99a4bd176e/analysis/Template_LFQ.Rmd" target="_blank">24ebe3a</a>
</td>
<td>
FabioBedin
</td>
<td>
2022-02-15
</td>
<td>
first push
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<style type="text/css">
p{
  font-size: 16px;
}
</style>
<pre class="r"><code>## Proteomics
library(&quot;MBQN&quot;)
library(&quot;DEP&quot;)
library(&quot;SummarizedExperiment&quot;)
library(&quot;preprocessCore&quot;)
library(&quot;enrichR&quot;)
library(&quot;gprofiler2&quot;)
library(&quot;org.Hs.eg.db&quot;)
library(&quot;rrvgo&quot;)

## Plots
library(&quot;DT&quot;)
library(&quot;patchwork&quot;)
library(&quot;datapasta&quot;)
library(&quot;ggstatsplot&quot;)
library(&quot;UpSetR&quot;)
library(&quot;ggrepel&quot;)
library(&quot;visdat&quot;)
library(&quot;naniar&quot;)

## Networks
library(&quot;STRINGdb&quot;)
library(&quot;RCy3&quot;)
library(&quot;RColorBrewer&quot;)

## Other
library(&quot;here&quot;)
library(&quot;openxlsx&quot;)
library(&quot;tidyverse&quot;)</code></pre>
<pre class="r"><code>source(here::here(&quot;code/custom_functions.R&quot;))</code></pre>
<pre class="r"><code>data &lt;- read.csv(here::here(paste0(&quot;data/&quot;, params$proteinGroups)), header = TRUE,stringsAsFactors = FALSE, sep = &quot;\t&quot;)
## qui metterei anche Razor...unique.peptides al posto di Peptides, perchè così non conto i peptidi che sono attribuiti a più di 1 PG
data &lt;- data[data$Reverse != &quot;+&quot; &amp; data$Potential.contaminant != &quot;+&quot; &amp; data$Only.identified.by.site != &quot;+&quot; &amp; data$Peptides &gt;= params$n_peptides,]

data_unique &lt;- make_unique(data, &quot;Gene.names&quot;, &quot;Protein.IDs&quot;, delim = &quot;;&quot;)</code></pre>
<pre class="r"><code>cond_1 &lt;- params$test_1

cond_2 &lt;- params$test_2

conditions&lt;-c(cond_1,cond_2)

test&lt;- paste(cond_1,cond_2,sep=&quot;_vs_&quot;)

expdesign &lt;- read.table(here::here(paste0(&quot;data/&quot;, params$expdesign)), header = T, stringsAsFactors = F)

expdesign &lt;- subset(expdesign, condition %in% conditions)

columns&lt;-match(paste(&quot;LFQ.intensity.&quot;,expdesign$label,sep=&quot;&quot;),colnames(data_unique))

data_se &lt;- make_se(data_unique, columns, expdesign)

## define automaticaly the best statistical threshold to filter data besed on replicates:
if(max(expdesign$replicate)&lt;=3){
       threshold&lt;-0
     } else if(max(expdesign$replicate)&lt;6){
       threshold&lt;-1
     } else if (max(expdesign$replicate)&gt;=6){
       threshold&lt;-trunc(max(expdesign$replicate)*0.25) ## 0.25 serve per avere sempre il 75% di valid vales
     }

data_filt &lt;- filter_missval(data_se, thr = threshold)</code></pre>
<div id="turbo_bio-vs-wt_bio" class="section level1">
<h1><strong>TURBO_BIO</strong> vs <strong>WT_BIO</strong></h1>
<hr />
</div>
<div id="introduction" class="section level1">
<h1><strong>Introduction</strong></h1>
<p>First, we need to wrangling the original dataset. From the <strong>proteinGroups</strong> obtained by <a href="https://www.maxquant.org/">MaxQuant</a> software, I remove proteins that are marked form the software as potential contaminant, only identify by site and reverse.<br />
We che also filter the initial datasets by excluding all proteins identified by less than 1 peptides.<br />
Now, we generate some quality control plots to investigate the structure of the dataset and observe the effect of filters.</p>
<div id="normalization-strategy" class="section level4">
<h4><strong>Normalization strategy:</strong></h4>
<pre class="r"><code>if(params$Normalization == &quot;VSN&quot;){
  data_filt &lt;- normalize_vsn(data_filt)
  print(&quot;VSN normalization is used&quot;)
} else if(params$Normalization == &quot;MBQN&quot;){
  data_filt &lt;- normalize_MBQN(data_filt)
  print(&quot;MBQN normalization is used&quot;)
} else {
  print(&quot;Max-LFQ normalization is used&quot;)
}</code></pre>
<p>[1] “Max-LFQ normalization is used”</p>
</div>
<div id="quality-control-plots" class="section level2 tabset tabset-fade">
<h2><strong>1. Quality control plots</strong></h2>
<div id="proteins-per-samples" class="section level3 tabset">
<h3>Proteins per samples</h3>
<p>This plot show the number of proteins identify in each samples <em>after and before User define</em> filters:</p>
<div id="filtered" class="section level4">
<h4>Filtered</h4>
<pre class="r"><code>plot_numbers_lables(data_filt)</code></pre>
<p><img src="figure/Template_LFQ.Rmd/proteins-filt-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="not-filtered" class="section level4">
<h4>Not-Filtered</h4>
<pre class="r"><code>plot_numbers_lables(data_se)</code></pre>
<p><img src="figure/Template_LFQ.Rmd/proteins-NOfilt-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="distributions" class="section level3 tabset">
<h3>Distributions</h3>
<p>This plot is useful for checking the distributions of the samples <em>after and before User define</em> filters:</p>
<div id="filtered-1" class="section level4">
<h4>Filtered</h4>
<pre class="r"><code>plot_normalization(data_filt)</code></pre>
<p><img src="figure/Template_LFQ.Rmd/normalization-filt-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="not-filtered-1" class="section level4">
<h4>Not-Filtered</h4>
<pre class="r"><code>plot_normalization(data_se)</code></pre>
<p><img src="figure/Template_LFQ.Rmd/normalization-NOfilt-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="missing-data" class="section level3 tabset">
<h3>Missing data</h3>
<p>This plot we can explore the missing data pattern <em>after and before User define</em> filters. Notice that the percentages of missingness are provided in the data. These are accurate to 1 decimal place. Also the dataset is arranged by columns with most missingness.</p>
<div id="filtered-2" class="section level4">
<h4>Filtered</h4>
<pre class="r"><code>assay(data_filt) %&gt;% 
  as.data.frame() %&gt;% 
  vis_miss(., sort_miss = TRUE)</code></pre>
<p><img src="figure/Template_LFQ.Rmd/missing-data-filt-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="not-filtered-2" class="section level4">
<h4>Not-Filtered</h4>
<pre class="r"><code>assay(data_se) %&gt;% 
  as.data.frame() %&gt;% 
  vis_miss(., sort_miss = TRUE)</code></pre>
<p><img src="figure/Template_LFQ.Rmd/missing-data-NOfilt-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="sample-cvs" class="section level3 tabset">
<h3>Sample CVs</h3>
<p>The coefficient of variation (CV) is a statistical measure of the dispersion of data points in a data series around the mean. The coefficient of variation represents the ratio of the standard deviation to the mean, and it is a useful statistic for comparing the degree of variation from one data series to another, even if the means are drastically different from one another.</p>
<div id="filtered-3" class="section level4">
<h4>Filtered</h4>
<pre class="r"><code>plot_cvs(data_filt)</code></pre>
<p><img src="figure/Template_LFQ.Rmd/CVs-filt-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="not-filtered-3" class="section level4">
<h4>Not-Filtered</h4>
<pre class="r"><code>plot_cvs(data_se)</code></pre>
<p><img src="figure/Template_LFQ.Rmd/CVs-NOfilt-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
</div>
</div>
<div id="upset-plot" class="section level2 tabset tabset-fade">
<h2><strong>2. Upset plot</strong></h2>
<p>With this plot we can identify the numper of prteins in common within the 2 conditions or in common between all condition.</p>
<div id="turbo_bio" class="section level3">
<h3>TURBO_BIO</h3>
<pre class="r"><code>define_set &lt;- assay(data_filt) %&gt;%  as.data.frame() %&gt;% select(starts_with(cond_1)) %&gt;% colnames()
n_sets &lt;- length(define_set)

assay(data_filt) %&gt;% 
  as.data.frame() %&gt;% 
  select(starts_with(cond_1)) %&gt;% 
  rownames_to_column() %&gt;% 
  pivot_longer(!rowname, names_to = &quot;samples&quot;, values_to = &quot;intensity&quot;) %&gt;% 
  mutate(intensity = if_else(is.na(intensity), 0, 1)) %&gt;% 
  pivot_wider(names_from = samples, values_from = intensity) %&gt;% 
  as.data.frame() %&gt;% 
  upset(nsets = n_sets,
        sets = define_set,
        order.by = &quot;freq&quot;,
        keep.order = T,
        text.scale = 2.5,
        point.size = 4,
        line.size = 0.5, 
        sets.bar.color = &quot;coral2&quot;,
        main.bar.color  = &quot;gray44&quot;)</code></pre>
<p><img src="figure/Template_LFQ.Rmd/upset-cond1-1.png" width="1440" style="display: block; margin: auto;" /></p>
</div>
<div id="wt_bio" class="section level3">
<h3>WT_BIO</h3>
<pre class="r"><code>define_set &lt;- assay(data_filt) %&gt;%  as.data.frame() %&gt;% select(starts_with(cond_2)) %&gt;% colnames()
n_sets &lt;- length(define_set)

assay(data_filt) %&gt;% 
  as.data.frame() %&gt;% 
  select(starts_with(cond_2)) %&gt;% 
  rownames_to_column() %&gt;% 
  pivot_longer(!rowname, names_to = &quot;samples&quot;, values_to = &quot;intensity&quot;) %&gt;% 
  mutate(intensity = if_else(is.na(intensity), 0, 1)) %&gt;% 
  pivot_wider(names_from = samples, values_from = intensity) %&gt;% 
  as.data.frame() %&gt;% 
  upset(nsets = n_sets,
        sets = define_set,
        order.by = &quot;freq&quot;,
        keep.order = T,
        text.scale = 2.5,
        point.size = 4,
        line.size = 0.5, 
        sets.bar.color = &quot;turquoise3&quot;,
        main.bar.color  = &quot;gray44&quot;)</code></pre>
<p><img src="figure/Template_LFQ.Rmd/upset-cond2-1.png" width="1440" style="display: block; margin: auto;" /></p>
</div>
<div id="both-conditions" class="section level3">
<h3>Both conditions</h3>
<pre class="r"><code>define_set &lt;- assay(data_filt) %&gt;% colnames()
n_sets &lt;- length(define_set)
n_each &lt;- n_sets / 2

assay(data_filt) %&gt;% 
  as.data.frame() %&gt;% 
  rownames_to_column() %&gt;% 
  pivot_longer(!rowname, names_to = &quot;samples&quot;, values_to = &quot;intensity&quot;) %&gt;% 
  mutate(intensity = if_else(is.na(intensity), 0, 1)) %&gt;% 
  pivot_wider(names_from = samples, values_from = intensity) %&gt;% 
  as.data.frame() %&gt;% 
  upset(nsets = n_sets,
        sets = define_set,
        order.by = &quot;freq&quot;,
        keep.order = T,
        text.scale = 2.5,
        point.size = 4,
        line.size = 0.5, 
        sets.bar.color = rep(c(&quot;red3&quot;, &quot;royalblue1&quot;), each = n_each),
        main.bar.color  = &quot;gray44&quot;)</code></pre>
<p><img src="figure/Template_LFQ.Rmd/upset-both-1.png" width="1440" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="imputation" class="section level2">
<h2><strong>3. Imputation</strong></h2>
<p>An idealized version of a label-free discovery mass spectrometry proteomics experiment would provide absolute abundance measurements for a whole proteome, across varying conditions. Unfortunately, this ideal is not realized. Measurements are made on peptides requiring an inferential step to obtain protein level estimates. The inference is complicated by experimental factors that necessitate relative abundance estimation and result in widespread non-ignorable missing data. Relative abundance on the log scale takes the form of parameter contrasts. In a complete-case analysis, contrast estimates may be biased by missing data and a substantial amount of useful information will often go unused.</p>
<pre class="r"><code>data_imputed &lt;- mix_imputation_mean(data_filt, cond1 = cond_1, cond2 = cond_2)

plot_imputation(data_filt, data_imputed)</code></pre>
<p><img src="figure/Template_LFQ.Rmd/imputation-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>data_filt &lt;- data_imputed</code></pre>
</div>
<div id="differential-enrichment-analysis" class="section level2">
<h2><strong>4. Differential enrichment analysis</strong></h2>
<p>Protein-wise linear models combined with empirical Bayes statistics are used for the differential enrichment analysis (or differential expression analysis) and False discovery rate (FDR) is ajusted with <em>Benjamini-Hochberg</em> procedure. Significant proteins are are define with <strong>FDR = 0.05</strong> and <strong>Fold change = 1</strong></p>
<pre class="r"><code>data_diff &lt;- test_diff_BH(data_filt, type = &quot;manual&quot;, test = test)

dep &lt;- add_rejections(data_diff, alpha = params$FDR, lfc = params$Fold_change)

results&lt;-get_results(dep)</code></pre>
<p>There are <strong>177</strong> significant proteins in <strong>TURBO_BIO</strong> vs <strong>WT_BIO</strong> comparison.</p>
<p>The results from the previous analysis can be easily visualized by a number of functions. These visualizations assist in the determination of the optimal cutoffs to be used, highlight the most interesting samples and contrasts, and pinpoint differentially enriched/expressed proteins.</p>
<div id="visualization-of-the-results" class="section level3 tabset tabset-fade">
<h3><strong>4.1. Visualization of the results</strong></h3>
<div id="pca" class="section level4">
<h4>PCA</h4>
<p>The PCA plot can be used to get a high-level overview of the data. This can be very useful to observe batch effects, such as clear differences between replicates.</p>
<pre class="r"><code>if(params$Imputation){
  if(nrow(dep) &gt;= 500){
    n_value &lt;- 500
  }else {
    n_value &lt;- nrow(dep)
  }
  plot_pca(dep, x = 1, y = 2, n = n_value, point_size = 4, indicate = &quot;condition&quot;, label = T, label_size = 3)
} else {
  var &lt;- apply(assay(dep), 1, sd)
  df &lt;- assay(dep)[order(var, decreasing = TRUE)[seq_len(nrow(dep))],]
  n_value &lt;- df %&gt;% as.data.frame() %&gt;% drop_na() %&gt;% nrow()
  if(n_value &gt;= 500){
    n_value &lt;- 500
  }
  plot_pca(dep, x = 1, y = 2, n = n_value, point_size = 4, indicate = &quot;condition&quot;, label = T, label_size = 3)
}</code></pre>
<p><img src="figure/Template_LFQ.Rmd/PCA-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="volcano" class="section level4">
<h4>Volcano</h4>
<p>Volcano plots allows to inspect the enrichment of proteins between the two samples (x axis) and their corresponding adjusted p value (y axis).</p>
<pre class="r"><code>plot_volcano(dep, contrast=test, add_names=T, label_size=5, adjusted = F)</code></pre>
<p><img src="figure/Template_LFQ.Rmd/volcano-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="volcano-and-unique" class="section level4">
<h4>Volcano and unique</h4>
<pre class="r"><code>plot_volcano_2(dep, contrast=test, add_names=T, label_size=5, adjusted = F, unique_vec = uni_name)</code></pre>
<p><img src="figure/Template_LFQ.Rmd/volcano-unique-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="correlation" class="section level4">
<h4>Correlation</h4>
<pre class="r"><code>plot_cor(dep, significant = TRUE, lower = 0, upper = 1, pal = &quot;Reds&quot;)</code></pre>
<p><img src="figure/Template_LFQ.Rmd/correlation-matrix-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="cluster-heatmap" class="section level4">
<h4>Cluster heatmap</h4>
<p>The heatmap representation gives an overview of all significant proteins (rows) in all samples (columns). This allows to see general trends, for example if one sample or replicate is really different compared to the others. Additionally, the clustering of samples (columns) can indicate closer related samples and clustering of proteins (rows) indicates similarly behaving proteins.</p>
<pre class="r"><code>if(results %&gt;% filter(significant) %&gt;% nrow() &lt;= 100){
  plot_heatmap(dep, type = &quot;centered&quot;, kmeans = TRUE, k = 5, show_row_names = T, indicate = &quot;condition&quot;, col_limit = 5)
}else {
  plot_heatmap(dep, type = &quot;centered&quot;, kmeans = TRUE, k = 5, show_row_names = F, indicate = &quot;condition&quot;, col_limit = 5)
}</code></pre>
<p><img src="figure/Template_LFQ.Rmd/cluster-heatmap-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="result-table" class="section level3">
<h3><strong>4.2. Result table</strong></h3>
<p>In this table are summarized the results of DEP analysis, sorted by p.value. Proteins that are UP regulated are colored with a red box in Fold change column, in the orhter hand proteins that are DOWN regulated are colored with blue.</p>
<pre class="r"><code>results %&gt;% 
  select(starts_with(&quot;name&quot;) | starts_with(test)) %&gt;% 
  filter(across(ends_with(&quot;significant&quot;))) %&gt;% 
  arrange(across(ends_with(&quot;p.adj&quot;))) %&gt;%
  mutate(across(2:3, format, scientific = T, digits = 2)) %&gt;%
  dplyr::rename_with(~ tolower(gsub(paste0(test,&quot;_&quot;), &quot;&quot;, .x)), .cols = starts_with(test)) %&gt;% 
  mutate(significant = str_to_title(significant)) %&gt;%
  dplyr::rename(Fold.Change = ratio, Gene.name = name) %&gt;% 
  DT::datatable(options = list(
  columnDefs = list(list(className = &#39;dt-center&#39;, targets = 1:5)),
  pageLength = 10)) %&gt;%
  formatStyle(&#39;Fold.Change&#39;, backgroundColor = styleInterval(c(-1, 1), c(&#39;lightblue&#39;, &#39;white&#39;, &#39;tomato&#39;))) %&gt;% 
  formatStyle(&#39;significant&#39;, color = styleEqual(c(&quot;True&quot;, &quot;False&quot;), c(&#39;green&#39;, &#39;red&#39;)))</code></pre>
<div id="htmlwidget-0dd1ba32e0541469b5ed" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-0dd1ba32e0541469b5ed">{"x":{"filter":"none","data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33","34","35","36","37","38","39","40","41","42","43","44","45","46","47","48","49","50","51","52","53","54","55","56","57","58","59","60","61","62","63","64","65","66","67","68","69","70","71","72","73","74","75","76","77","78","79","80","81","82","83","84","85","86","87","88","89","90","91","92","93","94","95","96","97","98","99","100","101","102","103","104","105","106","107","108","109","110","111","112","113","114","115","116","117","118","119","120","121","122","123","124","125","126","127","128","129","130","131","132","133","134","135","136","137","138","139","140","141","142","143","144","145","146","147","148","149","150","151","152","153","154","155","156","157","158","159","160","161","162","163","164","165","166","167","168","169","170","171","172","173","174","175","176","177"],["ZC3H4","CTR9","SYMPK","MYL12A","GPALPP1","PPIE","MED23","DKC1","HCFC1","TPM1.1","MRPS35","MRPS5","MRPS22","LEO1","RPRD1B","CPSF1","MRPS23","CLINT1","INTS3","HNRNPK","LRRFIP1","ZNF281","MRPS2","PTCD3","DAP3","ZNF616","MRPS27","DST","HELLS","PRPF3","MRPS25","MRPS9","PAF1","ALDH18A1","CSTF2","CDC73","WDR82","SUGP2","COIL","CPSF6","MRPS24","MRPS7","HNRNPL","FIP1L1","H2AFX","NHP2","TNK1","GULP1","NONO","HNRNPR","ZNF207","DBT","MRPS31","CMTR1","CPSF4","MYH14","KEAP1","JUNB","HIST1H4A","MYH14.1","FBXO30","SMN1","BUB3","NPM1","LDHB","CPNE7","CAPZB","PRMT1","NOP10","LRRFIP2","TOX4","KPNA3","MYH10","NECAP1","SYF2","CD44","MYO18A","DYNLL2","MRPS10","hCG_1984214","PHRF1","PTBP1","TBL2","CCDC101","ARID1A","TMOD3","NOSIP","FLII","MRPS28","TPM1","A0A1W2PNV4","CCDC12","HNRNPU","LDHA","PRPF4","SART3","ARPC3","MYL6","A0A2R8YHD2","PPAN-P2RY11","MYH9","ARHGEF2","ATP5B","NCL","ACTR2","PPARD","KHDRBS1","MRPS18B","MYL12B","TWF2","RBMX","RBPJ","ACTG1","ARPC4-TTLL3","LACTB","ISY1","ARPC2","GSN","MYO6","EIF2AK2","TPM4","CAPZA1","ITSN1","SPTAN1","ACTA1","ANXA1","DNAH14","MYL6B","SDC4","NECAP2","TPM2","SRSF5","KLHL7","RNF2","TPM3","MYO5A","SAFB","RPA1","ACTR3","MYL6.1","CCT8","GAPDH","CAPZA2","TPM3.1","CDK9","CWC15","DIMT1","HDAC1","SRSF4","TJP1","DLD","SSBP1","CORO1C","ACTN4","HNRNPA1","SDC1","TXN","WDR33","PFN1","RPA3","COPA","RPA2","ARPC1B","RBM22","ARPC5","SNRNP70","MPRIP","MYO1B","KAT6A","DHX9","ZNF106","TARDBP","CLIP2","PPP1R10","NFIC","FBXO11","AHNAK"],["4.3e-12","4.0e-11","2.9e-10","3.4e-09","7.2e-09","2.0e-08","3.0e-08","6.6e-08","6.0e-08","5.7e-08","1.0e-07","1.1e-07","1.6e-07","1.9e-07","2.1e-07","2.4e-07","3.0e-07","4.2e-07","6.9e-07","8.7e-07","1.0e-06","1.0e-06","1.4e-06","1.4e-06","1.6e-06","1.7e-06","1.8e-06","2.0e-06","2.0e-06","2.2e-06","2.8e-06","2.8e-06","3.3e-06","3.9e-06","4.1e-06","4.5e-06","4.8e-06","5.3e-06","5.7e-06","6.8e-06","1.0e-05","1.3e-05","1.5e-05","1.7e-05","1.9e-05","2.0e-05","2.3e-05","3.3e-05","3.4e-05","4.1e-05","4.1e-05","4.2e-05","4.3e-05","4.9e-05","5.0e-05","4.9e-05","5.4e-05","6.3e-05","6.5e-05","8.4e-05","8.9e-05","1.1e-04","1.3e-04","1.7e-04","2.0e-04","2.3e-04","2.6e-04","2.7e-04","3.1e-04","3.2e-04","3.5e-04","3.9e-04","3.9e-04","3.8e-04","3.9e-04","4.2e-04","5.0e-04","5.5e-04","5.6e-04","5.9e-04","6.3e-04","6.5e-04","6.5e-04","6.7e-04","7.1e-04","7.2e-04","7.9e-04","8.2e-04","8.2e-04","8.4e-04","8.6e-04","8.8e-04","8.7e-04","9.8e-04","1.0e-03","1.1e-03","1.2e-03","1.3e-03","1.4e-03","1.5e-03","1.5e-03","1.5e-03","1.5e-03","1.6e-03","1.6e-03","1.6e-03","1.7e-03","1.7e-03","1.8e-03","1.9e-03","2.0e-03","2.0e-03","2.2e-03","2.2e-03","2.2e-03","2.5e-03","2.5e-03","2.6e-03","2.7e-03","2.8e-03","2.8e-03","2.9e-03","3.0e-03","3.0e-03","3.1e-03","3.3e-03","3.3e-03","3.3e-03","3.4e-03","3.5e-03","3.6e-03","3.7e-03","3.8e-03","4.0e-03","4.2e-03","4.2e-03","4.3e-03","4.3e-03","4.4e-03","4.9e-03","5.5e-03","5.5e-03","5.8e-03","5.9e-03","6.0e-03","6.0e-03","6.1e-03","6.2e-03","7.1e-03","7.4e-03","7.8e-03","7.8e-03","8.0e-03","8.3e-03","8.7e-03","9.0e-03","9.2e-03","9.9e-03","1.0e-02","1.0e-02","1.2e-02","1.2e-02","1.2e-02","1.2e-02","1.2e-02","1.3e-02","1.4e-02","1.4e-02","1.4e-02","1.4e-02","1.5e-02","1.6e-02","1.6e-02","1.8e-02","1.9e-02","2.0e-02","2.0e-02"],["3.7e-09","1.7e-08","8.5e-08","7.5e-07","1.3e-06","2.8e-06","3.8e-06","5.7e-06","5.7e-06","5.7e-06","7.8e-06","7.8e-06","1.1e-05","1.2e-05","1.2e-05","1.3e-05","1.5e-05","2.0e-05","3.0e-05","3.6e-05","3.9e-05","3.9e-05","5.0e-05","5.0e-05","5.5e-05","5.6e-05","5.6e-05","5.9e-05","5.9e-05","6.2e-05","7.4e-05","7.4e-05","8.5e-05","9.8e-05","1.0e-04","1.1e-04","1.1e-04","1.2e-04","1.2e-04","1.4e-04","2.2e-04","2.6e-04","2.9e-04","3.1e-04","3.4e-04","3.5e-04","3.9e-04","5.5e-04","5.6e-04","6.4e-04","6.4e-04","6.5e-04","6.5e-04","6.9e-04","6.9e-04","6.9e-04","7.1e-04","8.2e-04","8.3e-04","1.0e-03","1.1e-03","1.3e-03","1.4e-03","1.8e-03","2.0e-03","2.2e-03","2.3e-03","2.4e-03","2.7e-03","2.8e-03","2.8e-03","2.9e-03","2.9e-03","2.9e-03","2.9e-03","3.0e-03","3.6e-03","3.8e-03","3.8e-03","4.0e-03","4.1e-03","4.1e-03","4.1e-03","4.2e-03","4.4e-03","4.5e-03","4.8e-03","4.9e-03","4.9e-03","5.0e-03","5.0e-03","5.0e-03","5.0e-03","5.5e-03","5.6e-03","6.2e-03","6.6e-03","7.0e-03","7.2e-03","7.5e-03","7.6e-03","7.7e-03","7.7e-03","7.8e-03","7.9e-03","7.9e-03","8.2e-03","8.2e-03","8.6e-03","8.8e-03","9.1e-03","9.4e-03","9.7e-03","9.7e-03","9.8e-03","1.1e-02","1.1e-02","1.1e-02","1.1e-02","1.1e-02","1.1e-02","1.2e-02","1.2e-02","1.2e-02","1.2e-02","1.3e-02","1.3e-02","1.3e-02","1.3e-02","1.3e-02","1.4e-02","1.4e-02","1.4e-02","1.5e-02","1.5e-02","1.5e-02","1.5e-02","1.5e-02","1.5e-02","1.7e-02","1.8e-02","1.8e-02","1.9e-02","1.9e-02","1.9e-02","1.9e-02","2.0e-02","2.0e-02","2.2e-02","2.2e-02","2.4e-02","2.4e-02","2.4e-02","2.4e-02","2.5e-02","2.6e-02","2.6e-02","2.8e-02","2.9e-02","2.9e-02","3.2e-02","3.2e-02","3.3e-02","3.3e-02","3.3e-02","3.4e-02","3.6e-02","3.6e-02","3.6e-02","3.7e-02","3.8e-02","3.9e-02","4.1e-02","4.4e-02","4.6e-02","4.6e-02","4.8e-02"],["True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True"],[10.1,4,6.88,5.51,-2.76,-2.67,5.22,1.21,2.46,3.57,4.09,2.04,2.25,3.93,4.68,1.62,4.03,-1.09,2.45,1.19,2.68,3.26,3.22,3.29,2.45,-1.06,5.15,3.69,4.81,2.52,1.94,3.61,3.65,-2.05,3.6,2.3,7.53,3.97,2.38,2.78,1.36,2.51,3.05,1.77,-1.17,1.14,2.26,-1.39,1.25,2.2,1.81,-1,2.42,2.37,1.2,1.93,-1.05,1.21,-1.36,2.09,2.64,3.07,1.28,1.84,-1.47,-1.23,2.48,1.03,1.39,2.45,2.68,-1.27,2.15,1.18,-1.34,2.05,2.48,3.26,3.09,2,1.27,1.59,1.68,3.26,1.28,1.64,4.25,3.07,2.12,3.18,1.46,-1.03,1.2,-1.25,2.44,2.08,1.46,1.97,4.36,2.11,1.99,1.32,-1.36,1.37,1.32,1.76,1.36,3.03,2.03,2.17,1.16,1.92,1.54,1.13,-2.02,-1.06,1.25,1.52,2.78,3.78,2.29,1.77,-2.02,2.26,1.68,2.72,3.6,1.8,1.5,-1.83,5.01,-2.48,-1.43,1.3,1.97,1.92,2.51,1.95,1.32,3.59,1.84,-1.11,1.08,3.61,1.33,-1.77,-1.73,1.75,-1.86,-1.5,-1.1,1.82,2.19,1.76,1.55,2.33,4.02,2.19,-1.7,1.64,3.22,1.67,1.1,1.69,2.38,1.43,1.62,1.14,1.11,1.63,-1.49,1.09,-1.27,2.97,1.93,-1.04,1.85]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>Gene.name<\/th>\n      <th>p.val<\/th>\n      <th>p.adj<\/th>\n      <th>significant<\/th>\n      <th>Fold.Change<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-center","targets":[1,2,3,4,5]},{"orderable":false,"targets":0}],"pageLength":10,"order":[],"autoWidth":false,"orderClasses":false,"rowCallback":"function(row, data, displayNum, displayIndex, dataIndex) {\nvar value=data[5]; $(this.api().cell(row, 5).node()).css({'background-color':isNaN(parseFloat(value)) ? '' : value <= -1 ? \"lightblue\" : value <= 1 ? \"white\" : \"tomato\"});\nvar value=data[4]; $(this.api().cell(row, 4).node()).css({'color':value == \"True\" ? \"green\" : value == \"False\" ? \"red\" : null});\n}"}},"evals":["options.rowCallback"],"jsHooks":[]}</script>
</div>
</div>
<div id="gene-ontology" class="section level2">
<h2><strong>5. Gene Ontology</strong></h2>
<p>Now I perfom a gene onology analysis (GO or by branch <a href="GO:MF" class="uri">GO:MF</a>, <a href="GO:BP" class="uri">GO:BP</a>, <a href="GO:CC" class="uri">GO:CC</a>) and a KEGG ontology.<br />
The enrichment results are visualized with an interactive <strong>Manhattan-like-plot</strong> named “Gostplot”.</p>
<div id="turbo_bio-1" class="section level3 tabset tabset-fade">
<h3><strong>5.1. TURBO_BIO</strong></h3>
<div id="gostplot" class="section level4">
<h4>Gostplot</h4>
<p>UP regulated proteins in TURBO_BIO were used to generate these gene ontologies.</p>
<pre class="r"><code>Gname &lt;- results %&gt;% filter(significant &amp; get(paste0(test, &quot;_ratio&quot;)) &gt; 0) %&gt;% pull(name)

if(!params$Imputation){
  Uni &lt;- unique_pair(data_filt_unique, conditions = conditions, table = T) %&gt;% 
  filter(!!sym(paste0(&quot;unique_&quot;, cond_1)) == &quot;+&quot;) %&gt;% 
  pull(name) %&gt;% unique()

  Gname &lt;- c(Gname, Uni)
}

if(length(Gname) == 0){
  print(&quot;NOT enough data.&quot;)
  GO &lt;- NULL
}else{
  GO &lt;- gost(query = Gname, organism = &quot;hsapiens&quot;, sources = c(&quot;GO:BP&quot;, &quot;GO:MF&quot;, &quot;GO:CC&quot;, &quot;KEGG&quot;), ordered_query = T, evcodes = TRUE)
  if(is.null(GO)){
    print(&quot;NOT enough data.&quot;)
  }else{
    gostplot(GO, capped = TRUE, interactive = TRUE)
  }
}</code></pre>
<div id="htmlwidget-4906484ed42468d86de4" style="width:960px;height:768px;" class="plotly html-widget"></div>
<script type="application/json" data-for="htmlwidget-4906484ed42468d86de4">{"x":{"data":[{"x":[105.120137124982,105.092857061508,86.6086654819669,113.549676738422,160.34277990004,134.711211690332,86.5969740261924,126.133580303748,86.5463110511694,107.037535872006,106.390608652482,176.59780057857,88.8768079022269,106.386711500557,163.596901757286,105.131828580757,136.005066129381,84.4106717963542,123.740729021893,177.97349520804,107.033638720081,88.9352651810995,111.071088114221,108.876991580533,110.841156150655,124.418833456816,137.127445883736,110.891819125678,122.778132496456,136.453238600738,109.594067534704,160.537637496282,161.367730856274,161.363833704349,176.021022093693,105.151314340381,157.427710260256,137.750990191712,84.414568948279,93.6780990736363,106.398402956331,123.370499589033,173.160512580856,112.185673564727,84.4730262271517,106.394505804406,107.583137141484,107.372690937543,89.4379977794046,97.2673759964188,76.1604011714567,136.500004423836,122.746955281057,135.697191127318,114.173221046398,136.176540814074,105.135725732681,97.1777415021473,109.617450446254,173.156615428931,131.722096163976,148.546101023533,152.232806744437,106.406197260181,91.5853284899943,124.414936304891,107.384382393317,108.619779553493,93.5144186927928,104.79667351522,84.4964091387008,134.715108842257,104.784982059445,128.382236964384,91.5775341861446,124.722811306954,112.875469455424,108.880888732458,136.464930056513,93.6313332505381,93.810602239081,177.977392359965,89.1807857523648,137.131343035661,164.746561575115,107.021947264307,86.5735911146433,84.0988996423665,122.754749584907,75.7784802828218,176.028816397542,97.9805547986655,131.293409452243,124.69553124348,94.5237810413277,106.788118148816,97.7623142908741,84.3794945809554,93.0194803983374,108.767871326637,134.719005994182,84.5275863540995,93.7677335679077,183.526936700945,114.157632438698,110.934687796851,136.437649993039,77.1619692161419,124.668251180006,150.732403253371,105.104548517283,93.635230402463,122.368931544347,75.7823774347467,105.159108644231,172.069310041899,176.601697730494,144.60218327559,183.519142397095,84.0326480596442,128.413414179782,84.4691290752268,162.22510427974,100.400686143994,159.263268816858,159.228194449534,159.224297297609,159.220400145684,89.4341006274798,150.603797239851,84.3327287578573,121.519352424731,86.3358648472277,111.421831787457,136.589638918108,93.2805895773021,140.416642108306,150.147830464645,119.933211591319,167.536922353304,136.433752841114,136.429855689189,178.690571162212,128.22635088739,175.8339588013,128.378339812459,133.066613578048,123.966763833534,91.9516607709298,107.399971001017,131.285615148393,163.98661694977,142.470441172699,147.758876334714,123.288659398611,177.98128951189,123.658888831471,136.008963281306,140.412744956381,136.168746510225,156.511879557917,149.719143752912,94.2860547739121,108.760077022788,84.8783300273356,150.736300405296,114.192706806022,158.304569443346,107.014152960457,107.364896633693,108.658751072742,134.941143653898,134.290319282449,137.688635760914,107.590931445334,143.612306686679,113.596442561521,169.033428692445,180.132517374404,180.140311678254,93.2844867292269,158.885245080148,98.6196877143401,193.429599741976,133.047127818423,101.191807984738,137.630178482042,133.062716426123,181.687480992418,135.790722773515,93.5105215408679,86.7138885839377,107.376588089467,84.2781686309094,119.859165704747,134.925555046199,141.593581989609,193.421805438127,111.398448875908,147.743287727015,91.569739882295,84.8627414196362,136.172643662149,189.509064905582,131.258335084919,87.1036037764223,126.324540748065,165.615626454356],"y":[17,17,17,14.8605378316866,14.3922876400625,14.3281464146053,13.6669798581921,13.5664425266806,12.050011731036,11.8724315188278,11.2832926780323,10.9408495726203,10.7759036585395,10.7509664021286,10.5963094877628,10.5570886915427,10.5331044863241,10.5008246247917,10.4020965663264,10.2957637512784,10.0344546334083,9.90650844311006,9.78844073681863,9.76937858838801,9.41879377099535,9.40343707668539,9.33350367692474,8.93956444680666,8.58079824807757,8.5733086193856,7.62117334996877,7.38542700269002,7.27861706135902,7.08736892284413,6.93878151937659,6.57690804818709,6.39292674386714,6.37625731594796,6.20221354417051,6.05974279839833,6.05703500356252,5.86486181661669,5.7259088347741,5.66201135354171,5.46991699292605,5.43585546032086,5.37702580621373,5.34615508976879,5.33506148597761,5.33387252759417,5.31367225019841,5.29493347911991,5.24609698761954,5.21849596253893,5.19510018640614,5.16349846451434,5.13613355325121,5.09870349133078,5.05262981629747,4.98656550212803,4.92143213153928,4.86390624080957,4.86126627300945,4.80988821078901,4.80750953270929,4.68323423360894,4.56539403371283,4.55927918418392,4.51623362068907,4.48178174429963,4.48035306108322,4.45617347881063,4.4457355794512,4.42277980016775,4.3950556899232,4.34141175015281,4.26218325317758,4.11631741212434,4.01111354653533,4.00810418880383,3.97007508406609,3.94133675534286,3.90462440583224,3.9009438943534,3.88060749131528,3.86947342533466,3.85518814639967,3.81261350588674,3.79775093418633,3.61495206832111,3.57112069912387,3.5626237594653,3.52724424144655,3.43547744092406,3.34694090271115,3.33555230943764,3.32346614719181,3.26060531751075,3.23365260846703,3.22703591849853,3.19675268470435,3.14899664234992,3.14817919260964,3.12372226513148,3.11889974202947,3.11150281426791,3.08454794364971,3.0676082743892,3.04824131500188,2.98597109462762,2.9852540805049,2.95328851084521,2.95077585214113,2.93600852404409,2.89529456373647,2.85183800901002,2.80871796086686,2.78937992545459,2.7774980689308,2.76714186336858,2.7649064852845,2.70635441853469,2.6984329166688,2.6943835070137,2.67629028918968,2.67629028918968,2.67629028918968,2.67629028918968,2.62968108814783,2.57372018196153,2.55717894400928,2.52743798494655,2.51720919977356,2.47864352837731,2.47864352837731,2.4707565474653,2.3934167936047,2.34217021199787,2.33092762206783,2.32680258160651,2.30468435317673,2.29529352630988,2.28691216747465,2.28691216747465,2.27754571047169,2.27377393897422,2.22869382868307,2.21980717732806,2.21859576393035,2.20680435038017,2.19057174051151,2.1022210243088,2.08003525830116,2.07669125893829,2.07197171133933,2.06708761351219,2.04696579957866,2.04390879799693,2.03773462821984,2.02874614254868,1.98749782500458,1.90187061277885,1.89297782858276,1.87744629250605,1.87339973690947,1.85869725095061,1.82669562413578,1.81228643237443,1.79995356250459,1.79938604927426,1.79618996608594,1.78647336511702,1.77702734389382,1.77702734389382,1.75801686807051,1.67247769910933,1.66628040252326,1.64518251187446,1.64518251187446,1.64518251187446,1.63633676923346,1.63450746207162,1.63272947776777,1.622806757874,1.6170990373372,1.60322262883066,1.55719288833874,1.54347056970066,1.52835048670238,1.52692785787361,1.51744407621183,1.49647316625382,1.49403096951642,1.46416949551323,1.45083966140769,1.44843620534149,1.44566327907624,1.40138499563042,1.40138499563042,1.37822416954875,1.37791087069061,1.35931983266589,1.3460199697696,1.33682776151387,1.33458135622563,1.33171789431611,1.32124993126304,1.30583831612673],"text":["GO:0030036 (711) <br> actin cytoskeleton organization <br> 1.414e-19","GO:0030029 (800) <br> actin filament-based process <br> 3.094e-19","GO:0007015 (444) <br> actin filament organization <br> 1.371e-18","GO:0034314 (39) <br> Arp2/3 complex-mediated actin nucleation <br> 1.379e-15","GO:0097435 (755) <br> supramolecular fiber organization <br> 4.052e-15","GO:0050684 (141) <br> regulation of mRNA processing <br> 4.697e-15","GO:0007010 (1435) <br> cytoskeleton organization <br> 2.153e-14","GO:0045010 (55) <br> actin nucleation <br> 2.714e-14","GO:0006996 (3805) <br> organelle organization <br> 8.912e-13","GO:0031124 (73) <br> mRNA 3'-end processing <br> 1.341e-12","GO:0030833 (140) <br> regulation of actin filament polymerization <br> 5.208e-12","GO:1902903 (373) <br> regulation of supramolecular fiber organization <br> 1.146e-11","GO:0008064 (159) <br> regulation of actin polymerization or depolymerization <br> 1.675e-11","GO:0030832 (160) <br> regulation of actin filament length <br> 1.774e-11","GO:0110053 (267) <br> regulation of actin filament organization <br> 2.533e-11","GO:0030041 (168) <br> actin filament polymerization <br> 2.773e-11","GO:0051128 (2246) <br> regulation of cellular component organization <br> 2.930e-11","GO:0006378 (52) <br> mRNA polyadenylation <br> 3.156e-11","GO:0043631 (53) <br> RNA polyadenylation <br> 3.962e-11","GO:1903311 (318) <br> regulation of mRNA metabolic process <br> 5.061e-11","GO:0031123 (126) <br> RNA 3'-end processing <br> 9.237e-11","GO:0008154 (198) <br> actin polymerization or depolymerization <br> 1.240e-10","GO:0033043 (1109) <br> regulation of organelle organization <br> 1.628e-10","GO:0032271 (205) <br> regulation of protein polymerization <br> 1.701e-10","GO:0032956 (350) <br> regulation of actin cytoskeleton organization <br> 3.812e-10","GO:0044087 (940) <br> regulation of cellular component biogenesis <br> 3.950e-10","GO:0051493 (523) <br> regulation of cytoskeleton organization <br> 4.640e-10","GO:0032970 (391) <br> regulation of actin filament-based process <br> 1.149e-09","GO:0043254 (402) <br> regulation of protein-containing complex assembly <br> 2.625e-09","GO:0051258 (278) <br> protein polymerization <br> 2.671e-09","GO:0032535 (355) <br> regulation of cellular component size <br> 2.392e-08","GO:0097581 (91) <br> lamellipodium organization <br> 4.117e-08","GO:0098789 (13) <br> pre-mRNA cleavage required for polyadenylation <br> 5.265e-08","GO:0098787 (14) <br> mRNA cleavage involved in mRNA processing <br> 8.178e-08","GO:1902743 (53) <br> regulation of lamellipodium organization <br> 1.151e-07","GO:0030048 (132) <br> actin filament-based movement <br> 2.649e-07","GO:0090066 (488) <br> regulation of anatomical structure size <br> 4.046e-07","GO:0051693 (44) <br> actin filament capping <br> 4.205e-07","GO:0006379 (20) <br> mRNA cleavage <br> 6.277e-07","GO:0010604 (3378) <br> positive regulation of macromolecule metabolic process <br> 8.715e-07","GO:0030835 (48) <br> negative regulation of actin filament depolymerization <br> 8.769e-07","GO:0043484 (151) <br> regulation of RNA splicing <br> 1.365e-06","GO:1901880 (77) <br> negative regulation of protein depolymerization <br> 1.880e-06","GO:0033523 (12) <br> histone H2B ubiquitination <br> 2.178e-06","GO:0006397 (1919) <br> mRNA processing <br> 3.389e-06","GO:0030834 (57) <br> regulation of actin filament depolymerization <br> 3.666e-06","GO:0031440 (28) <br> regulation of mRNA 3'-end processing <br> 4.197e-06","GO:0031325 (3176) <br> positive regulation of cellular metabolic process <br> 4.507e-06","GO:0009059 (4880) <br> macromolecule biosynthetic process <br> 4.623e-06","GO:0016071 (2153) <br> mRNA metabolic process <br> 4.636e-06","GO:0000723 (137) <br> telomere maintenance <br> 4.857e-06","GO:0051276 (1034) <br> chromosome organization <br> 5.071e-06","GO:0043242 (87) <br> negative regulation of protein-containing complex disassembly <br> 5.674e-06","GO:0051016 (29) <br> barbed-end actin filament capping <br> 6.046e-06","GO:0034645 (1564) <br> cellular macromolecule biosynthetic process <br> 6.381e-06","GO:0051173 (2978) <br> positive regulation of nitrogen compound metabolic process <br> 6.863e-06","GO:0030042 (62) <br> actin filament depolymerization <br> 7.309e-06","GO:0016043 (7793) <br> cellular component organization <br> 7.967e-06","GO:0032543 (79) <br> mitochondrial translation <br> 8.859e-06","GO:1901879 (93) <br> regulation of protein depolymerization <br> 1.031e-05","GO:0048024 (105) <br> regulation of mRNA splicing, via spliceosome <br> 1.198e-05","GO:0070358 (7) <br> actin polymerization-dependent cell motility <br> 1.368e-05","GO:0071840 (7994) <br> cellular component organization or biogenesis <br> 1.376e-05","GO:0030837 (68) <br> negative regulation of actin filament polymerization <br> 1.549e-05","GO:0009893 (3678) <br> positive regulation of metabolic process <br> 1.558e-05","GO:0044085 (4801) <br> cellular component biogenesis <br> 2.074e-05","GO:0031328 (1942) <br> positive regulation of cellular biosynthetic process <br> 2.720e-05","GO:0032200 (164) <br> telomere organization <br> 2.759e-05","GO:0010557 (1839) <br> positive regulation of macromolecule biosynthetic process <br> 3.046e-05","GO:0022607 (4556) <br> cellular component assembly <br> 3.298e-05","GO:0006403 (198) <br> RNA localization <br> 3.309e-05","GO:0050685 (32) <br> positive regulation of mRNA processing <br> 3.498e-05","GO:0022604 (305) <br> regulation of cell morphogenesis <br> 3.583e-05","GO:0045935 (1897) <br> positive regulation of nucleobase-containing compound metabolic process <br> 3.778e-05","GO:0009891 (1975) <br> positive regulation of biosynthetic process <br> 4.027e-05","GO:0044271 (4734) <br> cellular nitrogen compound biosynthetic process <br> 4.556e-05","GO:0034097 (922) <br> response to cytokine <br> 5.468e-05","GO:0032272 (83) <br> negative regulation of protein polymerization <br> 7.650e-05","GO:0051261 (120) <br> protein depolymerization <br> 9.747e-05","GO:0010591 (41) <br> regulation of lamellipodium assembly <br> 9.815e-05","GO:0010639 (352) <br> negative regulation of organelle organization <br> 1.071e-04","GO:1903312 (93) <br> negative regulation of mRNA metabolic process <br> 1.145e-04","GO:0008356 (19) <br> asymmetric cell division <br> 1.246e-04","GO:0051494 (164) <br> negative regulation of cytoskeleton organization <br> 1.256e-04","GO:0140053 (111) <br> mitochondrial gene expression <br> 1.316e-04","GO:0031120 (3) <br> snRNA pseudouridine synthesis <br> 1.351e-04","GO:0007004 (71) <br> telomere maintenance via telomerase <br> 1.396e-04","GO:0006278 (72) <br> RNA-dependent DNA biosynthetic process <br> 1.540e-04","GO:0043244 (127) <br> regulation of protein-containing complex disassembly <br> 1.593e-04","GO:0000380 (79) <br> alternative mRNA splicing, via spliceosome <br> 2.427e-04","GO:1902745 (37) <br> positive regulation of lamellipodium organization <br> 2.685e-04","GO:0016574 (48) <br> histone ubiquitination <br> 2.738e-04","GO:0046833 (5) <br> positive regulation of RNA export from nucleus <br> 2.970e-04","GO:0044260 (5746) <br> cellular macromolecule metabolic process <br> 3.669e-04","GO:0010833 (84) <br> telomere maintenance via telomere lengthening <br> 4.498e-04","GO:0031032 (193) <br> actomyosin structure organization <br> 4.618e-04","GO:0016344 (3) <br> meiotic chromosome movement towards spindle pole <br> 4.748e-04","GO:0006368 (61) <br> transcription elongation from RNA polymerase II promoter <br> 5.488e-04","GO:0010390 (29) <br> histone monoubiquitination <br> 5.839e-04","GO:0032241 (6) <br> positive regulation of nucleobase-containing compound transport <br> 5.929e-04","GO:0050686 (30) <br> negative regulation of mRNA processing <br> 6.357e-04","GO:0006412 (713) <br> translation <br> 7.096e-04","GO:0010628 (1117) <br> positive regulation of gene expression <br> 7.109e-04","GO:1904874 (15) <br> positive regulation of telomerase RNA localization to Cajal body <br> 7.521e-04","GO:0034641 (8767) <br> cellular nitrogen compound metabolic process <br> 7.605e-04","GO:0032984 (230) <br> protein-containing complex disassembly <br> 7.736e-04","GO:0051254 (1729) <br> positive regulation of RNA metabolic process <br> 8.231e-04","GO:0001711 (15) <br> endodermal cell fate commitment <br> 8.558e-04","GO:0044249 (5794) <br> cellular biosynthetic process <br> 8.949e-04","GO:0071345 (828) <br> cellular response to cytokine stimulus <br> 1.033e-03","GO:0030032 (73) <br> lamellipodium assembly <br> 1.035e-03","GO:0010592 (29) <br> positive regulation of lamellipodium assembly <br> 1.114e-03","GO:0043043 (738) <br> peptide biosynthetic process <br> 1.120e-03","GO:0000381 (62) <br> regulation of alternative mRNA splicing, via spliceosome <br> 1.159e-03","GO:0030050 (18) <br> vesicle transport along actin filament <br> 1.273e-03","GO:1901576 (5869) <br> organic substance biosynthetic process <br> 1.407e-03","GO:1902904 (166) <br> negative regulation of supramolecular fiber organization <br> 1.553e-03","GO:0061061 (616) <br> muscle structure development <br> 1.624e-03","GO:1904872 (18) <br> regulation of telomerase RNA localization to Cajal body <br> 1.669e-03","GO:0006259 (896) <br> DNA metabolic process <br> 1.709e-03","GO:0045944 (1185) <br> positive regulation of transcription by RNA polymerase II <br> 1.718e-03","GO:0006396 (3356) <br> RNA processing <br> 1.966e-03","GO:0099515 (20) <br> actin filament-based transport <br> 2.002e-03","GO:0019219 (3945) <br> regulation of nucleobase-containing compound metabolic process <br> 2.021e-03","GO:0090685 (19) <br> RNA localization to nucleus <br> 2.107e-03","GO:0090672 (19) <br> telomerase RNA localization <br> 2.107e-03","GO:0090671 (19) <br> telomerase RNA localization to Cajal body <br> 2.107e-03","GO:0090670 (19) <br> RNA localization to Cajal body <br> 2.107e-03","GO:0009058 (5956) <br> biosynthetic process <br> 2.346e-03","GO:0071310 (2385) <br> cellular response to organic substance <br> 2.669e-03","GO:0006354 (80) <br> DNA-templated transcription, elongation <br> 2.772e-03","GO:0042692 (357) <br> muscle cell differentiation <br> 2.969e-03","GO:0006928 (2074) <br> movement of cell or subcellular component <br> 3.039e-03","GO:0033206 (7) <br> meiotic cytokinesis <br> 3.322e-03","GO:0051305 (7) <br> chromosome movement towards spindle pole <br> 3.322e-03","GO:0010467 (8909) <br> gene expression <br> 3.383e-03","GO:0055002 (64) <br> striated muscle cell development <br> 4.042e-03","GO:0071103 (329) <br> DNA conformation change <br> 4.548e-03","GO:0040031 (7) <br> snRNA modification <br> 4.667e-03","GO:1900363 (18) <br> regulation of mRNA polyadenylation <br> 4.712e-03","GO:0051253 (1395) <br> negative regulation of RNA metabolic process <br> 4.958e-03","GO:0051252 (3700) <br> regulation of RNA metabolic process <br> 5.066e-03","GO:1903508 (1604) <br> positive regulation of nucleic acid-templated transcription <br> 5.165e-03","GO:0045893 (1604) <br> positive regulation of transcription, DNA-templated <br> 5.165e-03","GO:1902680 (1606) <br> positive regulation of RNA biosynthetic process <br> 5.278e-03","GO:0045934 (1496) <br> negative regulation of nucleobase-containing compound metabolic process <br> 5.324e-03","GO:0048523 (4818) <br> negative regulation of cellular process <br> 5.906e-03","GO:0043933 (3387) <br> protein-containing complex organization <br> 6.028e-03","GO:0010033 (3009) <br> response to organic substance <br> 6.045e-03","GO:0031333 (147) <br> negative regulation of protein-containing complex assembly <br> 6.211e-03","GO:0046831 (12) <br> regulation of RNA export from nucleus <br> 6.448e-03","GO:0120032 (190) <br> regulation of plasma membrane bounded cell projection assembly <br> 7.903e-03","GO:0060491 (192) <br> regulation of cell projection assembly <br> 8.317e-03","GO:0065008 (3741) <br> regulation of biological quality <br> 8.381e-03","GO:0043462 (72) <br> regulation of ATP-dependent activity <br> 8.473e-03","GO:1903313 (116) <br> positive regulation of mRNA metabolic process <br> 8.569e-03","GO:0043604 (867) <br> amide biosynthetic process <br> 8.975e-03","GO:0051129 (680) <br> negative regulation of cellular component organization <br> 9.038e-03","GO:0055001 (174) <br> muscle cell development <br> 9.168e-03","GO:0051171 (5529) <br> regulation of nitrogen compound metabolic process <br> 9.360e-03","GO:0080090 (5694) <br> regulation of primary metabolic process <br> 1.029e-02","GO:0070887 (3008) <br> cellular response to chemical stimulus <br> 1.254e-02","GO:0010769 (96) <br> regulation of cell morphogenesis involved in differentiation <br> 1.279e-02","GO:0032239 (15) <br> regulation of nucleobase-containing compound transport <br> 1.326e-02","GO:0006518 (895) <br> peptide metabolic process <br> 1.338e-02","GO:0071346 (132) <br> cellular response to interferon-gamma <br> 1.385e-02","GO:0034654 (3953) <br> nucleobase-containing compound biosynthetic process <br> 1.490e-02","GO:0090304 (7476) <br> nucleic acid metabolic process <br> 1.541e-02","GO:0031118 (10) <br> rRNA pseudouridine synthesis <br> 1.585e-02","GO:0031323 (5921) <br> regulation of cellular metabolic process <br> 1.587e-02","GO:0032210 (54) <br> regulation of telomere maintenance via telomerase <br> 1.599e-02","GO:0050794 (11067) <br> regulation of cellular process <br> 1.635e-02","GO:0048870 (1645) <br> cell motility <br> 1.671e-02","GO:0051674 (1645) <br> localization of cell <br> 1.671e-02","GO:0031442 (10) <br> positive regulation of mRNA 3'-end processing <br> 1.746e-02","GO:0060795 (32) <br> cell fate commitment involved in formation of primary germ layer <br> 2.126e-02","GO:0034341 (153) <br> response to interferon-gamma <br> 2.156e-02","GO:1900756 (2) <br> protein processing in phagocytic vesicle <br> 2.264e-02","GO:1903921 (2) <br> regulation of protein processing in phagocytic vesicle <br> 2.264e-02","GO:1903923 (2) <br> positive regulation of protein processing in phagocytic vesicle <br> 2.264e-02","GO:0010468 (5388) <br> regulation of gene expression <br> 2.310e-02","GO:0090501 (155) <br> RNA phosphodiester bond hydrolysis <br> 2.320e-02","GO:0018130 (4024) <br> heterocycle biosynthetic process <br> 2.330e-02","GO:2001168 (4) <br> positive regulation of histone H2B ubiquitination <br> 2.383e-02","GO:0048518 (6094) <br> positive regulation of biological process <br> 2.415e-02","GO:0019438 (4035) <br> aromatic compound biosynthetic process <br> 2.493e-02","GO:0051656 (409) <br> establishment of organelle localization <br> 2.772e-02","GO:0048522 (5561) <br> positive regulation of cellular process <br> 2.861e-02","GO:1904356 (63) <br> regulation of telomere maintenance via telomere lengthening <br> 2.962e-02","GO:0051052 (356) <br> regulation of DNA metabolic process <br> 2.972e-02","GO:0010556 (3926) <br> regulation of macromolecule biosynthetic process <br> 3.038e-02","GO:0007049 (1713) <br> cell cycle <br> 3.188e-02","GO:0031326 (4081) <br> regulation of cellular biosynthetic process <br> 3.206e-02","GO:0006333 (212) <br> chromatin assembly or disassembly <br> 3.434e-02","GO:0040011 (1821) <br> locomotion <br> 3.541e-02","GO:0050789 (12142) <br> regulation of biological process <br> 3.561e-02","GO:0060255 (6710) <br> regulation of macromolecule metabolic process <br> 3.584e-02","GO:2001166 (5) <br> regulation of histone H2B ubiquitination <br> 3.968e-02","GO:0033184 (5) <br> positive regulation of histone ubiquitination <br> 3.968e-02","GO:0065003 (3194) <br> protein-containing complex assembly <br> 4.186e-02","GO:0009889 (4141) <br> regulation of biosynthetic process <br> 4.189e-02","GO:0006513 (68) <br> protein monoubiquitination <br> 4.372e-02","GO:0051172 (2330) <br> negative regulation of nitrogen compound metabolic process <br> 4.508e-02","GO:2000112 (656) <br> regulation of cellular macromolecule biosynthetic process <br> 4.604e-02","GO:0046824 (61) <br> positive regulation of nucleocytoplasmic transport <br> 4.628e-02","GO:0007163 (222) <br> establishment or maintenance of cell polarity <br> 4.659e-02","GO:0045087 (892) <br> innate immune response <br> 4.773e-02","GO:0140694 (373) <br> non-membrane-bounded organelle assembly <br> 4.945e-02"],"key":["GO:0030036","GO:0030029","GO:0007015","GO:0034314","GO:0097435","GO:0050684","GO:0007010","GO:0045010","GO:0006996","GO:0031124","GO:0030833","GO:1902903","GO:0008064","GO:0030832","GO:0110053","GO:0030041","GO:0051128","GO:0006378","GO:0043631","GO:1903311","GO:0031123","GO:0008154","GO:0033043","GO:0032271","GO:0032956","GO:0044087","GO:0051493","GO:0032970","GO:0043254","GO:0051258","GO:0032535","GO:0097581","GO:0098789","GO:0098787","GO:1902743","GO:0030048","GO:0090066","GO:0051693","GO:0006379","GO:0010604","GO:0030835","GO:0043484","GO:1901880","GO:0033523","GO:0006397","GO:0030834","GO:0031440","GO:0031325","GO:0009059","GO:0016071","GO:0000723","GO:0051276","GO:0043242","GO:0051016","GO:0034645","GO:0051173","GO:0030042","GO:0016043","GO:0032543","GO:1901879","GO:0048024","GO:0070358","GO:0071840","GO:0030837","GO:0009893","GO:0044085","GO:0031328","GO:0032200","GO:0010557","GO:0022607","GO:0006403","GO:0050685","GO:0022604","GO:0045935","GO:0009891","GO:0044271","GO:0034097","GO:0032272","GO:0051261","GO:0010591","GO:0010639","GO:1903312","GO:0008356","GO:0051494","GO:0140053","GO:0031120","GO:0007004","GO:0006278","GO:0043244","GO:0000380","GO:1902745","GO:0016574","GO:0046833","GO:0044260","GO:0010833","GO:0031032","GO:0016344","GO:0006368","GO:0010390","GO:0032241","GO:0050686","GO:0006412","GO:0010628","GO:1904874","GO:0034641","GO:0032984","GO:0051254","GO:0001711","GO:0044249","GO:0071345","GO:0030032","GO:0010592","GO:0043043","GO:0000381","GO:0030050","GO:1901576","GO:1902904","GO:0061061","GO:1904872","GO:0006259","GO:0045944","GO:0006396","GO:0099515","GO:0019219","GO:0090685","GO:0090672","GO:0090671","GO:0090670","GO:0009058","GO:0071310","GO:0006354","GO:0042692","GO:0006928","GO:0033206","GO:0051305","GO:0010467","GO:0055002","GO:0071103","GO:0040031","GO:1900363","GO:0051253","GO:0051252","GO:1903508","GO:0045893","GO:1902680","GO:0045934","GO:0048523","GO:0043933","GO:0010033","GO:0031333","GO:0046831","GO:0120032","GO:0060491","GO:0065008","GO:0043462","GO:1903313","GO:0043604","GO:0051129","GO:0055001","GO:0051171","GO:0080090","GO:0070887","GO:0010769","GO:0032239","GO:0006518","GO:0071346","GO:0034654","GO:0090304","GO:0031118","GO:0031323","GO:0032210","GO:0050794","GO:0048870","GO:0051674","GO:0031442","GO:0060795","GO:0034341","GO:1900756","GO:1903921","GO:1903923","GO:0010468","GO:0090501","GO:0018130","GO:2001168","GO:0048518","GO:0019438","GO:0051656","GO:0048522","GO:1904356","GO:0051052","GO:0010556","GO:0007049","GO:0031326","GO:0006333","GO:0040011","GO:0050789","GO:0060255","GO:2001166","GO:0033184","GO:0065003","GO:0009889","GO:0006513","GO:0051172","GO:2000112","GO:0046824","GO:0007163","GO:0045087","GO:0140694"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(255,153,0,1)","opacity":0.8,"size":[18.9689234520263,19.1206656257896,18.3473821837242,14.5814139974581,19.0463669754254,16.708741750045,19.8513263318723,15.1893666851417,21.0018269113499,15.6667434572733,16.6979252377404,18.1105369965064,16.8899800403965,16.8993690490748,17.6447731536901,16.9722060406806,20.3897665416323,15.0924043121985,15.1254260222545,17.8902134152238,16.5367347658025,17.2145842446943,19.5332499620522,17.2652787333353,18.0230448205345,19.3257581587702,18.5665122992221,18.1749820385619,18.2127857347699,17.7018443366872,18.0425906994776,16.0254831918018,12.3542694385448,12.5223664034652,15.1254260222545,16.608155189633,18.4742232008518,14.798562659995,13.2899306801292,20.8655421968307,14.9525766293039,16.8124132451137,15.7545592284552,12.1689282166855,20.2026609694964,15.2506857256693,13.9611149660018,20.7945145333286,21.2832942259864,20.3396875973885,16.6649501000042,19.4457064749222,15.9531751392093,14.0285832754376,19.9561802351649,20.7200482713786,15.3937574422679,21.8008623810869,15.7965422855522,16.0603010299089,16.2528819106722,10.7944929897153,21.8285943750956,15.5489306475125,20.9630699561741,21.2649713126228,20.2169029716303,16.936282113548,20.1516574830331,21.2060372008953,17.2145842446943,14.2155212474336,17.8319964975192,20.1888657788203,20.2370242051305,21.2491783746142,19.301312316507,15.8769999436579,16.461396937013,14.671965682599,18.030899694876,16.0603010299089,13.1834053419274,16.936282113548,16.3400771582388,7.77039980235131,15.62075611391,15.6439317492578,16.5488994624023,15.7965422855522,14.4852668565195,14.9525766293039,9.77892631171347,21.4656446208725,15.8964279239824,17.177141715501,7.77039980235131,15.3662271239498,14.0285832754376,10.3487400178115,14.0933449312819,18.9725551526272,19.5422086045135,12.6760065098355,21.9287584012119,17.4318357596586,20.077497916752,12.6760065098355,21.4748804145049,19.1646449933625,15.6667434572733,14.0285832754376,19.0170406106578,15.3937574422679,13.0697969123388,21.4891497858458,16.9543645281488,18.7823225213627,13.0697969123388,19.2650749324978,19.615672030627,20.8580296361819,13.2899306801292,21.0429841272711,13.1834053419274,13.1834053419274,13.1834053419274,13.1834053419274,21.5054614715639,20.4606184422425,15.8170833598509,18.0503246746426,20.2952914186446,10.7944929897153,10.7944929897153,21.9461375999145,15.4473232090416,17.9374673218716,10.7944929897153,13.0697969123388,19.8167413392001,20.9698850002471,19.9868125912746,19.9868125912746,19.988322590352,19.9021198944811,21.2689411768662,20.8686004785421,20.7320503897643,16.7718916003282,12.1689282166855,17.154155960642,17.169524110889,20.9824712180928,15.6439317492578,16.4087842754834,19.223289317369,18.9111704501601,17.024344902898,21.4228418063926,21.4555461880038,20.7316652904454,16.110971780725,12.6760065098355,19.2636585520497,16.608155189633,21.045288742489,21.7555462670674,11.7306454967647,21.4989299313842,15.1577373033357,22.1791467424615,20.0173702355595,20.0173702355595,11.7306454967647,14.2155212474336,16.8322273404496,3.77952755905512,3.77952755905512,3.77952755905512,21.3940614406326,16.8517546521526,21.0655269573591,8.99752440324434,21.5308228872123,21.0686283768389,18.2362578787775,21.4292648753663,15.4207837047603,18.0464636422932,21.0374906409973,20.066290378378,21.081501044364,17.3140905024666,20.13985350238,22.2778367258612,21.6370474826684,9.77892631171347,9.77892631171347,20.8010368448869,21.0980610401899,15.5489306475125,20.4331258024996,18.8644595290412,15.3662271239498,17.3808104874969,19.2593991152911,18.1105369965064],"symbol":"circle","line":{"width":1.88976377952756,"color":"rgba(255,153,0,1)"}},"hoveron":"points","set":"SharedData0dd1ba32","name":"GO:BP","legendgroup":"GO:BP","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","_isNestedKey":false,"frame":null},{"x":[57.1046489358643,62.1290372645198,62.1446288497134,55.4324514238447,54.0915750971904,59.9579090263032,55.8456284314766,64.4326939768824,59.4745698852999,62.1485267460118,57.1163426247595,55.9859526982195,55.7598747129115,57.8452492325629,55.058253379197,55.4246556312479,54.087677200892,57.2371774100104,68.4475271642487,62.1329351608182,64.608099310311,59.4979572630904,62.9281060056946,64.5223455917459,69.6987518760395,64.428796080584,62.1368330571166,62.1251393682214,62.2108930867865,55.7130999573305,68.5683619494995,68.4514250605471,57.8374534399661,62.1212414719229,58.7534590700934,57.8257597510709,67.9758817121406,61.8639803162276,54.9491122828414,55.7286915425242,55.8417305351782,55.3622892904733,59.5018551593888,54.9257249050509,55.736487335121,55.9157905648481,54.559322653,67.0481823931181,57.3463185063659,56.0288295575021,69.2855748684076,55.6741209943464,57.6659460028359,57.8413513362645,59.7941973817699,55.7754662981052,55.3427998089812,61.1974400491989,66.4323147779687,61.9887129977769,55.783262090702,61.396232760418,68.884091549671,57.2410753063088,55.4168598386511,54.7815027420096,69.4843675796267,68.1084101862867,57.2449732026072,55.1985776459399,54.991989142124,57.3346248174707,59.447284611211,56.2471117502132,58.3207925809694,67.0325908079244,57.8296576473693,55.350595601578,62.140730953415,55.1479049940605,71.1994419509289,66.1321767629908,54.6294847863715,66.3660505408957,59.0263118109823,68.5722598457979,57.092955246969,55.9352800463401,66.1282788666924,55.2102713348351,65.2668437847429,59.4511825075094,59.4784677815983,56.1730517205434,57.3268290248739,66.0620146296194,55.2999229496987,67.9096174750675,58.157080936436,63.816826361733,54.1929204009491,55.0816407569875,58.3597715439535,65.7306934442542,59.9501132337064,58.3246904772678,65.220069029162,65.1927837550731,54.5632205492984,54.2474909491269,61.1506652936179,69.7728119057094,58.9912307442966,67.46135940075],"y":[17,17,17,17,17,17,17,15.9550544960812,15.9504430430876,15.9504430430876,15.8316061393429,15.635420224604,15.5403363763792,15.4896714781687,15.3643616056849,15.0873940862692,15.0873940862692,14.8510978332574,11.7606989415706,11.606325274139,11.2551007084142,11.2151019614337,10.5301847271003,10.4842576895513,10.409677688729,10.4063066906882,10.4063066906882,9.85599360240107,9.32952115899377,9.21042085993618,9.12954785268787,9.04363676721374,9.0347051383215,8.9285700077257,8.53719656787721,8.47250121175322,8.41517502332721,7.87322689166674,7.78539820098073,7.54947447089023,7.52090857332646,7.31237065464378,7.27400676636394,7.18758973859136,7.02410044058992,7.02313492810155,6.88776175417042,6.88776175417042,6.82563362423588,6.81164298047852,6.72216345315789,6.70073517724528,6.6118373250262,6.61052560985268,6.46364903958173,6.39571804311534,6.3889665838577,6.26987561737812,6.24047778103107,6.10850954035631,6.0380691934433,5.55718638739667,5.48594540520191,5.46727135054192,5.46329342048808,5.44955463876908,5.39588773885635,5.37416630999888,5.27058579007702,5.23307756809936,5.1753933798207,5.09531720278192,4.9662950361739,4.89988705127248,4.8224813536163,4.82245419941227,4.72616634878053,4.58998607259953,4.54139443407895,4.44519576591901,4.1572557667457,4.00406128089862,3.69233782051383,3.60746092706012,3.60746092706012,3.42124407018361,3.4203564324182,3.32915574213993,3.30777035905003,3.07087128406502,2.89495626250087,2.8326744249967,2.8326744249967,2.80598866549898,2.584105301846,2.58391918685851,2.29317821979429,2.2737746670181,2.09204479158853,1.98099727580793,1.9599489673968,1.91759383900312,1.87770979213178,1.74907191975009,1.65635488430682,1.59185187993008,1.56038865805759,1.54900302260684,1.5476759789856,1.54738035605233,1.41640594367627,1.40938790860257,1.36854187971577,1.3653167100713],"text":["GO:0015629 (501) <br> actin cytoskeleton <br> 9.760e-33","GO:0043228 (6362) <br> non-membrane-bounded organelle <br> 3.857e-26","GO:0043232 (6362) <br> intracellular non-membrane-bounded organelle <br> 3.857e-26","GO:0005763 (28) <br> mitochondrial small ribosomal subunit <br> 4.106e-23","GO:0000314 (28) <br> organellar small ribosomal subunit <br> 4.106e-23","GO:0032991 (7228) <br> protein-containing complex <br> 1.720e-22","GO:0005885 (11) <br> Arp2/3 protein complex <br> 5.906e-22","GO:0070013 (6569) <br> intracellular organelle lumen <br> 1.109e-16","GO:0031974 (6570) <br> membrane-enclosed lumen <br> 1.121e-16","GO:0043233 (6570) <br> organelle lumen <br> 1.121e-16","GO:0015935 (76) <br> small ribosomal subunit <br> 1.474e-16","GO:0005925 (420) <br> focal adhesion <br> 2.315e-16","GO:0005856 (2334) <br> cytoskeleton <br> 2.882e-16","GO:0030055 (428) <br> cell-substrate junction <br> 3.238e-16","GO:0005654 (4161) <br> nucleoplasm <br> 4.322e-16","GO:0005761 (86) <br> mitochondrial ribosome <br> 8.177e-16","GO:0000313 (86) <br> organellar ribosome <br> 8.177e-16","GO:0016459 (57) <br> myosin complex <br> 1.409e-15","GO:0099080 (1330) <br> supramolecular complex <br> 1.735e-12","GO:0043229 (15870) <br> intracellular organelle <br> 2.476e-12","GO:0070161 (839) <br> anchoring junction <br> 5.558e-12","GO:0031981 (5442) <br> nuclear lumen <br> 6.094e-12","GO:0044391 (189) <br> ribosomal subunit <br> 2.950e-11","GO:0070062 (2110) <br> extracellular exosome <br> 3.279e-11","GO:1903561 (2132) <br> extracellular vesicle <br> 3.893e-11","GO:0065010 (2133) <br> extracellular membrane-bounded organelle <br> 3.924e-11","GO:0043230 (2133) <br> extracellular organelle <br> 3.924e-11","GO:0043227 (15557) <br> membrane-bounded organelle <br> 1.393e-10","GO:0043292 (242) <br> contractile fiber <br> 4.683e-10","GO:0005840 (520) <br> ribosome <br> 6.160e-10","GO:0099512 (988) <br> supramolecular fiber <br> 7.421e-10","GO:0099081 (996) <br> supramolecular polymer <br> 9.044e-10","GO:0030027 (203) <br> lamellipodium <br> 9.232e-10","GO:0043226 (16674) <br> organelle <br> 1.179e-09","GO:0031252 (421) <br> cell leading edge <br> 2.903e-09","GO:0030016 (232) <br> myofibril <br> 3.369e-09","GO:0098798 (277) <br> mitochondrial protein-containing complex <br> 3.844e-09","GO:0042641 (79) <br> actomyosin <br> 1.339e-08","GO:0005622 (17758) <br> intracellular anatomical structure <br> 1.639e-08","GO:0005847 (24) <br> mRNA cleavage and polyadenylation specificity factor complex <br> 2.822e-08","GO:0005884 (114) <br> actin filament <br> 3.014e-08","GO:0005743 (500) <br> mitochondrial inner membrane <br> 4.871e-08","GO:0031982 (3942) <br> vesicle <br> 5.321e-08","GO:0005615 (3353) <br> extracellular space <br> 6.492e-08","GO:0005849 (28) <br> mRNA cleavage factor complex <br> 9.460e-08","GO:0005903 (105) <br> brush border <br> 9.481e-08","GO:0001725 (71) <br> stress fiber <br> 1.295e-07","GO:0097517 (71) <br> contractile actin filament bundle <br> 1.295e-07","GO:0016604 (869) <br> nuclear body <br> 1.494e-07","GO:0005938 (319) <br> cell cortex <br> 1.543e-07","GO:0140513 (2671) <br> nuclear protein-containing complex <br> 1.896e-07","GO:0005829 (5373) <br> cytosol <br> 1.992e-07","GO:0019866 (562) <br> organelle inner membrane <br> 2.444e-07","GO:0030054 (2076) <br> cell junction <br> 2.452e-07","GO:0032432 (79) <br> actin filament bundle <br> 3.438e-07","GO:0005862 (4) <br> muscle thin filament tropomyosin <br> 4.021e-07","GO:0005737 (12193) <br> cytoplasm <br> 4.084e-07","GO:0035861 (73) <br> site of double-strand break <br> 5.372e-07","GO:0090734 (97) <br> site of DNA damage <br> 5.748e-07","GO:0042995 (2312) <br> cell projection <br> 7.789e-07","GO:0005865 (22) <br> striated muscle thin filament <br> 9.161e-07","GO:0036379 (26) <br> myofilament <br> 2.772e-06","GO:0120025 (2205) <br> plasma membrane bounded cell projection <br> 3.266e-06","GO:0016460 (20) <br> myosin II complex <br> 3.410e-06","GO:0005759 (483) <br> mitochondrial matrix <br> 3.441e-06","GO:0005576 (4303) <br> extracellular region <br> 3.552e-06","GO:1902494 (1424) <br> catalytic complex <br> 4.019e-06","GO:0098862 (161) <br> cluster of actin-based cell projections <br> 4.225e-06","GO:0016461 (8) <br> unconventional myosin complex <br> 5.363e-06","GO:0005694 (1795) <br> chromosome <br> 5.847e-06","GO:0005634 (10053) <br> nucleus <br> 6.677e-06","GO:0016593 (8) <br> Cdc73/Paf1 complex <br> 8.029e-06","GO:0031966 (745) <br> mitochondrial membrane <br> 1.081e-05","GO:0008290 (8) <br> F-actin capping protein complex <br> 1.259e-05","GO:0030863 (110) <br> cortical cytoskeleton <br> 1.505e-05","GO:0097513 (3) <br> myosin II filament <br> 1.505e-05","GO:0030017 (211) <br> sarcomere <br> 1.879e-05","GO:0005740 (796) <br> mitochondrial envelope <br> 2.570e-05","GO:0043231 (14514) <br> intracellular membrane-bounded organelle <br> 2.875e-05","GO:0005681 (195) <br> spliceosomal complex <br> 3.588e-05","GO:1990904 (2377) <br> ribonucleoprotein complex <br> 6.962e-05","GO:0072589 (4) <br> box H/ACA scaRNP complex <br> 9.907e-05","GO:0002102 (30) <br> podosome <br> 2.031e-04","GO:0090661 (5) <br> box H/ACA telomerase RNP complex <br> 2.469e-04","GO:0031429 (5) <br> box H/ACA snoRNP complex <br> 2.469e-04","GO:0099513 (757) <br> polymeric cytoskeletal fiber <br> 3.791e-04","GO:0015030 (102) <br> Cajal body <br> 3.799e-04","GO:0005911 (495) <br> cell-cell junction <br> 4.686e-04","GO:0072588 (6) <br> box H/ACA RNP complex <br> 4.923e-04","GO:0005697 (23) <br> telomerase holoenzyme complex <br> 8.494e-04","GO:0071013 (85) <br> catalytic step 2 spliceosome <br> 1.274e-03","GO:0031967 (1245) <br> organelle envelope <br> 1.470e-03","GO:0031975 (1245) <br> envelope <br> 1.470e-03","GO:0008023 (56) <br> transcription elongation factor complex <br> 1.563e-03","GO:0016591 (91) <br> RNA polymerase II, holoenzyme <br> 2.606e-03","GO:0072357 (7) <br> PTW/PP1 phosphatase complex <br> 2.607e-03","GO:0005726 (4) <br> perichromatin fibrils <br> 5.091e-03","GO:0098687 (359) <br> chromosomal region <br> 5.324e-03","GO:0030478 (4) <br> actin cap <br> 8.090e-03","GO:0055029 (116) <br> nuclear DNA-directed RNA polymerase complex <br> 1.045e-02","GO:0000428 (117) <br> DNA-directed RNA polymerase complex <br> 1.097e-02","GO:0005662 (13) <br> DNA replication factor A complex <br> 1.209e-02","GO:0030880 (121) <br> RNA polymerase complex <br> 1.325e-02","GO:0071203 (12) <br> WASH complex <br> 1.782e-02","GO:0032982 (22) <br> myosin filament <br> 2.206e-02","GO:0030864 (80) <br> cortical actin cytoskeleton <br> 2.559e-02","GO:0071001 (3) <br> U4/U6 snRNP <br> 2.752e-02","GO:0070937 (6) <br> CRD-mediated mRNA stability complex <br> 2.825e-02","GO:0001726 (181) <br> ruffle <br> 2.834e-02","GO:0000781 (169) <br> chromosome, telomeric region <br> 2.835e-02","GO:0035770 (260) <br> ribonucleoprotein granule <br> 3.833e-02","GO:1904813 (124) <br> ficolin-1-rich granule lumen <br> 3.896e-02","GO:0031410 (2447) <br> cytoplasmic vesicle <br> 4.280e-02","GO:0097708 (2449) <br> intracellular vesicle <br> 4.312e-02"],"key":["GO:0015629","GO:0043228","GO:0043232","GO:0005763","GO:0000314","GO:0032991","GO:0005885","GO:0070013","GO:0031974","GO:0043233","GO:0015935","GO:0005925","GO:0005856","GO:0030055","GO:0005654","GO:0005761","GO:0000313","GO:0016459","GO:0099080","GO:0043229","GO:0070161","GO:0031981","GO:0044391","GO:0070062","GO:1903561","GO:0065010","GO:0043230","GO:0043227","GO:0043292","GO:0005840","GO:0099512","GO:0099081","GO:0030027","GO:0043226","GO:0031252","GO:0030016","GO:0098798","GO:0042641","GO:0005622","GO:0005847","GO:0005884","GO:0005743","GO:0031982","GO:0005615","GO:0005849","GO:0005903","GO:0001725","GO:0097517","GO:0016604","GO:0005938","GO:0140513","GO:0005829","GO:0019866","GO:0030054","GO:0032432","GO:0005862","GO:0005737","GO:0035861","GO:0090734","GO:0042995","GO:0005865","GO:0036379","GO:0120025","GO:0016460","GO:0005759","GO:0005576","GO:1902494","GO:0098862","GO:0016461","GO:0005694","GO:0005634","GO:0016593","GO:0031966","GO:0008290","GO:0030863","GO:0097513","GO:0030017","GO:0005740","GO:0043231","GO:0005681","GO:1990904","GO:0072589","GO:0002102","GO:0090661","GO:0031429","GO:0099513","GO:0015030","GO:0005911","GO:0072588","GO:0005697","GO:0071013","GO:0031967","GO:0031975","GO:0008023","GO:0016591","GO:0072357","GO:0005726","GO:0098687","GO:0030478","GO:0055029","GO:0000428","GO:0005662","GO:0030880","GO:0071203","GO:0032982","GO:0030864","GO:0071001","GO:0070937","GO:0001726","GO:0000781","GO:0035770","GO:1904813","GO:0031410","GO:0097708"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(16,150,24,1)","opacity":0.8,"size":[18.5093206302223,21.5783776479263,21.5783776479263,13.9611149660018,13.9611149660018,21.7186491899641,11.9626902987181,21.6136744323874,21.6138420676856,21.6138420676856,15.7331002252975,18.2722688771406,20.4351485853431,18.2978168250027,21.1035242849928,15.9345090534973,15.9345090534973,15.2506857256693,19.7581982152725,22.5599778805315,19.1814836382837,21.4051772962856,17.1464044185986,20.3157434393389,20.3280586048311,20.3286151448313,20.3286151448313,22.5391342674345,17.5048066811551,18.5588694915346,19.3885502335527,19.3986943920346,17.2509927405727,22.611590191448,18.2754913372617,17.4442857448196,17.6967597289842,15.7965422855522,22.6771653543307,13.6592813242164,16.3817083524515,18.5066562827429,21.042118612013,20.8570011251474,13.9611149660018,16.2528819106722,15.62075611391,15.62075611391,19.2262193222329,17.8945828748266,20.5934358537654,21.3909526731361,18.6617323675032,20.2964375921402,15.7965422855522,8.99752440324434,22.2822864937662,15.6667434572733,16.1274657222001,20.4239770277874,13.4847654999701,13.817141287718,20.3679678765959,13.2899306801292,18.4604517488511,21.1415267205783,19.8419198881927,16.9086929071744,11.1588894645275,20.1225803364186,22.07628187225,11.1588894645275,19.0292040076885,11.1588894645275,16.3259182730869,7.77039980235131,17.3072275895117,19.1142469792459,22.4663374189607,17.1922464142135,20.4566619508423,8.99752440324434,14.0933449312819,9.77892631171347,9.77892631171347,19.049769984315,16.2071544572567,18.4932467877116,10.3487400178115,13.574309418746,15.915595430735,19.6768124493713,19.6768124493713,15.2203410437363,16.0254831918018,10.7944929897153,8.99752440324434,18.0580112889306,8.99752440324434,16.4087842754834,16.4221263012614,12.3542694385448,16.4742428340656,12.1689282166855,13.4847654999701,15.8170833598509,7.77039980235131,10.3487400178115,17.0827037144223,16.9810383690236,17.6070890538082,16.5120774547391,20.4908077740003,20.4917679449807],"symbol":"circle","line":{"width":1.88976377952756,"color":"rgba(16,150,24,1)"}},"hoveron":"points","set":"SharedData0dd1ba32","name":"GO:CC","legendgroup":"GO:CC","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","_isNestedKey":false,"frame":null},{"x":[37.3800599541577,3.80836613997898,9.6154039601701,29.7295799266173,3.69144591541137,8.59040332479408,8.58260864315624,3.53944962347349,3.71093261950597,2.11302288374869,3.78887943588438,2.53783303301099,40.2134267295127,28.3577159583574,48.6628616249317,42.056868936862,9.30361669465648,2.78336550460296,2.78726284542188,29.9049602634687,3.55893632756809,3.7343166644195,28.572069703398,37.3332918643307,25.2203565991266,40.1588639580478,48.4134318125208,10.106468903354,3.61739643985189,9.74401620719447,42.349169498281,28.3031531868925,27.6990653599599,9.33869276202677,42.5518312208648,3.69924059704921,3.70703527868705,23.0495377629881,3.54334696429241,20.9917418105982],"y":[17,17,13.7583775247094,12.7265582230965,12.1563222658129,11.6217550511438,11.4983987633708,7.70333249230183,7.70229316144422,7.61179958440454,7.51733781619453,6.30301928144294,6.26755990040909,5.88266128829516,5.8743412255317,5.61606223453653,5.50862272314159,4.84395844918614,4.84395844918614,4.75830377848075,4.58535672238116,4.32282491907508,3.7962212323534,3.66591760843163,3.33626812756784,2.85165797266868,2.73275015970215,2.41834029705726,2.37136679465484,2.0361379981528,2.03115117875263,1.77136775277471,1.66168671007849,1.57874635753683,1.55614546868692,1.51132349209264,1.44743150753695,1.34824131708693,1.31974062016778,1.309356719105],"text":["GO:0051015 (218) <br> actin filament binding <br> 1.825e-24","GO:0003779 (449) <br> actin binding <br> 7.910e-22","GO:0008092 (995) <br> cytoskeletal protein binding <br> 1.744e-14","GO:0044877 (1275) <br> protein-containing complex binding <br> 1.877e-13","GO:0003723 (3142) <br> RNA binding <br> 6.977e-13","GO:0005200 (104) <br> structural constituent of cytoskeleton <br> 2.389e-12","GO:0005198 (980) <br> structural molecule activity <br> 3.174e-12","GO:0003676 (5447) <br> nucleic acid binding <br> 1.980e-08","GO:0003729 (350) <br> mRNA binding <br> 1.985e-08","GO:0000146 (31) <br> microfilament motor activity <br> 2.445e-08","GO:0003774 (115) <br> cytoskeletal motor activity <br> 3.039e-08","GO:0000993 (36) <br> RNA polymerase II complex binding <br> 4.977e-07","GO:0070063 (58) <br> RNA polymerase binding <br> 5.401e-07","GO:0043175 (41) <br> RNA polymerase core enzyme binding <br> 1.310e-06","GO:1901363 (7416) <br> heterocyclic compound binding <br> 1.336e-06","GO:0097159 (7497) <br> organic cyclic compound binding <br> 2.421e-06","GO:0005515 (14799) <br> protein binding <br> 3.100e-06","GO:0001098 (57) <br> basal transcription machinery binding <br> 1.432e-05","GO:0001099 (57) <br> basal RNA polymerase II transcription machinery binding <br> 1.432e-05","GO:0045296 (334) <br> cadherin binding <br> 1.745e-05","GO:0003682 (580) <br> chromatin binding <br> 2.598e-05","GO:0003735 (449) <br> structural constituent of ribosome <br> 4.755e-05","GO:0043531 (34) <br> ADP binding <br> 1.599e-04","GO:0050839 (550) <br> cell adhesion molecule binding <br> 2.158e-04","GO:0034513 (5) <br> box H/ACA snoRNA binding <br> 4.610e-04","GO:0070034 (22) <br> telomerase RNA binding <br> 1.407e-03","GO:0140657 (550) <br> ATP-dependent activity <br> 1.850e-03","GO:0008307 (43) <br> structural constituent of muscle <br> 3.816e-03","GO:0003697 (123) <br> single-stranded DNA binding <br> 4.252e-03","GO:0008134 (602) <br> transcription factor binding <br> 9.202e-03","GO:0098505 (10) <br> G-rich strand telomeric DNA binding <br> 9.308e-03","GO:0043047 (12) <br> single-stranded telomeric DNA binding <br> 1.693e-02","GO:0042162 (36) <br> telomeric DNA binding <br> 2.179e-02","GO:0005524 (1502) <br> ATP binding <br> 2.638e-02","GO:0098847 (14) <br> sequence-specific single stranded DNA binding <br> 2.779e-02","GO:0003725 (74) <br> double-stranded RNA binding <br> 3.081e-02","GO:0003727 (84) <br> single-stranded RNA binding <br> 3.569e-02","GO:0032559 (1565) <br> adenyl ribonucleotide binding <br> 4.485e-02","GO:0003677 (2494) <br> DNA binding <br> 4.789e-02","GO:0030554 (1576) <br> adenyl nucleotide binding <br> 4.905e-02"],"key":["GO:0051015","GO:0003779","GO:0008092","GO:0044877","GO:0003723","GO:0005200","GO:0005198","GO:0003676","GO:0003729","GO:0000146","GO:0003774","GO:0000993","GO:0070063","GO:0043175","GO:1901363","GO:0097159","GO:0005515","GO:0001098","GO:0001099","GO:0045296","GO:0003682","GO:0003735","GO:0043531","GO:0050839","GO:0034513","GO:0070034","GO:0140657","GO:0008307","GO:0003697","GO:0008134","GO:0098505","GO:0043047","GO:0042162","GO:0005524","GO:0098847","GO:0003725","GO:0003727","GO:0032559","GO:0003677","GO:0030554"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(220,57,18,1)","opacity":0.8,"size":[17.354529519854,18.3624720287095,19.3974311963365,19.7062024214885,20.7820862063444,16.2378047873792,19.3783169237459,21.4062006044941,18.0230448205345,14.1555983333967,16.3953124390496,14.4348828193235,15.2804246607373,14.671965682599,21.7467399236646,21.7586107865689,22.48676457789,15.2506857256693,15.2506857256693,17.9583759839834,18.7032804941857,18.3624720287095,14.3289994886174,18.6332205816173,9.77892631171347,13.4847654999701,18.6332205816173,14.7575093437122,16.4995810417852,18.7521961985983,11.7306454967647,12.1689282166855,14.4348828193235,19.906995199969,12.5223664034652,15.6892019866473,15.8964279239824,19.9569562634249,20.5131526676116,19.9654575586114],"symbol":"circle","line":{"width":1.88976377952756,"color":"rgba(220,57,18,1)"}},"hoveron":"points","set":"SharedData0dd1ba32","name":"GO:MF","legendgroup":"GO:MF","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","_isNestedKey":false,"frame":null},{"x":[199.020054891063,199.429989113158,199.4495097904,199.445605654952,199.207453392592,198.879506014916,199.465126332194,199.453413925849,199.113754141827,198.664778565247,198.719436461526],"y":[9.23022832674179,8.27154904895142,7.74410990918767,6.97541275260356,6.63981851586111,6.56275798910939,6.51389555953643,6.15125742572339,5.794011500714,2.57636162336089,1.54046491784237],"text":["KEGG:04530 (169) <br> Tight junction <br> 5.885e-10","KEGG:05100 (77) <br> Bacterial invasion of epithelial cells <br> 5.351e-09","KEGG:05131 (247) <br> Shigellosis <br> 1.803e-08","KEGG:05130 (196) <br> Pathogenic Escherichia coli infection <br> 1.058e-07","KEGG:04810 (216) <br> Regulation of actin cytoskeleton <br> 2.292e-07","KEGG:04144 (251) <br> Endocytosis <br> 2.737e-07","KEGG:05135 (136) <br> Yersinia infection <br> 3.063e-07","KEGG:05132 (249) <br> Salmonella infection <br> 7.059e-07","KEGG:04666 (96) <br> Fc gamma R-mediated phagocytosis <br> 1.607e-06","KEGG:03015 (97) <br> mRNA surveillance pathway <br> 2.652e-03","KEGG:03430 (23) <br> Mismatch repair <br> 2.881e-02"],"key":["KEGG:04530","KEGG:05100","KEGG:05131","KEGG:05130","KEGG:04810","KEGG:04144","KEGG:05135","KEGG:05132","KEGG:04666","KEGG:03015","KEGG:03430"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(221,68,119,1)","opacity":0.8,"size":[16.9810383690236,15.7545592284552,17.5340397903573,17.1997344570363,17.3411881917645,17.5569598245829,16.6537786100701,17.5455506022041,16.110971780725,16.1274657222001,13.574309418746],"symbol":"circle","line":{"width":1.88976377952756,"color":"rgba(221,68,119,1)"}},"hoveron":"points","set":"SharedData0dd1ba32","name":"KEGG","legendgroup":"KEGG","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","_isNestedKey":false,"frame":null},{"x":[2,49.9528814359943],"y":[-1,-1],"text":"","type":"scatter","mode":"lines","line":{"width":11.3385826771654,"color":"rgba(220,57,18,1)","dash":"solid"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[53.8499055266887,71.2657061880019],"y":[-1,-1],"text":"","type":"scatter","mode":"lines","line":{"width":11.3385826771654,"color":"rgba(16,150,24,1)","dash":"solid"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[75.1627302786963,193.967406707605],"y":[-1,-1],"text":"","type":"scatter","mode":"lines","line":{"width":11.3385826771654,"color":"rgba(255,153,0,1)","dash":"solid"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[197.864430798299,200.003897024091],"y":[-1,-1],"text":"","type":"scatter","mode":"lines","line":{"width":11.3385826771654,"color":"rgba(221,68,119,1)","dash":"solid"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[180],"y":[16.2],"text":"values above this threshold are capped","hovertext":"","textfont":{"size":7.55905511811024,"color":"rgba(190,190,190,1)"},"type":"scatter","mode":"text","hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[0,210],"y":[16,16],"text":"","type":"scatter","mode":"lines","line":{"width":0.755905511811024,"color":"rgba(190,190,190,1)","dash":"dash"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null}],"layout":{"margin":{"t":29.2835201328352,"r":6.6417600664176,"b":51.8931100989696,"l":61.0377750103778},"plot_bgcolor":"rgba(255,255,255,1)","paper_bgcolor":"rgba(255,255,255,1)","font":{"color":"rgba(0,0,0,1)","family":"","size":14.6118721461187},"xaxis":{"domain":[0,1],"automargin":true,"type":"linear","autorange":false,"range":[0,210],"tickmode":"array","ticktext":["GO:MF","GO:CC","GO:BP","KEGG"],"tickvals":[25.9764407179972,62.5578058573453,134.565068493151,198.934163911195],"categoryorder":"array","categoryarray":["GO:MF","GO:CC","GO:BP","KEGG"],"nticks":null,"ticks":"","tickcolor":null,"ticklen":3.65296803652968,"tickwidth":0,"showticklabels":true,"tickfont":{"color":"rgba(77,77,77,1)","family":"","size":10.6268161062682},"tickangle":-45,"showline":true,"linecolor":"rgba(190,190,190,1)","linewidth":0.132835201328352,"showgrid":false,"gridcolor":null,"gridwidth":0,"zeroline":false,"anchor":"y","title":"","hoverformat":".2f"},"yaxis":{"domain":[0,1],"automargin":true,"type":"linear","autorange":false,"range":[-1,18.5],"tickmode":"array","ticktext":["0","2","4","6","8","10","12","14",">16"],"tickvals":[0,2,4,6,8,10,12,14,16],"categoryorder":"array","categoryarray":["0","2","4","6","8","10","12","14",">16"],"nticks":null,"ticks":"outside","tickcolor":"rgba(190,190,190,1)","ticklen":3.65296803652968,"tickwidth":0.66417600664176,"showticklabels":true,"tickfont":{"color":"rgba(77,77,77,1)","family":"","size":11.689497716895},"tickangle":-0,"showline":true,"linecolor":"rgba(190,190,190,1)","linewidth":0.66417600664176,"showgrid":false,"gridcolor":null,"gridwidth":0,"zeroline":false,"anchor":"x","title":"","hoverformat":".2f"},"annotations":[{"text":"-log10(p-adj)","x":-0.0108706932337069,"y":0.5,"showarrow":false,"ax":0,"ay":0,"font":{"color":"rgba(0,0,0,1)","family":"","size":13.2835201328352},"xref":"paper","yref":"paper","textangle":-90,"xanchor":"right","yanchor":"center","annotationType":"axis"},{"text":"query_1","x":0.5,"y":1,"showarrow":false,"ax":0,"ay":0,"font":{"color":"rgba(169,169,169,1)","family":"","size":13.2835201328352},"xref":"paper","yref":"paper","textangle":-0,"xanchor":"center","yanchor":"bottom"}],"shapes":[{"type":"rect","fillcolor":null,"line":{"color":null,"width":0,"linetype":[]},"yref":"paper","xref":"paper","x0":0,"x1":1,"y0":0,"y1":1},{"type":"rect","fillcolor":null,"line":{"color":null,"width":0,"linetype":[]},"yref":"paper","xref":"paper","x0":0,"x1":1,"y0":0,"y1":24.9730178497302,"yanchor":1,"ysizemode":"pixel"}],"showlegend":false,"legend":{"bgcolor":"rgba(255,255,255,1)","bordercolor":"transparent","borderwidth":1.88976377952756,"font":{"color":"rgba(0,0,0,1)","family":"","size":11.689497716895}},"hovermode":"closest","barmode":"relative","dragmode":"zoom"},"config":{"doubleClick":"reset","showSendToCloud":false},"source":"A","attrs":{"7383e006ccf":{"x":{},"y":{},"text":{},"colour":{},"size":{},"alpha":{},"type":"scatter"},"7381fdb392d":{"x":{},"xend":{},"y":{},"yend":{}},"7385d8c35e1":{"x":{},"xend":{},"y":{},"yend":{}},"7387c161186":{"x":{},"xend":{},"y":{},"yend":{}},"7384f60529b":{"x":{},"xend":{},"y":{},"yend":{}},"738476a4f45":{"x":{},"y":{}},"7385839282e":{"yintercept":{}}},"cur_data":"7383e006ccf","visdat":{"7383e006ccf":["function (y) ","x"],"7381fdb392d":["function (y) ","x"],"7385d8c35e1":["function (y) ","x"],"7387c161186":["function (y) ","x"],"7384f60529b":["function (y) ","x"],"738476a4f45":["function (y) ","x"],"7385839282e":["function (y) ","x"]},"highlight":{"on":"plotly_click","off":"plotly_doubleclick","persistent":false,"dynamic":false,"color":null,"selectize":false,"defaultValues":null,"opacityDim":0.2,"selected":{"opacity":1},"debounce":0,"ctGroups":["SharedData0dd1ba32"]},"shinyEvents":["plotly_hover","plotly_click","plotly_selected","plotly_relayout","plotly_brushed","plotly_brushing","plotly_clickannotation","plotly_doubleclick","plotly_deselect","plotly_afterplot","plotly_sunburstclick"],"base_url":"https://plot.ly"},"evals":[],"jsHooks":[]}</script>
</div>
<div id="table-results" class="section level4">
<h4>Table results</h4>
<p>The results can also be visualized with a table:</p>
<pre class="r"><code>if(is.null(GO)){
  print(&quot;NOT enough data.&quot;)
}else{
  GO$result %&gt;% 
    as.data.frame() %&gt;% 
    select(starts_with(c(&quot;source&quot;, &quot;term_name&quot;, &quot;p_value&quot;, &quot;term_size&quot;, &quot;intersection_size&quot;))) %&gt;%
    arrange(p_value) %&gt;%  
    mutate(p_value = format(p_value, scientific = T, digits = 2)) %&gt;% 
    DT::datatable(options = list(
    columnDefs = list(list(className = &#39;dt-center&#39;, targets = 1:6)),
    pageLength = 10)) %&gt;%
    formatStyle(&#39;source&#39;, backgroundColor = styleEqual(c(&quot;GO:BP&quot;, &quot;GO:MF&quot;, &quot;GO:CC&quot;, &quot;KEGG&quot;), c(&#39;orange&#39;, &#39;tomato&#39;, &#39;lightgreen&#39;, &#39;violet&#39;)))
}</code></pre>
<div id="htmlwidget-a421697d58abce2053a2" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-a421697d58abce2053a2">{"x":{"filter":"none","data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33","34","35","36","37","38","39","40","41","42","43","44","45","46","47","48","49","50","51","52","53","54","55","56","57","58","59","60","61","62","63","64","65","66","67","68","69","70","71","72","73","74","75","76","77","78","79","80","81","82","83","84","85","86","87","88","89","90","91","92","93","94","95","96","97","98","99","100","101","102","103","104","105","106","107","108","109","110","111","112","113","114","115","116","117","118","119","120","121","122","123","124","125","126","127","128","129","130","131","132","133","134","135","136","137","138","139","140","141","142","143","144","145","146","147","148","149","150","151","152","153","154","155","156","157","158","159","160","161","162","163","164","165","166","167","168","169","170","171","172","173","174","175","176","177","178","179","180","181","182","183","184","185","186","187","188","189","190","191","192","193","194","195","196","197","198","199","200","201","202","203","204","205","206","207","208","209","210","211","212","213","214","215","216","217","218","219","220","221","222","223","224","225","226","227","228","229","230","231","232","233","234","235","236","237","238","239","240","241","242","243","244","245","246","247","248","249","250","251","252","253","254","255","256","257","258","259","260","261","262","263","264","265","266","267","268","269","270","271","272","273","274","275","276","277","278","279","280","281","282","283","284","285","286","287","288","289","290","291","292","293","294","295","296","297","298","299","300","301","302","303","304","305","306","307","308","309","310","311","312","313","314","315","316","317","318","319","320","321","322","323","324","325","326","327","328","329","330","331","332","333","334","335","336","337","338","339","340","341","342","343","344","345","346","347","348","349","350","351","352","353","354","355","356","357","358","359","360","361","362","363","364","365","366","367","368","369","370","371","372","373"],["GO:CC","GO:CC","GO:CC","GO:MF","GO:CC","GO:CC","GO:CC","GO:CC","GO:MF","GO:BP","GO:BP","GO:BP","GO:CC","GO:CC","GO:CC","GO:CC","GO:CC","GO:CC","GO:CC","GO:CC","GO:CC","GO:CC","GO:BP","GO:CC","GO:BP","GO:BP","GO:MF","GO:BP","GO:BP","GO:MF","GO:MF","GO:BP","GO:BP","GO:CC","GO:MF","GO:CC","GO:MF","GO:BP","GO:CC","GO:CC","GO:BP","GO:BP","GO:BP","GO:BP","GO:BP","GO:BP","GO:CC","GO:BP","GO:CC","GO:CC","GO:CC","GO:CC","GO:BP","GO:BP","GO:BP","GO:BP","GO:CC","GO:BP","GO:BP","GO:BP","GO:BP","GO:BP","GO:CC","KEGG","GO:CC","GO:CC","GO:CC","GO:CC","GO:BP","GO:CC","GO:BP","GO:BP","GO:CC","GO:CC","GO:CC","KEGG","GO:CC","GO:CC","KEGG","GO:MF","GO:MF","GO:BP","GO:MF","GO:CC","GO:CC","GO:MF","GO:BP","GO:CC","GO:BP","GO:CC","GO:CC","GO:BP","GO:CC","GO:CC","KEGG","GO:BP","GO:CC","GO:CC","GO:CC","GO:CC","GO:CC","GO:CC","KEGG","GO:CC","GO:CC","GO:BP","KEGG","KEGG","GO:CC","GO:CC","GO:BP","GO:CC","GO:BP","GO:MF","GO:CC","GO:MF","GO:CC","GO:BP","KEGG","GO:CC","GO:BP","GO:BP","GO:CC","GO:MF","GO:MF","GO:BP","KEGG","GO:BP","GO:BP","GO:MF","GO:CC","GO:MF","GO:CC","GO:BP","GO:CC","GO:CC","GO:CC","GO:BP","GO:CC","GO:BP","GO:CC","GO:BP","GO:BP","GO:BP","GO:BP","GO:BP","GO:CC","GO:BP","GO:CC","GO:BP","GO:BP","GO:CC","GO:BP","GO:BP","GO:BP","GO:CC","GO:BP","GO:BP","GO:CC","GO:BP","GO:CC","GO:BP","GO:BP","GO:MF","GO:MF","GO:CC","GO:CC","GO:BP","GO:BP","GO:MF","GO:CC","GO:BP","GO:CC","GO:MF","GO:BP","GO:BP","GO:CC","GO:BP","GO:BP","GO:BP","GO:BP","GO:BP","GO:CC","GO:BP","GO:BP","GO:BP","GO:MF","GO:BP","GO:CC","GO:BP","GO:BP","GO:BP","GO:CC","GO:BP","GO:BP","GO:BP","GO:BP","GO:BP","GO:BP","GO:BP","GO:BP","GO:BP","GO:MF","GO:CC","GO:MF","GO:BP","GO:CC","GO:CC","GO:BP","GO:BP","GO:BP","GO:BP","GO:CC","GO:CC","GO:BP","GO:MF","GO:BP","GO:CC","GO:BP","GO:CC","GO:BP","GO:BP","GO:BP","GO:BP","GO:BP","GO:BP","GO:BP","GO:BP","GO:BP","GO:BP","GO:CC","GO:BP","GO:BP","GO:BP","GO:BP","GO:BP","GO:BP","GO:BP","GO:BP","GO:CC","GO:BP","GO:MF","GO:CC","GO:CC","GO:BP","GO:CC","GO:BP","GO:BP","GO:BP","GO:BP","GO:MF","GO:BP","GO:BP","GO:BP","GO:BP","GO:BP","GO:BP","GO:BP","GO:BP","GO:CC","GO:CC","KEGG","GO:BP","GO:BP","GO:BP","GO:BP","GO:BP","GO:BP","GO:BP","GO:MF","GO:BP","GO:MF","GO:BP","GO:BP","GO:BP","GO:BP","GO:BP","GO:CC","GO:BP","GO:BP","GO:BP","GO:CC","GO:BP","GO:BP","GO:BP","GO:BP","GO:BP","GO:BP","GO:BP","GO:CC","GO:BP","GO:BP","GO:BP","GO:BP","GO:BP","GO:BP","GO:BP","GO:MF","GO:MF","GO:BP","GO:BP","GO:CC","GO:CC","GO:CC","GO:BP","GO:BP","GO:CC","GO:BP","GO:BP","GO:BP","GO:BP","GO:BP","GO:BP","GO:BP","GO:BP","GO:BP","GO:BP","GO:BP","GO:MF","GO:BP","GO:CC","GO:BP","GO:BP","GO:MF","GO:CC","GO:BP","GO:BP","GO:BP","GO:BP","GO:BP","GO:BP","GO:BP","GO:BP","GO:BP","GO:CC","GO:MF","GO:CC","GO:BP","GO:MF","GO:CC","GO:CC","GO:CC","GO:BP","KEGG","GO:BP","GO:BP","GO:BP","GO:MF","GO:BP","GO:BP","GO:BP","GO:BP","GO:BP","GO:MF","GO:BP","GO:CC","GO:CC","GO:BP","GO:BP","GO:BP","GO:BP","GO:CC","GO:CC","GO:BP","GO:MF","GO:BP","GO:BP","GO:BP","GO:BP","GO:BP","GO:MF","GO:MF","GO:BP"],[836,2125,2129,9079,407,63,1568,513,465,7688,7681,2938,2716,1444,2130,839,549,491,1026,311,405,62,9851,870,21858,15281,1955,2935,13080,7116,435,2922,8180,3746,1692,2126,1690,8014,2761,1450,26029,3520,8013,22693,7691,15613,2330,2374,2739,4067,2715,2127,12466,26382,8179,3535,2124,9215,8652,9156,12640,15901,2146,297,479,3777,3747,1024,9169,2123,12219,15728,1259,1021,3625,402,2057,283,407,396,440,8836,30,483,512,460,21908,389,22121,1451,277,22120,485,531,406,25881,183,3387,898,560,3961,469,345,980,1025,7696,261,411,1526,495,21110,384,16061,139,1886,9806,3229,2375,408,2089,4752,8016,497,6764,11974,12371,321,25147,9501,10279,1937,1875,3858,2390,871,403,240,8015,4012,8320,3659,8266,3664,5673,257,15740,872,12211,347,15534,10011,294,15657,7692,5650,895,8842,25146,1437,14514,616,18831,19777,202,203,1148,3383,8018,4215,7161,1022,12639,386,401,8269,8586,2128,4710,7605,2396,15282,7602,334,13657,4213,12718,446,9678,4452,8653,15731,4740,3152,4786,26383,3598,15902,22988,8176,2929,2294,12213,6819,201,9067,159,3212,1329,25883,5856,14404,12711,3778,833,4969,5959,8116,536,5800,3151,2366,4583,8624,15283,2404,4775,27807,10007,9180,15724,350,514,12704,19392,7684,4741,12114,160,7698,2930,24867,9792,1438,1445,26030,597,17819,27805,2277,13665,11910,2389,22341,6477,21581,21572,21571,21570,3663,893,3134,206,19359,2354,11896,2868,9305,15763,4650,2081,16745,416,19242,11489,23704,15723,15722,373,26566,13617,25833,3608,13656,14859,12524,4309,8273,14402,22793,1106,17272,18629,12350,26384,12445,15614,16744,1988,10354,15655,20875,2558,89,317,19132,4908,1158,8622,2494,19393,10016,21335,8174,8264,8596,15340,15173,16045,6750,8322,3049,17565,9863,6595,1566,24088,26936,26938,4651,21484,6020,30348,14854,6680,1149,1884,2918,16030,10406,2911,184,103,14858,220,27335,15558,4709,437,2965,8267,2340,11470,15336,439,17047,1874,4086,30346,9299,18625,4211,1320,3493,2490,5402,15656,29342,14395,3065,13129,397,4874,23211],["actin cytoskeleton","non-membrane-bounded organelle","intracellular non-membrane-bounded organelle","actin filament binding","mitochondrial small ribosomal subunit","organellar small ribosomal subunit","protein-containing complex","Arp2/3 protein complex","actin binding","actin cytoskeleton organization","actin filament-based process","actin filament organization","intracellular organelle lumen","membrane-enclosed lumen","organelle lumen","small ribosomal subunit","focal adhesion","cytoskeleton","cell-substrate junction","nucleoplasm","mitochondrial ribosome","organellar ribosome","Arp2/3 complex-mediated actin nucleation","myosin complex","supramolecular fiber organization","regulation of mRNA processing","cytoskeletal protein binding","cytoskeleton organization","actin nucleation","protein-containing complex binding","RNA binding","organelle organization","mRNA 3'-end processing","supramolecular complex","structural constituent of cytoskeleton","intracellular organelle","structural molecule activity","regulation of actin filament polymerization","anchoring junction","nuclear lumen","regulation of supramolecular fiber organization","regulation of actin polymerization or depolymerization","regulation of actin filament length","regulation of actin filament organization","actin filament polymerization","regulation of cellular component organization","ribosomal subunit","mRNA polyadenylation","extracellular exosome","extracellular vesicle","extracellular membrane-bounded organelle","extracellular organelle","RNA polyadenylation","regulation of mRNA metabolic process","RNA 3'-end processing","actin polymerization or depolymerization","membrane-bounded organelle","regulation of organelle organization","regulation of protein polymerization","regulation of actin cytoskeleton organization","regulation of cellular component biogenesis","regulation of cytoskeleton organization","contractile fiber","Tight junction","ribosome","supramolecular fiber","supramolecular polymer","lamellipodium","regulation of actin filament-based process","organelle","regulation of protein-containing complex assembly","protein polymerization","cell leading edge","myofibril","mitochondrial protein-containing complex","Bacterial invasion of epithelial cells","actomyosin","intracellular anatomical structure","Shigellosis","nucleic acid binding","mRNA binding","regulation of cellular component size","microfilament motor activity","mRNA cleavage and polyadenylation specificity factor complex","actin filament","cytoskeletal motor activity","lamellipodium organization","mitochondrial inner membrane","pre-mRNA cleavage required for polyadenylation","vesicle","extracellular space","mRNA cleavage involved in mRNA processing","mRNA cleavage factor complex","brush border","Pathogenic Escherichia coli infection","regulation of lamellipodium organization","stress fiber","contractile actin filament bundle","nuclear body","cell cortex","nuclear protein-containing complex","cytosol","Regulation of actin cytoskeleton","organelle inner membrane","cell junction","actin filament-based movement","Endocytosis","Yersinia infection","actin filament bundle","muscle thin filament tropomyosin","regulation of anatomical structure size","cytoplasm","actin filament capping","RNA polymerase II complex binding","site of double-strand break","RNA polymerase binding","site of DNA damage","mRNA cleavage","Salmonella infection","cell projection","positive regulation of macromolecule metabolic process","negative regulation of actin filament depolymerization","striated muscle thin filament","RNA polymerase core enzyme binding","heterocyclic compound binding","regulation of RNA splicing","Fc gamma R-mediated phagocytosis","negative regulation of protein depolymerization","histone H2B ubiquitination","organic cyclic compound binding","myofilament","protein binding","plasma membrane bounded cell projection","mRNA processing","myosin II complex","mitochondrial matrix","extracellular region","regulation of actin filament depolymerization","catalytic complex","regulation of mRNA 3'-end processing","cluster of actin-based cell projections","positive regulation of cellular metabolic process","macromolecule biosynthetic process","mRNA metabolic process","telomere maintenance","chromosome organization","unconventional myosin complex","negative regulation of protein-containing complex disassembly","chromosome","barbed-end actin filament capping","cellular macromolecule biosynthetic process","nucleus","positive regulation of nitrogen compound metabolic process","actin filament depolymerization","cellular component organization","Cdc73/Paf1 complex","mitochondrial translation","regulation of protein depolymerization","mitochondrial membrane","regulation of mRNA splicing, via spliceosome","F-actin capping protein complex","actin polymerization-dependent cell motility","cellular component organization or biogenesis","basal transcription machinery binding","basal RNA polymerase II transcription machinery binding","cortical cytoskeleton","myosin II filament","negative regulation of actin filament polymerization","positive regulation of metabolic process","cadherin binding","sarcomere","cellular component biogenesis","mitochondrial envelope","chromatin binding","positive regulation of cellular biosynthetic process","telomere organization","intracellular membrane-bounded organelle","positive regulation of macromolecule biosynthetic process","cellular component assembly","RNA localization","positive regulation of mRNA processing","regulation of cell morphogenesis","spliceosomal complex","positive regulation of nucleobase-containing compound metabolic process","positive regulation of biosynthetic process","cellular nitrogen compound biosynthetic process","structural constituent of ribosome","response to cytokine","ribonucleoprotein complex","negative regulation of protein polymerization","protein depolymerization","regulation of lamellipodium assembly","box H/ACA scaRNP complex","negative regulation of organelle organization","negative regulation of mRNA metabolic process","asymmetric cell division","negative regulation of cytoskeleton organization","mitochondrial gene expression","snRNA pseudouridine synthesis","telomere maintenance via telomerase","RNA-dependent DNA biosynthetic process","regulation of protein-containing complex disassembly","ADP binding","podosome","cell adhesion molecule binding","alternative mRNA splicing, via spliceosome","box H/ACA telomerase RNP complex","box H/ACA snoRNP complex","positive regulation of lamellipodium organization","histone ubiquitination","positive regulation of RNA export from nucleus","cellular macromolecule metabolic process","polymeric cytoskeletal fiber","Cajal body","telomere maintenance via telomere lengthening","box H/ACA snoRNA binding","actomyosin structure organization","cell-cell junction","meiotic chromosome movement towards spindle pole","box H/ACA RNP complex","transcription elongation from RNA polymerase II promoter","histone monoubiquitination","positive regulation of nucleobase-containing compound transport","negative regulation of mRNA processing","translation","positive regulation of gene expression","positive regulation of telomerase RNA localization to Cajal body","cellular nitrogen compound metabolic process","protein-containing complex disassembly","positive regulation of RNA metabolic process","telomerase holoenzyme complex","endodermal cell fate commitment","cellular biosynthetic process","cellular response to cytokine stimulus","lamellipodium assembly","positive regulation of lamellipodium assembly","peptide biosynthetic process","regulation of alternative mRNA splicing, via spliceosome","vesicle transport along actin filament","catalytic step 2 spliceosome","organic substance biosynthetic process","telomerase RNA binding","organelle envelope","envelope","negative regulation of supramolecular fiber organization","transcription elongation factor complex","muscle structure development","regulation of telomerase RNA localization to Cajal body","DNA metabolic process","positive regulation of transcription by RNA polymerase II","ATP-dependent activity","RNA processing","actin filament-based transport","regulation of nucleobase-containing compound metabolic process","RNA localization to nucleus","telomerase RNA localization","telomerase RNA localization to Cajal body","RNA localization to Cajal body","biosynthetic process","RNA polymerase II, holoenzyme","PTW/PP1 phosphatase complex","mRNA surveillance pathway","cellular response to organic substance","DNA-templated transcription, elongation","muscle cell differentiation","movement of cell or subcellular component","meiotic cytokinesis","chromosome movement towards spindle pole","gene expression","structural constituent of muscle","striated muscle cell development","single-stranded DNA binding","DNA conformation change","snRNA modification","regulation of mRNA polyadenylation","negative regulation of RNA metabolic process","regulation of RNA metabolic process","perichromatin fibrils","positive regulation of nucleic acid-templated transcription","positive regulation of transcription, DNA-templated","positive regulation of RNA biosynthetic process","chromosomal region","negative regulation of nucleobase-containing compound metabolic process","negative regulation of cellular process","protein-containing complex organization","response to organic substance","negative regulation of protein-containing complex assembly","regulation of RNA export from nucleus","regulation of plasma membrane bounded cell projection assembly","actin cap","regulation of cell projection assembly","regulation of biological quality","regulation of ATP-dependent activity","positive regulation of mRNA metabolic process","amide biosynthetic process","negative regulation of cellular component organization","muscle cell development","transcription factor binding","G-rich strand telomeric DNA binding","regulation of nitrogen compound metabolic process","regulation of primary metabolic process","nuclear DNA-directed RNA polymerase complex","DNA-directed RNA polymerase complex","DNA replication factor A complex","cellular response to chemical stimulus","regulation of cell morphogenesis involved in differentiation","RNA polymerase complex","regulation of nucleobase-containing compound transport","peptide metabolic process","cellular response to interferon-gamma","nucleobase-containing compound biosynthetic process","nucleic acid metabolic process","rRNA pseudouridine synthesis","regulation of cellular metabolic process","regulation of telomere maintenance via telomerase","regulation of cellular process","cell motility","localization of cell","single-stranded telomeric DNA binding","positive regulation of mRNA 3'-end processing","WASH complex","cell fate commitment involved in formation of primary germ layer","response to interferon-gamma","telomeric DNA binding","myosin filament","protein processing in phagocytic vesicle","regulation of protein processing in phagocytic vesicle","positive regulation of protein processing in phagocytic vesicle","regulation of gene expression","RNA phosphodiester bond hydrolysis","heterocycle biosynthetic process","positive regulation of histone H2B ubiquitination","positive regulation of biological process","aromatic compound biosynthetic process","cortical actin cytoskeleton","ATP binding","U4/U6 snRNP","establishment of organelle localization","sequence-specific single stranded DNA binding","CRD-mediated mRNA stability complex","ruffle","chromosome, telomeric region","positive regulation of cellular process","Mismatch repair","regulation of telomere maintenance via telomere lengthening","regulation of DNA metabolic process","regulation of macromolecule biosynthetic process","double-stranded RNA binding","cell cycle","regulation of cellular biosynthetic process","chromatin assembly or disassembly","locomotion","regulation of biological process","single-stranded RNA binding","regulation of macromolecule metabolic process","ribonucleoprotein granule","ficolin-1-rich granule lumen","regulation of histone H2B ubiquitination","positive regulation of histone ubiquitination","protein-containing complex assembly","regulation of biosynthetic process","cytoplasmic vesicle","intracellular vesicle","protein monoubiquitination","adenyl ribonucleotide binding","negative regulation of nitrogen compound metabolic process","regulation of cellular macromolecule biosynthetic process","positive regulation of nucleocytoplasmic transport","establishment or maintenance of cell polarity","innate immune response","DNA binding","adenyl nucleotide binding","non-membrane-bounded organelle assembly"],["9.8e-33","3.9e-26","3.9e-26","1.8e-24","4.1e-23","4.1e-23","1.7e-22","5.9e-22","7.9e-22","1.4e-19","3.1e-19","1.4e-18","1.1e-16","1.1e-16","1.1e-16","1.5e-16","2.3e-16","2.9e-16","3.2e-16","4.3e-16","8.2e-16","8.2e-16","1.4e-15","1.4e-15","4.1e-15","4.7e-15","1.7e-14","2.2e-14","2.7e-14","1.9e-13","7.0e-13","8.9e-13","1.3e-12","1.7e-12","2.4e-12","2.5e-12","3.2e-12","5.2e-12","5.6e-12","6.1e-12","1.1e-11","1.7e-11","1.8e-11","2.5e-11","2.8e-11","2.9e-11","2.9e-11","3.2e-11","3.3e-11","3.9e-11","3.9e-11","3.9e-11","4.0e-11","5.1e-11","9.2e-11","1.2e-10","1.4e-10","1.6e-10","1.7e-10","3.8e-10","3.9e-10","4.6e-10","4.7e-10","5.9e-10","6.2e-10","7.4e-10","9.0e-10","9.2e-10","1.1e-09","1.2e-09","2.6e-09","2.7e-09","2.9e-09","3.4e-09","3.8e-09","5.4e-09","1.3e-08","1.6e-08","1.8e-08","2.0e-08","2.0e-08","2.4e-08","2.4e-08","2.8e-08","3.0e-08","3.0e-08","4.1e-08","4.9e-08","5.3e-08","5.3e-08","6.5e-08","8.2e-08","9.5e-08","9.5e-08","1.1e-07","1.2e-07","1.3e-07","1.3e-07","1.5e-07","1.5e-07","1.9e-07","2.0e-07","2.3e-07","2.4e-07","2.5e-07","2.6e-07","2.7e-07","3.1e-07","3.4e-07","4.0e-07","4.0e-07","4.1e-07","4.2e-07","5.0e-07","5.4e-07","5.4e-07","5.7e-07","6.3e-07","7.1e-07","7.8e-07","8.7e-07","8.8e-07","9.2e-07","1.3e-06","1.3e-06","1.4e-06","1.6e-06","1.9e-06","2.2e-06","2.4e-06","2.8e-06","3.1e-06","3.3e-06","3.4e-06","3.4e-06","3.4e-06","3.6e-06","3.7e-06","4.0e-06","4.2e-06","4.2e-06","4.5e-06","4.6e-06","4.6e-06","4.9e-06","5.1e-06","5.4e-06","5.7e-06","5.8e-06","6.0e-06","6.4e-06","6.7e-06","6.9e-06","7.3e-06","8.0e-06","8.0e-06","8.9e-06","1.0e-05","1.1e-05","1.2e-05","1.3e-05","1.4e-05","1.4e-05","1.4e-05","1.4e-05","1.5e-05","1.5e-05","1.5e-05","1.6e-05","1.7e-05","1.9e-05","2.1e-05","2.6e-05","2.6e-05","2.7e-05","2.8e-05","2.9e-05","3.0e-05","3.3e-05","3.3e-05","3.5e-05","3.6e-05","3.6e-05","3.8e-05","4.0e-05","4.6e-05","4.8e-05","5.5e-05","7.0e-05","7.7e-05","9.7e-05","9.8e-05","9.9e-05","1.1e-04","1.1e-04","1.2e-04","1.3e-04","1.3e-04","1.4e-04","1.4e-04","1.5e-04","1.6e-04","1.6e-04","2.0e-04","2.2e-04","2.4e-04","2.5e-04","2.5e-04","2.7e-04","2.7e-04","3.0e-04","3.7e-04","3.8e-04","3.8e-04","4.5e-04","4.6e-04","4.6e-04","4.7e-04","4.7e-04","4.9e-04","5.5e-04","5.8e-04","5.9e-04","6.4e-04","7.1e-04","7.1e-04","7.5e-04","7.6e-04","7.7e-04","8.2e-04","8.5e-04","8.6e-04","8.9e-04","1.0e-03","1.0e-03","1.1e-03","1.1e-03","1.2e-03","1.3e-03","1.3e-03","1.4e-03","1.4e-03","1.5e-03","1.5e-03","1.6e-03","1.6e-03","1.6e-03","1.7e-03","1.7e-03","1.7e-03","1.9e-03","2.0e-03","2.0e-03","2.0e-03","2.1e-03","2.1e-03","2.1e-03","2.1e-03","2.3e-03","2.6e-03","2.6e-03","2.7e-03","2.7e-03","2.8e-03","3.0e-03","3.0e-03","3.3e-03","3.3e-03","3.4e-03","3.8e-03","4.0e-03","4.3e-03","4.5e-03","4.7e-03","4.7e-03","5.0e-03","5.1e-03","5.1e-03","5.2e-03","5.2e-03","5.3e-03","5.3e-03","5.3e-03","5.9e-03","6.0e-03","6.0e-03","6.2e-03","6.4e-03","7.9e-03","8.1e-03","8.3e-03","8.4e-03","8.5e-03","8.6e-03","9.0e-03","9.0e-03","9.2e-03","9.2e-03","9.3e-03","9.4e-03","1.0e-02","1.0e-02","1.1e-02","1.2e-02","1.3e-02","1.3e-02","1.3e-02","1.3e-02","1.3e-02","1.4e-02","1.5e-02","1.5e-02","1.6e-02","1.6e-02","1.6e-02","1.6e-02","1.7e-02","1.7e-02","1.7e-02","1.7e-02","1.8e-02","2.1e-02","2.2e-02","2.2e-02","2.2e-02","2.3e-02","2.3e-02","2.3e-02","2.3e-02","2.3e-02","2.3e-02","2.4e-02","2.4e-02","2.5e-02","2.6e-02","2.6e-02","2.8e-02","2.8e-02","2.8e-02","2.8e-02","2.8e-02","2.8e-02","2.9e-02","2.9e-02","3.0e-02","3.0e-02","3.0e-02","3.1e-02","3.2e-02","3.2e-02","3.4e-02","3.5e-02","3.6e-02","3.6e-02","3.6e-02","3.8e-02","3.9e-02","4.0e-02","4.0e-02","4.2e-02","4.2e-02","4.3e-02","4.3e-02","4.4e-02","4.5e-02","4.5e-02","4.6e-02","4.6e-02","4.7e-02","4.8e-02","4.8e-02","4.9e-02","4.9e-02"],[501,6362,6362,218,28,28,7228,11,449,711,800,444,6569,6570,6570,76,420,2334,428,4161,86,86,39,57,755,141,995,1435,55,1275,3142,3805,73,1330,104,15870,980,140,839,5442,373,159,160,267,168,2246,189,52,2110,2132,2133,2133,53,318,126,198,15557,1109,205,350,940,523,242,169,520,988,996,203,391,16674,402,278,421,232,277,77,79,17758,247,5447,350,355,31,24,114,115,91,500,13,3942,3353,14,28,105,196,53,71,71,869,319,2671,5373,216,562,2076,132,251,136,79,4,488,12193,44,36,73,58,97,20,249,2312,3378,48,22,41,7416,151,96,77,12,7497,26,14799,2205,1919,20,483,4303,57,1424,28,161,3176,4880,2153,137,1034,8,87,1795,29,1564,10053,2978,62,7793,8,79,93,745,105,8,7,7994,57,57,110,3,68,3678,334,211,4801,796,580,1942,164,14514,1839,4556,198,32,305,195,1897,1975,4734,449,922,2377,83,120,41,4,352,93,19,164,111,3,71,72,127,34,30,550,79,5,5,37,48,5,5746,757,102,84,5,193,495,3,6,61,29,6,30,713,1117,15,8767,230,1729,23,15,5794,828,73,29,738,62,18,85,5869,22,1245,1245,166,56,616,18,896,1185,550,3356,20,3945,19,19,19,19,5956,91,7,97,2385,80,357,2074,7,7,8909,43,64,123,329,7,18,1395,3700,4,1604,1604,1606,359,1496,4818,3387,3009,147,12,190,4,192,3741,72,116,867,680,174,602,10,5529,5694,116,117,13,3008,96,121,15,895,132,3953,7476,10,5921,54,11067,1645,1645,12,10,12,32,153,36,22,2,2,2,5388,155,4024,4,6094,4035,80,1502,3,409,14,6,181,169,5561,23,63,356,3926,74,1713,4081,212,1821,12142,84,6710,260,124,5,5,3194,4141,2447,2449,68,1565,2330,656,61,222,892,2494,1576,373],[41,101,101,27,13,13,94,8,32,17,37,15,85,85,85,13,18,42,18,72,13,13,8,12,15,17,36,17,8,31,60,64,13,37,8,133,26,9,12,76,11,9,9,10,9,28,13,11,17,17,17,17,11,19,14,9,119,14,9,10,18,11,16,9,17,29,29,9,10,133,13,9,16,15,13,7,10,135,9,73,17,9,7,7,11,10,7,15,5,18,17,5,7,9,8,6,9,9,23,13,41,20,8,15,12,10,9,7,9,4,9,76,8,7,5,8,9,5,8,16,48,8,6,7,84,11,6,9,5,84,6,61,18,34,5,13,17,8,26,6,9,45,55,36,10,24,4,9,29,5,30,87,43,8,22,4,8,9,15,9,3,3,22,7,7,5,3,8,48,14,11,29,15,18,31,10,108,30,28,10,6,11,10,32,31,52,12,14,31,8,9,4,3,14,8,3,10,8,3,7,7,9,5,4,13,7,3,3,4,6,3,60,18,7,7,3,10,5,2,3,6,5,3,5,16,22,4,74,7,28,4,4,56,12,4,3,16,6,4,6,56,4,16,16,9,5,16,4,18,21,12,41,4,43,4,4,4,4,56,6,3,9,18,6,12,24,2,2,70,5,6,7,11,3,3,25,26,2,24,24,24,11,26,55,21,9,8,3,5,2,5,14,6,7,16,10,3,13,3,55,56,6,6,3,21,4,6,3,16,3,41,67,3,57,4,48,8,8,3,3,2,4,3,4,3,2,2,2,31,5,41,2,61,41,5,18,2,11,3,2,6,7,57,4,4,10,46,4,25,26,8,8,50,5,35,8,4,2,2,19,26,11,11,5,18,32,14,5,7,5,33,18,11]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>source<\/th>\n      <th>source_order<\/th>\n      <th>term_name<\/th>\n      <th>p_value<\/th>\n      <th>term_size<\/th>\n      <th>intersection_size<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-center","targets":[1,2,3,4,5,6]},{"orderable":false,"targets":0}],"pageLength":10,"order":[],"autoWidth":false,"orderClasses":false,"rowCallback":"function(row, data, displayNum, displayIndex, dataIndex) {\nvar value=data[1]; $(this.api().cell(row, 1).node()).css({'background-color':value == \"GO:BP\" ? \"orange\" : value == \"GO:MF\" ? \"tomato\" : value == \"GO:CC\" ? \"lightgreen\" : value == \"KEGG\" ? \"violet\" : null});\n}"}},"evals":["options.rowCallback"],"jsHooks":[]}</script>
</div>
<div id="treemap" class="section level4 tabset tabset-fade">
<h4>Treemap</h4>
<div id="bp" class="section level5">
<h5>BP</h5>
<pre class="r"><code>if(sum(GO$result$source == &quot;GO:BP&quot;) &gt;= 5 &amp; !is.null(GO)){
  simMatrix &lt;- GO$result %&gt;% 
  as.data.frame() %&gt;%
  filter(source == &quot;GO:BP&quot;) %&gt;% 
  pull(term_id) %&gt;% 
  calculateSimMatrix(orgdb=&quot;org.Hs.eg.db&quot;, ont=&quot;BP&quot;, method=&quot;Rel&quot;)

scores &lt;- setNames(-log10(GO$result$p_value), GO$result$term_id)

reducedTerms &lt;- reduceSimMatrix(simMatrix,
                                scores,
                                threshold=0.7,
                                orgdb=&quot;org.Hs.eg.db&quot;)

treemapPlot(reducedTerms)
BP &lt;- T
}else{
  print(&quot;NOT enough data.&quot;)
  BP &lt;- F
}</code></pre>
<p><img src="figure/Template_LFQ.Rmd/BP-cond1-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="mf" class="section level5">
<h5>MF</h5>
<pre class="r"><code>if(sum(GO$result$source == &quot;GO:MF&quot;) &gt;= 5 &amp; !is.null(GO)){
simMatrix &lt;- GO$result %&gt;% 
  as.data.frame() %&gt;%
  filter(source == &quot;GO:MF&quot;) %&gt;% 
  pull(term_id) %&gt;% 
  calculateSimMatrix(orgdb=&quot;org.Hs.eg.db&quot;, ont=&quot;MF&quot;, method=&quot;Rel&quot;)

scores &lt;- setNames(-log10(GO$result$p_value), GO$result$term_id)

reducedTerms &lt;- reduceSimMatrix(simMatrix,
                                scores,
                                threshold=0.7,
                                orgdb=&quot;org.Hs.eg.db&quot;)

treemapPlot(reducedTerms)
MF &lt;- T
}else{
  print(&quot;NOT enough data.&quot;)
  MF &lt;- F
}</code></pre>
<p><img src="figure/Template_LFQ.Rmd/MF-cond1-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="cc" class="section level5">
<h5>CC</h5>
<pre class="r"><code>if(sum(GO$result$source == &quot;GO:CC&quot;) &gt;= 5 &amp; !is.null(GO)){
simMatrix &lt;- GO$result %&gt;% 
  as.data.frame() %&gt;%
  filter(source == &quot;GO:CC&quot;) %&gt;% 
  pull(term_id) %&gt;% 
  calculateSimMatrix(orgdb=&quot;org.Hs.eg.db&quot;, ont=&quot;CC&quot;, method=&quot;Rel&quot;)

scores &lt;- setNames(-log10(GO$result$p_value), GO$result$term_id)

reducedTerms &lt;- reduceSimMatrix(simMatrix,
                                scores,
                                threshold=0.7,
                                orgdb=&quot;org.Hs.eg.db&quot;)

treemapPlot(reducedTerms)
CC &lt;- T
}else{
  print(&quot;NOT enough data.&quot;)
  CC &lt;- F
}</code></pre>
<p><img src="figure/Template_LFQ.Rmd/CC-cond1-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="scatterplot" class="section level4">
<h4>ScatterPlot</h4>
<pre class="r"><code>if(BP | MF | CC){
scatterPlot(simMatrix, reducedTerms, size = &quot;score&quot;, labelSize = 5)
}else{
  print(&quot;NOT enough data.&quot;)
}</code></pre>
<p><img src="figure/Template_LFQ.Rmd/scatter-cond1-1.png" width="1440" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="wt_bio-1" class="section level3 tabset tabset-fade">
<h3><strong>5.2. WT_BIO</strong></h3>
<div id="gostplot-1" class="section level4">
<h4>Gostplot</h4>
<p>UP regulated proteins in WT_BIO were used to generate these gene ontologies.</p>
<pre class="r"><code>Gname &lt;- results %&gt;% filter(significant &amp; get(paste0(test, &quot;_ratio&quot;)) &lt; 0) %&gt;% pull(name)

if(!params$Imputation){
  Uni &lt;- unique_pair(data_filt_unique, conditions = conditions, table = T) %&gt;% 
  filter(!!sym(paste0(&quot;unique_&quot;, cond_2)) == &quot;+&quot;) %&gt;% 
  pull(name) %&gt;% unique()

  Gname &lt;- c(Gname, Uni)
}

if(length(Gname) == 0){
  print(&quot;NOT enough data.&quot;)
  GO &lt;- NULL
}else{
  GO &lt;- gost(query = Gname, organism = &quot;hsapiens&quot;, sources = c(&quot;GO:BP&quot;, &quot;GO:MF&quot;, &quot;GO:CC&quot;, &quot;KEGG&quot;), ordered_query = T, evcodes = TRUE)
  if(is.null(GO)){
    print(&quot;NOT enough data.&quot;)
  }else{
    gostplot(GO, capped = TRUE, interactive = TRUE)
  }
}</code></pre>
<div id="htmlwidget-454929a56e939b13f92b" style="width:960px;height:768px;" class="plotly html-widget"></div>
<script type="application/json" data-for="htmlwidget-454929a56e939b13f92b">{"x":{"data":[{"x":[55.1595986829558,65.2707416810414,65.2668437847429,55.1479049940605,65.2434564069525,63.384159872609,63.3802619763106,57.3580121952612,69.4843675796267,54.4345899714508,55.6741209943464],"y":[4.68178558174179,3.98270035353399,3.11459457869228,3.08657779913768,2.83667375224298,2.83407910026355,2.52862439311546,2.48627077597791,2.12403473496373,1.68133532451407,1.63961532185031],"text":["GO:0005684 (93) <br> U2-type spliceosomal complex <br> 2.081e-05","GO:0071014 (13) <br> post-mRNA release spliceosomal complex <br> 1.041e-04","GO:0071013 (85) <br> catalytic step 2 spliceosome <br> 7.681e-04","GO:0005681 (195) <br> spliceosomal complex <br> 8.193e-04","GO:0071007 (30) <br> U2-type catalytic step 2 spliceosome <br> 1.457e-03","GO:0045240 (10) <br> dihydrolipoyl dehydrogenase complex <br> 1.465e-03","GO:0045239 (14) <br> tricarboxylic acid cycle enzyme complex <br> 2.961e-03","GO:0016607 (414) <br> nuclear speck <br> 3.264e-03","GO:1902494 (1424) <br> catalytic complex <br> 7.516e-03","GO:0000974 (12) <br> Prp19 complex <br> 2.083e-02","GO:0005829 (5373) <br> cytosol <br> 2.293e-02"],"key":["GO:0005684","GO:0071014","GO:0071013","GO:0005681","GO:0071007","GO:0045240","GO:0045239","GO:0016607","GO:1902494","GO:0000974","GO:0005829"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(16,150,24,1)","opacity":0.8,"size":[17.5074972017388,14.1064766145913,17.3706006684143,18.5863375962292,15.6713538108912,13.5640691028775,14.2546003500892,19.6081227065093,21.155038613015,13.9440663201669,22.6771653543307],"symbol":"circle","line":{"width":1.88976377952756,"color":"rgba(16,150,24,1)"}},"hoveron":"points","set":"SharedData9b6b2f20","name":"GO:CC","legendgroup":"GO:CC","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","_isNestedKey":false,"frame":null},{"x":[16.018734925656,17.043735561032,6.12728392723651,6.11948924559867,5.75703654943909,5.76093389025801,19.584801774968,16.5721573219427],"y":[3.48791003065907,3.20529383560814,2.30233067298589,2.15647549388583,1.79598145668636,1.79598145668636,1.49495146102237,1.31886020196734],"text":["GO:0016620 (38) <br> oxidoreductase activity, acting on the aldehyde or oxo group of donors, NAD or NADP as acceptor <br> 3.252e-04","GO:0016903 (47) <br> oxidoreductase activity, acting on the aldehyde or oxo group of donors <br> 6.233e-04","GO:0004459 (6) <br> L-lactate dehydrogenase activity <br> 4.985e-03","GO:0004457 (7) <br> lactate dehydrogenase activity <br> 6.975e-03","GO:0004349 (1) <br> glutamate 5-kinase activity <br> 1.600e-02","GO:0004350 (1) <br> glutamate-5-semialdehyde dehydrogenase activity <br> 1.600e-02","GO:0019202 (2) <br> amino acid kinase activity <br> 3.199e-02","GO:0016774 (3) <br> phosphotransferase activity, carboxyl group as acceptor <br> 4.799e-02"],"key":["GO:0016620","GO:0016903","GO:0004459","GO:0004457","GO:0004349","GO:0004350","GO:0019202","GO:0016774"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(220,57,18,1)","opacity":0.8,"size":[16.0776632806394,16.4318813336882,12.4107536193322,12.7743797009708,3.77952755905512,3.77952755905512,9.14793373990032,10.5380982941344],"symbol":"circle","line":{"width":1.88976377952756,"color":"rgba(220,57,18,1)"}},"hoveron":"points","set":"SharedData9b6b2f20","name":"GO:MF","legendgroup":"GO:MF","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","_isNestedKey":false,"frame":null},{"x":[198.68429924249,198.282173291292,197.868334933748,198.243131936806],"y":[4.66214658381085,4.16169947519863,2.85888594482239,1.87073434272434],"text":["KEGG:03040 (146) <br> Spliceosome <br> 2.177e-05","KEGG:00640 (32) <br> Propanoate metabolism <br> 6.891e-05","KEGG:00010 (67) <br> Glycolysis / Gluconeogenesis <br> 1.384e-03","KEGG:00620 (47) <br> Pyruvate metabolism <br> 1.347e-02"],"key":["KEGG:03040","KEGG:00640","KEGG:00010","KEGG:00620"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(221,68,119,1)","opacity":0.8,"size":[18.1742904059275,15.783648710255,17.0016070396028,16.4318813336882],"symbol":"circle","line":{"width":1.88976377952756,"color":"rgba(221,68,119,1)"}},"hoveron":"points","set":"SharedData9b6b2f20","name":"KEGG","legendgroup":"KEGG","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","_isNestedKey":false,"frame":null},{"x":[2,49.9528814359943],"y":[-1,-1],"text":"","type":"scatter","mode":"lines","line":{"width":11.3385826771654,"color":"rgba(220,57,18,1)","dash":"solid"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[53.8499055266887,71.2657061880019],"y":[-1,-1],"text":"","type":"scatter","mode":"lines","line":{"width":11.3385826771654,"color":"rgba(16,150,24,1)","dash":"solid"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[75.1627302786963,193.967406707605],"y":[-1,-1],"text":"","type":"scatter","mode":"lines","line":{"width":11.3385826771654,"color":"rgba(255,153,0,1)","dash":"solid"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[197.864430798299,200.003897024091],"y":[-1,-1],"text":"","type":"scatter","mode":"lines","line":{"width":11.3385826771654,"color":"rgba(221,68,119,1)","dash":"solid"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[180],"y":[16.2],"text":"values above this threshold are capped","hovertext":"","textfont":{"size":7.55905511811024,"color":"rgba(190,190,190,1)"},"type":"scatter","mode":"text","hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[0,210],"y":[16,16],"text":"","type":"scatter","mode":"lines","line":{"width":0.755905511811024,"color":"rgba(190,190,190,1)","dash":"dash"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null}],"layout":{"margin":{"t":29.2835201328352,"r":6.6417600664176,"b":51.8931100989696,"l":61.0377750103778},"plot_bgcolor":"rgba(255,255,255,1)","paper_bgcolor":"rgba(255,255,255,1)","font":{"color":"rgba(0,0,0,1)","family":"","size":14.6118721461187},"xaxis":{"domain":[0,1],"automargin":true,"type":"linear","autorange":false,"range":[0,210],"tickmode":"array","ticktext":["GO:MF","GO:CC","GO:BP","KEGG"],"tickvals":[25.9764407179972,62.5578058573453,134.565068493151,198.934163911195],"categoryorder":"array","categoryarray":["GO:MF","GO:CC","GO:BP","KEGG"],"nticks":null,"ticks":"","tickcolor":null,"ticklen":3.65296803652968,"tickwidth":0,"showticklabels":true,"tickfont":{"color":"rgba(77,77,77,1)","family":"","size":10.6268161062682},"tickangle":-45,"showline":true,"linecolor":"rgba(190,190,190,1)","linewidth":0.132835201328352,"showgrid":false,"gridcolor":null,"gridwidth":0,"zeroline":false,"anchor":"y","title":"","hoverformat":".2f"},"yaxis":{"domain":[0,1],"automargin":true,"type":"linear","autorange":false,"range":[-1,18.5],"tickmode":"array","ticktext":["0","2","4","6","8","10","12","14",">16"],"tickvals":[0,2,4,6,8,10,12,14,16],"categoryorder":"array","categoryarray":["0","2","4","6","8","10","12","14",">16"],"nticks":null,"ticks":"outside","tickcolor":"rgba(190,190,190,1)","ticklen":3.65296803652968,"tickwidth":0.66417600664176,"showticklabels":true,"tickfont":{"color":"rgba(77,77,77,1)","family":"","size":11.689497716895},"tickangle":-0,"showline":true,"linecolor":"rgba(190,190,190,1)","linewidth":0.66417600664176,"showgrid":false,"gridcolor":null,"gridwidth":0,"zeroline":false,"anchor":"x","title":"","hoverformat":".2f"},"annotations":[{"text":"-log10(p-adj)","x":-0.0108706932337069,"y":0.5,"showarrow":false,"ax":0,"ay":0,"font":{"color":"rgba(0,0,0,1)","family":"","size":13.2835201328352},"xref":"paper","yref":"paper","textangle":-90,"xanchor":"right","yanchor":"center","annotationType":"axis"},{"text":"query_1","x":0.5,"y":1,"showarrow":false,"ax":0,"ay":0,"font":{"color":"rgba(169,169,169,1)","family":"","size":13.2835201328352},"xref":"paper","yref":"paper","textangle":-0,"xanchor":"center","yanchor":"bottom"}],"shapes":[{"type":"rect","fillcolor":null,"line":{"color":null,"width":0,"linetype":[]},"yref":"paper","xref":"paper","x0":0,"x1":1,"y0":0,"y1":1},{"type":"rect","fillcolor":null,"line":{"color":null,"width":0,"linetype":[]},"yref":"paper","xref":"paper","x0":0,"x1":1,"y0":0,"y1":24.9730178497302,"yanchor":1,"ysizemode":"pixel"}],"showlegend":false,"legend":{"bgcolor":"rgba(255,255,255,1)","bordercolor":"transparent","borderwidth":1.88976377952756,"font":{"color":"rgba(0,0,0,1)","family":"","size":11.689497716895}},"hovermode":"closest","barmode":"relative","dragmode":"zoom"},"config":{"doubleClick":"reset","showSendToCloud":false},"source":"A","attrs":{"73847e26474":{"x":{},"y":{},"text":{},"colour":{},"size":{},"alpha":{},"type":"scatter"},"7387dae54d6":{"x":{},"xend":{},"y":{},"yend":{}},"73860cb73ba":{"x":{},"xend":{},"y":{},"yend":{}},"73869d1929":{"x":{},"xend":{},"y":{},"yend":{}},"7387aa1142":{"x":{},"xend":{},"y":{},"yend":{}},"7386cbe4ba0":{"x":{},"y":{}},"73833942ac0":{"yintercept":{}}},"cur_data":"73847e26474","visdat":{"73847e26474":["function (y) ","x"],"7387dae54d6":["function (y) ","x"],"73860cb73ba":["function (y) ","x"],"73869d1929":["function (y) ","x"],"7387aa1142":["function (y) ","x"],"7386cbe4ba0":["function (y) ","x"],"73833942ac0":["function (y) ","x"]},"highlight":{"on":"plotly_click","off":"plotly_doubleclick","persistent":false,"dynamic":false,"color":null,"selectize":false,"defaultValues":null,"opacityDim":0.2,"selected":{"opacity":1},"debounce":0,"ctGroups":["SharedData9b6b2f20"]},"shinyEvents":["plotly_hover","plotly_click","plotly_selected","plotly_relayout","plotly_brushed","plotly_brushing","plotly_clickannotation","plotly_doubleclick","plotly_deselect","plotly_afterplot","plotly_sunburstclick"],"base_url":"https://plot.ly"},"evals":[],"jsHooks":[]}</script>
</div>
<div id="table-results-1" class="section level4">
<h4>Table results</h4>
<p>The results can also be visualized with a table:</p>
<pre class="r"><code>if(is.null(GO)){
  print(&quot;NOT enough data.&quot;)
}else{
  GO$result %&gt;% 
    as.data.frame() %&gt;% 
    select(starts_with(c(&quot;source&quot;, &quot;term_name&quot;, &quot;p_value&quot;, &quot;term_size&quot;, &quot;intersection_size&quot;))) %&gt;% 
    arrange(p_value) %&gt;%  
    mutate(p_value = format(p_value, scientific = T, digits = 2)) %&gt;% 
    DT::datatable(options = list(
    columnDefs = list(list(className = &#39;dt-center&#39;, targets = 1:6)),
    pageLength = 10)) %&gt;%
    formatStyle(&#39;source&#39;, backgroundColor = styleEqual(c(&quot;GO:BP&quot;, &quot;GO:MF&quot;, &quot;GO:CC&quot;, &quot;KEGG&quot;), c(&#39;orange&#39;, &#39;tomato&#39;, &#39;lightgreen&#39;, &#39;violet&#39;)))
}</code></pre>
<div id="htmlwidget-7856457827ab1876f7c5" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-7856457827ab1876f7c5">{"x":{"filter":"none","data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23"],["GO:CC","KEGG","KEGG","GO:CC","GO:MF","GO:MF","GO:CC","GO:CC","KEGG","GO:CC","GO:CC","GO:CC","GO:CC","GO:MF","GO:MF","GO:CC","KEGG","GO:MF","GO:MF","GO:CC","GO:CC","GO:MF","GO:MF"],[337,211,108,2931,3598,3861,2930,334,2,2924,2447,2446,901,1060,1058,4012,98,965,966,151,469,4513,3740],["U2-type spliceosomal complex","Spliceosome","Propanoate metabolism","post-mRNA release spliceosomal complex","oxidoreductase activity, acting on the aldehyde or oxo group of donors, NAD or NADP as acceptor","oxidoreductase activity, acting on the aldehyde or oxo group of donors","catalytic step 2 spliceosome","spliceosomal complex","Glycolysis / Gluconeogenesis","U2-type catalytic step 2 spliceosome","dihydrolipoyl dehydrogenase complex","tricarboxylic acid cycle enzyme complex","nuclear speck","L-lactate dehydrogenase activity","lactate dehydrogenase activity","catalytic complex","Pyruvate metabolism","glutamate 5-kinase activity","glutamate-5-semialdehyde dehydrogenase activity","Prp19 complex","cytosol","amino acid kinase activity","phosphotransferase activity, carboxyl group as acceptor"],["2.1e-05","2.2e-05","6.9e-05","1.0e-04","3.3e-04","6.2e-04","7.7e-04","8.2e-04","1.4e-03","1.5e-03","1.5e-03","3.0e-03","3.3e-03","5.0e-03","7.0e-03","7.5e-03","1.3e-02","1.6e-02","1.6e-02","2.1e-02","2.3e-02","3.2e-02","4.8e-02"],[93,146,32,13,38,47,85,195,67,30,10,14,414,6,7,1424,47,1,1,12,5373,2,3],[5,7,4,3,3,3,4,5,4,3,2,2,6,2,2,9,3,1,1,2,17,1,1]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>source<\/th>\n      <th>source_order<\/th>\n      <th>term_name<\/th>\n      <th>p_value<\/th>\n      <th>term_size<\/th>\n      <th>intersection_size<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-center","targets":[1,2,3,4,5,6]},{"orderable":false,"targets":0}],"pageLength":10,"order":[],"autoWidth":false,"orderClasses":false,"rowCallback":"function(row, data, displayNum, displayIndex, dataIndex) {\nvar value=data[1]; $(this.api().cell(row, 1).node()).css({'background-color':value == \"GO:BP\" ? \"orange\" : value == \"GO:MF\" ? \"tomato\" : value == \"GO:CC\" ? \"lightgreen\" : value == \"KEGG\" ? \"violet\" : null});\n}"}},"evals":["options.rowCallback"],"jsHooks":[]}</script>
</div>
<div id="treemap-1" class="section level4 tabset tabset-fade">
<h4>Treemap</h4>
<div id="bp-1" class="section level5">
<h5>BP</h5>
<pre class="r"><code>if(sum(GO$result$source == &quot;GO:BP&quot;) &gt;= 5 &amp; !is.null(GO)){
  simMatrix &lt;- GO$result %&gt;% 
  as.data.frame() %&gt;%
  filter(source == &quot;GO:BP&quot;) %&gt;% 
  pull(term_id) %&gt;% 
  calculateSimMatrix(orgdb=&quot;org.Hs.eg.db&quot;, ont=&quot;BP&quot;, method=&quot;Rel&quot;)

scores &lt;- setNames(-log10(GO$result$p_value), GO$result$term_id)

reducedTerms &lt;- reduceSimMatrix(simMatrix,
                                scores,
                                threshold=0.7,
                                orgdb=&quot;org.Hs.eg.db&quot;)

treemapPlot(reducedTerms)
BP &lt;- T
}else{
  print(&quot;NOT enough data.&quot;)
  BP &lt;- F
}</code></pre>
<pre><code>[1] &quot;NOT enough data.&quot;</code></pre>
</div>
<div id="mf-1" class="section level5">
<h5>MF</h5>
<pre class="r"><code>if(sum(GO$result$source == &quot;GO:MF&quot;) &gt;= 5 &amp; !is.null(GO)){
simMatrix &lt;- GO$result %&gt;% 
  as.data.frame() %&gt;%
  filter(source == &quot;GO:MF&quot;) %&gt;% 
  pull(term_id) %&gt;% 
  calculateSimMatrix(orgdb=&quot;org.Hs.eg.db&quot;, ont=&quot;MF&quot;, method=&quot;Rel&quot;)

scores &lt;- setNames(-log10(GO$result$p_value), GO$result$term_id)

reducedTerms &lt;- reduceSimMatrix(simMatrix,
                                scores,
                                threshold=0.7,
                                orgdb=&quot;org.Hs.eg.db&quot;)

treemapPlot(reducedTerms)
MF &lt;- T
}else{
  print(&quot;NOT enough data.&quot;)
  MF &lt;- F
}</code></pre>
<p><img src="figure/Template_LFQ.Rmd/MF-cond2-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="cc-1" class="section level5">
<h5>CC</h5>
<pre class="r"><code>if(sum(GO$result$source == &quot;GO:CC&quot;) &gt;= 5 &amp; !is.null(GO)){
simMatrix &lt;- GO$result %&gt;% 
  as.data.frame() %&gt;%
  filter(source == &quot;GO:CC&quot;) %&gt;% 
  pull(term_id) %&gt;% 
  calculateSimMatrix(orgdb=&quot;org.Hs.eg.db&quot;, ont=&quot;CC&quot;, method=&quot;Rel&quot;)

scores &lt;- setNames(-log10(GO$result$p_value), GO$result$term_id)

reducedTerms &lt;- reduceSimMatrix(simMatrix,
                                scores,
                                threshold=0.7,
                                orgdb=&quot;org.Hs.eg.db&quot;)

treemapPlot(reducedTerms)
CC &lt;- T
}else{
  print(&quot;NOT enough data.&quot;)
  CC &lt;- F
}</code></pre>
<p><img src="figure/Template_LFQ.Rmd/CC-cond2-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="scatterplot-1" class="section level4">
<h4>ScatterPlot</h4>
<pre class="r"><code>if(BP | MF | CC){
scatterPlot(simMatrix, reducedTerms, size = &quot;score&quot;, labelSize = 5)
}else{
  print(&quot;NOT enough data.&quot;)
}</code></pre>
<p><img src="figure/Template_LFQ.Rmd/scatter-cond2-1.png" width="1440" style="display: block; margin: auto;" /></p>
</div>
</div>
</div>
<div id="network-analysis" class="section level2">
<h2><strong>6. Network analysis</strong></h2>
<p>Overview from <strong>STRING Database</strong>: In STRING, each protein-protein interaction is annotated with one or more ‘scores’. Importantly, these scores do <strong>not</strong> indicate the strength or the specificity of the interaction. Instead, they are indicators of <strong>confidence</strong>, i.e. how likely STRING judges an interaction to be true, given the available evidence. All scores rank from 0 to 1, with 1 being the highest possible confidence. A score of 0.5 would indicate that roughly every second interaction might be erroneous (i.e., a false positive).</p>
<pre class="r"><code># if(leng_dep != 0){
#   
#   string_db &lt;- STRINGdb$new(version=&quot;11&quot;, species=9606, score_threshold=400)
# 
#   string_mapped &lt;- string_db$map(string_res, &quot;Gene.name&quot;, removeUnmappedRows = TRUE, quiet = T)
# 
#   hits &lt;- string_mapped$STRING_id[1:leng_dep]
# 
#   string_db$plot_network(hits, add_link = T)
# }else{
#   print(&quot;NOT enough data.&quot;)
# }</code></pre>
<div id="cytoscape-network" class="section level3">
<h3><strong>6.1. Cytoscape Network</strong></h3>
<p>Cytoscape is an open source software platform for visualizing complex networks and integrating these with any type of attribute data. In this presentation all proteins showing at least one confidence level from the STRING database are shown. The <strong>color</strong> code is relative to <strong>fold change</strong> and the <strong>size</strong> is directly proportional to the <strong>significance (p-value)</strong>.</p>
<pre class="r"><code>print(&quot;Enable Cytoscape option to see this results.&quot;)</code></pre>
<pre><code>[1] &quot;Enable Cytoscape option to see this results.&quot;</code></pre>
<pre class="r"><code>if(n_edge != 0){
  knitr::include_graphics(&quot;assets/network_FC_pvalue.png&quot;,  error = F)
}else{
  print(&quot;NOT enough data.&quot;)
}</code></pre>
</div>
<div id="cytoscape-enrichmentmap" class="section level3">
<h3><strong>6.2. Cytoscape EnrichmentMap</strong></h3>
<p>An <strong>enrichment map</strong> is a different sort of network. Instead of nodes representing genes, nodes represent pathways or functions. Edges between these pathways or functions represent shared genes or pathway crosstalk. An enrichment map is a way to visualize your enrichment results to help reduce redundancy and uncover main themes.</p>
<pre class="r"><code>print(&quot;Enable Cytoscape option to see this results.&quot;)</code></pre>
<pre><code>[1] &quot;Enable Cytoscape option to see this results.&quot;</code></pre>
<pre class="r"><code>Gname &lt;- results %&gt;% filter(significant) %&gt;% pull(name)

if(!params$Imputation){
  
  Uni &lt;- unique_pair(data_filt_unique, conditions = conditions, table = T) %&gt;% 
  pull(name) %&gt;% 
  unique()

  Gname &lt;- c(Gname, Uni)
}

if(length(Gname) == 0){
  
  print(&quot;NOT enough data.&quot;)
  
}else{
  
  GO &lt;- gost(query = Gname, organism = &quot;hsapiens&quot;, sources = c(&quot;GO:BP&quot;, &quot;GO:MF&quot;, &quot;GO:CC&quot;, &quot;KEGG&quot;), ordered_query = T, evcodes = TRUE)
  
  if(is.null(GO)){
    
    print(&quot;NOT enough data.&quot;)

    enrich_map = F
    
  }else{
    
    enrich_map = T
    
    gem &lt;- GO$result[,c(&quot;term_id&quot;, &quot;term_name&quot;, &quot;p_value&quot;, &quot;intersection&quot;)]
    colnames(gem) &lt;- c(&quot;GO.ID&quot;, &quot;Description&quot;, &quot;p.Val&quot;, &quot;Genes&quot;)
    gem$FDR &lt;- gem$p.Val
    gem$Phenotype = &quot;+1&quot;
    gem &lt;- gem[,c(&quot;GO.ID&quot;, &quot;Description&quot;, &quot;p.Val&quot;, &quot;FDR&quot;, &quot;Phenotype&quot;, &quot;Genes&quot;)]
    head(gem)
  
    write.table(gem, here::here(&quot;output/enrichmentmap.txt&quot;),col.name=TRUE,sep=&quot;\t&quot;,row.names=FALSE,quote=FALSE)
    
  }
}</code></pre>
<pre class="r"><code>if(enrich_map){
  knitr::include_graphics(&quot;assets/network_enrichmentmap.png&quot;,  error = F)
}else{
  print(&quot;NOT enough data.&quot;)
}</code></pre>
</div>
</div>
<div id="section" class="section level2 unnumbered">
<h2></h2>
<pre class="r"><code>saveWorkbook(excel, here::here(paste0(&quot;output/results_table_&quot;, test, &quot;.xlsx&quot;)), overwrite = T)</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.0.3 (2020-10-10)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 19042)

Matrix products: default

locale:
[1] LC_COLLATE=Italian_Italy.1252  LC_CTYPE=Italian_Italy.1252   
[3] LC_MONETARY=Italian_Italy.1252 LC_NUMERIC=C                  
[5] LC_TIME=Italian_Italy.1252    

attached base packages:
[1] parallel  stats4    stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] forcats_0.5.1               stringr_1.4.0              
 [3] dplyr_1.0.7                 purrr_0.3.4                
 [5] readr_1.4.0                 tidyr_1.1.3                
 [7] tibble_3.1.2                tidyverse_1.3.1            
 [9] openxlsx_4.2.4              here_1.0.1                 
[11] RColorBrewer_1.1-2          RCy3_2.10.2                
[13] STRINGdb_2.2.2              naniar_0.6.1               
[15] visdat_0.5.3                ggrepel_0.9.1              
[17] ggplot2_3.3.5               UpSetR_1.4.0               
[19] ggstatsplot_0.8.0           datapasta_3.1.0            
[21] patchwork_1.1.1             DT_0.18                    
[23] rrvgo_1.2.0                 org.Hs.eg.db_3.12.0        
[25] AnnotationDbi_1.52.0        gprofiler2_0.2.0           
[27] enrichR_3.0                 preprocessCore_1.52.1      
[29] SummarizedExperiment_1.20.0 Biobase_2.50.0             
[31] GenomicRanges_1.42.0        GenomeInfoDb_1.26.2        
[33] IRanges_2.24.1              S4Vectors_0.28.1           
[35] BiocGenerics_0.36.0         MatrixGenerics_1.2.1       
[37] matrixStats_0.59.0          DEP_1.12.0                 
[39] MBQN_2.2.0                  workflowr_1.6.2            

loaded via a namespace (and not attached):
  [1] rappdirs_0.3.3            R.methodsS3_1.8.1        
  [3] coda_0.19-4               bit64_4.0.5              
  [5] knitr_1.33                DelayedArray_0.16.2      
  [7] R.utils_2.10.1            data.table_1.14.0        
  [9] RCurl_1.98-1.3            doParallel_1.0.16        
 [11] generics_0.1.0            RSQLite_2.2.7            
 [13] correlation_0.6.1         chron_2.3-56             
 [15] bit_4.0.4                 lubridate_1.7.10         
 [17] xml2_1.3.2                httpuv_1.6.1             
 [19] assertthat_0.2.1          WRS2_1.1-2               
 [21] xfun_0.24                 hms_1.1.0                
 [23] jquerylib_0.1.4           evaluate_0.14            
 [25] promises_1.2.0.1          fansi_0.5.0              
 [27] readxl_1.3.1              caTools_1.18.2           
 [29] dbplyr_2.1.1              igraph_1.2.6             
 [31] DBI_1.1.1                 htmlwidgets_1.5.3        
 [33] reshape_0.8.8             kSamples_1.2-9           
 [35] Rmpfr_0.8-4               paletteer_1.3.0          
 [37] hash_2.2.6.1              ellipsis_0.3.2           
 [39] crosstalk_1.1.1           backports_1.2.1          
 [41] insight_0.14.2            gridBase_0.4-7           
 [43] vctrs_0.3.8               imputeLCMD_2.0           
 [45] Cairo_1.5-12.2            cachem_1.0.5             
 [47] withr_2.4.2               treemap_2.4-2            
 [49] cluster_2.1.0             lazyeval_0.2.2           
 [51] crayon_1.4.1              labeling_0.4.2           
 [53] pkgconfig_2.0.3           SuppDists_1.1-9.5        
 [55] slam_0.1-48               statsExpressions_1.1.0   
 [57] ProtGenerics_1.22.0       wordcloud_2.6            
 [59] rlang_0.4.11              RJSONIO_1.3-1.4          
 [61] lifecycle_1.0.0           MatrixModels_0.5-0       
 [63] sandwich_3.0-1            affyio_1.60.0            
 [65] BiocFileCache_1.14.0      modelr_0.1.8             
 [67] cellranger_1.1.0          rprojroot_2.0.2          
 [69] graph_1.68.0              Matrix_1.2-18            
 [71] mc2d_0.1-21               zoo_1.8-9                
 [73] reprex_2.0.0              whisker_0.4              
 [75] GlobalOptions_0.1.2       pheatmap_1.0.12          
 [77] png_0.1-7                 viridisLite_0.4.0        
 [79] rjson_0.2.20              mzR_2.24.1               
 [81] PMCMRplus_1.9.0           parameters_0.14.0        
 [83] bitops_1.0-7              shinydashboard_0.7.1     
 [85] R.oo_1.24.0               KernSmooth_2.23-17       
 [87] blob_1.2.1                shape_1.4.6              
 [89] multcompView_0.1-8        tmvtnorm_1.4-10          
 [91] ggsignif_0.6.2            scales_1.1.1             
 [93] memoise_2.0.0             magrittr_2.0.1           
 [95] plyr_1.8.6                gplots_3.1.1             
 [97] zlibbioc_1.36.0           compiler_4.0.3           
 [99] plotrix_3.8-1             pcaMethods_1.82.0        
[101] clue_0.3-59               cli_3.0.0                
[103] affy_1.68.0               XVector_0.30.0           
[105] pbapply_1.4-3             MASS_7.3-53              
[107] tidyselect_1.1.1          vsn_3.58.0               
[109] stringi_1.6.2             highr_0.9                
[111] yaml_2.2.1                GOSemSim_2.16.1          
[113] norm_1.0-9.5              MALDIquant_1.19.3        
[115] grid_4.0.3                sass_0.4.0               
[117] tools_4.0.3               rstudioapi_0.13          
[119] circlize_0.4.13           foreach_1.5.1            
[121] git2r_0.28.0              gridExtra_2.3            
[123] ipmisc_6.0.2              pairwiseComparisons_3.1.6
[125] farver_2.1.0              mzID_1.28.0              
[127] digest_0.6.27             BiocManager_1.30.16      
[129] shiny_1.6.0               proto_1.0.0              
[131] BWStest_0.2.2             Rcpp_1.0.6               
[133] broom_0.7.8               BayesFactor_0.9.12-4.2   
[135] performance_0.7.2         later_1.2.0              
[137] ncdf4_1.17                httr_1.4.2               
[139] MSnbase_2.15.7            ComplexHeatmap_2.6.2     
[141] effectsize_0.4.5          colorspace_2.0-2         
[143] rvest_1.0.0               XML_3.99-0.6             
[145] fs_1.5.0                  rematch2_2.1.2           
[147] gmm_1.6-6                 plotly_4.9.4.1           
[149] xtable_1.8-4              gmp_0.6-2                
[151] jsonlite_1.7.2            zeallot_0.1.0            
[153] NLP_0.2-1                 R6_2.5.0                 
[155] tm_0.7-8                  gsubfn_0.7               
[157] pillar_1.6.1              htmltools_0.5.1.1        
[159] mime_0.11                 glue_1.4.2               
[161] fastmap_1.1.0             BiocParallel_1.24.1      
[163] codetools_0.2-16          mvtnorm_1.1-2            
[165] utf8_1.2.1                lattice_0.20-41          
[167] bslib_0.2.5.1             sqldf_0.4-11             
[169] curl_4.3.2                gtools_3.9.2             
[171] zip_2.2.0                 GO.db_3.12.1             
[173] limma_3.46.0              rmarkdown_2.9            
[175] munsell_0.5.0             GetoptLong_1.0.5         
[177] GenomeInfoDbData_1.2.4    iterators_1.0.13         
[179] impute_1.64.0             haven_2.4.1              
[181] gtable_0.3.0              bayestestR_0.10.0        </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
