<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Fabio Bedin | MS-Unit at IEO" />


<title>LFQ_Analysis</title>

<script src="site_libs/jquery-3.5.1/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/pagedtable-1.1/css/pagedtable.css" rel="stylesheet" />
<script src="site_libs/pagedtable-1.1/js/pagedtable.js"></script>
<script src="site_libs/htmlwidgets-1.5.3/htmlwidgets.js"></script>
<link href="site_libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="site_libs/datatables-binding-0.18/datatables.js"></script>
<link href="site_libs/dt-core-1.10.20/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="site_libs/dt-core-1.10.20/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="site_libs/dt-core-1.10.20/js/jquery.dataTables.min.js"></script>
<link href="site_libs/crosstalk-1.1.1/css/crosstalk.css" rel="stylesheet" />
<script src="site_libs/crosstalk-1.1.1/js/crosstalk.min.js"></script>
<script src="site_libs/plotly-binding-4.9.4.1/plotly.js"></script>
<script src="site_libs/typedarray-0.1/typedarray.min.js"></script>
<link href="site_libs/plotly-htmlwidgets-css-1.57.1/plotly-htmlwidgets.css" rel="stylesheet" />
<script src="site_libs/plotly-main-1.57.1/plotly-latest.min.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Template_LFQ</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">LFQ_Analysis</h1>
<h4 class="author">Fabio Bedin | MS-Unit at IEO</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-02-15
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>GN_MR_ZC3H4/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20220215code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20220215)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20220215code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20220215)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrong24ebe3a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> 24ebe3a </a>
</p>
</div>
<div id="strongRepositoryversionstrong24ebe3a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version 24ebe3a. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/

Untracked files:
    Untracked:  code/custom_functions.R
    Untracked:  data/expdesign.tsv
    Untracked:  data/proteinGroups.txt
    Untracked:  output/results_table_TURBO._vs_WT..xlsx
    Untracked:  output/results_table_TURBO_BIO_vs_TURBO..xlsx
    Untracked:  output/results_table_TURBO_BIO_vs_WT_BIO.xlsx
    Untracked:  output/results_table_WT_BIO_vs_WT..xlsx

Unstaged changes:
    Modified:   analysis/_site.yml
    Deleted:    analysis/about.Rmd
    Deleted:    analysis/license.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/Template_LFQ.Rmd</code>) and HTML (<code>docs/Template_LFQ.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
24ebe3a
</td>
<td>
FabioBedin
</td>
<td>
2022-02-15
</td>
<td>
first push
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<style type="text/css">
p{
  font-size: 16px;
}
</style>
<pre class="r"><code>## Proteomics
library(&quot;MBQN&quot;)
library(&quot;DEP&quot;)
library(&quot;SummarizedExperiment&quot;)
library(&quot;preprocessCore&quot;)
library(&quot;enrichR&quot;)
library(&quot;gprofiler2&quot;)
library(&quot;org.Hs.eg.db&quot;)
library(&quot;rrvgo&quot;)

## Plots
library(&quot;DT&quot;)
library(&quot;patchwork&quot;)
library(&quot;datapasta&quot;)
library(&quot;ggstatsplot&quot;)
library(&quot;UpSetR&quot;)
library(&quot;ggrepel&quot;)
library(&quot;visdat&quot;)
library(&quot;naniar&quot;)

## Networks
library(&quot;STRINGdb&quot;)
library(&quot;RCy3&quot;)
library(&quot;RColorBrewer&quot;)

## Other
library(&quot;here&quot;)
library(&quot;openxlsx&quot;)
library(&quot;tidyverse&quot;)</code></pre>
<pre class="r"><code>source(here::here(&quot;code/custom_functions.R&quot;))</code></pre>
<pre class="r"><code>data &lt;- read.csv(here::here(paste0(&quot;data/&quot;, params$proteinGroups)), header = TRUE,stringsAsFactors = FALSE, sep = &quot;\t&quot;)
## qui metterei anche Razor...unique.peptides al posto di Peptides, perchè così non conto i peptidi che sono attribuiti a più di 1 PG
data &lt;- data[data$Reverse != &quot;+&quot; &amp; data$Potential.contaminant != &quot;+&quot; &amp; data$Only.identified.by.site != &quot;+&quot; &amp; data$Peptides &gt;= params$n_peptides,]

data_unique &lt;- make_unique(data, &quot;Gene.names&quot;, &quot;Protein.IDs&quot;, delim = &quot;;&quot;)</code></pre>
<pre class="r"><code>cond_1 &lt;- params$test_1

cond_2 &lt;- params$test_2

conditions&lt;-c(cond_1,cond_2)

test&lt;- paste(cond_1,cond_2,sep=&quot;_vs_&quot;)

expdesign &lt;- read.table(here::here(paste0(&quot;data/&quot;, params$expdesign)), header = T, stringsAsFactors = F)

expdesign &lt;- subset(expdesign, condition %in% conditions)

columns&lt;-match(paste(&quot;LFQ.intensity.&quot;,expdesign$label,sep=&quot;&quot;),colnames(data_unique))

data_se &lt;- make_se(data_unique, columns, expdesign)

## define automaticaly the best statistical threshold to filter data besed on replicates:
if(max(expdesign$replicate)&lt;=3){
       threshold&lt;-0
     } else if(max(expdesign$replicate)&lt;6){
       threshold&lt;-1
     } else if (max(expdesign$replicate)&gt;=6){
       threshold&lt;-trunc(max(expdesign$replicate)*0.25) ## 0.25 serve per avere sempre il 75% di valid vales
     }

data_filt &lt;- filter_missval(data_se, thr = threshold)</code></pre>
<div id="turbo_bio-vs-wt_bio" class="section level1">
<h1><strong>TURBO_BIO</strong> vs <strong>WT_BIO</strong></h1>
<hr />
</div>
<div id="introduction" class="section level1">
<h1><strong>Introduction</strong></h1>
<p>First, we need to wrangling the original dataset. From the <strong>proteinGroups</strong> obtained by <a href="https://www.maxquant.org/">MaxQuant</a> software, I remove proteins that are marked form the software as potential contaminant, only identify by site and reverse.<br />
We che also filter the initial datasets by excluding all proteins identified by less than 1 peptides.<br />
Now, we generate some quality control plots to investigate the structure of the dataset and observe the effect of filters.</p>
<div id="normalization-strategy" class="section level4">
<h4><strong>Normalization strategy:</strong></h4>
<pre class="r"><code>if(params$Normalization == &quot;VSN&quot;){
  data_filt &lt;- normalize_vsn(data_filt)
  print(&quot;VSN normalization is used&quot;)
} else if(params$Normalization == &quot;MBQN&quot;){
  data_filt &lt;- normalize_MBQN(data_filt)
  print(&quot;MBQN normalization is used&quot;)
} else {
  print(&quot;Max-LFQ normalization is used&quot;)
}</code></pre>
<p>[1] “Max-LFQ normalization is used”</p>
</div>
<div id="quality-control-plots" class="section level2 tabset tabset-fade">
<h2><strong>1. Quality control plots</strong></h2>
<div id="proteins-per-samples" class="section level3 tabset">
<h3>Proteins per samples</h3>
<p>This plot show the number of proteins identify in each samples <em>after and before User define</em> filters:</p>
<div id="filtered" class="section level4">
<h4>Filtered</h4>
<pre class="r"><code>plot_numbers_lables(data_filt)</code></pre>
<p><img src="figure/Template_LFQ.Rmd/proteins-filt-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="not-filtered" class="section level4">
<h4>Not-Filtered</h4>
<pre class="r"><code>plot_numbers_lables(data_se)</code></pre>
<p><img src="figure/Template_LFQ.Rmd/proteins-NOfilt-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="distributions" class="section level3 tabset">
<h3>Distributions</h3>
<p>This plot is useful for checking the distributions of the samples <em>after and before User define</em> filters:</p>
<div id="filtered-1" class="section level4">
<h4>Filtered</h4>
<pre class="r"><code>plot_normalization(data_filt)</code></pre>
<p><img src="figure/Template_LFQ.Rmd/normalization-filt-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="not-filtered-1" class="section level4">
<h4>Not-Filtered</h4>
<pre class="r"><code>plot_normalization(data_se)</code></pre>
<p><img src="figure/Template_LFQ.Rmd/normalization-NOfilt-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="missing-data" class="section level3 tabset">
<h3>Missing data</h3>
<p>This plot we can explore the missing data pattern <em>after and before User define</em> filters. Notice that the percentages of missingness are provided in the data. These are accurate to 1 decimal place. Also the dataset is arranged by columns with most missingness.</p>
<div id="filtered-2" class="section level4">
<h4>Filtered</h4>
<pre class="r"><code>assay(data_filt) %&gt;% 
  as.data.frame() %&gt;% 
  vis_miss(., sort_miss = TRUE)</code></pre>
<p><img src="figure/Template_LFQ.Rmd/missing-data-filt-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="not-filtered-2" class="section level4">
<h4>Not-Filtered</h4>
<pre class="r"><code>assay(data_se) %&gt;% 
  as.data.frame() %&gt;% 
  vis_miss(., sort_miss = TRUE)</code></pre>
<p><img src="figure/Template_LFQ.Rmd/missing-data-NOfilt-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="sample-cvs" class="section level3 tabset">
<h3>Sample CVs</h3>
<p>The coefficient of variation (CV) is a statistical measure of the dispersion of data points in a data series around the mean. The coefficient of variation represents the ratio of the standard deviation to the mean, and it is a useful statistic for comparing the degree of variation from one data series to another, even if the means are drastically different from one another.</p>
<div id="filtered-3" class="section level4">
<h4>Filtered</h4>
<pre class="r"><code>plot_cvs(data_filt)</code></pre>
<p><img src="figure/Template_LFQ.Rmd/CVs-filt-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="not-filtered-3" class="section level4">
<h4>Not-Filtered</h4>
<pre class="r"><code>plot_cvs(data_se)</code></pre>
<p><img src="figure/Template_LFQ.Rmd/CVs-NOfilt-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
</div>
</div>
<div id="upset-plot" class="section level2 tabset tabset-fade">
<h2><strong>2. Upset plot</strong></h2>
<p>With this plot we can identify the numper of prteins in common within the 2 conditions or in common between all condition.</p>
<div id="turbo_bio" class="section level3">
<h3>TURBO_BIO</h3>
<pre class="r"><code>define_set &lt;- assay(data_filt) %&gt;%  as.data.frame() %&gt;% select(starts_with(cond_1)) %&gt;% colnames()
n_sets &lt;- length(define_set)

assay(data_filt) %&gt;% 
  as.data.frame() %&gt;% 
  select(starts_with(cond_1)) %&gt;% 
  rownames_to_column() %&gt;% 
  pivot_longer(!rowname, names_to = &quot;samples&quot;, values_to = &quot;intensity&quot;) %&gt;% 
  mutate(intensity = if_else(is.na(intensity), 0, 1)) %&gt;% 
  pivot_wider(names_from = samples, values_from = intensity) %&gt;% 
  as.data.frame() %&gt;% 
  upset(nsets = n_sets,
        sets = define_set,
        order.by = &quot;freq&quot;,
        keep.order = T,
        text.scale = 2.5,
        point.size = 4,
        line.size = 0.5, 
        sets.bar.color = &quot;coral2&quot;,
        main.bar.color  = &quot;gray44&quot;)</code></pre>
<p><img src="figure/Template_LFQ.Rmd/upset-cond1-1.png" width="1440" style="display: block; margin: auto;" /></p>
</div>
<div id="wt_bio" class="section level3">
<h3>WT_BIO</h3>
<pre class="r"><code>define_set &lt;- assay(data_filt) %&gt;%  as.data.frame() %&gt;% select(starts_with(cond_2)) %&gt;% colnames()
n_sets &lt;- length(define_set)

assay(data_filt) %&gt;% 
  as.data.frame() %&gt;% 
  select(starts_with(cond_2)) %&gt;% 
  rownames_to_column() %&gt;% 
  pivot_longer(!rowname, names_to = &quot;samples&quot;, values_to = &quot;intensity&quot;) %&gt;% 
  mutate(intensity = if_else(is.na(intensity), 0, 1)) %&gt;% 
  pivot_wider(names_from = samples, values_from = intensity) %&gt;% 
  as.data.frame() %&gt;% 
  upset(nsets = n_sets,
        sets = define_set,
        order.by = &quot;freq&quot;,
        keep.order = T,
        text.scale = 2.5,
        point.size = 4,
        line.size = 0.5, 
        sets.bar.color = &quot;turquoise3&quot;,
        main.bar.color  = &quot;gray44&quot;)</code></pre>
<p><img src="figure/Template_LFQ.Rmd/upset-cond2-1.png" width="1440" style="display: block; margin: auto;" /></p>
</div>
<div id="both-conditions" class="section level3">
<h3>Both conditions</h3>
<pre class="r"><code>define_set &lt;- assay(data_filt) %&gt;% colnames()
n_sets &lt;- length(define_set)
n_each &lt;- n_sets / 2

assay(data_filt) %&gt;% 
  as.data.frame() %&gt;% 
  rownames_to_column() %&gt;% 
  pivot_longer(!rowname, names_to = &quot;samples&quot;, values_to = &quot;intensity&quot;) %&gt;% 
  mutate(intensity = if_else(is.na(intensity), 0, 1)) %&gt;% 
  pivot_wider(names_from = samples, values_from = intensity) %&gt;% 
  as.data.frame() %&gt;% 
  upset(nsets = n_sets,
        sets = define_set,
        order.by = &quot;freq&quot;,
        keep.order = T,
        text.scale = 2.5,
        point.size = 4,
        line.size = 0.5, 
        sets.bar.color = rep(c(&quot;red3&quot;, &quot;royalblue1&quot;), each = n_each),
        main.bar.color  = &quot;gray44&quot;)</code></pre>
<p><img src="figure/Template_LFQ.Rmd/upset-both-1.png" width="1440" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="imputation" class="section level2">
<h2><strong>3. Imputation</strong></h2>
<p>An idealized version of a label-free discovery mass spectrometry proteomics experiment would provide absolute abundance measurements for a whole proteome, across varying conditions. Unfortunately, this ideal is not realized. Measurements are made on peptides requiring an inferential step to obtain protein level estimates. The inference is complicated by experimental factors that necessitate relative abundance estimation and result in widespread non-ignorable missing data. Relative abundance on the log scale takes the form of parameter contrasts. In a complete-case analysis, contrast estimates may be biased by missing data and a substantial amount of useful information will often go unused.</p>
<pre class="r"><code>data_imputed &lt;- mix_imputation_mean(data_filt, cond1 = cond_1, cond2 = cond_2)

plot_imputation(data_filt, data_imputed)</code></pre>
<p><img src="figure/Template_LFQ.Rmd/imputation-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>data_filt &lt;- data_imputed</code></pre>
</div>
<div id="differential-enrichment-analysis" class="section level2">
<h2><strong>4. Differential enrichment analysis</strong></h2>
<p>Protein-wise linear models combined with empirical Bayes statistics are used for the differential enrichment analysis (or differential expression analysis) and False discovery rate (FDR) is ajusted with <em>Benjamini-Hochberg</em> procedure. Significant proteins are are define with <strong>FDR = 0.05</strong> and <strong>Fold change = 1</strong></p>
<pre class="r"><code>data_diff &lt;- test_diff_BH(data_filt, type = &quot;manual&quot;, test = test)

dep &lt;- add_rejections(data_diff, alpha = params$FDR, lfc = params$Fold_change)

results&lt;-get_results(dep)</code></pre>
<p>There are <strong>137</strong> significant proteins in <strong>TURBO_BIO</strong> vs <strong>WT_BIO</strong> comparison.</p>
<p>The results from the previous analysis can be easily visualized by a number of functions. These visualizations assist in the determination of the optimal cutoffs to be used, highlight the most interesting samples and contrasts, and pinpoint differentially enriched/expressed proteins.</p>
<div id="visualization-of-the-results" class="section level3 tabset tabset-fade">
<h3><strong>4.1. Visualization of the results</strong></h3>
<div id="pca" class="section level4">
<h4>PCA</h4>
<p>The PCA plot can be used to get a high-level overview of the data. This can be very useful to observe batch effects, such as clear differences between replicates.</p>
<pre class="r"><code>if(params$Imputation){
  if(nrow(dep) &gt;= 500){
    n_value &lt;- 500
  }else {
    n_value &lt;- nrow(dep)
  }
  plot_pca(dep, x = 1, y = 2, n = n_value, point_size = 4, indicate = &quot;condition&quot;, label = T, label_size = 3)
} else {
  var &lt;- apply(assay(dep), 1, sd)
  df &lt;- assay(dep)[order(var, decreasing = TRUE)[seq_len(nrow(dep))],]
  n_value &lt;- df %&gt;% as.data.frame() %&gt;% drop_na() %&gt;% nrow()
  if(n_value &gt;= 500){
    n_value &lt;- 500
  }
  plot_pca(dep, x = 1, y = 2, n = n_value, point_size = 4, indicate = &quot;condition&quot;, label = T, label_size = 3)
}</code></pre>
<p><img src="figure/Template_LFQ.Rmd/PCA-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="volcano" class="section level4">
<h4>Volcano</h4>
<p>Volcano plots allows to inspect the enrichment of proteins between the two samples (x axis) and their corresponding adjusted p value (y axis).</p>
<pre class="r"><code>plot_volcano(dep, contrast=test, add_names=T, label_size=5, adjusted = F)</code></pre>
<p><img src="figure/Template_LFQ.Rmd/volcano-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="volcano-and-unique" class="section level4">
<h4>Volcano and unique</h4>
<pre class="r"><code>plot_volcano_2(dep, contrast=test, add_names=T, label_size=5, adjusted = F, unique_vec = uni_name)</code></pre>
<p><img src="figure/Template_LFQ.Rmd/volcano-unique-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="correlation" class="section level4">
<h4>Correlation</h4>
<pre class="r"><code>plot_cor(dep, significant = TRUE, lower = 0, upper = 1, pal = &quot;Reds&quot;)</code></pre>
<p><img src="figure/Template_LFQ.Rmd/correlation-matrix-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="cluster-heatmap" class="section level4">
<h4>Cluster heatmap</h4>
<p>The heatmap representation gives an overview of all significant proteins (rows) in all samples (columns). This allows to see general trends, for example if one sample or replicate is really different compared to the others. Additionally, the clustering of samples (columns) can indicate closer related samples and clustering of proteins (rows) indicates similarly behaving proteins.</p>
<pre class="r"><code>if(results %&gt;% filter(significant) %&gt;% nrow() &lt;= 100){
  plot_heatmap(dep, type = &quot;centered&quot;, kmeans = TRUE, k = 5, show_row_names = T, indicate = &quot;condition&quot;, col_limit = 5)
}else {
  plot_heatmap(dep, type = &quot;centered&quot;, kmeans = TRUE, k = 5, show_row_names = F, indicate = &quot;condition&quot;, col_limit = 5)
}</code></pre>
<p><img src="figure/Template_LFQ.Rmd/cluster-heatmap-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="result-table" class="section level3">
<h3><strong>4.2. Result table</strong></h3>
<p>In this table are summarized the results of DEP analysis, sorted by p.value. Proteins that are UP regulated are colored with a red box in Fold change column, in the orhter hand proteins that are DOWN regulated are colored with blue.</p>
<pre class="r"><code>results %&gt;% 
  select(starts_with(&quot;name&quot;) | starts_with(test)) %&gt;% 
  filter(across(ends_with(&quot;significant&quot;))) %&gt;% 
  arrange(across(ends_with(&quot;p.adj&quot;))) %&gt;%
  mutate(across(2:3, format, scientific = T, digits = 2)) %&gt;%
  dplyr::rename_with(~ tolower(gsub(paste0(test,&quot;_&quot;), &quot;&quot;, .x)), .cols = starts_with(test)) %&gt;% 
  mutate(significant = str_to_title(significant)) %&gt;%
  dplyr::rename(Fold.Change = ratio, Gene.name = name) %&gt;% 
  DT::datatable(options = list(
  columnDefs = list(list(className = &#39;dt-center&#39;, targets = 1:5)),
  pageLength = 10)) %&gt;%
  formatStyle(&#39;Fold.Change&#39;, backgroundColor = styleInterval(c(-1, 1), c(&#39;lightblue&#39;, &#39;white&#39;, &#39;tomato&#39;))) %&gt;% 
  formatStyle(&#39;significant&#39;, color = styleEqual(c(&quot;True&quot;, &quot;False&quot;), c(&#39;green&#39;, &#39;red&#39;)))</code></pre>
<div id="htmlwidget-e0541469b5ed4906484e" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-e0541469b5ed4906484e">{"x":{"filter":"none","data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33","34","35","36","37","38","39","40","41","42","43","44","45","46","47","48","49","50","51","52","53","54","55","56","57","58","59","60","61","62","63","64","65","66","67","68","69","70","71","72","73","74","75","76","77","78","79","80","81","82","83","84","85","86","87","88","89","90","91","92","93","94","95","96","97","98","99","100","101","102","103","104","105","106","107","108","109","110","111","112","113","114","115","116","117","118","119","120","121","122","123","124","125","126","127","128","129","130","131","132","133","134","135","136","137"],["ZC3H4","SYMPK","CTR9","MRPS35","ZNF281","HCFC1","GPALPP1","CDC73","INTS3","MED23","RPRD1B","PPIE","CSTF2","CPSF6","MYL12A","LEO1","SART3","TNK1","TPM1.1","MRPS2","HNRNPR","JUNB","CD44","COIL","DKC1","MRPS5","MRPS22","CPSF1","CLINT1","HELLS","HNRNPK","DAP3","CDK9","PRPF3","ZNF616","ALDH18A1","MRPS9","MRPS25","DST","MRPS18B","MRPS24","MRPS7","MRPS27","HNRNPL","PAF1","TJP1","FIP1L1","H2AFX","NHP2","FBXO30","MRPS10","GULP1","NONO","ZNF207","DBT","CPSF4","A0A2R8YHD2","KEAP1","HIST1H4A","DYNLL2","PTCD3","MYH14.1","MYL6.1","SUGP2","UTP3","BUB3","NPM1","DNAH14","WDR82","LDHB","PRPF4","CPNE7","CAPZB","PRMT1","NOP10","LRRFIP2","MYH10","NECAP1","SYF2","KPNA3","MYO18A","PPP1R10","PTBP1","TMOD3","MRPS23","FLII","TPM1","A0A1W2PNV4","HNRNPU","CCDC12","SRSF5","LDHA","TPM2","ARPC3","MYL6","TPM3.1","MYH9","ACTR2","PPARD","KHDRBS1","MYL12B","SAFB","RBMX","ACTG1","ARPC4-TTLL3","ISY1","ARPC2","GSN","TPM4","CAPZA1","EIF2AK2","ACTA1","MYL6B","TPM3","MYO5A","RPA1","ACTR3","LRRFIP1","KAT6A","GAPDH","MYO6","CAPZA2","SSBP1","DLD","ACTN4","ATP5B","RPA3","RPA2","ARPC1B","ARPC5","MPRIP","MYO1B","SPTAN1","FBXO11","AHNAK","MYO1D","DHX30"],["4.3e-18","5.6e-15","2.1e-13","1.1e-12","6.5e-12","4.8e-11","1.7e-10","6.7e-10","7.7e-10","7.5e-10","6.0e-10","9.4e-10","1.2e-09","1.9e-09","2.0e-09","2.9e-09","2.8e-09","3.2e-09","3.2e-09","6.3e-09","8.9e-09","9.3e-09","2.2e-08","2.3e-08","3.5e-08","4.0e-08","6.1e-08","1.0e-07","2.3e-07","2.4e-07","4.5e-07","6.7e-07","7.6e-07","9.2e-07","9.6e-07","1.1e-06","1.2e-06","1.2e-06","1.5e-06","1.5e-06","5.3e-06","5.9e-06","6.1e-06","7.0e-06","7.2e-06","7.5e-06","8.2e-06","1.0e-05","1.1e-05","1.5e-05","1.4e-05","1.8e-05","1.9e-05","2.1e-05","2.5e-05","2.8e-05","3.0e-05","3.1e-05","3.6e-05","3.9e-05","4.4e-05","4.5e-05","5.4e-05","6.2e-05","7.1e-05","7.7e-05","9.7e-05","1.0e-04","1.0e-04","1.1e-04","1.2e-04","1.4e-04","1.5e-04","1.7e-04","1.8e-04","1.9e-04","2.3e-04","2.4e-04","2.4e-04","2.4e-04","3.0e-04","3.6e-04","4.1e-04","4.5e-04","4.7e-04","5.1e-04","5.2e-04","5.5e-04","5.7e-04","5.8e-04","5.8e-04","6.4e-04","6.5e-04","8.0e-04","8.6e-04","9.2e-04","9.7e-04","1.1e-03","1.1e-03","1.1e-03","1.2e-03","1.3e-03","1.3e-03","1.5e-03","1.5e-03","1.7e-03","1.8e-03","1.8e-03","1.9e-03","2.0e-03","2.1e-03","2.1e-03","2.3e-03","2.9e-03","3.0e-03","3.0e-03","3.1e-03","3.5e-03","3.9e-03","4.0e-03","4.2e-03","4.2e-03","5.7e-03","5.8e-03","6.1e-03","6.4e-03","7.6e-03","8.9e-03","9.3e-03","9.4e-03","1.1e-02","1.1e-02","1.1e-02","1.6e-02","1.6e-02","2.0e-02","2.0e-02"],["3.8e-15","2.5e-12","6.2e-11","2.4e-10","1.1e-09","7.0e-09","2.2e-08","6.1e-08","6.1e-08","6.1e-08","6.1e-08","6.8e-08","8.3e-08","1.1e-07","1.1e-07","1.5e-07","1.5e-07","1.5e-07","1.5e-07","2.7e-07","3.7e-07","3.7e-07","8.2e-07","8.2e-07","1.2e-06","1.4e-06","2.0e-06","3.2e-06","7.0e-06","7.0e-06","1.2e-05","1.8e-05","1.9e-05","2.3e-05","2.3e-05","2.6e-05","2.6e-05","2.7e-05","3.2e-05","3.2e-05","1.1e-04","1.2e-04","1.2e-04","1.4e-04","1.4e-04","1.4e-04","1.5e-04","1.7e-04","1.8e-04","2.3e-04","2.3e-04","2.7e-04","2.9e-04","3.1e-04","3.6e-04","4.0e-04","4.1e-04","4.2e-04","4.6e-04","5.0e-04","5.5e-04","5.5e-04","6.3e-04","7.2e-04","8.2e-04","8.5e-04","9.9e-04","1.0e-03","1.0e-03","1.1e-03","1.2e-03","1.3e-03","1.3e-03","1.4e-03","1.5e-03","1.5e-03","1.8e-03","1.8e-03","1.8e-03","1.8e-03","2.1e-03","2.4e-03","2.6e-03","2.8e-03","2.8e-03","3.0e-03","3.1e-03","3.2e-03","3.3e-03","3.3e-03","3.3e-03","3.5e-03","3.6e-03","4.2e-03","4.5e-03","4.7e-03","5.0e-03","5.3e-03","5.3e-03","5.6e-03","5.8e-03","6.2e-03","6.3e-03","6.7e-03","6.7e-03","7.6e-03","7.6e-03","7.6e-03","8.0e-03","8.3e-03","8.6e-03","8.7e-03","9.1e-03","1.1e-02","1.1e-02","1.1e-02","1.1e-02","1.2e-02","1.3e-02","1.4e-02","1.4e-02","1.4e-02","1.8e-02","1.8e-02","1.9e-02","1.9e-02","2.3e-02","2.5e-02","2.6e-02","2.6e-02","2.9e-02","2.9e-02","3.1e-02","3.9e-02","4.0e-02","4.6e-02","4.6e-02"],["True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True"],[9.3,6.52,3.68,4.07,3.05,2.33,-2.55,1.85,2.75,5.16,3.92,-2.83,2.81,2.07,4.36,3.22,1.96,2.5,2.82,3.08,1.99,1.46,1.5,2.06,1.21,2.04,2.25,1.62,-1.09,4.43,1.19,2.45,1.35,2.52,-1.06,-1.68,3.61,1.94,1.88,1.56,1.36,2.51,3.89,3.05,2.68,-1.25,1.77,-1.17,1.14,1.34,1.85,-1.39,1.25,1.81,-1,1.2,2.73,-1.05,-1.36,2.51,1.67,2.09,2.17,1.75,1.34,1.28,1.84,2.26,2.44,-1.47,1.04,-1.23,2.48,1.03,1.39,2.45,2.15,1.18,-1.34,-1.27,2.48,1.05,1.59,1.64,2.36,3.07,3.18,1.46,1.2,-1.03,-1.35,-1.25,3.13,1.46,1.97,2.37,1.99,1.32,1.76,1.36,2.03,1.7,1.16,1.54,1.13,-1.06,1.25,1.52,2.29,1.77,3.07,1.68,1.8,1.97,1.92,1.95,1.32,1.58,1.02,-1.11,2.25,1.08,1.82,-1.1,1.76,-1.01,1.64,1.67,1.1,1.23,1.62,1.14,1.36,-1.04,1.85,3.22,1.02]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>Gene.name<\/th>\n      <th>p.val<\/th>\n      <th>p.adj<\/th>\n      <th>significant<\/th>\n      <th>Fold.Change<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-center","targets":[1,2,3,4,5]},{"orderable":false,"targets":0}],"pageLength":10,"order":[],"autoWidth":false,"orderClasses":false,"rowCallback":"function(row, data, displayNum, displayIndex, dataIndex) {\nvar value=data[5]; $(this.api().cell(row, 5).node()).css({'background-color':isNaN(parseFloat(value)) ? '' : value <= -1 ? \"lightblue\" : value <= 1 ? \"white\" : \"tomato\"});\nvar value=data[4]; $(this.api().cell(row, 4).node()).css({'color':value == \"True\" ? \"green\" : value == \"False\" ? \"red\" : null});\n}"}},"evals":["options.rowCallback"],"jsHooks":[]}</script>
</div>
</div>
<div id="gene-ontology" class="section level2">
<h2><strong>5. Gene Ontology</strong></h2>
<p>Now I perfom a gene onology analysis (GO or by branch <a href="GO:MF" class="uri">GO:MF</a>, <a href="GO:BP" class="uri">GO:BP</a>, <a href="GO:CC" class="uri">GO:CC</a>) and a KEGG ontology.<br />
The enrichment results are visualized with an interactive <strong>Manhattan-like-plot</strong> named “Gostplot”.</p>
<div id="turbo_bio-1" class="section level3 tabset tabset-fade">
<h3><strong>5.1. TURBO_BIO</strong></h3>
<div id="gostplot" class="section level4">
<h4>Gostplot</h4>
<p>UP regulated proteins in TURBO_BIO were used to generate these gene ontologies.</p>
<pre class="r"><code>Gname &lt;- results %&gt;% filter(significant &amp; get(paste0(test, &quot;_ratio&quot;)) &gt; 0) %&gt;% pull(name)

if(!params$Imputation){
  Uni &lt;- unique_pair(data_filt_unique, conditions = conditions, table = T) %&gt;% 
  filter(!!sym(paste0(&quot;unique_&quot;, cond_1)) == &quot;+&quot;) %&gt;% 
  pull(name) %&gt;% unique()

  Gname &lt;- c(Gname, Uni)
}

if(length(Gname) == 0){
  print(&quot;NOT enough data.&quot;)
  GO &lt;- NULL
}else{
  GO &lt;- gost(query = Gname, organism = &quot;hsapiens&quot;, sources = c(&quot;GO:BP&quot;, &quot;GO:MF&quot;, &quot;GO:CC&quot;, &quot;KEGG&quot;), ordered_query = T, evcodes = TRUE)
  if(is.null(GO)){
    print(&quot;NOT enough data.&quot;)
  }else{
    gostplot(GO, capped = TRUE, interactive = TRUE)
  }
}</code></pre>
<div id="htmlwidget-d42468d86de4a421697d" style="width:960px;height:768px;" class="plotly html-widget"></div>
<script type="application/json" data-for="htmlwidget-d42468d86de4a421697d">{"x":{"data":[{"x":[86.6086654819669,105.120137124982,105.092857061508,113.549676738422,126.133580303748,160.34277990004,86.5969740261924,106.390608652482,107.037535872006,88.8768079022269,106.386711500557,163.596901757286,105.131828580757,86.5463110511694,88.9352651810995,108.876991580533,110.841156150655,107.033638720081,84.4106717963542,176.59780057857,123.740729021893,134.711211690332,110.891819125678,124.418833456816,122.778132496456,136.453238600738,137.127445883736,109.594067534704,136.005066129381,161.367730856274,161.363833704349,157.427710260256,105.151314340381,111.071088114221,160.537637496282,84.414568948279,177.97349520804,112.185673564727,176.021022093693,107.583137141484,135.697191127318,137.750990191712,106.398402956331,148.546101023533,105.159108644231,124.414936304891,162.22510427974,106.394505804406,104.79667351522,105.135725732681,93.6313332505381,106.406197260181,109.617450446254,107.021947264307,173.160512580856,84.3794945809554,108.880888732458,76.1604011714567,134.719005994182,122.746955281057,97.1777415021473,77.1619692161419,136.500004423836,93.635230402463,97.7623142908741,173.156615428931,105.104548517283,152.232806744437,97.2673759964188,164.746561575115,165.615626454356,114.173221046398,84.4730262271517,176.028816397542,108.619779553493,84.3327287578573,97.9805547986655,137.131343035661,89.4379977794046,167.536922353304,136.464930056513,112.875469455424,119.933211591319,136.589638918108,111.421831787457,93.810602239081,122.754749584907,177.977392359965,137.630178482042,123.370499589033,147.743287727015,93.0194803983374,104.784982059445,106.788118148816,150.732403253371,162.23289858359,140.412744956381,158.885245080148,131.722096163976,107.399971001017,106.164573840841,107.014152960457,93.6780990736363,107.590931445334,75.7823774347467,143.612306686679,86.3358648472277,84.0326480596442,84.5275863540995,150.736300405296,193.429599741976,144.60218327559,121.519352424731,180.140311678254,169.033428692445,180.132517374404,134.715108842257,87.1036037764223,123.966763833534,176.601697730494,84.4964091387008,94.2860547739121,124.722811306954,122.368931544347,113.596442561521,86.5735911146433,105.451395038594,163.98661694977,163.99441125362,84.0988996423665,147.758876334714,142.470441172699,193.421805438127,111.398448875908,140.416642108306,89.1807857523648,135.693293975393,75.7784802828218,183.526936700945,93.5144186927928],"y":[17,17,17,15.8130710541489,14.5180927907503,14.2224630216539,12.1749113172082,12.0696286479396,11.7956799410721,11.5611740746676,11.5361807168406,11.5137182713206,11.3418541242094,10.7607476540662,10.6895894480114,10.5520663055579,10.3314710426953,10.2964826159526,10.0784977187212,10.0546273225671,9.98936604594197,9.95600488533194,9.84989900626654,9.77120498066744,9.61217212452504,9.35188916097631,8.59153580318206,8.39541028409215,8.06346631660445,7.65026318004734,7.45889441586924,7.15963349269262,7.13647431274877,6.84481053412945,6.73723915601616,6.5730153736647,6.57275311485853,5.96893825706056,5.88713641877792,5.75360713334394,5.56629288809527,5.56380273200795,5.28691760979447,5.16158023261802,5.11287723613125,4.91563188639408,4.85772500072856,4.74687421858576,4.59176276808551,4.4857401298738,4.26922846471992,4.20109675382855,4.12994328285104,4.02457524472655,3.82167766014847,3.62901032797251,3.59460000758298,3.54147697066365,3.53132399765291,3.45291284957732,3.40652988282516,3.30112625510229,3.29914154078795,3.28350715385895,3.27770865663113,3.25314744791308,3.24479165589488,3.21334809164684,3.11603770610928,3.10801979353353,3.04107082160686,3.03729034117497,2.98545191513499,2.9567228486095,2.94306935015029,2.9208894710101,2.68119794542195,2.66511592696634,2.63861216962111,2.6013189836542,2.50061645074512,2.49750782342482,2.48522194688654,2.43288603781664,2.43288603781664,2.43013614808414,2.335676850237,2.29322089036552,2.20115796947984,2.17549851479072,2.09031187584382,2.07839080686838,2.07362431677066,2.07242670186964,2.03422293511993,2.02551326369666,1.99197713765916,1.98907026821918,1.9862807712572,1.97845200678312,1.9569900867628,1.95364302530007,1.93265256365178,1.92087060749857,1.92042862825096,1.90206750521659,1.85893000732663,1.85653333475938,1.82371915065612,1.81293976038994,1.78760690634769,1.7858949045573,1.76142919696206,1.74555305699203,1.74555305699203,1.74555305699203,1.73917574140091,1.7339055576861,1.72240463264491,1.7215285544467,1.71185804175529,1.69857975894506,1.67357073312644,1.66418443945335,1.62052291196259,1.60609291602551,1.59644918485773,1.58578912145045,1.58162068501412,1.57629522715829,1.57600988878112,1.56788523447483,1.56608870928848,1.56608870928848,1.56278716471579,1.5229355904635,1.44331020315294,1.40302513903476,1.38069993818932,1.32121038798304],"text":["GO:0007015 (444) <br> actin filament organization <br> 3.395e-18","GO:0030036 (711) <br> actin cytoskeleton organization <br> 1.766e-17","GO:0030029 (800) <br> actin filament-based process <br> 1.777e-17","GO:0034314 (39) <br> Arp2/3 complex-mediated actin nucleation <br> 1.538e-16","GO:0045010 (55) <br> actin nucleation <br> 3.033e-15","GO:0097435 (755) <br> supramolecular fiber organization <br> 5.992e-15","GO:0007010 (1435) <br> cytoskeleton organization <br> 6.685e-13","GO:0030833 (140) <br> regulation of actin filament polymerization <br> 8.519e-13","GO:0031124 (73) <br> mRNA 3'-end processing <br> 1.601e-12","GO:0008064 (159) <br> regulation of actin polymerization or depolymerization <br> 2.747e-12","GO:0030832 (160) <br> regulation of actin filament length <br> 2.910e-12","GO:0110053 (267) <br> regulation of actin filament organization <br> 3.064e-12","GO:0030041 (168) <br> actin filament polymerization <br> 4.551e-12","GO:0006996 (3805) <br> organelle organization <br> 1.735e-11","GO:0008154 (198) <br> actin polymerization or depolymerization <br> 2.044e-11","GO:0032271 (205) <br> regulation of protein polymerization <br> 2.805e-11","GO:0032956 (350) <br> regulation of actin cytoskeleton organization <br> 4.662e-11","GO:0031123 (126) <br> RNA 3'-end processing <br> 5.053e-11","GO:0006378 (52) <br> mRNA polyadenylation <br> 8.346e-11","GO:1902903 (373) <br> regulation of supramolecular fiber organization <br> 8.818e-11","GO:0043631 (53) <br> RNA polyadenylation <br> 1.025e-10","GO:0050684 (141) <br> regulation of mRNA processing <br> 1.107e-10","GO:0032970 (391) <br> regulation of actin filament-based process <br> 1.413e-10","GO:0044087 (940) <br> regulation of cellular component biogenesis <br> 1.694e-10","GO:0043254 (402) <br> regulation of protein-containing complex assembly <br> 2.442e-10","GO:0051258 (278) <br> protein polymerization <br> 4.447e-10","GO:0051493 (523) <br> regulation of cytoskeleton organization <br> 2.561e-09","GO:0032535 (355) <br> regulation of cellular component size <br> 4.023e-09","GO:0051128 (2246) <br> regulation of cellular component organization <br> 8.640e-09","GO:0098789 (13) <br> pre-mRNA cleavage required for polyadenylation <br> 2.237e-08","GO:0098787 (14) <br> mRNA cleavage involved in mRNA processing <br> 3.476e-08","GO:0090066 (488) <br> regulation of anatomical structure size <br> 6.924e-08","GO:0030048 (132) <br> actin filament-based movement <br> 7.303e-08","GO:0033043 (1109) <br> regulation of organelle organization <br> 1.430e-07","GO:0097581 (91) <br> lamellipodium organization <br> 1.831e-07","GO:0006379 (20) <br> mRNA cleavage <br> 2.673e-07","GO:1903311 (318) <br> regulation of mRNA metabolic process <br> 2.675e-07","GO:0033523 (12) <br> histone H2B ubiquitination <br> 1.074e-06","GO:1902743 (53) <br> regulation of lamellipodium organization <br> 1.297e-06","GO:0031440 (28) <br> regulation of mRNA 3'-end processing <br> 1.764e-06","GO:0051016 (29) <br> barbed-end actin filament capping <br> 2.715e-06","GO:0051693 (44) <br> actin filament capping <br> 2.730e-06","GO:0030835 (48) <br> negative regulation of actin filament depolymerization <br> 5.165e-06","GO:0070358 (7) <br> actin polymerization-dependent cell motility <br> 6.893e-06","GO:0030050 (18) <br> vesicle transport along actin filament <br> 7.711e-06","GO:0044085 (4801) <br> cellular component biogenesis <br> 1.214e-05","GO:0099515 (20) <br> actin filament-based transport <br> 1.388e-05","GO:0030834 (57) <br> regulation of actin filament depolymerization <br> 1.791e-05","GO:0022607 (4556) <br> cellular component assembly <br> 2.560e-05","GO:0030042 (62) <br> actin filament depolymerization <br> 3.268e-05","GO:0010591 (41) <br> regulation of lamellipodium assembly <br> 5.380e-05","GO:0030837 (68) <br> negative regulation of actin filament polymerization <br> 6.294e-05","GO:0032543 (79) <br> mitochondrial translation <br> 7.414e-05","GO:0031120 (3) <br> snRNA pseudouridine synthesis <br> 9.450e-05","GO:1901880 (77) <br> negative regulation of protein depolymerization <br> 1.508e-04","GO:0006368 (61) <br> transcription elongation from RNA polymerase II promoter <br> 2.350e-04","GO:0032272 (83) <br> negative regulation of protein polymerization <br> 2.543e-04","GO:0000723 (137) <br> telomere maintenance <br> 2.874e-04","GO:0050686 (30) <br> negative regulation of mRNA processing <br> 2.942e-04","GO:0043242 (87) <br> negative regulation of protein-containing complex disassembly <br> 3.524e-04","GO:0016043 (7793) <br> cellular component organization <br> 3.922e-04","GO:0001711 (15) <br> endodermal cell fate commitment <br> 4.999e-04","GO:0051276 (1034) <br> chromosome organization <br> 5.022e-04","GO:0010592 (29) <br> positive regulation of lamellipodium assembly <br> 5.206e-04","GO:0016344 (3) <br> meiotic chromosome movement towards spindle pole <br> 5.276e-04","GO:1901879 (93) <br> regulation of protein depolymerization <br> 5.583e-04","GO:0030032 (73) <br> lamellipodium assembly <br> 5.691e-04","GO:0071840 (7994) <br> cellular component organization or biogenesis <br> 6.119e-04","GO:0016071 (2153) <br> mRNA metabolic process <br> 7.655e-04","GO:0140053 (111) <br> mitochondrial gene expression <br> 7.798e-04","GO:0140694 (373) <br> non-membrane-bounded organelle assembly <br> 9.098e-04","GO:0034645 (1564) <br> cellular macromolecule biosynthetic process <br> 9.177e-04","GO:0006397 (1919) <br> mRNA processing <br> 1.034e-03","GO:1902745 (37) <br> positive regulation of lamellipodium organization <br> 1.105e-03","GO:0032200 (164) <br> telomere organization <br> 1.140e-03","GO:0006354 (80) <br> DNA-templated transcription, elongation <br> 1.200e-03","GO:0016574 (48) <br> histone ubiquitination <br> 2.084e-03","GO:0051494 (164) <br> negative regulation of cytoskeleton organization <br> 2.162e-03","GO:0009059 (4880) <br> macromolecule biosynthetic process <br> 2.298e-03","GO:1900363 (18) <br> regulation of mRNA polyadenylation <br> 2.504e-03","GO:0051261 (120) <br> protein depolymerization <br> 3.158e-03","GO:0034097 (922) <br> response to cytokine <br> 3.180e-03","GO:0040031 (7) <br> snRNA modification <br> 3.272e-03","GO:0051305 (7) <br> chromosome movement towards spindle pole <br> 3.691e-03","GO:0033206 (7) <br> meiotic cytokinesis <br> 3.691e-03","GO:0010639 (352) <br> negative regulation of organelle organization <br> 3.714e-03","GO:0043244 (127) <br> regulation of protein-containing complex disassembly <br> 4.617e-03","GO:1903312 (93) <br> negative regulation of mRNA metabolic process <br> 5.091e-03","GO:0051656 (409) <br> establishment of organelle localization <br> 6.293e-03","GO:0043484 (151) <br> regulation of RNA splicing <br> 6.676e-03","GO:0065003 (3194) <br> protein-containing complex assembly <br> 8.122e-03","GO:0010390 (29) <br> histone monoubiquitination <br> 8.349e-03","GO:0022604 (305) <br> regulation of cell morphogenesis <br> 8.441e-03","GO:0031032 (193) <br> actomyosin structure organization <br> 8.464e-03","GO:0071345 (828) <br> cellular response to cytokine stimulus <br> 9.242e-03","GO:0099518 (70) <br> vesicle cytoskeletal trafficking <br> 9.429e-03","GO:0055001 (174) <br> muscle cell development <br> 1.019e-02","GO:0090501 (155) <br> RNA phosphodiester bond hydrolysis <br> 1.025e-02","GO:0048024 (105) <br> regulation of mRNA splicing, via spliceosome <br> 1.032e-02","GO:0031333 (147) <br> negative regulation of protein-containing complex assembly <br> 1.051e-02","GO:0030705 (194) <br> cytoskeleton-dependent intracellular transport <br> 1.104e-02","GO:0031118 (10) <br> rRNA pseudouridine synthesis <br> 1.113e-02","GO:0010604 (3378) <br> positive regulation of macromolecule metabolic process <br> 1.168e-02","GO:0031442 (10) <br> positive regulation of mRNA 3'-end processing <br> 1.200e-02","GO:0000381 (62) <br> regulation of alternative mRNA splicing, via spliceosome <br> 1.201e-02","GO:0060795 (32) <br> cell fate commitment involved in formation of primary germ layer <br> 1.253e-02","GO:0006928 (2074) <br> movement of cell or subcellular component <br> 1.384e-02","GO:0006259 (896) <br> DNA metabolic process <br> 1.391e-02","GO:0006412 (713) <br> translation <br> 1.501e-02","GO:0071346 (132) <br> cellular response to interferon-gamma <br> 1.538e-02","GO:2001168 (4) <br> positive regulation of histone H2B ubiquitination <br> 1.631e-02","GO:0061061 (616) <br> muscle structure development <br> 1.637e-02","GO:0042692 (357) <br> muscle cell differentiation <br> 1.732e-02","GO:1903923 (2) <br> positive regulation of protein processing in phagocytic vesicle <br> 1.797e-02","GO:1900756 (2) <br> protein processing in phagocytic vesicle <br> 1.797e-02","GO:1903921 (2) <br> regulation of protein processing in phagocytic vesicle <br> 1.797e-02","GO:0050685 (32) <br> positive regulation of mRNA processing <br> 1.823e-02","GO:0007163 (222) <br> establishment or maintenance of cell polarity <br> 1.845e-02","GO:0043933 (3387) <br> protein-containing complex organization <br> 1.895e-02","GO:1902904 (166) <br> negative regulation of supramolecular fiber organization <br> 1.899e-02","GO:0006403 (198) <br> RNA localization <br> 1.942e-02","GO:0010769 (96) <br> regulation of cell morphogenesis involved in differentiation <br> 2.002e-02","GO:0044271 (4734) <br> cellular nitrogen compound biosynthetic process <br> 2.120e-02","GO:0043043 (738) <br> peptide biosynthetic process <br> 2.167e-02","GO:0034341 (153) <br> response to interferon-gamma <br> 2.396e-02","GO:0007004 (71) <br> telomere maintenance via telomerase <br> 2.477e-02","GO:0030239 (63) <br> myofibril assembly <br> 2.533e-02","GO:0120032 (190) <br> regulation of plasma membrane bounded cell projection assembly <br> 2.595e-02","GO:0120034 (105) <br> positive regulation of plasma membrane bounded cell projection assembly <br> 2.620e-02","GO:0006278 (72) <br> RNA-dependent DNA biosynthetic process <br> 2.653e-02","GO:0065008 (3741) <br> regulation of biological quality <br> 2.655e-02","GO:0060491 (192) <br> regulation of cell projection assembly <br> 2.705e-02","GO:2001166 (5) <br> regulation of histone H2B ubiquitination <br> 2.716e-02","GO:0033184 (5) <br> positive regulation of histone ubiquitination <br> 2.716e-02","GO:0055002 (64) <br> striated muscle cell development <br> 2.737e-02","GO:0008356 (19) <br> asymmetric cell division <br> 3.000e-02","GO:0051014 (17) <br> actin filament severing <br> 3.603e-02","GO:0000380 (79) <br> alternative mRNA splicing, via spliceosome <br> 3.953e-02","GO:1904874 (15) <br> positive regulation of telomerase RNA localization to Cajal body <br> 4.162e-02","GO:0010557 (1839) <br> positive regulation of macromolecule biosynthetic process <br> 4.773e-02"],"key":["GO:0007015","GO:0030036","GO:0030029","GO:0034314","GO:0045010","GO:0097435","GO:0007010","GO:0030833","GO:0031124","GO:0008064","GO:0030832","GO:0110053","GO:0030041","GO:0006996","GO:0008154","GO:0032271","GO:0032956","GO:0031123","GO:0006378","GO:1902903","GO:0043631","GO:0050684","GO:0032970","GO:0044087","GO:0043254","GO:0051258","GO:0051493","GO:0032535","GO:0051128","GO:0098789","GO:0098787","GO:0090066","GO:0030048","GO:0033043","GO:0097581","GO:0006379","GO:1903311","GO:0033523","GO:1902743","GO:0031440","GO:0051016","GO:0051693","GO:0030835","GO:0070358","GO:0030050","GO:0044085","GO:0099515","GO:0030834","GO:0022607","GO:0030042","GO:0010591","GO:0030837","GO:0032543","GO:0031120","GO:1901880","GO:0006368","GO:0032272","GO:0000723","GO:0050686","GO:0043242","GO:0016043","GO:0001711","GO:0051276","GO:0010592","GO:0016344","GO:1901879","GO:0030032","GO:0071840","GO:0016071","GO:0140053","GO:0140694","GO:0034645","GO:0006397","GO:1902745","GO:0032200","GO:0006354","GO:0016574","GO:0051494","GO:0009059","GO:1900363","GO:0051261","GO:0034097","GO:0040031","GO:0051305","GO:0033206","GO:0010639","GO:0043244","GO:1903312","GO:0051656","GO:0043484","GO:0065003","GO:0010390","GO:0022604","GO:0031032","GO:0071345","GO:0099518","GO:0055001","GO:0090501","GO:0048024","GO:0031333","GO:0030705","GO:0031118","GO:0010604","GO:0031442","GO:0000381","GO:0060795","GO:0006928","GO:0006259","GO:0006412","GO:0071346","GO:2001168","GO:0061061","GO:0042692","GO:1903923","GO:1900756","GO:1903921","GO:0050685","GO:0007163","GO:0043933","GO:1902904","GO:0006403","GO:0010769","GO:0044271","GO:0043043","GO:0034341","GO:0007004","GO:0030239","GO:0120032","GO:0120034","GO:0006278","GO:0065008","GO:0060491","GO:2001166","GO:0033184","GO:0055002","GO:0008356","GO:0051014","GO:0000380","GO:1904874","GO:0010557"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(255,153,0,1)","opacity":0.8,"size":[18.3473821837242,18.9689234520263,19.1206656257896,14.5814139974581,15.1893666851417,19.0463669754254,19.8513263318723,16.6979252377404,15.6667434572733,16.8899800403965,16.8993690490748,17.6447731536901,16.9722060406806,21.0018269113499,17.2145842446943,17.2652787333353,18.0230448205345,16.5367347658025,15.0924043121985,18.1105369965064,15.1254260222545,16.708741750045,18.1749820385619,19.3257581587702,18.2127857347699,17.7018443366872,18.5665122992221,18.0425906994776,20.3897665416323,12.3542694385448,12.5223664034652,18.4742232008518,16.608155189633,19.5332499620522,16.0254831918018,13.2899306801292,17.8902134152238,12.1689282166855,15.1254260222545,13.9611149660018,14.0285832754376,14.798562659995,14.9525766293039,10.7944929897153,13.0697969123388,21.2649713126228,13.2899306801292,15.2506857256693,21.2060372008953,15.3937574422679,14.671965682599,15.5489306475125,15.7965422855522,7.77039980235131,15.7545592284552,15.3662271239498,15.8769999436579,16.6649501000042,14.0933449312819,15.9531751392093,21.8008623810869,12.6760065098355,19.4457064749222,14.0285832754376,7.77039980235131,16.0603010299089,15.6667434572733,21.8285943750956,20.3396875973885,16.3400771582388,18.1105369965064,19.9561802351649,20.2026609694964,14.4852668565195,16.936282113548,15.8170833598509,14.9525766293039,16.936282113548,21.2832942259864,13.0697969123388,16.461396937013,19.301312316507,10.7944929897153,10.7944929897153,10.7944929897153,18.030899694876,16.5488994624023,16.0603010299089,18.2362578787775,16.8124132451137,20.8010368448869,14.0285832754376,17.8319964975192,17.177141715501,19.1646449933625,15.5972053139566,17.024344902898,16.8517546521526,16.2528819106722,16.7718916003282,17.1847156572066,11.7306454967647,20.8655421968307,11.7306454967647,15.3937574422679,14.2155212474336,20.2952914186446,19.2650749324978,18.9725551526272,16.608155189633,8.99752440324434,18.7823225213627,18.0503246746426,3.77952755905512,3.77952755905512,3.77952755905512,14.2155212474336,17.3808104874969,20.8686004785421,16.9543645281488,17.2145842446943,16.110971780725,21.2491783746142,19.0170406106578,16.8322273404496,15.62075611391,15.4207837047603,17.154155960642,16.2528819106722,15.6439317492578,20.9824712180928,17.169524110889,9.77892631171347,9.77892631171347,15.4473232090416,13.1834053419274,12.9481625134338,15.7965422855522,12.6760065098355,20.1516574830331],"symbol":"circle","line":{"width":1.88976377952756,"color":"rgba(255,153,0,1)"}},"hoveron":"points","set":"SharedDatae0541469","name":"GO:BP","legendgroup":"GO:BP","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","_isNestedKey":false,"frame":null},{"x":[57.1046489358643,62.1290372645198,62.1446288497134,55.8456284314766,59.9579090263032,55.4324514238447,54.0915750971904,57.2371774100104,55.7598747129115,57.1163426247595,55.9859526982195,57.8452492325629,55.4246556312479,54.087677200892,55.058253379197,64.4326939768824,62.1485267460118,59.4745698852999,64.5223455917459,69.6987518760395,64.428796080584,62.1368330571166,64.608099310311,57.8374534399661,62.9281060056946,58.7534590700934,62.1329351608182,68.4475271642487,55.9157905648481,61.8639803162276,62.2108930867865,55.7130999573305,66.4323147779687,55.7286915425242,54.9257249050509,61.1974400491989,67.9758817121406,55.7754662981052,62.1251393682214,55.783262090702,55.736487335121,56.0288295575021,59.4979572630904,57.8257597510709,67.0481823931181,54.559322653,62.1212414719229,57.3463185063659,55.3622892904733,61.396232760418,68.5683619494995,68.4514250605471,68.1084101862867,59.5018551593888,55.8417305351782,59.7941973817699,55.6741209943464,57.8413513362645,57.6659460028359,54.9491122828414,57.2449732026072,69.2855748684076,57.3346248174707,54.7815027420096,55.4168598386511,58.3207925809694,56.2471117502132,55.3427998089812,57.8296576473693,66.1321767629908,59.447284611211,54.6294847863715,61.9887129977769,57.092955246969,66.3660505408957,59.0263118109823,57.2410753063088,55.350595601578,66.1282788666924,55.2102713348351,54.991989142124,69.4843675796267,56.1730517205434,68.884091549671,55.1985776459399,57.3268290248739,71.1994419509289,63.816826361733,54.1929204009491,67.9096174750675,59.4784677815983,59.4511825075094,58.3597715439535,62.140730953415,58.157080936436,55.0816407569875,68.5722598457979,67.0325908079244,58.3246904772678,65.7306934442542,65.220069029162,68.092818601093,62.3356257683357,58.3753631291472,55.3233103274892],"y":[17,17,17,17,17,17,17,15.6121889861784,13.6588937566749,13.0039163292951,12.8620812179926,12.7409101609106,12.3830916056421,12.3830916056421,10.1655644165558,10.0576968062643,10.0541388932568,10.0541388932568,9.88305401700201,9.81662294843794,9.81362008875186,9.81362008875186,9.2107289210518,9.15825871089636,8.56127006504285,8.36254604902101,8.21736780861945,7.82367109382595,7.58581822963759,7.30755306560455,7.29730956126022,7.02859346008813,6.97780441304571,6.94694040844154,6.93712623628708,6.80007383250788,6.77994142901429,6.77780487023088,6.75793376338309,6.60795726390821,6.5849108908398,6.49490340929789,6.46412443639803,6.3847797407665,6.22837632457651,6.22837632457651,6.21062813315339,6.17737840413775,6.17196431425469,6.12535316490274,6.01839712921513,5.95421428288045,5.9257300827245,5.9201963530965,5.86944371675687,5.85253234823514,5.77996093228244,5.68459841540185,5.56430110266795,5.53134589506196,5.51510989053433,5.43426801045904,5.37914730237622,5.37311654716528,5.29952302006685,5.21901287156297,5.12696240040017,4.82430773141366,4.61508435583019,4.20762961043086,4.13526701001932,4.05667230047011,3.98536572481668,3.9625814023111,3.81083335332732,3.81083335332732,3.80971734437223,3.8081002350186,3.51094694993575,3.33924722517746,3.18852456772376,3.16378823794807,3.1513495561379,3.10368411595856,3.0931866917064,3.05604320083542,2.64947993932401,2.44484598416872,2.42347471066242,2.37640338649346,2.3740612699305,2.3740612699305,2.33994414594076,2.28201528867979,2.2471794402786,2.16470892488036,2.09848303358028,2.0401765893952,2.02402146849762,1.89679074082826,1.72217082541183,1.49406288809879,1.445584377142,1.32956541699251,1.32175147280235],"text":["GO:0015629 (501) <br> actin cytoskeleton <br> 1.281e-31","GO:0043228 (6362) <br> non-membrane-bounded organelle <br> 7.792e-24","GO:0043232 (6362) <br> intracellular non-membrane-bounded organelle <br> 7.792e-24","GO:0005885 (11) <br> Arp2/3 protein complex <br> 5.869e-23","GO:0032991 (7228) <br> protein-containing complex <br> 1.071e-19","GO:0005763 (28) <br> mitochondrial small ribosomal subunit <br> 4.139e-19","GO:0000314 (28) <br> organellar small ribosomal subunit <br> 4.139e-19","GO:0016459 (57) <br> myosin complex <br> 2.442e-16","GO:0005856 (2334) <br> cytoskeleton <br> 2.193e-14","GO:0015935 (76) <br> small ribosomal subunit <br> 9.910e-14","GO:0005925 (420) <br> focal adhesion <br> 1.374e-13","GO:0030055 (428) <br> cell-substrate junction <br> 1.816e-13","GO:0005761 (86) <br> mitochondrial ribosome <br> 4.139e-13","GO:0000313 (86) <br> organellar ribosome <br> 4.139e-13","GO:0005654 (4161) <br> nucleoplasm <br> 6.830e-11","GO:0070013 (6569) <br> intracellular organelle lumen <br> 8.756e-11","GO:0043233 (6570) <br> organelle lumen <br> 8.828e-11","GO:0031974 (6570) <br> membrane-enclosed lumen <br> 8.828e-11","GO:0070062 (2110) <br> extracellular exosome <br> 1.309e-10","GO:1903561 (2132) <br> extracellular vesicle <br> 1.525e-10","GO:0065010 (2133) <br> extracellular membrane-bounded organelle <br> 1.536e-10","GO:0043230 (2133) <br> extracellular organelle <br> 1.536e-10","GO:0070161 (839) <br> anchoring junction <br> 6.156e-10","GO:0030027 (203) <br> lamellipodium <br> 6.946e-10","GO:0044391 (189) <br> ribosomal subunit <br> 2.746e-09","GO:0031252 (421) <br> cell leading edge <br> 4.340e-09","GO:0043229 (15870) <br> intracellular organelle <br> 6.062e-09","GO:0099080 (1330) <br> supramolecular complex <br> 1.501e-08","GO:0005903 (105) <br> brush border <br> 2.595e-08","GO:0042641 (79) <br> actomyosin <br> 4.925e-08","GO:0043292 (242) <br> contractile fiber <br> 5.043e-08","GO:0005840 (520) <br> ribosome <br> 9.363e-08","GO:0090734 (97) <br> site of DNA damage <br> 1.052e-07","GO:0005847 (24) <br> mRNA cleavage and polyadenylation specificity factor complex <br> 1.130e-07","GO:0005615 (3353) <br> extracellular space <br> 1.156e-07","GO:0035861 (73) <br> site of double-strand break <br> 1.585e-07","GO:0098798 (277) <br> mitochondrial protein-containing complex <br> 1.660e-07","GO:0005862 (4) <br> muscle thin filament tropomyosin <br> 1.668e-07","GO:0043227 (15557) <br> membrane-bounded organelle <br> 1.746e-07","GO:0005865 (22) <br> striated muscle thin filament <br> 2.466e-07","GO:0005849 (28) <br> mRNA cleavage factor complex <br> 2.601e-07","GO:0005938 (319) <br> cell cortex <br> 3.200e-07","GO:0031981 (5442) <br> nuclear lumen <br> 3.435e-07","GO:0030016 (232) <br> myofibril <br> 4.123e-07","GO:0097517 (71) <br> contractile actin filament bundle <br> 5.910e-07","GO:0001725 (71) <br> stress fiber <br> 5.910e-07","GO:0043226 (16674) <br> organelle <br> 6.157e-07","GO:0016604 (869) <br> nuclear body <br> 6.647e-07","GO:0005743 (500) <br> mitochondrial inner membrane <br> 6.730e-07","GO:0036379 (26) <br> myofilament <br> 7.493e-07","GO:0099512 (988) <br> supramolecular fiber <br> 9.585e-07","GO:0099081 (996) <br> supramolecular polymer <br> 1.111e-06","GO:0098862 (161) <br> cluster of actin-based cell projections <br> 1.187e-06","GO:0031982 (3942) <br> vesicle <br> 1.202e-06","GO:0005884 (114) <br> actin filament <br> 1.351e-06","GO:0032432 (79) <br> actin filament bundle <br> 1.404e-06","GO:0005829 (5373) <br> cytosol <br> 1.660e-06","GO:0030054 (2076) <br> cell junction <br> 2.067e-06","GO:0019866 (562) <br> organelle inner membrane <br> 2.727e-06","GO:0005622 (17758) <br> intracellular anatomical structure <br> 2.942e-06","GO:0016461 (8) <br> unconventional myosin complex <br> 3.054e-06","GO:0140513 (2671) <br> nuclear protein-containing complex <br> 3.679e-06","GO:0016593 (8) <br> Cdc73/Paf1 complex <br> 4.177e-06","GO:0005576 (4303) <br> extracellular region <br> 4.235e-06","GO:0005759 (483) <br> mitochondrial matrix <br> 5.017e-06","GO:0030863 (110) <br> cortical cytoskeleton <br> 6.039e-06","GO:0008290 (8) <br> F-actin capping protein complex <br> 7.465e-06","GO:0005737 (12193) <br> cytoplasm <br> 1.499e-05","GO:0030017 (211) <br> sarcomere <br> 2.426e-05","GO:0072589 (4) <br> box H/ACA scaRNP complex <br> 6.200e-05","GO:0031966 (745) <br> mitochondrial membrane <br> 7.324e-05","GO:0002102 (30) <br> podosome <br> 8.777e-05","GO:0042995 (2312) <br> cell projection <br> 1.034e-04","GO:0015030 (102) <br> Cajal body <br> 1.090e-04","GO:0090661 (5) <br> box H/ACA telomerase RNP complex <br> 1.546e-04","GO:0031429 (5) <br> box H/ACA snoRNP complex <br> 1.546e-04","GO:0016460 (20) <br> myosin II complex <br> 1.550e-04","GO:0005740 (796) <br> mitochondrial envelope <br> 1.556e-04","GO:0072588 (6) <br> box H/ACA RNP complex <br> 3.084e-04","GO:0005697 (23) <br> telomerase holoenzyme complex <br> 4.579e-04","GO:0005634 (10053) <br> nucleus <br> 6.479e-04","GO:1902494 (1424) <br> catalytic complex <br> 6.858e-04","GO:0008023 (56) <br> transcription elongation factor complex <br> 7.057e-04","GO:0120025 (2205) <br> plasma membrane bounded cell projection <br> 7.876e-04","GO:0005694 (1795) <br> chromosome <br> 8.069e-04","GO:0016591 (91) <br> RNA polymerase II, holoenzyme <br> 8.789e-04","GO:1990904 (2377) <br> ribonucleoprotein complex <br> 2.241e-03","GO:0055029 (116) <br> nuclear DNA-directed RNA polymerase complex <br> 3.590e-03","GO:0000428 (117) <br> DNA-directed RNA polymerase complex <br> 3.772e-03","GO:0098687 (359) <br> chromosomal region <br> 4.203e-03","GO:0031975 (1245) <br> envelope <br> 4.226e-03","GO:0031967 (1245) <br> organelle envelope <br> 4.226e-03","GO:0030880 (121) <br> RNA polymerase complex <br> 4.571e-03","GO:0043231 (14514) <br> intracellular membrane-bounded organelle <br> 5.224e-03","GO:0030478 (4) <br> actin cap <br> 5.660e-03","GO:0005662 (13) <br> DNA replication factor A complex <br> 6.844e-03","GO:0099513 (757) <br> polymeric cytoskeletal fiber <br> 7.971e-03","GO:0097513 (3) <br> myosin II filament <br> 9.116e-03","GO:0030864 (80) <br> cortical actin cytoskeleton <br> 9.462e-03","GO:0071203 (12) <br> WASH complex <br> 1.268e-02","GO:0071001 (3) <br> U4/U6 snRNP <br> 1.896e-02","GO:0098858 (213) <br> actin-based cell projection <br> 3.206e-02","GO:0043601 (22) <br> nuclear replisome <br> 3.584e-02","GO:0030894 (24) <br> replisome <br> 4.682e-02","GO:0005732 (28) <br> sno(s)RNA-containing ribonucleoprotein complex <br> 4.767e-02"],"key":["GO:0015629","GO:0043228","GO:0043232","GO:0005885","GO:0032991","GO:0005763","GO:0000314","GO:0016459","GO:0005856","GO:0015935","GO:0005925","GO:0030055","GO:0005761","GO:0000313","GO:0005654","GO:0070013","GO:0043233","GO:0031974","GO:0070062","GO:1903561","GO:0065010","GO:0043230","GO:0070161","GO:0030027","GO:0044391","GO:0031252","GO:0043229","GO:0099080","GO:0005903","GO:0042641","GO:0043292","GO:0005840","GO:0090734","GO:0005847","GO:0005615","GO:0035861","GO:0098798","GO:0005862","GO:0043227","GO:0005865","GO:0005849","GO:0005938","GO:0031981","GO:0030016","GO:0097517","GO:0001725","GO:0043226","GO:0016604","GO:0005743","GO:0036379","GO:0099512","GO:0099081","GO:0098862","GO:0031982","GO:0005884","GO:0032432","GO:0005829","GO:0030054","GO:0019866","GO:0005622","GO:0016461","GO:0140513","GO:0016593","GO:0005576","GO:0005759","GO:0030863","GO:0008290","GO:0005737","GO:0030017","GO:0072589","GO:0031966","GO:0002102","GO:0042995","GO:0015030","GO:0090661","GO:0031429","GO:0016460","GO:0005740","GO:0072588","GO:0005697","GO:0005634","GO:1902494","GO:0008023","GO:0120025","GO:0005694","GO:0016591","GO:1990904","GO:0055029","GO:0000428","GO:0098687","GO:0031975","GO:0031967","GO:0030880","GO:0043231","GO:0030478","GO:0005662","GO:0099513","GO:0097513","GO:0030864","GO:0071203","GO:0071001","GO:0098858","GO:0043601","GO:0030894","GO:0005732"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(16,150,24,1)","opacity":0.8,"size":[18.5093206302223,21.5783776479263,21.5783776479263,11.9626902987181,21.7186491899641,13.9611149660018,13.9611149660018,15.2506857256693,20.4351485853431,15.7331002252975,18.2722688771406,18.2978168250027,15.9345090534973,15.9345090534973,21.1035242849928,21.6136744323874,21.6138420676856,21.6138420676856,20.3157434393389,20.3280586048311,20.3286151448313,20.3286151448313,19.1814836382837,17.2509927405727,17.1464044185986,18.2754913372617,22.5599778805315,19.7581982152725,16.2528819106722,15.7965422855522,17.5048066811551,18.5588694915346,16.1274657222001,13.6592813242164,20.8570011251474,15.6667434572733,17.6967597289842,8.99752440324434,22.5391342674345,13.4847654999701,13.9611149660018,17.8945828748266,21.4051772962856,17.4442857448196,15.62075611391,15.62075611391,22.611590191448,19.2262193222329,18.5066562827429,13.817141287718,19.3885502335527,19.3986943920346,16.9086929071744,21.042118612013,16.3817083524515,15.7965422855522,21.3909526731361,20.2964375921402,18.6617323675032,22.6771653543307,11.1588894645275,20.5934358537654,11.1588894645275,21.1415267205783,18.4604517488511,16.3259182730869,11.1588894645275,22.2822864937662,17.3072275895117,8.99752440324434,19.0292040076885,14.0933449312819,20.4239770277874,16.2071544572567,9.77892631171347,9.77892631171347,13.2899306801292,19.1142469792459,10.3487400178115,13.574309418746,22.07628187225,19.8419198881927,15.2203410437363,20.3679678765959,20.1225803364186,16.0254831918018,20.4566619508423,16.4087842754834,16.4221263012614,18.0580112889306,19.6768124493713,19.6768124493713,16.4742428340656,22.4663374189607,8.99752440324434,12.3542694385448,19.049769984315,7.77039980235131,15.8170833598509,12.1689282166855,7.77039980235131,17.3209176655616,13.4847654999701,13.6592813242164,13.9611149660018],"symbol":"circle","line":{"width":1.88976377952756,"color":"rgba(16,150,24,1)"}},"hoveron":"points","set":"SharedDatae0541469","name":"GO:CC","legendgroup":"GO:CC","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","_isNestedKey":false,"frame":null},{"x":[37.3800599541577,3.80836613997898,9.6154039601701,8.59040332479408,29.7295799266173,8.58260864315624,3.69144591541137,3.78887943588438,2.11302288374869,3.71093261950597,2.53783303301099,28.3577159583574,9.30361669465648,3.53944962347349,3.7343166644195,28.572069703398,2.78726284542188,2.78336550460296,40.2134267295127,25.2203565991266,48.6628616249317,29.9049602634687,42.056868936862,40.1588639580478,10.106468903354,37.3332918643307,48.4134318125208,3.55893632756809,9.33869276202677,3.69924059704921,23.0495377629881,20.9917418105982,9.3075140354754],"y":[17,17,14.3445899875308,12.5601122912091,11.7397459158317,11.2651688926473,9.27064479347112,8.06913126632822,7.98320433057369,5.59178584443694,5.08723329896467,4.73195018526841,4.23907090789404,4.14781669338061,4.05964637334626,4.0392540382649,3.85075990028387,3.85075990028387,3.80494174143804,3.48059026700451,3.37462609872653,3.32045994333767,3.19174321475702,3.06094030368932,2.82548611525982,2.72702746417682,2.48656169297965,2.19532258964055,1.78533825183376,1.74593776323495,1.55912723019342,1.5209151251484,1.51792044018356],"text":["GO:0051015 (218) <br> actin filament binding <br> 3.491e-24","GO:0003779 (449) <br> actin binding <br> 1.522e-22","GO:0008092 (995) <br> cytoskeletal protein binding <br> 4.523e-15","GO:0005200 (104) <br> structural constituent of cytoskeleton <br> 2.754e-13","GO:0044877 (1275) <br> protein-containing complex binding <br> 1.821e-12","GO:0005198 (980) <br> structural molecule activity <br> 5.430e-12","GO:0003723 (3142) <br> RNA binding <br> 5.362e-10","GO:0003774 (115) <br> cytoskeletal motor activity <br> 8.528e-09","GO:0000146 (31) <br> microfilament motor activity <br> 1.039e-08","GO:0003729 (350) <br> mRNA binding <br> 2.560e-06","GO:0000993 (36) <br> RNA polymerase II complex binding <br> 8.180e-06","GO:0043175 (41) <br> RNA polymerase core enzyme binding <br> 1.854e-05","GO:0005515 (14799) <br> protein binding <br> 5.767e-05","GO:0003676 (5447) <br> nucleic acid binding <br> 7.115e-05","GO:0003735 (449) <br> structural constituent of ribosome <br> 8.717e-05","GO:0043531 (34) <br> ADP binding <br> 9.136e-05","GO:0001099 (57) <br> basal RNA polymerase II transcription machinery binding <br> 1.410e-04","GO:0001098 (57) <br> basal transcription machinery binding <br> 1.410e-04","GO:0070063 (58) <br> RNA polymerase binding <br> 1.567e-04","GO:0034513 (5) <br> box H/ACA snoRNA binding <br> 3.307e-04","GO:1901363 (7416) <br> heterocyclic compound binding <br> 4.221e-04","GO:0045296 (334) <br> cadherin binding <br> 4.781e-04","GO:0097159 (7497) <br> organic cyclic compound binding <br> 6.431e-04","GO:0070034 (22) <br> telomerase RNA binding <br> 8.691e-04","GO:0008307 (43) <br> structural constituent of muscle <br> 1.495e-03","GO:0050839 (550) <br> cell adhesion molecule binding <br> 1.875e-03","GO:0140657 (550) <br> ATP-dependent activity <br> 3.262e-03","GO:0003682 (580) <br> chromatin binding <br> 6.378e-03","GO:0005524 (1502) <br> ATP binding <br> 1.639e-02","GO:0003725 (74) <br> double-stranded RNA binding <br> 1.795e-02","GO:0032559 (1565) <br> adenyl ribonucleotide binding <br> 2.760e-02","GO:0030554 (1576) <br> adenyl nucleotide binding <br> 3.014e-02","GO:0005516 (203) <br> calmodulin binding <br> 3.034e-02"],"key":["GO:0051015","GO:0003779","GO:0008092","GO:0005200","GO:0044877","GO:0005198","GO:0003723","GO:0003774","GO:0000146","GO:0003729","GO:0000993","GO:0043175","GO:0005515","GO:0003676","GO:0003735","GO:0043531","GO:0001099","GO:0001098","GO:0070063","GO:0034513","GO:1901363","GO:0045296","GO:0097159","GO:0070034","GO:0008307","GO:0050839","GO:0140657","GO:0003682","GO:0005524","GO:0003725","GO:0032559","GO:0030554","GO:0005516"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(220,57,18,1)","opacity":0.8,"size":[17.354529519854,18.3624720287095,19.3974311963365,16.2378047873792,19.7062024214885,19.3783169237459,20.7820862063444,16.3953124390496,14.1555983333967,18.0230448205345,14.4348828193235,14.671965682599,22.48676457789,21.4062006044941,18.3624720287095,14.3289994886174,15.2506857256693,15.2506857256693,15.2804246607373,9.77892631171347,21.7467399236646,17.9583759839834,21.7586107865689,13.4847654999701,14.7575093437122,18.6332205816173,18.6332205816173,18.7032804941857,19.906995199969,15.6892019866473,19.9569562634249,19.9654575586114,17.2509927405727],"symbol":"circle","line":{"width":1.88976377952756,"color":"rgba(220,57,18,1)"}},"hoveron":"points","set":"SharedDatae0541469","name":"GO:MF","legendgroup":"GO:MF","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","_isNestedKey":false,"frame":null},{"x":[199.429989113158,199.020054891063,199.207453392592,198.879506014916,199.465126332194,199.4495097904,199.453413925849,199.113754141827,199.445605654952,198.664778565247,198.719436461526],"y":[9.13972201683113,8.52073968621208,7.66367979147811,7.42056433558076,7.37359039895833,7.19816717139133,7.17023922805415,6.50468429114254,6.25883761782357,2.35586972242224,1.93672207051992],"text":["KEGG:05100 (77) <br> Bacterial invasion of epithelial cells <br> 7.249e-10","KEGG:04530 (169) <br> Tight junction <br> 3.015e-09","KEGG:04810 (216) <br> Regulation of actin cytoskeleton <br> 2.169e-08","KEGG:04144 (251) <br> Endocytosis <br> 3.797e-08","KEGG:05135 (136) <br> Yersinia infection <br> 4.231e-08","KEGG:05131 (247) <br> Shigellosis <br> 6.336e-08","KEGG:05132 (249) <br> Salmonella infection <br> 6.757e-08","KEGG:04666 (96) <br> Fc gamma R-mediated phagocytosis <br> 3.128e-07","KEGG:05130 (196) <br> Pathogenic Escherichia coli infection <br> 5.510e-07","KEGG:03015 (97) <br> mRNA surveillance pathway <br> 4.407e-03","KEGG:03430 (23) <br> Mismatch repair <br> 1.157e-02"],"key":["KEGG:05100","KEGG:04530","KEGG:04810","KEGG:04144","KEGG:05135","KEGG:05131","KEGG:05132","KEGG:04666","KEGG:05130","KEGG:03015","KEGG:03430"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(221,68,119,1)","opacity":0.8,"size":[15.7545592284552,16.9810383690236,17.3411881917645,17.5569598245829,16.6537786100701,17.5340397903573,17.5455506022041,16.110971780725,17.1997344570363,16.1274657222001,13.574309418746],"symbol":"circle","line":{"width":1.88976377952756,"color":"rgba(221,68,119,1)"}},"hoveron":"points","set":"SharedDatae0541469","name":"KEGG","legendgroup":"KEGG","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","_isNestedKey":false,"frame":null},{"x":[2,49.9528814359943],"y":[-1,-1],"text":"","type":"scatter","mode":"lines","line":{"width":11.3385826771654,"color":"rgba(220,57,18,1)","dash":"solid"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[53.8499055266887,71.2657061880019],"y":[-1,-1],"text":"","type":"scatter","mode":"lines","line":{"width":11.3385826771654,"color":"rgba(16,150,24,1)","dash":"solid"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[75.1627302786963,193.967406707605],"y":[-1,-1],"text":"","type":"scatter","mode":"lines","line":{"width":11.3385826771654,"color":"rgba(255,153,0,1)","dash":"solid"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[197.864430798299,200.003897024091],"y":[-1,-1],"text":"","type":"scatter","mode":"lines","line":{"width":11.3385826771654,"color":"rgba(221,68,119,1)","dash":"solid"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[180],"y":[16.2],"text":"values above this threshold are capped","hovertext":"","textfont":{"size":7.55905511811024,"color":"rgba(190,190,190,1)"},"type":"scatter","mode":"text","hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[0,210],"y":[16,16],"text":"","type":"scatter","mode":"lines","line":{"width":0.755905511811024,"color":"rgba(190,190,190,1)","dash":"dash"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null}],"layout":{"margin":{"t":29.2835201328352,"r":6.6417600664176,"b":51.8931100989696,"l":61.0377750103778},"plot_bgcolor":"rgba(255,255,255,1)","paper_bgcolor":"rgba(255,255,255,1)","font":{"color":"rgba(0,0,0,1)","family":"","size":14.6118721461187},"xaxis":{"domain":[0,1],"automargin":true,"type":"linear","autorange":false,"range":[0,210],"tickmode":"array","ticktext":["GO:MF","GO:CC","GO:BP","KEGG"],"tickvals":[25.9764407179972,62.5578058573453,134.565068493151,198.934163911195],"categoryorder":"array","categoryarray":["GO:MF","GO:CC","GO:BP","KEGG"],"nticks":null,"ticks":"","tickcolor":null,"ticklen":3.65296803652968,"tickwidth":0,"showticklabels":true,"tickfont":{"color":"rgba(77,77,77,1)","family":"","size":10.6268161062682},"tickangle":-45,"showline":true,"linecolor":"rgba(190,190,190,1)","linewidth":0.132835201328352,"showgrid":false,"gridcolor":null,"gridwidth":0,"zeroline":false,"anchor":"y","title":"","hoverformat":".2f"},"yaxis":{"domain":[0,1],"automargin":true,"type":"linear","autorange":false,"range":[-1,18.5],"tickmode":"array","ticktext":["0","2","4","6","8","10","12","14",">16"],"tickvals":[0,2,4,6,8,10,12,14,16],"categoryorder":"array","categoryarray":["0","2","4","6","8","10","12","14",">16"],"nticks":null,"ticks":"outside","tickcolor":"rgba(190,190,190,1)","ticklen":3.65296803652968,"tickwidth":0.66417600664176,"showticklabels":true,"tickfont":{"color":"rgba(77,77,77,1)","family":"","size":11.689497716895},"tickangle":-0,"showline":true,"linecolor":"rgba(190,190,190,1)","linewidth":0.66417600664176,"showgrid":false,"gridcolor":null,"gridwidth":0,"zeroline":false,"anchor":"x","title":"","hoverformat":".2f"},"annotations":[{"text":"-log10(p-adj)","x":-0.0108706932337069,"y":0.5,"showarrow":false,"ax":0,"ay":0,"font":{"color":"rgba(0,0,0,1)","family":"","size":13.2835201328352},"xref":"paper","yref":"paper","textangle":-90,"xanchor":"right","yanchor":"center","annotationType":"axis"},{"text":"query_1","x":0.5,"y":1,"showarrow":false,"ax":0,"ay":0,"font":{"color":"rgba(169,169,169,1)","family":"","size":13.2835201328352},"xref":"paper","yref":"paper","textangle":-0,"xanchor":"center","yanchor":"bottom"}],"shapes":[{"type":"rect","fillcolor":null,"line":{"color":null,"width":0,"linetype":[]},"yref":"paper","xref":"paper","x0":0,"x1":1,"y0":0,"y1":1},{"type":"rect","fillcolor":null,"line":{"color":null,"width":0,"linetype":[]},"yref":"paper","xref":"paper","x0":0,"x1":1,"y0":0,"y1":24.9730178497302,"yanchor":1,"ysizemode":"pixel"}],"showlegend":false,"legend":{"bgcolor":"rgba(255,255,255,1)","bordercolor":"transparent","borderwidth":1.88976377952756,"font":{"color":"rgba(0,0,0,1)","family":"","size":11.689497716895}},"hovermode":"closest","barmode":"relative","dragmode":"zoom"},"config":{"doubleClick":"reset","showSendToCloud":false},"source":"A","attrs":{"2c94537244c6":{"x":{},"y":{},"text":{},"colour":{},"size":{},"alpha":{},"type":"scatter"},"2c946be86162":{"x":{},"xend":{},"y":{},"yend":{}},"2c9430ad48c0":{"x":{},"xend":{},"y":{},"yend":{}},"2c9421906278":{"x":{},"xend":{},"y":{},"yend":{}},"2c946bb0734e":{"x":{},"xend":{},"y":{},"yend":{}},"2c946a9329a9":{"x":{},"y":{}},"2c94670a26c6":{"yintercept":{}}},"cur_data":"2c94537244c6","visdat":{"2c94537244c6":["function (y) ","x"],"2c946be86162":["function (y) ","x"],"2c9430ad48c0":["function (y) ","x"],"2c9421906278":["function (y) ","x"],"2c946bb0734e":["function (y) ","x"],"2c946a9329a9":["function (y) ","x"],"2c94670a26c6":["function (y) ","x"]},"highlight":{"on":"plotly_click","off":"plotly_doubleclick","persistent":false,"dynamic":false,"color":null,"selectize":false,"defaultValues":null,"opacityDim":0.2,"selected":{"opacity":1},"debounce":0,"ctGroups":["SharedDatae0541469"]},"shinyEvents":["plotly_hover","plotly_click","plotly_selected","plotly_relayout","plotly_brushed","plotly_brushing","plotly_clickannotation","plotly_doubleclick","plotly_deselect","plotly_afterplot","plotly_sunburstclick"],"base_url":"https://plot.ly"},"evals":[],"jsHooks":[]}</script>
</div>
<div id="table-results" class="section level4">
<h4>Table results</h4>
<p>The results can also be visualized with a table:</p>
<pre class="r"><code>if(is.null(GO)){
  print(&quot;NOT enough data.&quot;)
}else{
  GO$result %&gt;% 
    as.data.frame() %&gt;% 
    select(starts_with(c(&quot;source&quot;, &quot;term_name&quot;, &quot;p_value&quot;, &quot;term_size&quot;, &quot;intersection_size&quot;))) %&gt;%
    arrange(p_value) %&gt;%  
    mutate(p_value = format(p_value, scientific = T, digits = 2)) %&gt;% 
    DT::datatable(options = list(
    columnDefs = list(list(className = &#39;dt-center&#39;, targets = 1:6)),
    pageLength = 10)) %&gt;%
    formatStyle(&#39;source&#39;, backgroundColor = styleEqual(c(&quot;GO:BP&quot;, &quot;GO:MF&quot;, &quot;GO:CC&quot;, &quot;KEGG&quot;), c(&#39;orange&#39;, &#39;tomato&#39;, &#39;lightgreen&#39;, &#39;violet&#39;)))
}</code></pre>
<div id="htmlwidget-58abce2053a2454929a5" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-58abce2053a2454929a5">{"x":{"filter":"none","data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33","34","35","36","37","38","39","40","41","42","43","44","45","46","47","48","49","50","51","52","53","54","55","56","57","58","59","60","61","62","63","64","65","66","67","68","69","70","71","72","73","74","75","76","77","78","79","80","81","82","83","84","85","86","87","88","89","90","91","92","93","94","95","96","97","98","99","100","101","102","103","104","105","106","107","108","109","110","111","112","113","114","115","116","117","118","119","120","121","122","123","124","125","126","127","128","129","130","131","132","133","134","135","136","137","138","139","140","141","142","143","144","145","146","147","148","149","150","151","152","153","154","155","156","157","158","159","160","161","162","163","164","165","166","167","168","169","170","171","172","173","174","175","176","177","178","179","180","181","182","183","184","185","186","187","188","189","190","191","192","193","194","195","196","197","198","199","200","201","202","203","204","205","206","207","208","209","210","211","212","213","214","215","216","217","218","219","220","221","222","223","224","225","226","227","228","229","230","231","232","233","234","235","236","237","238","239","240","241","242","243","244","245","246","247","248","249","250","251","252","253","254","255","256","257","258","259","260","261","262","263","264","265","266","267","268","269","270","271","272","273","274","275","276","277","278","279","280","281","282","283","284","285","286","287","288","289"],["GO:CC","GO:MF","GO:CC","GO:CC","GO:CC","GO:MF","GO:CC","GO:CC","GO:CC","GO:BP","GO:BP","GO:BP","GO:BP","GO:CC","GO:BP","GO:MF","GO:BP","GO:CC","GO:CC","GO:CC","GO:CC","GO:MF","GO:CC","GO:CC","GO:BP","GO:BP","GO:BP","GO:MF","GO:BP","GO:BP","GO:BP","GO:BP","GO:MF","GO:BP","GO:BP","GO:BP","GO:BP","GO:BP","GO:CC","GO:BP","GO:CC","GO:BP","GO:CC","GO:CC","GO:BP","GO:BP","GO:CC","GO:BP","GO:CC","GO:CC","GO:CC","GO:BP","GO:BP","GO:BP","GO:MF","GO:CC","GO:CC","KEGG","GO:BP","GO:CC","KEGG","GO:BP","GO:CC","GO:CC","GO:MF","GO:BP","GO:MF","GO:CC","KEGG","GO:BP","GO:CC","GO:BP","KEGG","KEGG","GO:CC","GO:CC","KEGG","KEGG","GO:BP","GO:BP","GO:CC","GO:CC","GO:CC","GO:CC","GO:BP","GO:CC","GO:CC","GO:CC","GO:CC","GO:BP","GO:CC","GO:CC","GO:BP","GO:BP","KEGG","GO:CC","GO:CC","GO:CC","KEGG","GO:CC","GO:CC","GO:CC","GO:CC","GO:CC","GO:CC","GO:CC","GO:BP","GO:CC","GO:CC","GO:CC","GO:BP","GO:CC","GO:CC","GO:CC","GO:BP","GO:CC","GO:MF","GO:BP","GO:CC","GO:BP","GO:CC","GO:CC","GO:CC","GO:CC","GO:CC","GO:CC","GO:BP","GO:CC","GO:BP","GO:CC","GO:BP","GO:MF","GO:BP","GO:BP","GO:CC","GO:BP","GO:MF","GO:CC","GO:BP","GO:BP","GO:BP","GO:MF","GO:CC","GO:BP","GO:MF","GO:CC","GO:BP","GO:MF","GO:CC","GO:MF","GO:BP","GO:CC","GO:CC","GO:MF","GO:MF","GO:BP","GO:CC","GO:CC","GO:CC","GO:CC","GO:MF","GO:BP","GO:BP","GO:BP","GO:BP","GO:CC","GO:MF","GO:BP","GO:BP","GO:MF","GO:CC","GO:MF","GO:BP","GO:BP","GO:BP","GO:BP","GO:BP","GO:BP","GO:BP","GO:MF","GO:CC","GO:CC","GO:CC","GO:BP","GO:BP","GO:CC","GO:CC","GO:MF","GO:CC","GO:BP","GO:BP","GO:BP","GO:BP","GO:BP","GO:BP","GO:MF","GO:MF","GO:BP","GO:BP","GO:CC","GO:BP","GO:BP","GO:BP","GO:BP","GO:MF","GO:BP","GO:CC","GO:BP","GO:BP","GO:BP","GO:CC","GO:CC","GO:CC","GO:CC","KEGG","GO:CC","GO:BP","GO:BP","GO:CC","GO:CC","GO:BP","GO:MF","GO:BP","GO:CC","GO:CC","GO:BP","GO:BP","GO:BP","GO:BP","GO:CC","GO:BP","GO:BP","GO:CC","GO:BP","GO:BP","GO:BP","GO:BP","GO:BP","GO:BP","KEGG","GO:BP","GO:BP","GO:BP","GO:BP","GO:CC","GO:BP","GO:BP","GO:BP","GO:BP","GO:BP","GO:BP","GO:MF","GO:BP","GO:MF","GO:BP","GO:BP","GO:BP","GO:BP","GO:BP","GO:BP","GO:CC","GO:BP","GO:BP","GO:BP","GO:BP","GO:BP","GO:BP","GO:BP","GO:BP","GO:BP","GO:BP","GO:BP","GO:BP","GO:BP","GO:BP","GO:BP","GO:BP","GO:MF","GO:BP","GO:MF","GO:MF","GO:CC","GO:CC","GO:BP","GO:BP","GO:BP","GO:CC","GO:CC","GO:BP"],[836,9079,2125,2129,513,465,1568,407,63,2938,7688,7681,9851,870,13080,1955,21858,491,839,549,1026,1692,405,62,2935,8014,8180,7116,3520,8013,22693,7691,1690,2922,3535,8652,9156,8179,311,2374,2716,26029,2130,1444,12466,15281,2739,9169,4067,2715,2127,12640,12219,15728,435,2761,1024,402,15901,2330,297,8836,1259,2126,460,15613,30,3746,345,22121,531,22120,261,411,2057,2146,407,408,21110,7696,479,3229,483,277,9215,1886,3625,495,2124,21908,497,485,2375,26382,321,560,1450,1021,406,3387,183,2123,898,389,1937,3777,9501,3747,3659,1451,25881,512,1526,469,8320,1025,440,15534,980,16061,283,872,3961,895,240,403,8016,1148,18831,616,7698,139,12639,22341,384,8015,6764,1022,7605,7692,4740,1875,3152,8018,396,1437,8842,446,201,6819,8176,2089,833,203,202,25147,3212,1329,871,386,9806,2366,8653,257,15283,3151,5959,12211,5650,11974,350,7161,514,15740,4741,5800,25146,7684,19777,10279,294,4012,597,5673,22988,3858,347,9792,893,23211,10011,2390,25883,8586,2354,2081,9067,5856,15902,4452,3664,23704,15731,9678,11910,11489,2558,15763,9305,4786,89,3608,1445,1438,206,1158,12213,26383,2128,1106,16030,401,12371,317,3778,18625,4583,7602,8116,3383,19392,22343,1149,16744,21484,14514,8273,7956,8174,220,4752,8322,160,17565,3049,2868,2277,2404,19393,30348,17819,1884,11896,437,26938,24088,26936,15282,3065,12524,2918,26030,2396,4908,12718,12114,9863,2929,7773,22793,22795,2294,18629,17272,30346,9299,16745,5402,3598,4874,1876,3655,2178,15533,159,27807,1162,379,4710],["actin cytoskeleton","actin filament binding","non-membrane-bounded organelle","intracellular non-membrane-bounded organelle","Arp2/3 protein complex","actin binding","protein-containing complex","mitochondrial small ribosomal subunit","organellar small ribosomal subunit","actin filament organization","actin cytoskeleton organization","actin filament-based process","Arp2/3 complex-mediated actin nucleation","myosin complex","actin nucleation","cytoskeletal protein binding","supramolecular fiber organization","cytoskeleton","small ribosomal subunit","focal adhesion","cell-substrate junction","structural constituent of cytoskeleton","mitochondrial ribosome","organellar ribosome","cytoskeleton organization","regulation of actin filament polymerization","mRNA 3'-end processing","protein-containing complex binding","regulation of actin polymerization or depolymerization","regulation of actin filament length","regulation of actin filament organization","actin filament polymerization","structural molecule activity","organelle organization","actin polymerization or depolymerization","regulation of protein polymerization","regulation of actin cytoskeleton organization","RNA 3'-end processing","nucleoplasm","mRNA polyadenylation","intracellular organelle lumen","regulation of supramolecular fiber organization","organelle lumen","membrane-enclosed lumen","RNA polyadenylation","regulation of mRNA processing","extracellular exosome","regulation of actin filament-based process","extracellular vesicle","extracellular membrane-bounded organelle","extracellular organelle","regulation of cellular component biogenesis","regulation of protein-containing complex assembly","protein polymerization","RNA binding","anchoring junction","lamellipodium","Bacterial invasion of epithelial cells","regulation of cytoskeleton organization","ribosomal subunit","Tight junction","regulation of cellular component size","cell leading edge","intracellular organelle","cytoskeletal motor activity","regulation of cellular component organization","microfilament motor activity","supramolecular complex","Regulation of actin cytoskeleton","pre-mRNA cleavage required for polyadenylation","brush border","mRNA cleavage involved in mRNA processing","Endocytosis","Yersinia infection","actomyosin","contractile fiber","Shigellosis","Salmonella infection","regulation of anatomical structure size","actin filament-based movement","ribosome","site of DNA damage","mRNA cleavage and polyadenylation specificity factor complex","extracellular space","regulation of organelle organization","site of double-strand break","mitochondrial protein-containing complex","muscle thin filament tropomyosin","membrane-bounded organelle","lamellipodium organization","striated muscle thin filament","mRNA cleavage factor complex","mRNA cleavage","regulation of mRNA metabolic process","Fc gamma R-mediated phagocytosis","cell cortex","nuclear lumen","myofibril","Pathogenic Escherichia coli infection","contractile actin filament bundle","stress fiber","organelle","nuclear body","mitochondrial inner membrane","myofilament","supramolecular fiber","histone H2B ubiquitination","supramolecular polymer","cluster of actin-based cell projections","vesicle","regulation of lamellipodium organization","actin filament","actin filament bundle","cytosol","regulation of mRNA 3'-end processing","cell junction","mRNA binding","barbed-end actin filament capping","organelle inner membrane","actin filament capping","intracellular anatomical structure","unconventional myosin complex","nuclear protein-containing complex","Cdc73/Paf1 complex","extracellular region","mitochondrial matrix","negative regulation of actin filament depolymerization","cortical cytoskeleton","actin polymerization-dependent cell motility","F-actin capping protein complex","vesicle transport along actin filament","RNA polymerase II complex binding","cellular component biogenesis","actin filament-based transport","cytoplasm","regulation of actin filament depolymerization","RNA polymerase core enzyme binding","sarcomere","cellular component assembly","actin filament depolymerization","regulation of lamellipodium assembly","protein binding","box H/ACA scaRNP complex","negative regulation of actin filament polymerization","nucleic acid binding","mitochondrial membrane","mitochondrial translation","structural constituent of ribosome","podosome","ADP binding","snRNA pseudouridine synthesis","cell projection","Cajal body","basal RNA polymerase II transcription machinery binding","basal transcription machinery binding","negative regulation of protein depolymerization","box H/ACA telomerase RNP complex","box H/ACA snoRNP complex","myosin II complex","mitochondrial envelope","RNA polymerase binding","transcription elongation from RNA polymerase II promoter","negative regulation of protein polymerization","telomere maintenance","negative regulation of mRNA processing","box H/ACA RNP complex","box H/ACA snoRNA binding","negative regulation of protein-containing complex disassembly","cellular component organization","heterocyclic compound binding","telomerase holoenzyme complex","cadherin binding","endodermal cell fate commitment","chromosome organization","positive regulation of lamellipodium assembly","meiotic chromosome movement towards spindle pole","regulation of protein depolymerization","lamellipodium assembly","cellular component organization or biogenesis","organic cyclic compound binding","nucleus","catalytic complex","transcription elongation factor complex","mRNA metabolic process","mitochondrial gene expression","plasma membrane bounded cell projection","chromosome","telomerase RNA binding","RNA polymerase II, holoenzyme","non-membrane-bounded organelle assembly","cellular macromolecule biosynthetic process","mRNA processing","positive regulation of lamellipodium organization","telomere organization","DNA-templated transcription, elongation","structural constituent of muscle","cell adhesion molecule binding","histone ubiquitination","negative regulation of cytoskeleton organization","ribonucleoprotein complex","macromolecule biosynthetic process","regulation of mRNA polyadenylation","protein depolymerization","response to cytokine","ATP-dependent activity","snRNA modification","nuclear DNA-directed RNA polymerase complex","chromosome movement towards spindle pole","meiotic cytokinesis","negative regulation of organelle organization","DNA-directed RNA polymerase complex","chromosomal region","envelope","organelle envelope","mRNA surveillance pathway","RNA polymerase complex","regulation of protein-containing complex disassembly","negative regulation of mRNA metabolic process","intracellular membrane-bounded organelle","actin cap","establishment of organelle localization","chromatin binding","regulation of RNA splicing","DNA replication factor A complex","polymeric cytoskeletal fiber","protein-containing complex assembly","histone monoubiquitination","regulation of cell morphogenesis","actomyosin structure organization","myosin II filament","cellular response to cytokine stimulus","vesicle cytoskeletal trafficking","cortical actin cytoskeleton","muscle cell development","RNA phosphodiester bond hydrolysis","regulation of mRNA splicing, via spliceosome","negative regulation of protein-containing complex assembly","cytoskeleton-dependent intracellular transport","rRNA pseudouridine synthesis","Mismatch repair","positive regulation of macromolecule metabolic process","positive regulation of mRNA 3'-end processing","regulation of alternative mRNA splicing, via spliceosome","cell fate commitment involved in formation of primary germ layer","WASH complex","movement of cell or subcellular component","DNA metabolic process","translation","cellular response to interferon-gamma","positive regulation of histone H2B ubiquitination","muscle structure development","ATP binding","muscle cell differentiation","double-stranded RNA binding","positive regulation of protein processing in phagocytic vesicle","protein processing in phagocytic vesicle","regulation of protein processing in phagocytic vesicle","positive regulation of mRNA processing","establishment or maintenance of cell polarity","protein-containing complex organization","U4/U6 snRNP","negative regulation of supramolecular fiber organization","RNA localization","regulation of cell morphogenesis involved in differentiation","cellular nitrogen compound biosynthetic process","peptide biosynthetic process","response to interferon-gamma","telomere maintenance via telomerase","myofibril assembly","regulation of plasma membrane bounded cell projection assembly","positive regulation of plasma membrane bounded cell projection assembly","RNA-dependent DNA biosynthetic process","regulation of biological quality","regulation of cell projection assembly","regulation of histone H2B ubiquitination","positive regulation of histone ubiquitination","striated muscle cell development","adenyl ribonucleotide binding","asymmetric cell division","adenyl nucleotide binding","calmodulin binding","actin-based cell projection","nuclear replisome","actin filament severing","alternative mRNA splicing, via spliceosome","positive regulation of telomerase RNA localization to Cajal body","replisome","sno(s)RNA-containing ribonucleoprotein complex","positive regulation of macromolecule biosynthetic process"],["1.3e-31","3.5e-24","7.8e-24","7.8e-24","5.9e-23","1.5e-22","1.1e-19","4.1e-19","4.1e-19","3.4e-18","1.8e-17","1.8e-17","1.5e-16","2.4e-16","3.0e-15","4.5e-15","6.0e-15","2.2e-14","9.9e-14","1.4e-13","1.8e-13","2.8e-13","4.1e-13","4.1e-13","6.7e-13","8.5e-13","1.6e-12","1.8e-12","2.7e-12","2.9e-12","3.1e-12","4.6e-12","5.4e-12","1.7e-11","2.0e-11","2.8e-11","4.7e-11","5.1e-11","6.8e-11","8.3e-11","8.8e-11","8.8e-11","8.8e-11","8.8e-11","1.0e-10","1.1e-10","1.3e-10","1.4e-10","1.5e-10","1.5e-10","1.5e-10","1.7e-10","2.4e-10","4.4e-10","5.4e-10","6.2e-10","6.9e-10","7.2e-10","2.6e-09","2.7e-09","3.0e-09","4.0e-09","4.3e-09","6.1e-09","8.5e-09","8.6e-09","1.0e-08","1.5e-08","2.2e-08","2.2e-08","2.6e-08","3.5e-08","3.8e-08","4.2e-08","4.9e-08","5.0e-08","6.3e-08","6.8e-08","6.9e-08","7.3e-08","9.4e-08","1.1e-07","1.1e-07","1.2e-07","1.4e-07","1.6e-07","1.7e-07","1.7e-07","1.7e-07","1.8e-07","2.5e-07","2.6e-07","2.7e-07","2.7e-07","3.1e-07","3.2e-07","3.4e-07","4.1e-07","5.5e-07","5.9e-07","5.9e-07","6.2e-07","6.6e-07","6.7e-07","7.5e-07","9.6e-07","1.1e-06","1.1e-06","1.2e-06","1.2e-06","1.3e-06","1.4e-06","1.4e-06","1.7e-06","1.8e-06","2.1e-06","2.6e-06","2.7e-06","2.7e-06","2.7e-06","2.9e-06","3.1e-06","3.7e-06","4.2e-06","4.2e-06","5.0e-06","5.2e-06","6.0e-06","6.9e-06","7.5e-06","7.7e-06","8.2e-06","1.2e-05","1.4e-05","1.5e-05","1.8e-05","1.9e-05","2.4e-05","2.6e-05","3.3e-05","5.4e-05","5.8e-05","6.2e-05","6.3e-05","7.1e-05","7.3e-05","7.4e-05","8.7e-05","8.8e-05","9.1e-05","9.4e-05","1.0e-04","1.1e-04","1.4e-04","1.4e-04","1.5e-04","1.5e-04","1.5e-04","1.5e-04","1.6e-04","1.6e-04","2.3e-04","2.5e-04","2.9e-04","2.9e-04","3.1e-04","3.3e-04","3.5e-04","3.9e-04","4.2e-04","4.6e-04","4.8e-04","5.0e-04","5.0e-04","5.2e-04","5.3e-04","5.6e-04","5.7e-04","6.1e-04","6.4e-04","6.5e-04","6.9e-04","7.1e-04","7.7e-04","7.8e-04","7.9e-04","8.1e-04","8.7e-04","8.8e-04","9.1e-04","9.2e-04","1.0e-03","1.1e-03","1.1e-03","1.2e-03","1.5e-03","1.9e-03","2.1e-03","2.2e-03","2.2e-03","2.3e-03","2.5e-03","3.2e-03","3.2e-03","3.3e-03","3.3e-03","3.6e-03","3.7e-03","3.7e-03","3.7e-03","3.8e-03","4.2e-03","4.2e-03","4.2e-03","4.4e-03","4.6e-03","4.6e-03","5.1e-03","5.2e-03","5.7e-03","6.3e-03","6.4e-03","6.7e-03","6.8e-03","8.0e-03","8.1e-03","8.3e-03","8.4e-03","8.5e-03","9.1e-03","9.2e-03","9.4e-03","9.5e-03","1.0e-02","1.0e-02","1.0e-02","1.1e-02","1.1e-02","1.1e-02","1.2e-02","1.2e-02","1.2e-02","1.2e-02","1.3e-02","1.3e-02","1.4e-02","1.4e-02","1.5e-02","1.5e-02","1.6e-02","1.6e-02","1.6e-02","1.7e-02","1.8e-02","1.8e-02","1.8e-02","1.8e-02","1.8e-02","1.8e-02","1.9e-02","1.9e-02","1.9e-02","1.9e-02","2.0e-02","2.1e-02","2.2e-02","2.4e-02","2.5e-02","2.5e-02","2.6e-02","2.6e-02","2.7e-02","2.7e-02","2.7e-02","2.7e-02","2.7e-02","2.7e-02","2.8e-02","3.0e-02","3.0e-02","3.0e-02","3.2e-02","3.6e-02","3.6e-02","4.0e-02","4.2e-02","4.7e-02","4.8e-02","4.8e-02"],[501,218,6362,6362,11,449,7228,28,28,444,711,800,39,57,55,995,755,2334,76,420,428,104,86,86,1435,140,73,1275,159,160,267,168,980,3805,198,205,350,126,4161,52,6569,373,6570,6570,53,141,2110,391,2132,2133,2133,940,402,278,3142,839,203,77,523,189,169,355,421,15870,115,2246,31,1330,216,13,105,14,251,136,79,242,247,249,488,132,520,97,24,3353,1109,73,277,4,15557,91,22,28,20,318,96,319,5442,232,196,71,71,16674,869,500,26,988,12,996,161,3942,53,114,79,5373,28,2076,350,29,562,44,17758,8,2671,8,4303,483,48,110,7,8,18,36,4801,20,12193,57,41,211,4556,62,41,14799,4,68,5447,745,79,449,30,34,3,2312,102,57,57,77,5,5,20,796,58,61,83,137,30,6,5,87,7793,7416,23,334,15,1034,29,3,93,73,7994,7497,10053,1424,56,2153,111,2205,1795,22,91,373,1564,1919,37,164,80,43,550,48,164,2377,4880,18,120,922,550,7,116,7,7,352,117,359,1245,1245,97,121,127,93,14514,4,409,580,151,13,757,3194,29,305,193,3,828,70,80,174,155,105,147,194,10,23,3378,10,62,32,12,2074,896,713,132,4,616,1502,357,74,2,2,2,32,222,3387,3,166,198,96,4734,738,153,71,63,190,105,72,3741,192,5,5,64,1565,19,1576,203,213,22,17,79,15,24,28,1839],[37,25,84,84,8,30,84,11,11,14,15,32,8,12,8,33,14,37,11,15,15,8,11,11,15,9,12,33,9,9,10,9,24,55,9,9,10,13,55,10,70,10,70,70,10,13,15,10,15,15,15,18,13,9,48,10,8,7,10,11,8,9,15,106,10,23,7,28,8,5,9,5,9,7,9,13,8,8,9,10,14,9,5,15,11,5,11,4,95,6,6,5,5,14,6,12,57,12,7,8,8,98,20,13,6,22,5,22,9,15,5,9,8,17,6,12,13,5,13,7,108,4,34,4,15,12,7,5,3,3,5,6,26,5,65,7,6,10,25,7,4,52,3,7,51,13,7,11,4,5,3,11,7,6,6,7,3,3,4,13,6,6,7,8,5,3,3,7,18,60,4,11,4,20,3,2,7,4,18,60,39,20,5,28,7,10,15,4,6,12,23,26,3,8,6,5,13,5,8,25,43,3,7,11,11,3,6,2,2,11,6,10,14,14,5,6,7,6,85,2,10,13,7,3,14,18,4,5,8,2,10,5,5,3,5,6,5,7,3,4,34,3,5,4,2,21,15,13,3,2,13,17,10,4,2,2,2,4,7,18,2,5,7,3,40,13,3,5,5,4,3,5,12,4,2,2,5,17,2,17,7,6,3,3,5,3,3,3,22]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>source<\/th>\n      <th>source_order<\/th>\n      <th>term_name<\/th>\n      <th>p_value<\/th>\n      <th>term_size<\/th>\n      <th>intersection_size<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-center","targets":[1,2,3,4,5,6]},{"orderable":false,"targets":0}],"pageLength":10,"order":[],"autoWidth":false,"orderClasses":false,"rowCallback":"function(row, data, displayNum, displayIndex, dataIndex) {\nvar value=data[1]; $(this.api().cell(row, 1).node()).css({'background-color':value == \"GO:BP\" ? \"orange\" : value == \"GO:MF\" ? \"tomato\" : value == \"GO:CC\" ? \"lightgreen\" : value == \"KEGG\" ? \"violet\" : null});\n}"}},"evals":["options.rowCallback"],"jsHooks":[]}</script>
</div>
<div id="treemap" class="section level4 tabset tabset-fade">
<h4>Treemap</h4>
<div id="bp" class="section level5">
<h5>BP</h5>
<pre class="r"><code>if(sum(GO$result$source == &quot;GO:BP&quot;) &gt;= 5 &amp; !is.null(GO)){
  simMatrix &lt;- GO$result %&gt;% 
  as.data.frame() %&gt;%
  filter(source == &quot;GO:BP&quot;) %&gt;% 
  pull(term_id) %&gt;% 
  calculateSimMatrix(orgdb=&quot;org.Hs.eg.db&quot;, ont=&quot;BP&quot;, method=&quot;Rel&quot;)

scores &lt;- setNames(-log10(GO$result$p_value), GO$result$term_id)

reducedTerms &lt;- reduceSimMatrix(simMatrix,
                                scores,
                                threshold=0.7,
                                orgdb=&quot;org.Hs.eg.db&quot;)

treemapPlot(reducedTerms)
BP &lt;- T
}else{
  print(&quot;NOT enough data.&quot;)
  BP &lt;- F
}</code></pre>
<p><img src="figure/Template_LFQ.Rmd/BP-cond1-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="mf" class="section level5">
<h5>MF</h5>
<pre class="r"><code>if(sum(GO$result$source == &quot;GO:MF&quot;) &gt;= 5 &amp; !is.null(GO)){
simMatrix &lt;- GO$result %&gt;% 
  as.data.frame() %&gt;%
  filter(source == &quot;GO:MF&quot;) %&gt;% 
  pull(term_id) %&gt;% 
  calculateSimMatrix(orgdb=&quot;org.Hs.eg.db&quot;, ont=&quot;MF&quot;, method=&quot;Rel&quot;)

scores &lt;- setNames(-log10(GO$result$p_value), GO$result$term_id)

reducedTerms &lt;- reduceSimMatrix(simMatrix,
                                scores,
                                threshold=0.7,
                                orgdb=&quot;org.Hs.eg.db&quot;)

treemapPlot(reducedTerms)
MF &lt;- T
}else{
  print(&quot;NOT enough data.&quot;)
  MF &lt;- F
}</code></pre>
<p><img src="figure/Template_LFQ.Rmd/MF-cond1-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="cc" class="section level5">
<h5>CC</h5>
<pre class="r"><code>if(sum(GO$result$source == &quot;GO:CC&quot;) &gt;= 5 &amp; !is.null(GO)){
simMatrix &lt;- GO$result %&gt;% 
  as.data.frame() %&gt;%
  filter(source == &quot;GO:CC&quot;) %&gt;% 
  pull(term_id) %&gt;% 
  calculateSimMatrix(orgdb=&quot;org.Hs.eg.db&quot;, ont=&quot;CC&quot;, method=&quot;Rel&quot;)

scores &lt;- setNames(-log10(GO$result$p_value), GO$result$term_id)

reducedTerms &lt;- reduceSimMatrix(simMatrix,
                                scores,
                                threshold=0.7,
                                orgdb=&quot;org.Hs.eg.db&quot;)

treemapPlot(reducedTerms)
CC &lt;- T
}else{
  print(&quot;NOT enough data.&quot;)
  CC &lt;- F
}</code></pre>
<p><img src="figure/Template_LFQ.Rmd/CC-cond1-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="scatterplot" class="section level4">
<h4>ScatterPlot</h4>
<pre class="r"><code>if(BP | MF | CC){
scatterPlot(simMatrix, reducedTerms, size = &quot;score&quot;, labelSize = 5)
}else{
  print(&quot;NOT enough data.&quot;)
}</code></pre>
<p><img src="figure/Template_LFQ.Rmd/scatter-cond1-1.png" width="1440" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="wt_bio-1" class="section level3 tabset tabset-fade">
<h3><strong>5.2. WT_BIO</strong></h3>
<div id="gostplot-1" class="section level4">
<h4>Gostplot</h4>
<p>UP regulated proteins in WT_BIO were used to generate these gene ontologies.</p>
<pre class="r"><code>Gname &lt;- results %&gt;% filter(significant &amp; get(paste0(test, &quot;_ratio&quot;)) &lt; 0) %&gt;% pull(name)

if(!params$Imputation){
  Uni &lt;- unique_pair(data_filt_unique, conditions = conditions, table = T) %&gt;% 
  filter(!!sym(paste0(&quot;unique_&quot;, cond_2)) == &quot;+&quot;) %&gt;% 
  pull(name) %&gt;% unique()

  Gname &lt;- c(Gname, Uni)
}

if(length(Gname) == 0){
  print(&quot;NOT enough data.&quot;)
  GO &lt;- NULL
}else{
  GO &lt;- gost(query = Gname, organism = &quot;hsapiens&quot;, sources = c(&quot;GO:BP&quot;, &quot;GO:MF&quot;, &quot;GO:CC&quot;, &quot;KEGG&quot;), ordered_query = T, evcodes = TRUE)
  if(is.null(GO)){
    print(&quot;NOT enough data.&quot;)
  }else{
    gostplot(GO, capped = TRUE, interactive = TRUE)
  }
}</code></pre>
<div id="htmlwidget-6e939b13f92b78564578" style="width:960px;height:768px;" class="plotly html-widget"></div>
<script type="application/json" data-for="htmlwidget-6e939b13f92b78564578">{"x":{"data":[{"x":[65.2707416810414,55.1595986829558,63.384159872609,63.3802619763106,55.1479049940605,54.4345899714508,65.2668437847429,69.4843675796267,65.2434564069525],"y":[4.66164441320706,3.83015819717275,3.33983142173426,3.03421636901412,2.55197436670909,2.16661022864313,2.13962532912372,1.8580721955122,1.35151185070269],"text":["GO:0071014 (13) <br> post-mRNA release spliceosomal complex <br> 2.179e-05","GO:0005684 (93) <br> U2-type spliceosomal complex <br> 1.479e-04","GO:0045240 (10) <br> dihydrolipoyl dehydrogenase complex <br> 4.573e-04","GO:0045239 (14) <br> tricarboxylic acid cycle enzyme complex <br> 9.242e-04","GO:0005681 (195) <br> spliceosomal complex <br> 2.806e-03","GO:0000974 (12) <br> Prp19 complex <br> 6.814e-03","GO:0071013 (85) <br> catalytic step 2 spliceosome <br> 7.251e-03","GO:1902494 (1424) <br> catalytic complex <br> 1.387e-02","GO:0071007 (30) <br> U2-type catalytic step 2 spliceosome <br> 4.451e-02"],"key":["GO:0071014","GO:0005684","GO:0045240","GO:0045239","GO:0005681","GO:0000974","GO:0071013","GO:1902494","GO:0071007"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(16,150,24,1)","opacity":0.8,"size":[15.01113635466,18.7100926035205,14.4212129468264,15.1722360019527,19.8834414017618,14.83449861959,18.561203681971,22.6771653543307,16.7130996729143],"symbol":"circle","line":{"width":1.88976377952756,"color":"rgba(16,150,24,1)"}},"hoveron":"points","set":"SharedData8cfb95dc","name":"GO:CC","legendgroup":"GO:CC","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","_isNestedKey":false,"frame":null},{"x":[16.018734925656,17.043735561032,6.12728392723651,6.11948924559867,5.76093389025801,5.75703654943909,19.584801774968,16.5721573219427,15.8277652255289,42.3335801350053],"y":[4.04163922082856,3.75858662739019,2.68914603172084,2.54320468741982,1.88211156268931,1.88211156268931,1.58108156702533,1.40499030797029,1.3723597695853,1.34795359915564],"text":["GO:0016620 (38) <br> oxidoreductase activity, acting on the aldehyde or oxo group of donors, NAD or NADP as acceptor <br> 9.086e-05","GO:0016903 (47) <br> oxidoreductase activity, acting on the aldehyde or oxo group of donors <br> 1.743e-04","GO:0004459 (6) <br> L-lactate dehydrogenase activity <br> 2.046e-03","GO:0004457 (7) <br> lactate dehydrogenase activity <br> 2.863e-03","GO:0004350 (1) <br> glutamate-5-semialdehyde dehydrogenase activity <br> 1.312e-02","GO:0004349 (1) <br> glutamate 5-kinase activity <br> 1.312e-02","GO:0019202 (2) <br> amino acid kinase activity <br> 2.624e-02","GO:0016774 (3) <br> phosphotransferase activity, carboxyl group as acceptor <br> 3.936e-02","GO:0016491 (756) <br> oxidoreductase activity <br> 4.243e-02","GO:0097718 (33) <br> disordered domain specific binding <br> 4.488e-02"],"key":["GO:0016620","GO:0016903","GO:0004459","GO:0004457","GO:0004350","GO:0004349","GO:0019202","GO:0016774","GO:0016491","GO:0097718"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(220,57,18,1)","opacity":0.8,"size":[17.1550025998698,17.5402508061686,13.1668649046861,13.562345300191,3.77952755905512,3.77952755905512,9.61821599984264,11.1301615270943,21.8344312577662,16.8930633760062],"symbol":"circle","line":{"width":1.88976377952756,"color":"rgba(220,57,18,1)"}},"hoveron":"points","set":"SharedData8cfb95dc","name":"GO:MF","legendgroup":"GO:MF","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","_isNestedKey":false,"frame":null},{"x":[198.282173291292,197.868334933748,198.68429924249,198.243131936806,197.981554861755],"y":[4.76626449541645,3.45433554201663,3.08637602655062,2.29577571985055,1.49385073286675],"text":["KEGG:00640 (32) <br> Propanoate metabolism <br> 1.713e-05","KEGG:00010 (67) <br> Glycolysis / Gluconeogenesis <br> 3.513e-04","KEGG:03040 (146) <br> Spliceosome <br> 8.196e-04","KEGG:00620 (47) <br> Pyruvate metabolism <br> 5.061e-03","KEGG:00280 (48) <br> Valine, leucine and isoleucine degradation <br> 3.207e-02"],"key":["KEGG:00640","KEGG:00010","KEGG:03040","KEGG:00620","KEGG:00280"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(221,68,119,1)","opacity":0.8,"size":[16.8352318120983,18.1598855322838,19.4352981193742,17.5402508061686,17.5778227641315],"symbol":"circle","line":{"width":1.88976377952756,"color":"rgba(221,68,119,1)"}},"hoveron":"points","set":"SharedData8cfb95dc","name":"KEGG","legendgroup":"KEGG","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","_isNestedKey":false,"frame":null},{"x":[2,49.9528814359943],"y":[-1,-1],"text":"","type":"scatter","mode":"lines","line":{"width":11.3385826771654,"color":"rgba(220,57,18,1)","dash":"solid"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[53.8499055266887,71.2657061880019],"y":[-1,-1],"text":"","type":"scatter","mode":"lines","line":{"width":11.3385826771654,"color":"rgba(16,150,24,1)","dash":"solid"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[75.1627302786963,193.967406707605],"y":[-1,-1],"text":"","type":"scatter","mode":"lines","line":{"width":11.3385826771654,"color":"rgba(255,153,0,1)","dash":"solid"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[197.864430798299,200.003897024091],"y":[-1,-1],"text":"","type":"scatter","mode":"lines","line":{"width":11.3385826771654,"color":"rgba(221,68,119,1)","dash":"solid"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[180],"y":[16.2],"text":"values above this threshold are capped","hovertext":"","textfont":{"size":7.55905511811024,"color":"rgba(190,190,190,1)"},"type":"scatter","mode":"text","hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[0,210],"y":[16,16],"text":"","type":"scatter","mode":"lines","line":{"width":0.755905511811024,"color":"rgba(190,190,190,1)","dash":"dash"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null}],"layout":{"margin":{"t":29.2835201328352,"r":6.6417600664176,"b":51.8931100989696,"l":61.0377750103778},"plot_bgcolor":"rgba(255,255,255,1)","paper_bgcolor":"rgba(255,255,255,1)","font":{"color":"rgba(0,0,0,1)","family":"","size":14.6118721461187},"xaxis":{"domain":[0,1],"automargin":true,"type":"linear","autorange":false,"range":[0,210],"tickmode":"array","ticktext":["GO:MF","GO:CC","GO:BP","KEGG"],"tickvals":[25.9764407179972,62.5578058573453,134.565068493151,198.934163911195],"categoryorder":"array","categoryarray":["GO:MF","GO:CC","GO:BP","KEGG"],"nticks":null,"ticks":"","tickcolor":null,"ticklen":3.65296803652968,"tickwidth":0,"showticklabels":true,"tickfont":{"color":"rgba(77,77,77,1)","family":"","size":10.6268161062682},"tickangle":-45,"showline":true,"linecolor":"rgba(190,190,190,1)","linewidth":0.132835201328352,"showgrid":false,"gridcolor":null,"gridwidth":0,"zeroline":false,"anchor":"y","title":"","hoverformat":".2f"},"yaxis":{"domain":[0,1],"automargin":true,"type":"linear","autorange":false,"range":[-1,18.5],"tickmode":"array","ticktext":["0","2","4","6","8","10","12","14",">16"],"tickvals":[0,2,4,6,8,10,12,14,16],"categoryorder":"array","categoryarray":["0","2","4","6","8","10","12","14",">16"],"nticks":null,"ticks":"outside","tickcolor":"rgba(190,190,190,1)","ticklen":3.65296803652968,"tickwidth":0.66417600664176,"showticklabels":true,"tickfont":{"color":"rgba(77,77,77,1)","family":"","size":11.689497716895},"tickangle":-0,"showline":true,"linecolor":"rgba(190,190,190,1)","linewidth":0.66417600664176,"showgrid":false,"gridcolor":null,"gridwidth":0,"zeroline":false,"anchor":"x","title":"","hoverformat":".2f"},"annotations":[{"text":"-log10(p-adj)","x":-0.0108706932337069,"y":0.5,"showarrow":false,"ax":0,"ay":0,"font":{"color":"rgba(0,0,0,1)","family":"","size":13.2835201328352},"xref":"paper","yref":"paper","textangle":-90,"xanchor":"right","yanchor":"center","annotationType":"axis"},{"text":"query_1","x":0.5,"y":1,"showarrow":false,"ax":0,"ay":0,"font":{"color":"rgba(169,169,169,1)","family":"","size":13.2835201328352},"xref":"paper","yref":"paper","textangle":-0,"xanchor":"center","yanchor":"bottom"}],"shapes":[{"type":"rect","fillcolor":null,"line":{"color":null,"width":0,"linetype":[]},"yref":"paper","xref":"paper","x0":0,"x1":1,"y0":0,"y1":1},{"type":"rect","fillcolor":null,"line":{"color":null,"width":0,"linetype":[]},"yref":"paper","xref":"paper","x0":0,"x1":1,"y0":0,"y1":24.9730178497302,"yanchor":1,"ysizemode":"pixel"}],"showlegend":false,"legend":{"bgcolor":"rgba(255,255,255,1)","bordercolor":"transparent","borderwidth":1.88976377952756,"font":{"color":"rgba(0,0,0,1)","family":"","size":11.689497716895}},"hovermode":"closest","barmode":"relative","dragmode":"zoom"},"config":{"doubleClick":"reset","showSendToCloud":false},"source":"A","attrs":{"2c942eef37a6":{"x":{},"y":{},"text":{},"colour":{},"size":{},"alpha":{},"type":"scatter"},"2c94c831d57":{"x":{},"xend":{},"y":{},"yend":{}},"2c947fe02a45":{"x":{},"xend":{},"y":{},"yend":{}},"2c94114d6057":{"x":{},"xend":{},"y":{},"yend":{}},"2c945ec364d0":{"x":{},"xend":{},"y":{},"yend":{}},"2c9464c5c63":{"x":{},"y":{}},"2c94481030a4":{"yintercept":{}}},"cur_data":"2c942eef37a6","visdat":{"2c942eef37a6":["function (y) ","x"],"2c94c831d57":["function (y) ","x"],"2c947fe02a45":["function (y) ","x"],"2c94114d6057":["function (y) ","x"],"2c945ec364d0":["function (y) ","x"],"2c9464c5c63":["function (y) ","x"],"2c94481030a4":["function (y) ","x"]},"highlight":{"on":"plotly_click","off":"plotly_doubleclick","persistent":false,"dynamic":false,"color":null,"selectize":false,"defaultValues":null,"opacityDim":0.2,"selected":{"opacity":1},"debounce":0,"ctGroups":["SharedData8cfb95dc"]},"shinyEvents":["plotly_hover","plotly_click","plotly_selected","plotly_relayout","plotly_brushed","plotly_brushing","plotly_clickannotation","plotly_doubleclick","plotly_deselect","plotly_afterplot","plotly_sunburstclick"],"base_url":"https://plot.ly"},"evals":[],"jsHooks":[]}</script>
</div>
<div id="table-results-1" class="section level4">
<h4>Table results</h4>
<p>The results can also be visualized with a table:</p>
<pre class="r"><code>if(is.null(GO)){
  print(&quot;NOT enough data.&quot;)
}else{
  GO$result %&gt;% 
    as.data.frame() %&gt;% 
    select(starts_with(c(&quot;source&quot;, &quot;term_name&quot;, &quot;p_value&quot;, &quot;term_size&quot;, &quot;intersection_size&quot;))) %&gt;% 
    arrange(p_value) %&gt;%  
    mutate(p_value = format(p_value, scientific = T, digits = 2)) %&gt;% 
    DT::datatable(options = list(
    columnDefs = list(list(className = &#39;dt-center&#39;, targets = 1:6)),
    pageLength = 10)) %&gt;%
    formatStyle(&#39;source&#39;, backgroundColor = styleEqual(c(&quot;GO:BP&quot;, &quot;GO:MF&quot;, &quot;GO:CC&quot;, &quot;KEGG&quot;), c(&#39;orange&#39;, &#39;tomato&#39;, &#39;lightgreen&#39;, &#39;violet&#39;)))
}</code></pre>
<div id="htmlwidget-27ab1876f7c57926b229" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-27ab1876f7c57926b229">{"x":{"filter":"none","data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24"],["KEGG","GO:CC","GO:MF","GO:CC","GO:MF","KEGG","GO:CC","KEGG","GO:CC","GO:MF","GO:CC","GO:MF","KEGG","GO:CC","GO:CC","GO:MF","GO:MF","GO:CC","GO:MF","KEGG","GO:MF","GO:MF","GO:CC","GO:MF"],[108,2931,3598,337,3861,2,2447,211,2446,1060,334,1058,98,151,2930,966,965,4012,4513,31,3740,3549,2924,10350],["Propanoate metabolism","post-mRNA release spliceosomal complex","oxidoreductase activity, acting on the aldehyde or oxo group of donors, NAD or NADP as acceptor","U2-type spliceosomal complex","oxidoreductase activity, acting on the aldehyde or oxo group of donors","Glycolysis / Gluconeogenesis","dihydrolipoyl dehydrogenase complex","Spliceosome","tricarboxylic acid cycle enzyme complex","L-lactate dehydrogenase activity","spliceosomal complex","lactate dehydrogenase activity","Pyruvate metabolism","Prp19 complex","catalytic step 2 spliceosome","glutamate-5-semialdehyde dehydrogenase activity","glutamate 5-kinase activity","catalytic complex","amino acid kinase activity","Valine, leucine and isoleucine degradation","phosphotransferase activity, carboxyl group as acceptor","oxidoreductase activity","U2-type catalytic step 2 spliceosome","disordered domain specific binding"],["1.7e-05","2.2e-05","9.1e-05","1.5e-04","1.7e-04","3.5e-04","4.6e-04","8.2e-04","9.2e-04","2.0e-03","2.8e-03","2.9e-03","5.1e-03","6.8e-03","7.3e-03","1.3e-02","1.3e-02","1.4e-02","2.6e-02","3.2e-02","3.9e-02","4.2e-02","4.5e-02","4.5e-02"],[32,13,38,93,47,67,10,146,14,6,195,7,47,12,85,1,1,1424,2,48,3,756,30,33],[4,3,3,4,3,4,2,5,2,2,4,2,3,2,3,1,1,7,1,2,1,5,2,2]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>source<\/th>\n      <th>source_order<\/th>\n      <th>term_name<\/th>\n      <th>p_value<\/th>\n      <th>term_size<\/th>\n      <th>intersection_size<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-center","targets":[1,2,3,4,5,6]},{"orderable":false,"targets":0}],"pageLength":10,"order":[],"autoWidth":false,"orderClasses":false,"rowCallback":"function(row, data, displayNum, displayIndex, dataIndex) {\nvar value=data[1]; $(this.api().cell(row, 1).node()).css({'background-color':value == \"GO:BP\" ? \"orange\" : value == \"GO:MF\" ? \"tomato\" : value == \"GO:CC\" ? \"lightgreen\" : value == \"KEGG\" ? \"violet\" : null});\n}"}},"evals":["options.rowCallback"],"jsHooks":[]}</script>
</div>
<div id="treemap-1" class="section level4 tabset tabset-fade">
<h4>Treemap</h4>
<div id="bp-1" class="section level5">
<h5>BP</h5>
<pre class="r"><code>if(sum(GO$result$source == &quot;GO:BP&quot;) &gt;= 5 &amp; !is.null(GO)){
  simMatrix &lt;- GO$result %&gt;% 
  as.data.frame() %&gt;%
  filter(source == &quot;GO:BP&quot;) %&gt;% 
  pull(term_id) %&gt;% 
  calculateSimMatrix(orgdb=&quot;org.Hs.eg.db&quot;, ont=&quot;BP&quot;, method=&quot;Rel&quot;)

scores &lt;- setNames(-log10(GO$result$p_value), GO$result$term_id)

reducedTerms &lt;- reduceSimMatrix(simMatrix,
                                scores,
                                threshold=0.7,
                                orgdb=&quot;org.Hs.eg.db&quot;)

treemapPlot(reducedTerms)
BP &lt;- T
}else{
  print(&quot;NOT enough data.&quot;)
  BP &lt;- F
}</code></pre>
<pre><code>[1] &quot;NOT enough data.&quot;</code></pre>
</div>
<div id="mf-1" class="section level5">
<h5>MF</h5>
<pre class="r"><code>if(sum(GO$result$source == &quot;GO:MF&quot;) &gt;= 5 &amp; !is.null(GO)){
simMatrix &lt;- GO$result %&gt;% 
  as.data.frame() %&gt;%
  filter(source == &quot;GO:MF&quot;) %&gt;% 
  pull(term_id) %&gt;% 
  calculateSimMatrix(orgdb=&quot;org.Hs.eg.db&quot;, ont=&quot;MF&quot;, method=&quot;Rel&quot;)

scores &lt;- setNames(-log10(GO$result$p_value), GO$result$term_id)

reducedTerms &lt;- reduceSimMatrix(simMatrix,
                                scores,
                                threshold=0.7,
                                orgdb=&quot;org.Hs.eg.db&quot;)

treemapPlot(reducedTerms)
MF &lt;- T
}else{
  print(&quot;NOT enough data.&quot;)
  MF &lt;- F
}</code></pre>
<p><img src="figure/Template_LFQ.Rmd/MF-cond2-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="cc-1" class="section level5">
<h5>CC</h5>
<pre class="r"><code>if(sum(GO$result$source == &quot;GO:CC&quot;) &gt;= 5 &amp; !is.null(GO)){
simMatrix &lt;- GO$result %&gt;% 
  as.data.frame() %&gt;%
  filter(source == &quot;GO:CC&quot;) %&gt;% 
  pull(term_id) %&gt;% 
  calculateSimMatrix(orgdb=&quot;org.Hs.eg.db&quot;, ont=&quot;CC&quot;, method=&quot;Rel&quot;)

scores &lt;- setNames(-log10(GO$result$p_value), GO$result$term_id)

reducedTerms &lt;- reduceSimMatrix(simMatrix,
                                scores,
                                threshold=0.7,
                                orgdb=&quot;org.Hs.eg.db&quot;)

treemapPlot(reducedTerms)
CC &lt;- T
}else{
  print(&quot;NOT enough data.&quot;)
  CC &lt;- F
}</code></pre>
<p><img src="figure/Template_LFQ.Rmd/CC-cond2-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="scatterplot-1" class="section level4">
<h4>ScatterPlot</h4>
<pre class="r"><code>if(BP | MF | CC){
scatterPlot(simMatrix, reducedTerms, size = &quot;score&quot;, labelSize = 5)
}else{
  print(&quot;NOT enough data.&quot;)
}</code></pre>
<p><img src="figure/Template_LFQ.Rmd/scatter-cond2-1.png" width="1440" style="display: block; margin: auto;" /></p>
</div>
</div>
</div>
<div id="network-analysis" class="section level2">
<h2><strong>6. Network analysis</strong></h2>
<p>Overview from <strong>STRING Database</strong>: In STRING, each protein-protein interaction is annotated with one or more ‘scores’. Importantly, these scores do <strong>not</strong> indicate the strength or the specificity of the interaction. Instead, they are indicators of <strong>confidence</strong>, i.e. how likely STRING judges an interaction to be true, given the available evidence. All scores rank from 0 to 1, with 1 being the highest possible confidence. A score of 0.5 would indicate that roughly every second interaction might be erroneous (i.e., a false positive).</p>
<pre class="r"><code># if(leng_dep != 0){
#   
#   string_db &lt;- STRINGdb$new(version=&quot;11&quot;, species=9606, score_threshold=400)
# 
#   string_mapped &lt;- string_db$map(string_res, &quot;Gene.name&quot;, removeUnmappedRows = TRUE, quiet = T)
# 
#   hits &lt;- string_mapped$STRING_id[1:leng_dep]
# 
#   string_db$plot_network(hits, add_link = T)
# }else{
#   print(&quot;NOT enough data.&quot;)
# }</code></pre>
<div id="cytoscape-network" class="section level3">
<h3><strong>6.1. Cytoscape Network</strong></h3>
<p>Cytoscape is an open source software platform for visualizing complex networks and integrating these with any type of attribute data. In this presentation all proteins showing at least one confidence level from the STRING database are shown. The <strong>color</strong> code is relative to <strong>fold change</strong> and the <strong>size</strong> is directly proportional to the <strong>significance (p-value)</strong>.</p>
<pre class="r"><code>print(&quot;Enable Cytoscape option to see this results.&quot;)</code></pre>
<pre><code>[1] &quot;Enable Cytoscape option to see this results.&quot;</code></pre>
<pre class="r"><code>if(n_edge != 0){
  knitr::include_graphics(&quot;assets/network_FC_pvalue.png&quot;,  error = F)
}else{
  print(&quot;NOT enough data.&quot;)
}</code></pre>
</div>
<div id="cytoscape-enrichmentmap" class="section level3">
<h3><strong>6.2. Cytoscape EnrichmentMap</strong></h3>
<p>An <strong>enrichment map</strong> is a different sort of network. Instead of nodes representing genes, nodes represent pathways or functions. Edges between these pathways or functions represent shared genes or pathway crosstalk. An enrichment map is a way to visualize your enrichment results to help reduce redundancy and uncover main themes.</p>
<pre class="r"><code>print(&quot;Enable Cytoscape option to see this results.&quot;)</code></pre>
<pre><code>[1] &quot;Enable Cytoscape option to see this results.&quot;</code></pre>
<pre class="r"><code>Gname &lt;- results %&gt;% filter(significant) %&gt;% pull(name)

if(!params$Imputation){
  
  Uni &lt;- unique_pair(data_filt_unique, conditions = conditions, table = T) %&gt;% 
  pull(name) %&gt;% 
  unique()

  Gname &lt;- c(Gname, Uni)
}

if(length(Gname) == 0){
  
  print(&quot;NOT enough data.&quot;)
  
}else{
  
  GO &lt;- gost(query = Gname, organism = &quot;hsapiens&quot;, sources = c(&quot;GO:BP&quot;, &quot;GO:MF&quot;, &quot;GO:CC&quot;, &quot;KEGG&quot;), ordered_query = T, evcodes = TRUE)
  
  if(is.null(GO)){
    
    print(&quot;NOT enough data.&quot;)

    enrich_map = F
    
  }else{
    
    enrich_map = T
    
    gem &lt;- GO$result[,c(&quot;term_id&quot;, &quot;term_name&quot;, &quot;p_value&quot;, &quot;intersection&quot;)]
    colnames(gem) &lt;- c(&quot;GO.ID&quot;, &quot;Description&quot;, &quot;p.Val&quot;, &quot;Genes&quot;)
    gem$FDR &lt;- gem$p.Val
    gem$Phenotype = &quot;+1&quot;
    gem &lt;- gem[,c(&quot;GO.ID&quot;, &quot;Description&quot;, &quot;p.Val&quot;, &quot;FDR&quot;, &quot;Phenotype&quot;, &quot;Genes&quot;)]
    head(gem)
  
    write.table(gem, here::here(&quot;output/enrichmentmap.txt&quot;),col.name=TRUE,sep=&quot;\t&quot;,row.names=FALSE,quote=FALSE)
    
  }
}</code></pre>
<pre class="r"><code>if(enrich_map){
  knitr::include_graphics(&quot;assets/network_enrichmentmap.png&quot;,  error = F)
}else{
  print(&quot;NOT enough data.&quot;)
}</code></pre>
</div>
</div>
<div id="section" class="section level2 unnumbered">
<h2></h2>
<pre class="r"><code>saveWorkbook(excel, here::here(paste0(&quot;output/results_table_&quot;, test, &quot;.xlsx&quot;)), overwrite = T)</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.0.3 (2020-10-10)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 19042)

Matrix products: default

locale:
[1] LC_COLLATE=Italian_Italy.1252  LC_CTYPE=Italian_Italy.1252   
[3] LC_MONETARY=Italian_Italy.1252 LC_NUMERIC=C                  
[5] LC_TIME=Italian_Italy.1252    

attached base packages:
[1] parallel  stats4    stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] forcats_0.5.1               stringr_1.4.0              
 [3] dplyr_1.0.7                 purrr_0.3.4                
 [5] readr_1.4.0                 tidyr_1.1.3                
 [7] tibble_3.1.2                tidyverse_1.3.1            
 [9] openxlsx_4.2.4              here_1.0.1                 
[11] RColorBrewer_1.1-2          RCy3_2.10.2                
[13] STRINGdb_2.2.2              naniar_0.6.1               
[15] visdat_0.5.3                ggrepel_0.9.1              
[17] ggplot2_3.3.5               UpSetR_1.4.0               
[19] ggstatsplot_0.8.0           datapasta_3.1.0            
[21] patchwork_1.1.1             DT_0.18                    
[23] rrvgo_1.2.0                 org.Hs.eg.db_3.12.0        
[25] AnnotationDbi_1.52.0        gprofiler2_0.2.0           
[27] enrichR_3.0                 preprocessCore_1.52.1      
[29] SummarizedExperiment_1.20.0 Biobase_2.50.0             
[31] GenomicRanges_1.42.0        GenomeInfoDb_1.26.2        
[33] IRanges_2.24.1              S4Vectors_0.28.1           
[35] BiocGenerics_0.36.0         MatrixGenerics_1.2.1       
[37] matrixStats_0.59.0          DEP_1.12.0                 
[39] MBQN_2.2.0                  workflowr_1.6.2            

loaded via a namespace (and not attached):
  [1] rappdirs_0.3.3            R.methodsS3_1.8.1        
  [3] coda_0.19-4               bit64_4.0.5              
  [5] knitr_1.33                DelayedArray_0.16.2      
  [7] R.utils_2.10.1            data.table_1.14.0        
  [9] RCurl_1.98-1.3            doParallel_1.0.16        
 [11] generics_0.1.0            RSQLite_2.2.7            
 [13] correlation_0.6.1         chron_2.3-56             
 [15] bit_4.0.4                 lubridate_1.7.10         
 [17] xml2_1.3.2                httpuv_1.6.1             
 [19] assertthat_0.2.1          WRS2_1.1-2               
 [21] xfun_0.24                 hms_1.1.0                
 [23] jquerylib_0.1.4           evaluate_0.14            
 [25] promises_1.2.0.1          fansi_0.5.0              
 [27] readxl_1.3.1              caTools_1.18.2           
 [29] dbplyr_2.1.1              igraph_1.2.6             
 [31] DBI_1.1.1                 htmlwidgets_1.5.3        
 [33] reshape_0.8.8             kSamples_1.2-9           
 [35] Rmpfr_0.8-4               paletteer_1.3.0          
 [37] hash_2.2.6.1              ellipsis_0.3.2           
 [39] crosstalk_1.1.1           backports_1.2.1          
 [41] insight_0.14.2            gridBase_0.4-7           
 [43] vctrs_0.3.8               imputeLCMD_2.0           
 [45] Cairo_1.5-12.2            cachem_1.0.5             
 [47] withr_2.4.2               treemap_2.4-2            
 [49] cluster_2.1.0             lazyeval_0.2.2           
 [51] crayon_1.4.1              labeling_0.4.2           
 [53] pkgconfig_2.0.3           SuppDists_1.1-9.5        
 [55] slam_0.1-48               statsExpressions_1.1.0   
 [57] ProtGenerics_1.22.0       wordcloud_2.6            
 [59] rlang_0.4.11              RJSONIO_1.3-1.4          
 [61] lifecycle_1.0.0           MatrixModels_0.5-0       
 [63] sandwich_3.0-1            affyio_1.60.0            
 [65] BiocFileCache_1.14.0      modelr_0.1.8             
 [67] cellranger_1.1.0          rprojroot_2.0.2          
 [69] graph_1.68.0              Matrix_1.2-18            
 [71] mc2d_0.1-21               zoo_1.8-9                
 [73] reprex_2.0.0              whisker_0.4              
 [75] GlobalOptions_0.1.2       pheatmap_1.0.12          
 [77] png_0.1-7                 viridisLite_0.4.0        
 [79] rjson_0.2.20              mzR_2.24.1               
 [81] PMCMRplus_1.9.0           parameters_0.14.0        
 [83] bitops_1.0-7              shinydashboard_0.7.1     
 [85] R.oo_1.24.0               KernSmooth_2.23-17       
 [87] blob_1.2.1                shape_1.4.6              
 [89] multcompView_0.1-8        tmvtnorm_1.4-10          
 [91] ggsignif_0.6.2            scales_1.1.1             
 [93] memoise_2.0.0             magrittr_2.0.1           
 [95] plyr_1.8.6                gplots_3.1.1             
 [97] zlibbioc_1.36.0           compiler_4.0.3           
 [99] plotrix_3.8-1             pcaMethods_1.82.0        
[101] clue_0.3-59               cli_3.0.0                
[103] affy_1.68.0               XVector_0.30.0           
[105] pbapply_1.4-3             MASS_7.3-53              
[107] tidyselect_1.1.1          vsn_3.58.0               
[109] stringi_1.6.2             highr_0.9                
[111] yaml_2.2.1                GOSemSim_2.16.1          
[113] norm_1.0-9.5              MALDIquant_1.19.3        
[115] grid_4.0.3                sass_0.4.0               
[117] tools_4.0.3               rstudioapi_0.13          
[119] circlize_0.4.13           foreach_1.5.1            
[121] git2r_0.28.0              gridExtra_2.3            
[123] ipmisc_6.0.2              pairwiseComparisons_3.1.6
[125] farver_2.1.0              mzID_1.28.0              
[127] digest_0.6.27             BiocManager_1.30.16      
[129] shiny_1.6.0               proto_1.0.0              
[131] BWStest_0.2.2             Rcpp_1.0.6               
[133] broom_0.7.8               BayesFactor_0.9.12-4.2   
[135] performance_0.7.2         later_1.2.0              
[137] ncdf4_1.17                httr_1.4.2               
[139] MSnbase_2.15.7            ComplexHeatmap_2.6.2     
[141] effectsize_0.4.5          colorspace_2.0-2         
[143] rvest_1.0.0               XML_3.99-0.6             
[145] fs_1.5.0                  rematch2_2.1.2           
[147] gmm_1.6-6                 plotly_4.9.4.1           
[149] xtable_1.8-4              gmp_0.6-2                
[151] jsonlite_1.7.2            zeallot_0.1.0            
[153] NLP_0.2-1                 R6_2.5.0                 
[155] tm_0.7-8                  gsubfn_0.7               
[157] pillar_1.6.1              htmltools_0.5.1.1        
[159] mime_0.11                 glue_1.4.2               
[161] fastmap_1.1.0             BiocParallel_1.24.1      
[163] codetools_0.2-16          mvtnorm_1.1-2            
[165] utf8_1.2.1                lattice_0.20-41          
[167] bslib_0.2.5.1             sqldf_0.4-11             
[169] curl_4.3.2                gtools_3.9.2             
[171] zip_2.2.0                 GO.db_3.12.1             
[173] limma_3.46.0              rmarkdown_2.9            
[175] munsell_0.5.0             GetoptLong_1.0.5         
[177] GenomeInfoDbData_1.2.4    iterators_1.0.13         
[179] impute_1.64.0             haven_2.4.1              
[181] gtable_0.3.0              bayestestR_0.10.0        </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
