<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Fabio Bedin | MS-Unit at IEO" />


<title>LFQ_Analysis</title>

<script src="site_libs/jquery-3.5.1/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/pagedtable-1.1/css/pagedtable.css" rel="stylesheet" />
<script src="site_libs/pagedtable-1.1/js/pagedtable.js"></script>
<script src="site_libs/htmlwidgets-1.5.3/htmlwidgets.js"></script>
<link href="site_libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="site_libs/datatables-binding-0.18/datatables.js"></script>
<link href="site_libs/dt-core-1.10.20/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="site_libs/dt-core-1.10.20/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="site_libs/dt-core-1.10.20/js/jquery.dataTables.min.js"></script>
<link href="site_libs/crosstalk-1.1.1/css/crosstalk.css" rel="stylesheet" />
<script src="site_libs/crosstalk-1.1.1/js/crosstalk.min.js"></script>
<script src="site_libs/plotly-binding-4.9.4.1/plotly.js"></script>
<script src="site_libs/typedarray-0.1/typedarray.min.js"></script>
<link href="site_libs/plotly-htmlwidgets-css-1.57.1/plotly-htmlwidgets.css" rel="stylesheet" />
<script src="site_libs/plotly-main-1.57.1/plotly-latest.min.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Template_LFQ</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">LFQ_Analysis</h1>
<h4 class="author">Fabio Bedin | MS-Unit at IEO</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-02-15
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>GN_MR_ZC3H4/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20220215code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20220215)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20220215code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20220215)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrong24ebe3a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> 24ebe3a </a>
</p>
</div>
<div id="strongRepositoryversionstrong24ebe3a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version 24ebe3a. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/

Untracked files:
    Untracked:  code/custom_functions.R
    Untracked:  data/expdesign.tsv
    Untracked:  data/proteinGroups.txt
    Untracked:  output/results_table_TURBO._vs_WT..xlsx
    Untracked:  output/results_table_TURBO_BIO_vs_TURBO..xlsx
    Untracked:  output/results_table_TURBO_BIO_vs_WT_BIO.xlsx
    Untracked:  output/results_table_WT_BIO_vs_WT..xlsx

Unstaged changes:
    Modified:   analysis/_site.yml
    Deleted:    analysis/about.Rmd
    Deleted:    analysis/license.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/Turbo_bio_vs_Turbo.Rmd</code>) and HTML (<code>docs/Turbo_bio_vs_Turbo.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
24ebe3a
</td>
<td>
FabioBedin
</td>
<td>
2022-02-15
</td>
<td>
first push
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<style type="text/css">
p{
  font-size: 16px;
}
</style>
<pre class="r"><code>## Proteomics
library(&quot;MBQN&quot;)
library(&quot;DEP&quot;)
library(&quot;SummarizedExperiment&quot;)
library(&quot;preprocessCore&quot;)
library(&quot;enrichR&quot;)
library(&quot;gprofiler2&quot;)
library(&quot;org.Hs.eg.db&quot;)
library(&quot;rrvgo&quot;)

## Plots
library(&quot;DT&quot;)
library(&quot;patchwork&quot;)
library(&quot;datapasta&quot;)
library(&quot;ggstatsplot&quot;)
library(&quot;UpSetR&quot;)
library(&quot;ggrepel&quot;)
library(&quot;visdat&quot;)
library(&quot;naniar&quot;)

## Networks
library(&quot;STRINGdb&quot;)
library(&quot;RCy3&quot;)
library(&quot;RColorBrewer&quot;)

## Other
library(&quot;here&quot;)
library(&quot;openxlsx&quot;)
library(&quot;tidyverse&quot;)</code></pre>
<pre class="r"><code>source(here::here(&quot;code/custom_functions.R&quot;))</code></pre>
<pre class="r"><code>data &lt;- read.csv(here::here(paste0(&quot;data/&quot;, params$proteinGroups)), header = TRUE,stringsAsFactors = FALSE, sep = &quot;\t&quot;)
## qui metterei anche Razor...unique.peptides al posto di Peptides, perchè così non conto i peptidi che sono attribuiti a più di 1 PG
data &lt;- data[data$Reverse != &quot;+&quot; &amp; data$Potential.contaminant != &quot;+&quot; &amp; data$Only.identified.by.site != &quot;+&quot; &amp; data$Peptides &gt;= params$n_peptides,]

data_unique &lt;- make_unique(data, &quot;Gene.names&quot;, &quot;Protein.IDs&quot;, delim = &quot;;&quot;)</code></pre>
<pre class="r"><code>cond_1 &lt;- params$test_1

cond_2 &lt;- params$test_2

conditions&lt;-c(cond_1,cond_2)

test&lt;- paste(cond_1,cond_2,sep=&quot;_vs_&quot;)

expdesign &lt;- read.table(here::here(paste0(&quot;data/&quot;, params$expdesign)), header = T, stringsAsFactors = F)

expdesign &lt;- subset(expdesign, condition %in% conditions)

columns&lt;-match(paste(&quot;LFQ.intensity.&quot;,expdesign$label,sep=&quot;&quot;),colnames(data_unique))

data_se &lt;- make_se(data_unique, columns, expdesign)

## define automaticaly the best statistical threshold to filter data besed on replicates:
if(max(expdesign$replicate)&lt;=3){
       threshold&lt;-0
     } else if(max(expdesign$replicate)&lt;6){
       threshold&lt;-1
     } else if (max(expdesign$replicate)&gt;=6){
       threshold&lt;-trunc(max(expdesign$replicate)*0.25) ## 0.25 serve per avere sempre il 75% di valid vales
     }

data_filt &lt;- filter_missval(data_se, thr = threshold)</code></pre>
<div id="turbo_bio-vs-turbo." class="section level1">
<h1><strong>TURBO_BIO</strong> vs <strong>TURBO.</strong></h1>
<hr />
</div>
<div id="introduction" class="section level1">
<h1><strong>Introduction</strong></h1>
<p>First, we need to wrangling the original dataset. From the <strong>proteinGroups</strong> obtained by <a href="https://www.maxquant.org/">MaxQuant</a> software, I remove proteins that are marked form the software as potential contaminant, only identify by site and reverse.<br />
We che also filter the initial datasets by excluding all proteins identified by less than 1 peptides.<br />
Now, we generate some quality control plots to investigate the structure of the dataset and observe the effect of filters.</p>
<div id="normalization-strategy" class="section level4">
<h4><strong>Normalization strategy:</strong></h4>
<pre class="r"><code>if(params$Normalization == &quot;VSN&quot;){
  data_filt &lt;- normalize_vsn(data_filt)
  print(&quot;VSN normalization is used&quot;)
} else if(params$Normalization == &quot;MBQN&quot;){
  data_filt &lt;- normalize_MBQN(data_filt)
  print(&quot;MBQN normalization is used&quot;)
} else {
  print(&quot;Max-LFQ normalization is used&quot;)
}</code></pre>
<p>[1] “Max-LFQ normalization is used”</p>
</div>
<div id="quality-control-plots" class="section level2 tabset tabset-fade">
<h2><strong>1. Quality control plots</strong></h2>
<div id="proteins-per-samples" class="section level3 tabset">
<h3>Proteins per samples</h3>
<p>This plot show the number of proteins identify in each samples <em>after and before User define</em> filters:</p>
<div id="filtered" class="section level4">
<h4>Filtered</h4>
<pre class="r"><code>plot_numbers_lables(data_filt)</code></pre>
<p><img src="figure/Turbo_bio_vs_Turbo.Rmd/proteins-filt-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="not-filtered" class="section level4">
<h4>Not-Filtered</h4>
<pre class="r"><code>plot_numbers_lables(data_se)</code></pre>
<p><img src="figure/Turbo_bio_vs_Turbo.Rmd/proteins-NOfilt-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="distributions" class="section level3 tabset">
<h3>Distributions</h3>
<p>This plot is useful for checking the distributions of the samples <em>after and before User define</em> filters:</p>
<div id="filtered-1" class="section level4">
<h4>Filtered</h4>
<pre class="r"><code>plot_normalization(data_filt)</code></pre>
<p><img src="figure/Turbo_bio_vs_Turbo.Rmd/normalization-filt-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="not-filtered-1" class="section level4">
<h4>Not-Filtered</h4>
<pre class="r"><code>plot_normalization(data_se)</code></pre>
<p><img src="figure/Turbo_bio_vs_Turbo.Rmd/normalization-NOfilt-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="missing-data" class="section level3 tabset">
<h3>Missing data</h3>
<p>This plot we can explore the missing data pattern <em>after and before User define</em> filters. Notice that the percentages of missingness are provided in the data. These are accurate to 1 decimal place. Also the dataset is arranged by columns with most missingness.</p>
<div id="filtered-2" class="section level4">
<h4>Filtered</h4>
<pre class="r"><code>assay(data_filt) %&gt;% 
  as.data.frame() %&gt;% 
  vis_miss(., sort_miss = TRUE)</code></pre>
<p><img src="figure/Turbo_bio_vs_Turbo.Rmd/missing-data-filt-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="not-filtered-2" class="section level4">
<h4>Not-Filtered</h4>
<pre class="r"><code>assay(data_se) %&gt;% 
  as.data.frame() %&gt;% 
  vis_miss(., sort_miss = TRUE)</code></pre>
<p><img src="figure/Turbo_bio_vs_Turbo.Rmd/missing-data-NOfilt-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="sample-cvs" class="section level3 tabset">
<h3>Sample CVs</h3>
<p>The coefficient of variation (CV) is a statistical measure of the dispersion of data points in a data series around the mean. The coefficient of variation represents the ratio of the standard deviation to the mean, and it is a useful statistic for comparing the degree of variation from one data series to another, even if the means are drastically different from one another.</p>
<div id="filtered-3" class="section level4">
<h4>Filtered</h4>
<pre class="r"><code>plot_cvs(data_filt)</code></pre>
<p><img src="figure/Turbo_bio_vs_Turbo.Rmd/CVs-filt-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="not-filtered-3" class="section level4">
<h4>Not-Filtered</h4>
<pre class="r"><code>plot_cvs(data_se)</code></pre>
<p><img src="figure/Turbo_bio_vs_Turbo.Rmd/CVs-NOfilt-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
</div>
</div>
<div id="upset-plot" class="section level2 tabset tabset-fade">
<h2><strong>2. Upset plot</strong></h2>
<p>With this plot we can identify the numper of prteins in common within the 2 conditions or in common between all condition.</p>
<div id="turbo_bio" class="section level3">
<h3>TURBO_BIO</h3>
<pre class="r"><code>define_set &lt;- assay(data_filt) %&gt;%  as.data.frame() %&gt;% select(starts_with(cond_1)) %&gt;% colnames()
n_sets &lt;- length(define_set)

assay(data_filt) %&gt;% 
  as.data.frame() %&gt;% 
  select(starts_with(cond_1)) %&gt;% 
  rownames_to_column() %&gt;% 
  pivot_longer(!rowname, names_to = &quot;samples&quot;, values_to = &quot;intensity&quot;) %&gt;% 
  mutate(intensity = if_else(is.na(intensity), 0, 1)) %&gt;% 
  pivot_wider(names_from = samples, values_from = intensity) %&gt;% 
  as.data.frame() %&gt;% 
  upset(nsets = n_sets,
        sets = define_set,
        order.by = &quot;freq&quot;,
        keep.order = T,
        text.scale = 2.5,
        point.size = 4,
        line.size = 0.5, 
        sets.bar.color = &quot;coral2&quot;,
        main.bar.color  = &quot;gray44&quot;)</code></pre>
<p><img src="figure/Turbo_bio_vs_Turbo.Rmd/upset-cond1-1.png" width="1440" style="display: block; margin: auto;" /></p>
</div>
<div id="turbo." class="section level3">
<h3>TURBO.</h3>
<pre class="r"><code>define_set &lt;- assay(data_filt) %&gt;%  as.data.frame() %&gt;% select(starts_with(cond_2)) %&gt;% colnames()
n_sets &lt;- length(define_set)

assay(data_filt) %&gt;% 
  as.data.frame() %&gt;% 
  select(starts_with(cond_2)) %&gt;% 
  rownames_to_column() %&gt;% 
  pivot_longer(!rowname, names_to = &quot;samples&quot;, values_to = &quot;intensity&quot;) %&gt;% 
  mutate(intensity = if_else(is.na(intensity), 0, 1)) %&gt;% 
  pivot_wider(names_from = samples, values_from = intensity) %&gt;% 
  as.data.frame() %&gt;% 
  upset(nsets = n_sets,
        sets = define_set,
        order.by = &quot;freq&quot;,
        keep.order = T,
        text.scale = 2.5,
        point.size = 4,
        line.size = 0.5, 
        sets.bar.color = &quot;turquoise3&quot;,
        main.bar.color  = &quot;gray44&quot;)</code></pre>
<p><img src="figure/Turbo_bio_vs_Turbo.Rmd/upset-cond2-1.png" width="1440" style="display: block; margin: auto;" /></p>
</div>
<div id="both-conditions" class="section level3">
<h3>Both conditions</h3>
<pre class="r"><code>define_set &lt;- assay(data_filt) %&gt;% colnames()
n_sets &lt;- length(define_set)
n_each &lt;- n_sets / 2

assay(data_filt) %&gt;% 
  as.data.frame() %&gt;% 
  rownames_to_column() %&gt;% 
  pivot_longer(!rowname, names_to = &quot;samples&quot;, values_to = &quot;intensity&quot;) %&gt;% 
  mutate(intensity = if_else(is.na(intensity), 0, 1)) %&gt;% 
  pivot_wider(names_from = samples, values_from = intensity) %&gt;% 
  as.data.frame() %&gt;% 
  upset(nsets = n_sets,
        sets = define_set,
        order.by = &quot;freq&quot;,
        keep.order = T,
        text.scale = 2.5,
        point.size = 4,
        line.size = 0.5, 
        sets.bar.color = rep(c(&quot;red3&quot;, &quot;royalblue1&quot;), each = n_each),
        main.bar.color  = &quot;gray44&quot;)</code></pre>
<p><img src="figure/Turbo_bio_vs_Turbo.Rmd/upset-both-1.png" width="1440" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="imputation" class="section level2">
<h2><strong>3. Imputation</strong></h2>
<p>An idealized version of a label-free discovery mass spectrometry proteomics experiment would provide absolute abundance measurements for a whole proteome, across varying conditions. Unfortunately, this ideal is not realized. Measurements are made on peptides requiring an inferential step to obtain protein level estimates. The inference is complicated by experimental factors that necessitate relative abundance estimation and result in widespread non-ignorable missing data. Relative abundance on the log scale takes the form of parameter contrasts. In a complete-case analysis, contrast estimates may be biased by missing data and a substantial amount of useful information will often go unused.</p>
<pre class="r"><code>data_imputed &lt;- mix_imputation_mean(data_filt, cond1 = cond_1, cond2 = cond_2)

plot_imputation(data_filt, data_imputed)</code></pre>
<p><img src="figure/Turbo_bio_vs_Turbo.Rmd/imputation-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>data_filt &lt;- data_imputed</code></pre>
</div>
<div id="differential-enrichment-analysis" class="section level2">
<h2><strong>4. Differential enrichment analysis</strong></h2>
<p>Protein-wise linear models combined with empirical Bayes statistics are used for the differential enrichment analysis (or differential expression analysis) and False discovery rate (FDR) is ajusted with <em>Benjamini-Hochberg</em> procedure. Significant proteins are are define with <strong>FDR = 0.05</strong> and <strong>Fold change = 1</strong></p>
<pre class="r"><code>data_diff &lt;- test_diff_BH(data_filt, type = &quot;manual&quot;, test = test)

dep &lt;- add_rejections(data_diff, alpha = params$FDR, lfc = params$Fold_change)

results&lt;-get_results(dep)</code></pre>
<p>There are <strong>66</strong> significant proteins in <strong>TURBO_BIO</strong> vs <strong>TURBO.</strong> comparison.</p>
<p>The results from the previous analysis can be easily visualized by a number of functions. These visualizations assist in the determination of the optimal cutoffs to be used, highlight the most interesting samples and contrasts, and pinpoint differentially enriched/expressed proteins.</p>
<div id="visualization-of-the-results" class="section level3 tabset tabset-fade">
<h3><strong>4.1. Visualization of the results</strong></h3>
<div id="pca" class="section level4">
<h4>PCA</h4>
<p>The PCA plot can be used to get a high-level overview of the data. This can be very useful to observe batch effects, such as clear differences between replicates.</p>
<pre class="r"><code>if(params$Imputation){
  if(nrow(dep) &gt;= 500){
    n_value &lt;- 500
  }else {
    n_value &lt;- nrow(dep)
  }
  plot_pca(dep, x = 1, y = 2, n = n_value, point_size = 4, indicate = &quot;condition&quot;, label = T, label_size = 3)
} else {
  var &lt;- apply(assay(dep), 1, sd)
  df &lt;- assay(dep)[order(var, decreasing = TRUE)[seq_len(nrow(dep))],]
  n_value &lt;- df %&gt;% as.data.frame() %&gt;% drop_na() %&gt;% nrow()
  if(n_value &gt;= 500){
    n_value &lt;- 500
  }
  plot_pca(dep, x = 1, y = 2, n = n_value, point_size = 4, indicate = &quot;condition&quot;, label = T, label_size = 3)
}</code></pre>
<p><img src="figure/Turbo_bio_vs_Turbo.Rmd/PCA-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="volcano" class="section level4">
<h4>Volcano</h4>
<p>Volcano plots allows to inspect the enrichment of proteins between the two samples (x axis) and their corresponding adjusted p value (y axis).</p>
<pre class="r"><code>plot_volcano(dep, contrast=test, add_names=T, label_size=5, adjusted = F)</code></pre>
<p><img src="figure/Turbo_bio_vs_Turbo.Rmd/volcano-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="volcano-and-unique" class="section level4">
<h4>Volcano and unique</h4>
<pre class="r"><code>plot_volcano_2(dep, contrast=test, add_names=T, label_size=5, adjusted = F, unique_vec = uni_name)</code></pre>
<p><img src="figure/Turbo_bio_vs_Turbo.Rmd/volcano-unique-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="correlation" class="section level4">
<h4>Correlation</h4>
<pre class="r"><code>plot_cor(dep, significant = TRUE, lower = 0, upper = 1, pal = &quot;Reds&quot;)</code></pre>
<p><img src="figure/Turbo_bio_vs_Turbo.Rmd/correlation-matrix-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="cluster-heatmap" class="section level4">
<h4>Cluster heatmap</h4>
<p>The heatmap representation gives an overview of all significant proteins (rows) in all samples (columns). This allows to see general trends, for example if one sample or replicate is really different compared to the others. Additionally, the clustering of samples (columns) can indicate closer related samples and clustering of proteins (rows) indicates similarly behaving proteins.</p>
<pre class="r"><code>if(results %&gt;% filter(significant) %&gt;% nrow() &lt;= 100){
  plot_heatmap(dep, type = &quot;centered&quot;, kmeans = TRUE, k = 5, show_row_names = T, indicate = &quot;condition&quot;, col_limit = 5)
}else {
  plot_heatmap(dep, type = &quot;centered&quot;, kmeans = TRUE, k = 5, show_row_names = F, indicate = &quot;condition&quot;, col_limit = 5)
}</code></pre>
<p><img src="figure/Turbo_bio_vs_Turbo.Rmd/cluster-heatmap-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="result-table" class="section level3">
<h3><strong>4.2. Result table</strong></h3>
<p>In this table are summarized the results of DEP analysis, sorted by p.value. Proteins that are UP regulated are colored with a red box in Fold change column, in the orhter hand proteins that are DOWN regulated are colored with blue.</p>
<pre class="r"><code>results %&gt;% 
  select(starts_with(&quot;name&quot;) | starts_with(test)) %&gt;% 
  filter(across(ends_with(&quot;significant&quot;))) %&gt;% 
  arrange(across(ends_with(&quot;p.adj&quot;))) %&gt;%
  mutate(across(2:3, format, scientific = T, digits = 2)) %&gt;%
  dplyr::rename_with(~ tolower(gsub(paste0(test,&quot;_&quot;), &quot;&quot;, .x)), .cols = starts_with(test)) %&gt;% 
  mutate(significant = str_to_title(significant)) %&gt;%
  dplyr::rename(Fold.Change = ratio, Gene.name = name) %&gt;% 
  DT::datatable(options = list(
  columnDefs = list(list(className = &#39;dt-center&#39;, targets = 1:5)),
  pageLength = 10)) %&gt;%
  formatStyle(&#39;Fold.Change&#39;, backgroundColor = styleInterval(c(-1, 1), c(&#39;lightblue&#39;, &#39;white&#39;, &#39;tomato&#39;))) %&gt;% 
  formatStyle(&#39;significant&#39;, color = styleEqual(c(&quot;True&quot;, &quot;False&quot;), c(&#39;green&#39;, &#39;red&#39;)))</code></pre>
<div id="htmlwidget-2a0dd1ba32e0541469b5" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-2a0dd1ba32e0541469b5">{"x":{"filter":"none","data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33","34","35","36","37","38","39","40","41","42","43","44","45","46","47","48","49","50","51","52","53","54","55","56","57","58","59","60","61","62","63","64","65","66"],["CTR9","ZNF281","SSR4","SYMPK","MYL12A","INTS3","MRPS23","DYNLL2","LEO1","MRPS22","MRPS27","GPALPP1","MRPS9","MRPS35","MRPS7","XAB2","TPM1.1","TNK1","MRPS18B","DAP3","PTCD3","C12orf43","SUGP2","MRPS5","BLMH","MED23","MRPS34","WDR82","MRPS24","MYH14","HELLS","ZNF616","PAF1","HNRNPL","TMEM263","SAFB","A0A2R8YHD2","TPM1","MRPS10","TPM3.1","KHDRBS1","S100A8","ACTN4","SPTBN1","RBMX","PIP","RPA3","RPA2","EIF2AK2","CAPZB","MYO1B","NOSIP","DNAH14","MYL6B","RPA1","DHX9","GSN","MYL12B","MYH14.1","MYL6","TPM2","ACTG1","MYH10","SSBP1","MYH9","MYL6.1"],["9.6e-12","5.3e-10","1.5e-09","1.9e-09","3.6e-08","8.1e-08","8.5e-08","1.3e-07","1.4e-07","1.5e-07","1.1e-07","1.6e-07","1.8e-07","2.2e-07","2.2e-07","2.3e-07","2.7e-07","5.2e-07","5.6e-06","7.5e-06","9.5e-06","1.2e-05","1.2e-05","2.2e-05","2.5e-05","2.7e-05","4.9e-05","8.2e-05","1.0e-04","1.0e-04","1.6e-04","2.0e-04","3.5e-04","7.3e-04","7.7e-04","8.0e-04","1.0e-03","1.1e-03","1.3e-03","1.5e-03","1.8e-03","1.9e-03","2.1e-03","2.1e-03","2.4e-03","2.5e-03","2.5e-03","2.5e-03","2.8e-03","3.0e-03","3.3e-03","3.3e-03","3.9e-03","4.0e-03","5.2e-03","5.5e-03","5.5e-03","6.0e-03","6.3e-03","6.2e-03","6.3e-03","6.6e-03","7.0e-03","7.0e-03","8.4e-03","8.3e-03"],["8.2e-09","2.3e-07","4.1e-07","4.1e-07","6.2e-06","1.0e-05","1.0e-05","1.1e-05","1.1e-05","1.1e-05","1.1e-05","1.1e-05","1.2e-05","1.2e-05","1.2e-05","1.2e-05","1.4e-05","2.5e-05","2.3e-04","2.9e-04","3.6e-04","4.2e-04","4.2e-04","7.3e-04","7.5e-04","8.0e-04","1.4e-03","2.3e-03","2.7e-03","2.7e-03","3.7e-03","4.4e-03","7.2e-03","1.2e-02","1.3e-02","1.3e-02","1.5e-02","1.6e-02","1.8e-02","2.0e-02","2.3e-02","2.3e-02","2.4e-02","2.4e-02","2.6e-02","2.7e-02","2.7e-02","2.7e-02","2.7e-02","2.8e-02","2.9e-02","2.9e-02","3.2e-02","3.2e-02","3.9e-02","4.0e-02","4.0e-02","4.1e-02","4.1e-02","4.1e-02","4.1e-02","4.2e-02","4.4e-02","4.4e-02","4.7e-02","4.7e-02"],["True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True"],[2.75,2.13,2.08,4.39,3.44,1.83,3.24,1.47,2.3,1.76,4.63,-1.5,3.31,1.75,1.8,1.53,1.9,1.57,1.16,1.76,1.84,-1.03,2.42,1.01,1.48,2.48,1.39,1.33,1.48,1.29,2.88,-1,1.75,1.28,-1.11,1.72,2.64,2.02,1.16,2.71,1.43,-1.32,2.26,3.23,1.58,-1.51,2.18,2.54,2.16,1.88,1.05,2,1.79,1.74,2.48,1.3,1.98,2.53,2.62,2.25,2.5,1.78,2.23,2.74,2.24,1.66]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>Gene.name<\/th>\n      <th>p.val<\/th>\n      <th>p.adj<\/th>\n      <th>significant<\/th>\n      <th>Fold.Change<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-center","targets":[1,2,3,4,5]},{"orderable":false,"targets":0}],"pageLength":10,"order":[],"autoWidth":false,"orderClasses":false,"rowCallback":"function(row, data, displayNum, displayIndex, dataIndex) {\nvar value=data[5]; $(this.api().cell(row, 5).node()).css({'background-color':isNaN(parseFloat(value)) ? '' : value <= -1 ? \"lightblue\" : value <= 1 ? \"white\" : \"tomato\"});\nvar value=data[4]; $(this.api().cell(row, 4).node()).css({'color':value == \"True\" ? \"green\" : value == \"False\" ? \"red\" : null});\n}"}},"evals":["options.rowCallback"],"jsHooks":[]}</script>
</div>
</div>
<div id="gene-ontology" class="section level2">
<h2><strong>5. Gene Ontology</strong></h2>
<p>Now I perfom a gene onology analysis (GO or by branch <a href="GO:MF" class="uri">GO:MF</a>, <a href="GO:BP" class="uri">GO:BP</a>, <a href="GO:CC" class="uri">GO:CC</a>) and a KEGG ontology.<br />
The enrichment results are visualized with an interactive <strong>Manhattan-like-plot</strong> named “Gostplot”.</p>
<div id="turbo_bio-1" class="section level3 tabset tabset-fade">
<h3><strong>5.1. TURBO_BIO</strong></h3>
<div id="gostplot" class="section level4">
<h4>Gostplot</h4>
<p>UP regulated proteins in TURBO_BIO were used to generate these gene ontologies.</p>
<pre class="r"><code>Gname &lt;- results %&gt;% filter(significant &amp; get(paste0(test, &quot;_ratio&quot;)) &gt; 0) %&gt;% pull(name)

if(!params$Imputation){
  Uni &lt;- unique_pair(data_filt_unique, conditions = conditions, table = T) %&gt;% 
  filter(!!sym(paste0(&quot;unique_&quot;, cond_1)) == &quot;+&quot;) %&gt;% 
  pull(name) %&gt;% unique()

  Gname &lt;- c(Gname, Uni)
}

if(length(Gname) == 0){
  print(&quot;NOT enough data.&quot;)
  GO &lt;- NULL
}else{
  GO &lt;- gost(query = Gname, organism = &quot;hsapiens&quot;, sources = c(&quot;GO:BP&quot;, &quot;GO:MF&quot;, &quot;GO:CC&quot;, &quot;KEGG&quot;), ordered_query = T, evcodes = TRUE)
  if(is.null(GO)){
    print(&quot;NOT enough data.&quot;)
  }else{
    gostplot(GO, capped = TRUE, interactive = TRUE)
  }
}</code></pre>
<div id="htmlwidget-ed4906484ed42468d86d" style="width:960px;height:768px;" class="plotly html-widget"></div>
<script type="application/json" data-for="htmlwidget-ed4906484ed42468d86d">{"x":{"data":[{"x":[105.151314340381,84.5275863540995,122.368931544347,109.617450446254,134.711211690332,123.658888831471,84.8783300273356,164.746561575115,105.092857061508,123.654991679546,114.173221046398,105.120137124982,177.97349520804,112.875469455424,172.034235674576,112.185673564727,169.033428692445,180.132517374404,180.140311678254,77.1619692161419,150.732403253371,124.69553124348,131.293409452243,84.4106717963542,123.740729021893,89.4379977794046,108.767871326637,104.784982059445,105.155211492306,86.6086654819669,111.569923560601,84.1417683135398,75.7784802828218,86.5463110511694,107.583137141484,93.0194803983374,86.3670420626265],"y":[7.13613455286493,6.85403256773687,6.68139865832011,6.5298944028773,5.89581443652719,5.87920224678614,5.7219162487249,5.48022718959647,4.80619335601769,4.43163910928272,4.38192091998525,3.48063082300623,3.14184297013043,3.04536310808623,2.64360529090097,2.55470058557965,2.31186274888118,2.31186274888118,2.31186274888118,2.24078520762918,1.96795969261705,1.93393235634427,1.92874161198672,1.81536451742825,1.78204428232763,1.76097126889163,1.75284315615751,1.7350628535833,1.6083078422459,1.5076487903985,1.46461076283103,1.45873312931092,1.43725683416872,1.42110289047811,1.39070104127179,1.34383376584216,1.31841134083081],"text":["GO:0030048 (132) <br> actin filament-based movement <br> 7.309e-08","GO:0006412 (713) <br> translation <br> 1.399e-07","GO:0043043 (738) <br> peptide biosynthetic process <br> 2.083e-07","GO:0032543 (79) <br> mitochondrial translation <br> 2.952e-07","GO:0050684 (141) <br> regulation of mRNA processing <br> 1.271e-06","GO:0043604 (867) <br> amide biosynthetic process <br> 1.321e-06","GO:0006518 (895) <br> peptide metabolic process <br> 1.897e-06","GO:0140053 (111) <br> mitochondrial gene expression <br> 3.310e-06","GO:0030029 (800) <br> actin filament-based process <br> 1.562e-05","GO:0043603 (1165) <br> cellular amide metabolic process <br> 3.701e-05","GO:0034645 (1564) <br> cellular macromolecule biosynthetic process <br> 4.150e-05","GO:0030036 (711) <br> actin cytoskeleton organization <br> 3.307e-04","GO:1903311 (318) <br> regulation of mRNA metabolic process <br> 7.214e-04","GO:0034097 (922) <br> response to cytokine <br> 9.008e-04","GO:1901566 (1698) <br> organonitrogen compound biosynthetic process <br> 2.272e-03","GO:0033523 (12) <br> histone H2B ubiquitination <br> 2.788e-03","GO:1900756 (2) <br> protein processing in phagocytic vesicle <br> 4.877e-03","GO:1903921 (2) <br> regulation of protein processing in phagocytic vesicle <br> 4.877e-03","GO:1903923 (2) <br> positive regulation of protein processing in phagocytic vesicle <br> 4.877e-03","GO:0001711 (15) <br> endodermal cell fate commitment <br> 5.744e-03","GO:0071345 (828) <br> cellular response to cytokine stimulus <br> 1.077e-02","GO:0044260 (5746) <br> cellular macromolecule metabolic process <br> 1.164e-02","GO:0046833 (5) <br> positive regulation of RNA export from nucleus <br> 1.178e-02","GO:0006378 (52) <br> mRNA polyadenylation <br> 1.530e-02","GO:0043631 (53) <br> RNA polyadenylation <br> 1.652e-02","GO:0009059 (4880) <br> macromolecule biosynthetic process <br> 1.734e-02","GO:0032241 (6) <br> positive regulation of nucleobase-containing compound transport <br> 1.767e-02","GO:0022604 (305) <br> regulation of cell morphogenesis <br> 1.841e-02","GO:0030049 (18) <br> muscle filament sliding <br> 2.464e-02","GO:0007015 (444) <br> actin filament organization <br> 3.107e-02","GO:0033275 (20) <br> actin-myosin filament sliding <br> 3.431e-02","GO:0006289 (58) <br> nucleotide-excision repair <br> 3.477e-02","GO:0000380 (79) <br> alternative mRNA splicing, via spliceosome <br> 3.654e-02","GO:0006996 (3805) <br> organelle organization <br> 3.792e-02","GO:0031440 (28) <br> regulation of mRNA 3'-end processing <br> 4.067e-02","GO:0010390 (29) <br> histone monoubiquitination <br> 4.531e-02","GO:0006936 (340) <br> muscle contraction <br> 4.804e-02"],"key":["GO:0030048","GO:0006412","GO:0043043","GO:0032543","GO:0050684","GO:0043604","GO:0006518","GO:0140053","GO:0030029","GO:0043603","GO:0034645","GO:0030036","GO:1903311","GO:0034097","GO:1901566","GO:0033523","GO:1900756","GO:1903921","GO:1903923","GO:0001711","GO:0071345","GO:0044260","GO:0046833","GO:0006378","GO:0043631","GO:0009059","GO:0032241","GO:0022604","GO:0030049","GO:0007015","GO:0033275","GO:0006289","GO:0000380","GO:0006996","GO:0031440","GO:0010390","GO:0006936"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(255,153,0,1)","opacity":0.8,"size":[17.1292036097827,19.2641891295964,19.304750980403,16.4079602522644,17.2190663439651,19.4929694976528,19.5298401215138,16.8901999906596,19.3992843320467,19.8324155495527,20.1638153904009,19.2608782960062,18.2814219128508,19.5642394508994,20.2551218810831,13.3029097283266,8.80930656837597,8.80930656837597,8.80930656837597,13.7213162710519,19.4394252088968,21.5541875017563,11.4438415281718,15.7884235332077,15.8173350333943,21.3856692069636,11.8663149965605,18.2288014576218,14.0505345805786,18.6955285437039,14.236059602826,15.9532356375401,16.4079602522644,21.1258351184584,14.8076606210532,14.865576968,18.3653584943448],"symbol":"circle","line":{"width":1.88976377952756,"color":"rgba(255,153,0,1)"}},"hoveron":"points","set":"SharedData2a0dd1ba","name":"GO:BP","legendgroup":"GO:BP","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","_isNestedKey":false,"frame":null},{"x":[55.4324514238447,54.0915750971904,57.1163426247595,55.4246556312479,54.087677200892,57.1046489358643,57.2371774100104,62.9281060056946,55.7130999573305,62.1290372645198,62.1446288497134,67.9758817121406,55.3622892904733,59.9579090263032,57.6659460028359,55.4168598386511,59.447284611211,57.2410753063088,55.350595601578,67.0481823931181,54.559322653,59.7941973817699,61.8639803162276,68.4475271642487,67.0325908079244,62.2108930867865,68.5683619494995,68.4514250605471,55.9157905648481,59.4784677815983,59.4511825075094,64.5223455917459,55.7598747129115,69.6987518760395,64.428796080584,62.1368330571166,55.7754662981052,62.1251393682214,57.8257597510709,58.3207925809694,71.1994419509289,62.1329351608182,57.2449732026072,68.1084101862867,57.3346248174707,56.0288295575021,64.4326939768824,59.4745698852999,62.1485267460118,55.3466977052796,57.8296576473693,62.1212414719229,55.0816407569875,55.3427998089812,59.9501132337064,55.9859526982195,57.8452492325629,55.8417305351782,62.3356257683357,54.9491122828414,58.3753631291472,55.783262090702,58.3246904772678,61.396232760418,66.4323147779687,54.9257249050509,62.3161362868436,64.608099310311,57.1553215877437,66.6350053854862,68.4865061272328,69.0789863645917,54.6294847863715,67.9018216824707,55.058253379197,56.1730517205434],"y":[17,17,17,17,17,13.6696215487135,13.0131705319409,12.8767451223991,12.5825043214512,12.1261827104679,12.1261827104679,11.0384654647977,9.71470684500736,9.39172055769546,9.11900456255404,8.4139184891172,7.69633586553802,7.41629725471299,7.365296609741,7.02398799230918,7.02398799230918,6.69191783073464,6.69191783073464,6.52519827481075,6.10919177353359,5.74767307898357,5.64899256089987,5.60201126925571,5.21271687097111,5.16811542483795,5.16811542483795,4.9732865509012,4.9210374496024,4.90061253434855,4.89733066044828,4.89733066044828,4.83366755342191,4.78351169336239,4.77270162326782,4.50212385346369,4.50012273085056,4.35028205229351,4.20526865584369,4.1065731159348,4.09829387921115,4.06483992872688,3.77486683622795,3.77312102083897,3.77312102083897,3.76203761312864,3.67676052589601,3.27551880379003,3.23245398946117,2.98822397144041,2.92995539784727,2.7161851052682,2.69154345726652,2.63813648021224,2.50684539960629,2.42627936618353,2.38938768107008,2.2617368554817,2.17785079562097,2.03728826796421,1.92565243504763,1.9086693224174,1.88646231916526,1.8130932629552,1.68803695537846,1.68427504580659,1.61005684128889,1.61005684128889,1.49164401717313,1.46033330171411,1.45788508488323,1.42964482503869],"text":["GO:0005763 (28) <br> mitochondrial small ribosomal subunit <br> 1.576e-23","GO:0000314 (28) <br> organellar small ribosomal subunit <br> 1.576e-23","GO:0015935 (76) <br> small ribosomal subunit <br> 4.052e-18","GO:0005761 (86) <br> mitochondrial ribosome <br> 1.718e-17","GO:0000313 (86) <br> organellar ribosome <br> 1.718e-17","GO:0015629 (501) <br> actin cytoskeleton <br> 2.140e-14","GO:0016459 (57) <br> myosin complex <br> 9.701e-14","GO:0044391 (189) <br> ribosomal subunit <br> 1.328e-13","GO:0005840 (520) <br> ribosome <br> 2.615e-13","GO:0043228 (6362) <br> non-membrane-bounded organelle <br> 7.479e-13","GO:0043232 (6362) <br> intracellular non-membrane-bounded organelle <br> 7.479e-13","GO:0098798 (277) <br> mitochondrial protein-containing complex <br> 9.152e-12","GO:0005743 (500) <br> mitochondrial inner membrane <br> 1.929e-10","GO:0032991 (7228) <br> protein-containing complex <br> 4.058e-10","GO:0019866 (562) <br> organelle inner membrane <br> 7.603e-10","GO:0005759 (483) <br> mitochondrial matrix <br> 3.856e-09","GO:0031966 (745) <br> mitochondrial membrane <br> 2.012e-08","GO:0016460 (20) <br> myosin II complex <br> 3.834e-08","GO:0005740 (796) <br> mitochondrial envelope <br> 4.312e-08","GO:0097517 (71) <br> contractile actin filament bundle <br> 9.463e-08","GO:0001725 (71) <br> stress fiber <br> 9.463e-08","GO:0032432 (79) <br> actin filament bundle <br> 2.033e-07","GO:0042641 (79) <br> actomyosin <br> 2.033e-07","GO:0099080 (1330) <br> supramolecular complex <br> 2.984e-07","GO:0097513 (3) <br> myosin II filament <br> 7.777e-07","GO:0043292 (242) <br> contractile fiber <br> 1.788e-06","GO:0099512 (988) <br> supramolecular fiber <br> 2.244e-06","GO:0099081 (996) <br> supramolecular polymer <br> 2.500e-06","GO:0005903 (105) <br> brush border <br> 6.127e-06","GO:0031975 (1245) <br> envelope <br> 6.790e-06","GO:0031967 (1245) <br> organelle envelope <br> 6.790e-06","GO:0070062 (2110) <br> extracellular exosome <br> 1.063e-05","GO:0005856 (2334) <br> cytoskeleton <br> 1.199e-05","GO:1903561 (2132) <br> extracellular vesicle <br> 1.257e-05","GO:0065010 (2133) <br> extracellular membrane-bounded organelle <br> 1.267e-05","GO:0043230 (2133) <br> extracellular organelle <br> 1.267e-05","GO:0005862 (4) <br> muscle thin filament tropomyosin <br> 1.467e-05","GO:0043227 (15557) <br> membrane-bounded organelle <br> 1.646e-05","GO:0030016 (232) <br> myofibril <br> 1.688e-05","GO:0030863 (110) <br> cortical cytoskeleton <br> 3.147e-05","GO:1990904 (2377) <br> ribonucleoprotein complex <br> 3.161e-05","GO:0043229 (15870) <br> intracellular organelle <br> 4.464e-05","GO:0016461 (8) <br> unconventional myosin complex <br> 6.233e-05","GO:0098862 (161) <br> cluster of actin-based cell projections <br> 7.824e-05","GO:0016593 (8) <br> Cdc73/Paf1 complex <br> 7.975e-05","GO:0005938 (319) <br> cell cortex <br> 8.613e-05","GO:0070013 (6569) <br> intracellular organelle lumen <br> 1.679e-04","GO:0031974 (6570) <br> membrane-enclosed lumen <br> 1.686e-04","GO:0043233 (6570) <br> organelle lumen <br> 1.686e-04","GO:0005739 (1674) <br> mitochondrion <br> 1.730e-04","GO:0030017 (211) <br> sarcomere <br> 2.105e-04","GO:0043226 (16674) <br> organelle <br> 5.303e-04","GO:0005662 (13) <br> DNA replication factor A complex <br> 5.855e-04","GO:0005737 (12193) <br> cytoplasm <br> 1.027e-03","GO:0032982 (22) <br> myosin filament <br> 1.175e-03","GO:0005925 (420) <br> focal adhesion <br> 1.922e-03","GO:0030055 (428) <br> cell-substrate junction <br> 2.034e-03","GO:0005884 (114) <br> actin filament <br> 2.301e-03","GO:0043601 (22) <br> nuclear replisome <br> 3.113e-03","GO:0005622 (17758) <br> intracellular anatomical structure <br> 3.747e-03","GO:0030894 (24) <br> replisome <br> 4.080e-03","GO:0005865 (22) <br> striated muscle thin filament <br> 5.473e-03","GO:0030864 (80) <br> cortical actin cytoskeleton <br> 6.640e-03","GO:0036379 (26) <br> myofilament <br> 9.177e-03","GO:0090734 (97) <br> site of DNA damage <br> 1.187e-02","GO:0005615 (3353) <br> extracellular space <br> 1.234e-02","GO:0043596 (35) <br> nuclear replication fork <br> 1.299e-02","GO:0070161 (839) <br> anchoring junction <br> 1.538e-02","GO:0016020 (9818) <br> membrane <br> 2.051e-02","GO:0097165 (6) <br> nuclear stress granule <br> 2.069e-02","GO:0099143 (1) <br> presynaptic actin cytoskeleton <br> 2.454e-02","GO:0120220 (1) <br> basal body patch <br> 2.454e-02","GO:0002102 (30) <br> podosome <br> 3.224e-02","GO:0098685 (79) <br> Schaffer collateral - CA1 synapse <br> 3.465e-02","GO:0005654 (4161) <br> nucleoplasm <br> 3.484e-02","GO:0008023 (56) <br> transcription elongation factor complex <br> 3.718e-02"],"key":["GO:0005763","GO:0000314","GO:0015935","GO:0005761","GO:0000313","GO:0015629","GO:0016459","GO:0044391","GO:0005840","GO:0043228","GO:0043232","GO:0098798","GO:0005743","GO:0032991","GO:0019866","GO:0005759","GO:0031966","GO:0016460","GO:0005740","GO:0097517","GO:0001725","GO:0032432","GO:0042641","GO:0099080","GO:0097513","GO:0043292","GO:0099512","GO:0099081","GO:0005903","GO:0031975","GO:0031967","GO:0070062","GO:0005856","GO:1903561","GO:0065010","GO:0043230","GO:0005862","GO:0043227","GO:0030016","GO:0030863","GO:1990904","GO:0043229","GO:0016461","GO:0098862","GO:0016593","GO:0005938","GO:0070013","GO:0031974","GO:0043233","GO:0005739","GO:0030017","GO:0043226","GO:0005662","GO:0005737","GO:0032982","GO:0005925","GO:0030055","GO:0005884","GO:0043601","GO:0005622","GO:0030894","GO:0005865","GO:0030864","GO:0036379","GO:0090734","GO:0005615","GO:0043596","GO:0070161","GO:0016020","GO:0097165","GO:0099143","GO:0120220","GO:0002102","GO:0098685","GO:0005654","GO:0008023"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(16,150,24,1)","opacity":0.8,"size":[14.8076606210532,14.8076606210532,16.3518900796216,16.5300567869452,16.5300567869452,18.8425751679818,15.9271363673814,17.6111842782733,18.8876038104453,21.6584390995261,21.6584390995261,18.1066675865569,18.8401543593181,21.7882302458994,18.9811360260309,18.7981810116601,19.3158437368831,14.236059602826,19.3934268963151,16.2527174191542,16.2527174191542,16.4079602522644,16.4079602522644,19.9822957222753,10.1117826215735,17.9335645209027,19.6439698152882,19.6532439049367,16.8126211235897,19.9077396571612,19.9077396571612,20.4940116687561,20.6038104257887,20.5053327583453,20.5058443926346,20.5058443926346,10.8927092497762,22.5489458921952,17.8790496128476,16.8775971961113,20.6236005193286,22.5683035768283,12.4913603540423,17.3979891958427,12.4913603540423,18.2853723884657,21.6910911315793,21.6912462183155,21.6912462183155,20.2393468497534,17.7557070639715,22.616243094119,13.4550755410977,22.3105369168068,14.4010961626078,18.6273853717876,18.6505580804722,16.9272679216699,14.4010961626078,22.6771653543307,14.5495547096524,14.4010961626078,16.4261245775268,14.684338682381,16.7011784771014,20.9922784979403,15.1709310584103,19.4547973269953,22.0959451721519,11.8663149965605,3.77952755905512,3.77952755905512,14.9212437498666,16.4079602522644,21.2196758163161,15.9005174408981],"symbol":"circle","line":{"width":1.88976377952756,"color":"rgba(16,150,24,1)"}},"hoveron":"points","set":"SharedData2a0dd1ba","name":"GO:CC","legendgroup":"GO:CC","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","_isNestedKey":false,"frame":null},{"x":[37.3800599541577,8.58260864315624,3.7343166644195,3.78887943588438,2.11302288374869,3.80836613997898,10.106468903354,29.7295799266173,2.53783303301099,3.69144591541137,28.3577159583574,2.78336550460296,2.78726284542188,40.2134267295127,48.4134318125208,9.6154039601701,3.61739643985189,3.71093261950597,38.0036344851849,9.3075140354754],"y":[8.36680978212238,6.93526711324444,6.76820950432779,6.174846311033,5.92749935876715,5.05115924422413,4.37364591885118,3.55560618159138,3.51061015025712,3.30385797175887,3.2782974797843,2.69681954006981,2.69681954006981,2.66638829841009,2.59092959705218,2.43730412778294,2.35075375728446,1.49019062622132,1.41892733919077,1.35419545426748],"text":["GO:0051015 (218) <br> actin filament binding <br> 4.297e-09","GO:0005198 (980) <br> structural molecule activity <br> 1.161e-07","GO:0003735 (449) <br> structural constituent of ribosome <br> 1.705e-07","GO:0003774 (115) <br> cytoskeletal motor activity <br> 6.686e-07","GO:0000146 (31) <br> microfilament motor activity <br> 1.182e-06","GO:0003779 (449) <br> actin binding <br> 8.889e-06","GO:0008307 (43) <br> structural constituent of muscle <br> 4.230e-05","GO:0044877 (1275) <br> protein-containing complex binding <br> 2.782e-04","GO:0000993 (36) <br> RNA polymerase II complex binding <br> 3.086e-04","GO:0003723 (3142) <br> RNA binding <br> 4.968e-04","GO:0043175 (41) <br> RNA polymerase core enzyme binding <br> 5.269e-04","GO:0001098 (57) <br> basal transcription machinery binding <br> 2.010e-03","GO:0001099 (57) <br> basal RNA polymerase II transcription machinery binding <br> 2.010e-03","GO:0070063 (58) <br> RNA polymerase binding <br> 2.156e-03","GO:0140657 (550) <br> ATP-dependent activity <br> 2.565e-03","GO:0008092 (995) <br> cytoskeletal protein binding <br> 3.653e-03","GO:0003697 (123) <br> single-stranded DNA binding <br> 4.459e-03","GO:0003729 (350) <br> mRNA binding <br> 3.235e-02","GO:0051959 (28) <br> dynein light intermediate chain binding <br> 3.811e-02","GO:0005516 (203) <br> calmodulin binding <br> 4.424e-02"],"key":["GO:0051015","GO:0005198","GO:0003735","GO:0003774","GO:0000146","GO:0003779","GO:0008307","GO:0044877","GO:0000993","GO:0003723","GO:0043175","GO:0001098","GO:0001099","GO:0070063","GO:0140657","GO:0008092","GO:0003697","GO:0003729","GO:0051959","GO:0005516"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(220,57,18,1)","opacity":0.8,"size":[17.7982573587569,19.6346148346981,18.709222965806,16.939384735405,14.9748217342642,18.709222965806,15.4960679489772,19.9346590379887,15.2159720292035,20.9232306479053,15.4216483730041,15.9271363673814,15.9271363673814,15.9532356375401,18.9552033869835,19.65208902109,17.0323171228931,18.4015814224809,14.8076606210532,17.7051462729756],"symbol":"circle","line":{"width":1.88976377952756,"color":"rgba(220,57,18,1)"}},"hoveron":"points","set":"SharedData2a0dd1ba","name":"GO:MF","legendgroup":"GO:MF","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","_isNestedKey":false,"frame":null},{"x":[199.020054891063,199.207453392592,198.719436461526,198.680395107041,198.723340596975,199.68766205276,199.695470323657,199.016150755614,198.945876317541,199.121562412724],"y":[5.54948197468462,3.74957543441686,3.20666624431807,2.40795367415823,2.18194537141429,2.03882633507394,1.99160739226963,1.77148184625341,1.6555645632333,1.35942062308011],"text":["KEGG:04530 (169) <br> Tight junction <br> 2.822e-06","KEGG:04810 (216) <br> Regulation of actin cytoskeleton <br> 1.780e-04","KEGG:03430 (23) <br> Mismatch repair <br> 6.213e-04","KEGG:03030 (36) <br> DNA replication <br> 3.909e-03","KEGG:03440 (41) <br> Homologous recombination <br> 6.577e-03","KEGG:05410 (90) <br> Hypertrophic cardiomyopathy <br> 9.145e-03","KEGG:05414 (95) <br> Dilated cardiomyopathy <br> 1.020e-02","KEGG:04520 (71) <br> Adherens junction <br> 1.692e-02","KEGG:04270 (134) <br> Vascular smooth muscle contraction <br> 2.210e-02","KEGG:04670 (114) <br> Leukocyte transendothelial migration <br> 4.371e-02"],"key":["KEGG:04530","KEGG:04810","KEGG:03430","KEGG:03030","KEGG:03440","KEGG:05410","KEGG:05414","KEGG:04520","KEGG:04270","KEGG:04670"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(221,68,119,1)","opacity":0.8,"size":[17.4628187386514,17.7862542708241,14.4771970381861,15.2159720292035,15.4216483730041,16.5949594227646,16.6717210962517,16.2527174191542,17.1497447320504,16.9272679216699],"symbol":"circle","line":{"width":1.88976377952756,"color":"rgba(221,68,119,1)"}},"hoveron":"points","set":"SharedData2a0dd1ba","name":"KEGG","legendgroup":"KEGG","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","_isNestedKey":false,"frame":null},{"x":[2,49.9528814359943],"y":[-1,-1],"text":"","type":"scatter","mode":"lines","line":{"width":11.3385826771654,"color":"rgba(220,57,18,1)","dash":"solid"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[53.8499055266887,71.2657061880019],"y":[-1,-1],"text":"","type":"scatter","mode":"lines","line":{"width":11.3385826771654,"color":"rgba(16,150,24,1)","dash":"solid"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[75.1627302786963,193.967406707605],"y":[-1,-1],"text":"","type":"scatter","mode":"lines","line":{"width":11.3385826771654,"color":"rgba(255,153,0,1)","dash":"solid"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[197.864430798299,200.003897024091],"y":[-1,-1],"text":"","type":"scatter","mode":"lines","line":{"width":11.3385826771654,"color":"rgba(221,68,119,1)","dash":"solid"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[180],"y":[16.2],"text":"values above this threshold are capped","hovertext":"","textfont":{"size":7.55905511811024,"color":"rgba(190,190,190,1)"},"type":"scatter","mode":"text","hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[0,210],"y":[16,16],"text":"","type":"scatter","mode":"lines","line":{"width":0.755905511811024,"color":"rgba(190,190,190,1)","dash":"dash"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null}],"layout":{"margin":{"t":29.2835201328352,"r":6.6417600664176,"b":51.8931100989696,"l":61.0377750103778},"plot_bgcolor":"rgba(255,255,255,1)","paper_bgcolor":"rgba(255,255,255,1)","font":{"color":"rgba(0,0,0,1)","family":"","size":14.6118721461187},"xaxis":{"domain":[0,1],"automargin":true,"type":"linear","autorange":false,"range":[0,210],"tickmode":"array","ticktext":["GO:MF","GO:CC","GO:BP","KEGG"],"tickvals":[25.9764407179972,62.5578058573453,134.565068493151,198.934163911195],"categoryorder":"array","categoryarray":["GO:MF","GO:CC","GO:BP","KEGG"],"nticks":null,"ticks":"","tickcolor":null,"ticklen":3.65296803652968,"tickwidth":0,"showticklabels":true,"tickfont":{"color":"rgba(77,77,77,1)","family":"","size":10.6268161062682},"tickangle":-45,"showline":true,"linecolor":"rgba(190,190,190,1)","linewidth":0.132835201328352,"showgrid":false,"gridcolor":null,"gridwidth":0,"zeroline":false,"anchor":"y","title":"","hoverformat":".2f"},"yaxis":{"domain":[0,1],"automargin":true,"type":"linear","autorange":false,"range":[-1,18.5],"tickmode":"array","ticktext":["0","2","4","6","8","10","12","14",">16"],"tickvals":[0,2,4,6,8,10,12,14,16],"categoryorder":"array","categoryarray":["0","2","4","6","8","10","12","14",">16"],"nticks":null,"ticks":"outside","tickcolor":"rgba(190,190,190,1)","ticklen":3.65296803652968,"tickwidth":0.66417600664176,"showticklabels":true,"tickfont":{"color":"rgba(77,77,77,1)","family":"","size":11.689497716895},"tickangle":-0,"showline":true,"linecolor":"rgba(190,190,190,1)","linewidth":0.66417600664176,"showgrid":false,"gridcolor":null,"gridwidth":0,"zeroline":false,"anchor":"x","title":"","hoverformat":".2f"},"annotations":[{"text":"-log10(p-adj)","x":-0.0108706932337069,"y":0.5,"showarrow":false,"ax":0,"ay":0,"font":{"color":"rgba(0,0,0,1)","family":"","size":13.2835201328352},"xref":"paper","yref":"paper","textangle":-90,"xanchor":"right","yanchor":"center","annotationType":"axis"},{"text":"query_1","x":0.5,"y":1,"showarrow":false,"ax":0,"ay":0,"font":{"color":"rgba(169,169,169,1)","family":"","size":13.2835201328352},"xref":"paper","yref":"paper","textangle":-0,"xanchor":"center","yanchor":"bottom"}],"shapes":[{"type":"rect","fillcolor":null,"line":{"color":null,"width":0,"linetype":[]},"yref":"paper","xref":"paper","x0":0,"x1":1,"y0":0,"y1":1},{"type":"rect","fillcolor":null,"line":{"color":null,"width":0,"linetype":[]},"yref":"paper","xref":"paper","x0":0,"x1":1,"y0":0,"y1":24.9730178497302,"yanchor":1,"ysizemode":"pixel"}],"showlegend":false,"legend":{"bgcolor":"rgba(255,255,255,1)","bordercolor":"transparent","borderwidth":1.88976377952756,"font":{"color":"rgba(0,0,0,1)","family":"","size":11.689497716895}},"hovermode":"closest","barmode":"relative","dragmode":"zoom"},"config":{"doubleClick":"reset","showSendToCloud":false},"source":"A","attrs":{"4f068a748b2":{"x":{},"y":{},"text":{},"colour":{},"size":{},"alpha":{},"type":"scatter"},"4f051c911ca":{"x":{},"xend":{},"y":{},"yend":{}},"4f0274f1c47":{"x":{},"xend":{},"y":{},"yend":{}},"4f055a868df":{"x":{},"xend":{},"y":{},"yend":{}},"4f06afc2e90":{"x":{},"xend":{},"y":{},"yend":{}},"4f0370c3180":{"x":{},"y":{}},"4f07b2c75d3":{"yintercept":{}}},"cur_data":"4f068a748b2","visdat":{"4f068a748b2":["function (y) ","x"],"4f051c911ca":["function (y) ","x"],"4f0274f1c47":["function (y) ","x"],"4f055a868df":["function (y) ","x"],"4f06afc2e90":["function (y) ","x"],"4f0370c3180":["function (y) ","x"],"4f07b2c75d3":["function (y) ","x"]},"highlight":{"on":"plotly_click","off":"plotly_doubleclick","persistent":false,"dynamic":false,"color":null,"selectize":false,"defaultValues":null,"opacityDim":0.2,"selected":{"opacity":1},"debounce":0,"ctGroups":["SharedData2a0dd1ba"]},"shinyEvents":["plotly_hover","plotly_click","plotly_selected","plotly_relayout","plotly_brushed","plotly_brushing","plotly_clickannotation","plotly_doubleclick","plotly_deselect","plotly_afterplot","plotly_sunburstclick"],"base_url":"https://plot.ly"},"evals":[],"jsHooks":[]}</script>
</div>
<div id="table-results" class="section level4">
<h4>Table results</h4>
<p>The results can also be visualized with a table:</p>
<pre class="r"><code>if(is.null(GO)){
  print(&quot;NOT enough data.&quot;)
}else{
  GO$result %&gt;% 
    as.data.frame() %&gt;% 
    select(starts_with(c(&quot;source&quot;, &quot;term_name&quot;, &quot;p_value&quot;, &quot;term_size&quot;, &quot;intersection_size&quot;))) %&gt;%
    arrange(p_value) %&gt;%  
    mutate(p_value = format(p_value, scientific = T, digits = 2)) %&gt;% 
    DT::datatable(options = list(
    columnDefs = list(list(className = &#39;dt-center&#39;, targets = 1:6)),
    pageLength = 10)) %&gt;%
    formatStyle(&#39;source&#39;, backgroundColor = styleEqual(c(&quot;GO:BP&quot;, &quot;GO:MF&quot;, &quot;GO:CC&quot;, &quot;KEGG&quot;), c(&#39;orange&#39;, &#39;tomato&#39;, &#39;lightgreen&#39;, &#39;violet&#39;)))
}</code></pre>
<div id="htmlwidget-e4a421697d58abce2053" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-e4a421697d58abce2053">{"x":{"filter":"none","data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33","34","35","36","37","38","39","40","41","42","43","44","45","46","47","48","49","50","51","52","53","54","55","56","57","58","59","60","61","62","63","64","65","66","67","68","69","70","71","72","73","74","75","76","77","78","79","80","81","82","83","84","85","86","87","88","89","90","91","92","93","94","95","96","97","98","99","100","101","102","103","104","105","106","107","108","109","110","111","112","113","114","115","116","117","118","119","120","121","122","123","124","125","126","127","128","129","130","131","132","133","134","135","136","137","138","139","140","141","142","143"],["GO:CC","GO:CC","GO:CC","GO:CC","GO:CC","GO:CC","GO:CC","GO:CC","GO:CC","GO:CC","GO:CC","GO:CC","GO:CC","GO:CC","GO:CC","GO:CC","GO:MF","GO:CC","GO:CC","GO:CC","GO:BP","GO:CC","GO:CC","GO:MF","GO:BP","GO:MF","GO:CC","GO:CC","GO:BP","GO:BP","GO:CC","GO:MF","GO:CC","GO:MF","GO:BP","GO:BP","GO:CC","GO:BP","GO:CC","GO:CC","KEGG","GO:BP","GO:CC","GO:CC","GO:CC","GO:MF","GO:CC","GO:CC","GO:CC","GO:CC","GO:CC","GO:CC","GO:BP","GO:CC","GO:CC","GO:CC","GO:CC","GO:BP","GO:BP","GO:MF","GO:CC","GO:CC","GO:CC","GO:CC","GO:CC","GO:CC","GO:CC","GO:CC","GO:CC","KEGG","GO:CC","GO:MF","GO:MF","GO:BP","GO:MF","GO:MF","GO:CC","GO:CC","KEGG","GO:BP","GO:BP","GO:CC","GO:CC","GO:CC","GO:MF","GO:MF","GO:CC","GO:MF","GO:BP","GO:CC","GO:MF","GO:BP","GO:CC","GO:MF","GO:CC","KEGG","GO:CC","GO:MF","GO:BP","GO:BP","GO:BP","GO:CC","GO:BP","KEGG","GO:CC","KEGG","GO:CC","KEGG","GO:BP","GO:BP","GO:BP","GO:CC","GO:CC","GO:CC","GO:BP","GO:CC","GO:BP","KEGG","GO:BP","GO:BP","GO:BP","GO:CC","GO:CC","KEGG","GO:CC","GO:CC","GO:BP","GO:BP","GO:CC","GO:MF","GO:BP","GO:CC","GO:BP","GO:CC","GO:BP","GO:CC","GO:BP","GO:MF","GO:BP","KEGG","GO:MF","GO:BP","GO:BP"],[407,63,839,405,62,836,870,2330,479,2125,2129,3625,389,1568,980,403,9079,1437,871,386,7696,3387,183,1690,2404,446,1526,2057,12114,8842,3746,460,3383,30,15281,12445,2146,2494,3777,3747,297,22988,531,1445,1438,465,2739,491,4067,2715,2127,495,7681,2124,1021,1148,4452,12444,10011,2081,2126,872,3659,895,560,2716,1444,2130,385,345,1022,7116,139,7688,435,6764,2123,317,220,26382,9678,384,1566,549,202,203,1026,9806,24858,512,11910,9501,2178,1955,283,210,1162,416,24088,26936,26938,497,514,221,1149,468,1937,470,19392,12711,14404,3229,277,2173,2374,2761,12466,296,3664,8624,7602,849,3281,278,3756,3908,7697,2938,201,440,9343,3606,2305,311,159,597,2922,9239,8320,323,1876,4583,2876],["mitochondrial small ribosomal subunit","organellar small ribosomal subunit","small ribosomal subunit","mitochondrial ribosome","organellar ribosome","actin cytoskeleton","myosin complex","ribosomal subunit","ribosome","non-membrane-bounded organelle","intracellular non-membrane-bounded organelle","mitochondrial protein-containing complex","mitochondrial inner membrane","protein-containing complex","organelle inner membrane","mitochondrial matrix","actin filament binding","mitochondrial membrane","myosin II complex","mitochondrial envelope","actin filament-based movement","contractile actin filament bundle","stress fiber","structural molecule activity","translation","structural constituent of ribosome","actin filament bundle","actomyosin","peptide biosynthetic process","mitochondrial translation","supramolecular complex","cytoskeletal motor activity","myosin II filament","microfilament motor activity","regulation of mRNA processing","amide biosynthetic process","contractile fiber","peptide metabolic process","supramolecular fiber","supramolecular polymer","Tight junction","mitochondrial gene expression","brush border","envelope","organelle envelope","actin binding","extracellular exosome","cytoskeleton","extracellular vesicle","extracellular membrane-bounded organelle","extracellular organelle","muscle thin filament tropomyosin","actin filament-based process","membrane-bounded organelle","myofibril","cortical cytoskeleton","ribonucleoprotein complex","cellular amide metabolic process","cellular macromolecule biosynthetic process","structural constituent of muscle","intracellular organelle","unconventional myosin complex","cluster of actin-based cell projections","Cdc73/Paf1 complex","cell cortex","intracellular organelle lumen","membrane-enclosed lumen","organelle lumen","mitochondrion","Regulation of actin cytoskeleton","sarcomere","protein-containing complex binding","RNA polymerase II complex binding","actin cytoskeleton organization","RNA binding","RNA polymerase core enzyme binding","organelle","DNA replication factor A complex","Mismatch repair","regulation of mRNA metabolic process","response to cytokine","cytoplasm","myosin filament","focal adhesion","basal transcription machinery binding","basal RNA polymerase II transcription machinery binding","cell-substrate junction","RNA polymerase binding","organonitrogen compound biosynthetic process","actin filament","ATP-dependent activity","histone H2B ubiquitination","nuclear replisome","cytoskeletal protein binding","intracellular anatomical structure","DNA replication","replisome","single-stranded DNA binding","protein processing in phagocytic vesicle","regulation of protein processing in phagocytic vesicle","positive regulation of protein processing in phagocytic vesicle","striated muscle thin filament","endodermal cell fate commitment","Homologous recombination","cortical actin cytoskeleton","Hypertrophic cardiomyopathy","myofilament","Dilated cardiomyopathy","cellular response to cytokine stimulus","cellular macromolecule metabolic process","positive regulation of RNA export from nucleus","site of DNA damage","extracellular space","nuclear replication fork","mRNA polyadenylation","anchoring junction","RNA polyadenylation","Adherens junction","macromolecule biosynthetic process","positive regulation of nucleobase-containing compound transport","regulation of cell morphogenesis","membrane","nuclear stress granule","Vascular smooth muscle contraction","presynaptic actin cytoskeleton","basal body patch","muscle filament sliding","actin filament organization","podosome","mRNA binding","actin-myosin filament sliding","Schaffer collateral - CA1 synapse","nucleotide-excision repair","nucleoplasm","alternative mRNA splicing, via spliceosome","transcription elongation factor complex","organelle organization","dynein light intermediate chain binding","regulation of mRNA 3'-end processing","Leukocyte transendothelial migration","calmodulin binding","histone monoubiquitination","muscle contraction"],["1.6e-23","1.6e-23","4.1e-18","1.7e-17","1.7e-17","2.1e-14","9.7e-14","1.3e-13","2.6e-13","7.5e-13","7.5e-13","9.2e-12","1.9e-10","4.1e-10","7.6e-10","3.9e-09","4.3e-09","2.0e-08","3.8e-08","4.3e-08","7.3e-08","9.5e-08","9.5e-08","1.2e-07","1.4e-07","1.7e-07","2.0e-07","2.0e-07","2.1e-07","3.0e-07","3.0e-07","6.7e-07","7.8e-07","1.2e-06","1.3e-06","1.3e-06","1.8e-06","1.9e-06","2.2e-06","2.5e-06","2.8e-06","3.3e-06","6.1e-06","6.8e-06","6.8e-06","8.9e-06","1.1e-05","1.2e-05","1.3e-05","1.3e-05","1.3e-05","1.5e-05","1.6e-05","1.6e-05","1.7e-05","3.1e-05","3.2e-05","3.7e-05","4.2e-05","4.2e-05","4.5e-05","6.2e-05","7.8e-05","8.0e-05","8.6e-05","1.7e-04","1.7e-04","1.7e-04","1.7e-04","1.8e-04","2.1e-04","2.8e-04","3.1e-04","3.3e-04","5.0e-04","5.3e-04","5.3e-04","5.9e-04","6.2e-04","7.2e-04","9.0e-04","1.0e-03","1.2e-03","1.9e-03","2.0e-03","2.0e-03","2.0e-03","2.2e-03","2.3e-03","2.3e-03","2.6e-03","2.8e-03","3.1e-03","3.7e-03","3.7e-03","3.9e-03","4.1e-03","4.5e-03","4.9e-03","4.9e-03","4.9e-03","5.5e-03","5.7e-03","6.6e-03","6.6e-03","9.1e-03","9.2e-03","1.0e-02","1.1e-02","1.2e-02","1.2e-02","1.2e-02","1.2e-02","1.3e-02","1.5e-02","1.5e-02","1.7e-02","1.7e-02","1.7e-02","1.8e-02","1.8e-02","2.1e-02","2.1e-02","2.2e-02","2.5e-02","2.5e-02","2.5e-02","3.1e-02","3.2e-02","3.2e-02","3.4e-02","3.5e-02","3.5e-02","3.5e-02","3.7e-02","3.7e-02","3.8e-02","3.8e-02","4.1e-02","4.4e-02","4.4e-02","4.5e-02","4.8e-02"],[28,28,76,86,86,501,57,189,520,6362,6362,277,500,7228,562,483,218,745,20,796,132,71,71,980,713,449,79,79,738,79,1330,115,3,31,141,867,242,895,988,996,169,111,105,1245,1245,449,2110,2334,2132,2133,2133,4,800,15557,232,110,2377,1165,1564,43,15870,8,161,8,319,6569,6570,6570,1674,216,211,1275,36,711,3142,41,16674,13,23,318,922,12193,22,420,57,57,428,58,1698,114,550,12,22,995,17758,36,24,123,2,2,2,22,15,41,80,90,26,95,828,5746,5,97,3353,35,52,839,53,71,4880,6,305,9818,6,134,1,1,18,444,30,350,20,79,58,4161,79,56,3805,28,28,114,203,29,340],[11,11,11,11,11,18,9,11,14,40,40,11,12,44,12,11,11,12,5,12,9,7,7,17,12,10,7,7,12,7,18,7,3,5,8,12,9,12,15,15,9,7,6,12,12,11,19,21,19,19,19,3,14,50,4,6,15,12,17,5,50,3,6,3,8,35,35,35,12,8,7,15,4,12,22,4,50,3,4,8,7,36,3,3,4,4,3,4,12,5,8,3,3,12,56,4,3,5,2,2,2,3,3,4,4,2,3,2,6,31,2,4,19,3,4,3,4,2,25,2,6,31,2,5,1,1,3,8,2,6,3,2,4,23,4,3,23,2,3,2,5,3,7]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>source<\/th>\n      <th>source_order<\/th>\n      <th>term_name<\/th>\n      <th>p_value<\/th>\n      <th>term_size<\/th>\n      <th>intersection_size<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-center","targets":[1,2,3,4,5,6]},{"orderable":false,"targets":0}],"pageLength":10,"order":[],"autoWidth":false,"orderClasses":false,"rowCallback":"function(row, data, displayNum, displayIndex, dataIndex) {\nvar value=data[1]; $(this.api().cell(row, 1).node()).css({'background-color':value == \"GO:BP\" ? \"orange\" : value == \"GO:MF\" ? \"tomato\" : value == \"GO:CC\" ? \"lightgreen\" : value == \"KEGG\" ? \"violet\" : null});\n}"}},"evals":["options.rowCallback"],"jsHooks":[]}</script>
</div>
<div id="treemap" class="section level4 tabset tabset-fade">
<h4>Treemap</h4>
<div id="bp" class="section level5">
<h5>BP</h5>
<pre class="r"><code>if(sum(GO$result$source == &quot;GO:BP&quot;) &gt;= 5 &amp; !is.null(GO)){
  simMatrix &lt;- GO$result %&gt;% 
  as.data.frame() %&gt;%
  filter(source == &quot;GO:BP&quot;) %&gt;% 
  pull(term_id) %&gt;% 
  calculateSimMatrix(orgdb=&quot;org.Hs.eg.db&quot;, ont=&quot;BP&quot;, method=&quot;Rel&quot;)

scores &lt;- setNames(-log10(GO$result$p_value), GO$result$term_id)

reducedTerms &lt;- reduceSimMatrix(simMatrix,
                                scores,
                                threshold=0.7,
                                orgdb=&quot;org.Hs.eg.db&quot;)

treemapPlot(reducedTerms)
BP &lt;- T
}else{
  print(&quot;NOT enough data.&quot;)
  BP &lt;- F
}</code></pre>
<p><img src="figure/Turbo_bio_vs_Turbo.Rmd/BP-cond1-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="mf" class="section level5">
<h5>MF</h5>
<pre class="r"><code>if(sum(GO$result$source == &quot;GO:MF&quot;) &gt;= 5 &amp; !is.null(GO)){
simMatrix &lt;- GO$result %&gt;% 
  as.data.frame() %&gt;%
  filter(source == &quot;GO:MF&quot;) %&gt;% 
  pull(term_id) %&gt;% 
  calculateSimMatrix(orgdb=&quot;org.Hs.eg.db&quot;, ont=&quot;MF&quot;, method=&quot;Rel&quot;)

scores &lt;- setNames(-log10(GO$result$p_value), GO$result$term_id)

reducedTerms &lt;- reduceSimMatrix(simMatrix,
                                scores,
                                threshold=0.7,
                                orgdb=&quot;org.Hs.eg.db&quot;)

treemapPlot(reducedTerms)
MF &lt;- T
}else{
  print(&quot;NOT enough data.&quot;)
  MF &lt;- F
}</code></pre>
<p><img src="figure/Turbo_bio_vs_Turbo.Rmd/MF-cond1-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="cc" class="section level5">
<h5>CC</h5>
<pre class="r"><code>if(sum(GO$result$source == &quot;GO:CC&quot;) &gt;= 5 &amp; !is.null(GO)){
simMatrix &lt;- GO$result %&gt;% 
  as.data.frame() %&gt;%
  filter(source == &quot;GO:CC&quot;) %&gt;% 
  pull(term_id) %&gt;% 
  calculateSimMatrix(orgdb=&quot;org.Hs.eg.db&quot;, ont=&quot;CC&quot;, method=&quot;Rel&quot;)

scores &lt;- setNames(-log10(GO$result$p_value), GO$result$term_id)

reducedTerms &lt;- reduceSimMatrix(simMatrix,
                                scores,
                                threshold=0.7,
                                orgdb=&quot;org.Hs.eg.db&quot;)

treemapPlot(reducedTerms)
CC &lt;- T
}else{
  print(&quot;NOT enough data.&quot;)
  CC &lt;- F
}</code></pre>
<p><img src="figure/Turbo_bio_vs_Turbo.Rmd/CC-cond1-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="scatterplot" class="section level4">
<h4>ScatterPlot</h4>
<pre class="r"><code>if(BP | MF | CC){
scatterPlot(simMatrix, reducedTerms, size = &quot;score&quot;, labelSize = 5)
}else{
  print(&quot;NOT enough data.&quot;)
}</code></pre>
<p><img src="figure/Turbo_bio_vs_Turbo.Rmd/scatter-cond1-1.png" width="1440" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="turbo.-1" class="section level3 tabset tabset-fade">
<h3><strong>5.2. TURBO.</strong></h3>
<div id="gostplot-1" class="section level4">
<h4>Gostplot</h4>
<p>UP regulated proteins in TURBO. were used to generate these gene ontologies.</p>
<pre class="r"><code>Gname &lt;- results %&gt;% filter(significant &amp; get(paste0(test, &quot;_ratio&quot;)) &lt; 0) %&gt;% pull(name)

if(!params$Imputation){
  Uni &lt;- unique_pair(data_filt_unique, conditions = conditions, table = T) %&gt;% 
  filter(!!sym(paste0(&quot;unique_&quot;, cond_2)) == &quot;+&quot;) %&gt;% 
  pull(name) %&gt;% unique()

  Gname &lt;- c(Gname, Uni)
}

if(length(Gname) == 0){
  print(&quot;NOT enough data.&quot;)
  GO &lt;- NULL
}else{
  GO &lt;- gost(query = Gname, organism = &quot;hsapiens&quot;, sources = c(&quot;GO:BP&quot;, &quot;GO:MF&quot;, &quot;GO:CC&quot;, &quot;KEGG&quot;), ordered_query = T, evcodes = TRUE)
  if(is.null(GO)){
    print(&quot;NOT enough data.&quot;)
  }else{
    gostplot(GO, capped = TRUE, interactive = TRUE)
  }
}</code></pre>
<pre><code>[1] &quot;NOT enough data.&quot;</code></pre>
</div>
<div id="table-results-1" class="section level4">
<h4>Table results</h4>
<p>The results can also be visualized with a table:</p>
<pre class="r"><code>if(is.null(GO)){
  print(&quot;NOT enough data.&quot;)
}else{
  GO$result %&gt;% 
    as.data.frame() %&gt;% 
    select(starts_with(c(&quot;source&quot;, &quot;term_name&quot;, &quot;p_value&quot;, &quot;term_size&quot;, &quot;intersection_size&quot;))) %&gt;% 
    arrange(p_value) %&gt;%  
    mutate(p_value = format(p_value, scientific = T, digits = 2)) %&gt;% 
    DT::datatable(options = list(
    columnDefs = list(list(className = &#39;dt-center&#39;, targets = 1:6)),
    pageLength = 10)) %&gt;%
    formatStyle(&#39;source&#39;, backgroundColor = styleEqual(c(&quot;GO:BP&quot;, &quot;GO:MF&quot;, &quot;GO:CC&quot;, &quot;KEGG&quot;), c(&#39;orange&#39;, &#39;tomato&#39;, &#39;lightgreen&#39;, &#39;violet&#39;)))
}</code></pre>
<pre><code>[1] &quot;NOT enough data.&quot;</code></pre>
</div>
<div id="treemap-1" class="section level4 tabset tabset-fade">
<h4>Treemap</h4>
<div id="bp-1" class="section level5">
<h5>BP</h5>
<pre class="r"><code>if(sum(GO$result$source == &quot;GO:BP&quot;) &gt;= 5 &amp; !is.null(GO)){
  simMatrix &lt;- GO$result %&gt;% 
  as.data.frame() %&gt;%
  filter(source == &quot;GO:BP&quot;) %&gt;% 
  pull(term_id) %&gt;% 
  calculateSimMatrix(orgdb=&quot;org.Hs.eg.db&quot;, ont=&quot;BP&quot;, method=&quot;Rel&quot;)

scores &lt;- setNames(-log10(GO$result$p_value), GO$result$term_id)

reducedTerms &lt;- reduceSimMatrix(simMatrix,
                                scores,
                                threshold=0.7,
                                orgdb=&quot;org.Hs.eg.db&quot;)

treemapPlot(reducedTerms)
BP &lt;- T
}else{
  print(&quot;NOT enough data.&quot;)
  BP &lt;- F
}</code></pre>
<pre><code>[1] &quot;NOT enough data.&quot;</code></pre>
</div>
<div id="mf-1" class="section level5">
<h5>MF</h5>
<pre class="r"><code>if(sum(GO$result$source == &quot;GO:MF&quot;) &gt;= 5 &amp; !is.null(GO)){
simMatrix &lt;- GO$result %&gt;% 
  as.data.frame() %&gt;%
  filter(source == &quot;GO:MF&quot;) %&gt;% 
  pull(term_id) %&gt;% 
  calculateSimMatrix(orgdb=&quot;org.Hs.eg.db&quot;, ont=&quot;MF&quot;, method=&quot;Rel&quot;)

scores &lt;- setNames(-log10(GO$result$p_value), GO$result$term_id)

reducedTerms &lt;- reduceSimMatrix(simMatrix,
                                scores,
                                threshold=0.7,
                                orgdb=&quot;org.Hs.eg.db&quot;)

treemapPlot(reducedTerms)
MF &lt;- T
}else{
  print(&quot;NOT enough data.&quot;)
  MF &lt;- F
}</code></pre>
<pre><code>[1] &quot;NOT enough data.&quot;</code></pre>
</div>
<div id="cc-1" class="section level5">
<h5>CC</h5>
<pre class="r"><code>if(sum(GO$result$source == &quot;GO:CC&quot;) &gt;= 5 &amp; !is.null(GO)){
simMatrix &lt;- GO$result %&gt;% 
  as.data.frame() %&gt;%
  filter(source == &quot;GO:CC&quot;) %&gt;% 
  pull(term_id) %&gt;% 
  calculateSimMatrix(orgdb=&quot;org.Hs.eg.db&quot;, ont=&quot;CC&quot;, method=&quot;Rel&quot;)

scores &lt;- setNames(-log10(GO$result$p_value), GO$result$term_id)

reducedTerms &lt;- reduceSimMatrix(simMatrix,
                                scores,
                                threshold=0.7,
                                orgdb=&quot;org.Hs.eg.db&quot;)

treemapPlot(reducedTerms)
CC &lt;- T
}else{
  print(&quot;NOT enough data.&quot;)
  CC &lt;- F
}</code></pre>
<pre><code>[1] &quot;NOT enough data.&quot;</code></pre>
</div>
</div>
<div id="scatterplot-1" class="section level4">
<h4>ScatterPlot</h4>
<pre class="r"><code>if(BP | MF | CC){
scatterPlot(simMatrix, reducedTerms, size = &quot;score&quot;, labelSize = 5)
}else{
  print(&quot;NOT enough data.&quot;)
}</code></pre>
<pre><code>[1] &quot;NOT enough data.&quot;</code></pre>
</div>
</div>
</div>
<div id="network-analysis" class="section level2">
<h2><strong>6. Network analysis</strong></h2>
<p>Overview from <strong>STRING Database</strong>: In STRING, each protein-protein interaction is annotated with one or more ‘scores’. Importantly, these scores do <strong>not</strong> indicate the strength or the specificity of the interaction. Instead, they are indicators of <strong>confidence</strong>, i.e. how likely STRING judges an interaction to be true, given the available evidence. All scores rank from 0 to 1, with 1 being the highest possible confidence. A score of 0.5 would indicate that roughly every second interaction might be erroneous (i.e., a false positive).</p>
<pre class="r"><code># if(leng_dep != 0){
#   
#   string_db &lt;- STRINGdb$new(version=&quot;11&quot;, species=9606, score_threshold=400)
# 
#   string_mapped &lt;- string_db$map(string_res, &quot;Gene.name&quot;, removeUnmappedRows = TRUE, quiet = T)
# 
#   hits &lt;- string_mapped$STRING_id[1:leng_dep]
# 
#   string_db$plot_network(hits, add_link = T)
# }else{
#   print(&quot;NOT enough data.&quot;)
# }</code></pre>
<div id="cytoscape-network" class="section level3">
<h3><strong>6.1. Cytoscape Network</strong></h3>
<p>Cytoscape is an open source software platform for visualizing complex networks and integrating these with any type of attribute data. In this presentation all proteins showing at least one confidence level from the STRING database are shown. The <strong>color</strong> code is relative to <strong>fold change</strong> and the <strong>size</strong> is directly proportional to the <strong>significance (p-value)</strong>.</p>
<pre class="r"><code>print(&quot;Enable Cytoscape option to see this results.&quot;)</code></pre>
<pre><code>[1] &quot;Enable Cytoscape option to see this results.&quot;</code></pre>
<pre class="r"><code>if(n_edge != 0){
  knitr::include_graphics(&quot;assets/network_FC_pvalue.png&quot;,  error = F)
}else{
  print(&quot;NOT enough data.&quot;)
}</code></pre>
</div>
<div id="cytoscape-enrichmentmap" class="section level3">
<h3><strong>6.2. Cytoscape EnrichmentMap</strong></h3>
<p>An <strong>enrichment map</strong> is a different sort of network. Instead of nodes representing genes, nodes represent pathways or functions. Edges between these pathways or functions represent shared genes or pathway crosstalk. An enrichment map is a way to visualize your enrichment results to help reduce redundancy and uncover main themes.</p>
<pre class="r"><code>print(&quot;Enable Cytoscape option to see this results.&quot;)</code></pre>
<pre><code>[1] &quot;Enable Cytoscape option to see this results.&quot;</code></pre>
<pre class="r"><code>Gname &lt;- results %&gt;% filter(significant) %&gt;% pull(name)

if(!params$Imputation){
  
  Uni &lt;- unique_pair(data_filt_unique, conditions = conditions, table = T) %&gt;% 
  pull(name) %&gt;% 
  unique()

  Gname &lt;- c(Gname, Uni)
}

if(length(Gname) == 0){
  
  print(&quot;NOT enough data.&quot;)
  
}else{
  
  GO &lt;- gost(query = Gname, organism = &quot;hsapiens&quot;, sources = c(&quot;GO:BP&quot;, &quot;GO:MF&quot;, &quot;GO:CC&quot;, &quot;KEGG&quot;), ordered_query = T, evcodes = TRUE)
  
  if(is.null(GO)){
    
    print(&quot;NOT enough data.&quot;)

    enrich_map = F
    
  }else{
    
    enrich_map = T
    
    gem &lt;- GO$result[,c(&quot;term_id&quot;, &quot;term_name&quot;, &quot;p_value&quot;, &quot;intersection&quot;)]
    colnames(gem) &lt;- c(&quot;GO.ID&quot;, &quot;Description&quot;, &quot;p.Val&quot;, &quot;Genes&quot;)
    gem$FDR &lt;- gem$p.Val
    gem$Phenotype = &quot;+1&quot;
    gem &lt;- gem[,c(&quot;GO.ID&quot;, &quot;Description&quot;, &quot;p.Val&quot;, &quot;FDR&quot;, &quot;Phenotype&quot;, &quot;Genes&quot;)]
    head(gem)
  
    write.table(gem, here::here(&quot;output/enrichmentmap.txt&quot;),col.name=TRUE,sep=&quot;\t&quot;,row.names=FALSE,quote=FALSE)
    
  }
}</code></pre>
<pre class="r"><code>if(enrich_map){
  knitr::include_graphics(&quot;assets/network_enrichmentmap.png&quot;,  error = F)
}else{
  print(&quot;NOT enough data.&quot;)
}</code></pre>
</div>
</div>
<div id="section" class="section level2 unnumbered">
<h2></h2>
<pre class="r"><code>saveWorkbook(excel, here::here(paste0(&quot;output/results_table_&quot;, test, &quot;.xlsx&quot;)), overwrite = T)</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.0.3 (2020-10-10)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 19042)

Matrix products: default

locale:
[1] LC_COLLATE=Italian_Italy.1252  LC_CTYPE=Italian_Italy.1252   
[3] LC_MONETARY=Italian_Italy.1252 LC_NUMERIC=C                  
[5] LC_TIME=Italian_Italy.1252    

attached base packages:
[1] parallel  stats4    stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] forcats_0.5.1               stringr_1.4.0              
 [3] dplyr_1.0.7                 purrr_0.3.4                
 [5] readr_1.4.0                 tidyr_1.1.3                
 [7] tibble_3.1.2                tidyverse_1.3.1            
 [9] openxlsx_4.2.4              here_1.0.1                 
[11] RColorBrewer_1.1-2          RCy3_2.10.2                
[13] STRINGdb_2.2.2              naniar_0.6.1               
[15] visdat_0.5.3                ggrepel_0.9.1              
[17] ggplot2_3.3.5               UpSetR_1.4.0               
[19] ggstatsplot_0.8.0           datapasta_3.1.0            
[21] patchwork_1.1.1             DT_0.18                    
[23] rrvgo_1.2.0                 org.Hs.eg.db_3.12.0        
[25] AnnotationDbi_1.52.0        gprofiler2_0.2.0           
[27] enrichR_3.0                 preprocessCore_1.52.1      
[29] SummarizedExperiment_1.20.0 Biobase_2.50.0             
[31] GenomicRanges_1.42.0        GenomeInfoDb_1.26.2        
[33] IRanges_2.24.1              S4Vectors_0.28.1           
[35] BiocGenerics_0.36.0         MatrixGenerics_1.2.1       
[37] matrixStats_0.59.0          DEP_1.12.0                 
[39] MBQN_2.2.0                  workflowr_1.6.2            

loaded via a namespace (and not attached):
  [1] rappdirs_0.3.3            R.methodsS3_1.8.1        
  [3] coda_0.19-4               bit64_4.0.5              
  [5] knitr_1.33                DelayedArray_0.16.2      
  [7] R.utils_2.10.1            data.table_1.14.0        
  [9] RCurl_1.98-1.3            doParallel_1.0.16        
 [11] generics_0.1.0            RSQLite_2.2.7            
 [13] correlation_0.6.1         chron_2.3-56             
 [15] bit_4.0.4                 lubridate_1.7.10         
 [17] xml2_1.3.2                httpuv_1.6.1             
 [19] assertthat_0.2.1          WRS2_1.1-2               
 [21] xfun_0.24                 hms_1.1.0                
 [23] jquerylib_0.1.4           evaluate_0.14            
 [25] promises_1.2.0.1          fansi_0.5.0              
 [27] readxl_1.3.1              caTools_1.18.2           
 [29] dbplyr_2.1.1              igraph_1.2.6             
 [31] DBI_1.1.1                 htmlwidgets_1.5.3        
 [33] reshape_0.8.8             kSamples_1.2-9           
 [35] Rmpfr_0.8-4               paletteer_1.3.0          
 [37] hash_2.2.6.1              ellipsis_0.3.2           
 [39] crosstalk_1.1.1           backports_1.2.1          
 [41] insight_0.14.2            gridBase_0.4-7           
 [43] vctrs_0.3.8               imputeLCMD_2.0           
 [45] Cairo_1.5-12.2            cachem_1.0.5             
 [47] withr_2.4.2               treemap_2.4-2            
 [49] cluster_2.1.0             lazyeval_0.2.2           
 [51] crayon_1.4.1              labeling_0.4.2           
 [53] pkgconfig_2.0.3           SuppDists_1.1-9.5        
 [55] slam_0.1-48               statsExpressions_1.1.0   
 [57] ProtGenerics_1.22.0       wordcloud_2.6            
 [59] rlang_0.4.11              RJSONIO_1.3-1.4          
 [61] lifecycle_1.0.0           MatrixModels_0.5-0       
 [63] sandwich_3.0-1            affyio_1.60.0            
 [65] BiocFileCache_1.14.0      modelr_0.1.8             
 [67] cellranger_1.1.0          rprojroot_2.0.2          
 [69] graph_1.68.0              Matrix_1.2-18            
 [71] mc2d_0.1-21               zoo_1.8-9                
 [73] reprex_2.0.0              whisker_0.4              
 [75] GlobalOptions_0.1.2       pheatmap_1.0.12          
 [77] png_0.1-7                 viridisLite_0.4.0        
 [79] rjson_0.2.20              mzR_2.24.1               
 [81] PMCMRplus_1.9.0           parameters_0.14.0        
 [83] bitops_1.0-7              shinydashboard_0.7.1     
 [85] R.oo_1.24.0               KernSmooth_2.23-17       
 [87] blob_1.2.1                shape_1.4.6              
 [89] multcompView_0.1-8        tmvtnorm_1.4-10          
 [91] ggsignif_0.6.2            scales_1.1.1             
 [93] memoise_2.0.0             magrittr_2.0.1           
 [95] plyr_1.8.6                gplots_3.1.1             
 [97] zlibbioc_1.36.0           compiler_4.0.3           
 [99] plotrix_3.8-1             pcaMethods_1.82.0        
[101] clue_0.3-59               cli_3.0.0                
[103] affy_1.68.0               XVector_0.30.0           
[105] pbapply_1.4-3             MASS_7.3-53              
[107] tidyselect_1.1.1          vsn_3.58.0               
[109] stringi_1.6.2             highr_0.9                
[111] yaml_2.2.1                GOSemSim_2.16.1          
[113] norm_1.0-9.5              MALDIquant_1.19.3        
[115] grid_4.0.3                sass_0.4.0               
[117] tools_4.0.3               rstudioapi_0.13          
[119] circlize_0.4.13           foreach_1.5.1            
[121] git2r_0.28.0              gridExtra_2.3            
[123] ipmisc_6.0.2              pairwiseComparisons_3.1.6
[125] farver_2.1.0              mzID_1.28.0              
[127] digest_0.6.27             BiocManager_1.30.16      
[129] shiny_1.6.0               proto_1.0.0              
[131] BWStest_0.2.2             Rcpp_1.0.6               
[133] broom_0.7.8               BayesFactor_0.9.12-4.2   
[135] performance_0.7.2         later_1.2.0              
[137] ncdf4_1.17                httr_1.4.2               
[139] MSnbase_2.15.7            ComplexHeatmap_2.6.2     
[141] effectsize_0.4.5          colorspace_2.0-2         
[143] rvest_1.0.0               XML_3.99-0.6             
[145] fs_1.5.0                  rematch2_2.1.2           
[147] gmm_1.6-6                 plotly_4.9.4.1           
[149] xtable_1.8-4              gmp_0.6-2                
[151] jsonlite_1.7.2            zeallot_0.1.0            
[153] NLP_0.2-1                 R6_2.5.0                 
[155] tm_0.7-8                  gsubfn_0.7               
[157] pillar_1.6.1              htmltools_0.5.1.1        
[159] mime_0.11                 glue_1.4.2               
[161] fastmap_1.1.0             BiocParallel_1.24.1      
[163] codetools_0.2-16          mvtnorm_1.1-2            
[165] utf8_1.2.1                lattice_0.20-41          
[167] bslib_0.2.5.1             sqldf_0.4-11             
[169] curl_4.3.2                gtools_3.9.2             
[171] zip_2.2.0                 GO.db_3.12.1             
[173] limma_3.46.0              rmarkdown_2.9            
[175] munsell_0.5.0             GetoptLong_1.0.5         
[177] GenomeInfoDbData_1.2.4    iterators_1.0.13         
[179] impute_1.64.0             haven_2.4.1              
[181] gtable_0.3.0              bayestestR_0.10.0        </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
