<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Fabio Bedin | MS-Unit at IEO" />


<title>LFQ_Analysis</title>

<script src="site_libs/jquery-3.5.1/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/pagedtable-1.1/css/pagedtable.css" rel="stylesheet" />
<script src="site_libs/pagedtable-1.1/js/pagedtable.js"></script>
<script src="site_libs/htmlwidgets-1.5.3/htmlwidgets.js"></script>
<link href="site_libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="site_libs/datatables-binding-0.18/datatables.js"></script>
<link href="site_libs/dt-core-1.10.20/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="site_libs/dt-core-1.10.20/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="site_libs/dt-core-1.10.20/js/jquery.dataTables.min.js"></script>
<link href="site_libs/crosstalk-1.1.1/css/crosstalk.css" rel="stylesheet" />
<script src="site_libs/crosstalk-1.1.1/js/crosstalk.min.js"></script>
<script src="site_libs/plotly-binding-4.9.4.1/plotly.js"></script>
<script src="site_libs/typedarray-0.1/typedarray.min.js"></script>
<link href="site_libs/plotly-htmlwidgets-css-1.57.1/plotly-htmlwidgets.css" rel="stylesheet" />
<script src="site_libs/plotly-main-1.57.1/plotly-latest.min.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Template_LFQ</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/FabioBedin/GN_MR_ZC3H4">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">LFQ_Analysis</h1>
<h4 class="author">Fabio Bedin | MS-Unit at IEO</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-02-16
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>GN_MR_ZC3H4/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20220215code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20220215)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20220215code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20220215)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomFabioBedinGNMRZC3H4tree3e779377b92dbb0fcbea3b5bb9238beb1772ab6dtargetblank3e77937a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/FabioBedin/GN_MR_ZC3H4/tree/3e779377b92dbb0fcbea3b5bb9238beb1772ab6d" target="_blank">3e77937</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomFabioBedinGNMRZC3H4tree3e779377b92dbb0fcbea3b5bb9238beb1772ab6dtargetblank3e77937a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/FabioBedin/GN_MR_ZC3H4/tree/3e779377b92dbb0fcbea3b5bb9238beb1772ab6d" target="_blank">3e77937</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/

Untracked files:
    Untracked:  code/custom_functions.R
    Untracked:  data/expdesign.tsv
    Untracked:  data/proteinGroups.txt
    Untracked:  output/results_table_TURBO._vs_WT..xlsx
    Untracked:  output/results_table_TURBO_BIO_vs_TURBO..xlsx
    Untracked:  output/results_table_TURBO_BIO_vs_WT_BIO.xlsx
    Untracked:  output/results_table_WT_BIO_vs_WT..xlsx

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/Turbo_vs_WT.Rmd</code>) and HTML (<code>docs/Turbo_vs_WT.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/FabioBedin/GN_MR_ZC3H4/blob/3e779377b92dbb0fcbea3b5bb9238beb1772ab6d/analysis/Turbo_vs_WT.Rmd" target="_blank">3e77937</a>
</td>
<td>
FabioBedin
</td>
<td>
2022-02-16
</td>
<td>
Update imputation
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/FabioBedin/GN_MR_ZC3H4/3e779377b92dbb0fcbea3b5bb9238beb1772ab6d/docs/Turbo_vs_WT.html" target="_blank">3e77937</a>
</td>
<td>
FabioBedin
</td>
<td>
2022-02-16
</td>
<td>
Update imputation
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/FabioBedin/GN_MR_ZC3H4/9ecde79132cfac88eaf9a4160cd1b06092b45f53/docs/Turbo_vs_WT.html" target="_blank">9ecde79</a>
</td>
<td>
FabioBedin
</td>
<td>
2022-02-15
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/FabioBedin/GN_MR_ZC3H4/blob/24ebe3abcc4514b8925fc1a3fc0b6a99a4bd176e/analysis/Turbo_vs_WT.Rmd" target="_blank">24ebe3a</a>
</td>
<td>
FabioBedin
</td>
<td>
2022-02-15
</td>
<td>
first push
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<style type="text/css">
p{
  font-size: 16px;
}

</style>
<pre class="r"><code>## Proteomics
library(&quot;MBQN&quot;)
library(&quot;DEP&quot;)
library(&quot;SummarizedExperiment&quot;)
library(&quot;preprocessCore&quot;)
library(&quot;enrichR&quot;)
library(&quot;gprofiler2&quot;)
library(&quot;org.Hs.eg.db&quot;)
library(&quot;rrvgo&quot;)

## Plots
library(&quot;DT&quot;)
library(&quot;patchwork&quot;)
library(&quot;datapasta&quot;)
library(&quot;ggstatsplot&quot;)
library(&quot;UpSetR&quot;)
library(&quot;ggrepel&quot;)
library(&quot;visdat&quot;)
library(&quot;naniar&quot;)

## Networks
library(&quot;STRINGdb&quot;)
library(&quot;RCy3&quot;)
library(&quot;RColorBrewer&quot;)

## Other
library(&quot;here&quot;)
library(&quot;openxlsx&quot;)
library(&quot;tidyverse&quot;)</code></pre>
<pre class="r"><code>source(here::here(&quot;code/custom_functions.R&quot;))</code></pre>
<pre class="r"><code>data &lt;- read.csv(here::here(paste0(&quot;data/&quot;, params$proteinGroups)), header = TRUE,stringsAsFactors = FALSE, sep = &quot;\t&quot;)
## qui metterei anche Razor...unique.peptides al posto di Peptides, perchè così non conto i peptidi che sono attribuiti a più di 1 PG
data &lt;- data[data$Reverse != &quot;+&quot; &amp; data$Potential.contaminant != &quot;+&quot; &amp; data$Only.identified.by.site != &quot;+&quot; &amp; data$Peptides &gt;= params$n_peptides,]

data_unique &lt;- make_unique(data, &quot;Gene.names&quot;, &quot;Protein.IDs&quot;, delim = &quot;;&quot;)</code></pre>
<pre class="r"><code>cond_1 &lt;- params$test_1

cond_2 &lt;- params$test_2

conditions&lt;-c(cond_1,cond_2)

test&lt;- paste(cond_1,cond_2,sep=&quot;_vs_&quot;)

expdesign &lt;- read.table(here::here(paste0(&quot;data/&quot;, params$expdesign)), header = T, stringsAsFactors = F)

expdesign &lt;- subset(expdesign, condition %in% conditions)

columns&lt;-match(paste(&quot;LFQ.intensity.&quot;,expdesign$label,sep=&quot;&quot;),colnames(data_unique))

data_se &lt;- make_se(data_unique, columns, expdesign)

## define automaticaly the best statistical threshold to filter data besed on replicates:
if(max(expdesign$replicate)&lt;=3){
       threshold&lt;-0
     } else if(max(expdesign$replicate)&lt;6){
       threshold&lt;-1
     } else if (max(expdesign$replicate)&gt;=6){
       threshold&lt;-trunc(max(expdesign$replicate)*0.25) ## 0.25 serve per avere sempre il 75% di valid vales
     }

data_filt &lt;- filter_missval(data_se, thr = threshold)</code></pre>
<div id="turbo.-vs-wt." class="section level1">
<h1><strong>TURBO.</strong> vs <strong>WT.</strong></h1>
<hr />
</div>
<div id="introduction" class="section level1">
<h1><strong>Introduction</strong></h1>
<p>First, we need to wrangling the original dataset. From the <strong>proteinGroups</strong> obtained by <a href="https://www.maxquant.org/">MaxQuant</a> software, I remove proteins that are marked form the software as potential contaminant, only identify by site and reverse.<br />
We che also filter the initial datasets by excluding all proteins identified by less than 1 peptides.<br />
Now, we generate some quality control plots to investigate the structure of the dataset and observe the effect of filters.</p>
<div id="normalization-strategy" class="section level4">
<h4><strong>Normalization strategy:</strong></h4>
<pre class="r"><code>if(params$Normalization == &quot;VSN&quot;){
  data_filt &lt;- normalize_vsn(data_filt)
  print(&quot;VSN normalization is used&quot;)
} else if(params$Normalization == &quot;MBQN&quot;){
  data_filt &lt;- normalize_MBQN(data_filt)
  print(&quot;MBQN normalization is used&quot;)
} else {
  print(&quot;Max-LFQ normalization is used&quot;)
}</code></pre>
<p>[1] “Max-LFQ normalization is used”</p>
</div>
<div id="quality-control-plots" class="section level2 tabset tabset-fade">
<h2><strong>1. Quality control plots</strong></h2>
<div id="proteins-per-samples" class="section level3 tabset">
<h3>Proteins per samples</h3>
<p>This plot show the number of proteins identify in each samples <em>after and before User define</em> filters:</p>
<div id="filtered" class="section level4">
<h4>Filtered</h4>
<pre class="r"><code>plot_numbers_lables(data_filt)</code></pre>
<p><img src="figure/Turbo_vs_WT.Rmd/proteins-filt-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="not-filtered" class="section level4">
<h4>Not-Filtered</h4>
<pre class="r"><code>plot_numbers_lables(data_se)</code></pre>
<p><img src="figure/Turbo_vs_WT.Rmd/proteins-NOfilt-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="distributions" class="section level3 tabset">
<h3>Distributions</h3>
<p>This plot is useful for checking the distributions of the samples <em>after and before User define</em> filters:</p>
<div id="filtered-1" class="section level4">
<h4>Filtered</h4>
<pre class="r"><code>plot_normalization(data_filt)</code></pre>
<p><img src="figure/Turbo_vs_WT.Rmd/normalization-filt-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="not-filtered-1" class="section level4">
<h4>Not-Filtered</h4>
<pre class="r"><code>plot_normalization(data_se)</code></pre>
<p><img src="figure/Turbo_vs_WT.Rmd/normalization-NOfilt-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="missing-data" class="section level3 tabset">
<h3>Missing data</h3>
<p>This plot we can explore the missing data pattern <em>after and before User define</em> filters. Notice that the percentages of missingness are provided in the data. These are accurate to 1 decimal place. Also the dataset is arranged by columns with most missingness.</p>
<div id="filtered-2" class="section level4">
<h4>Filtered</h4>
<pre class="r"><code>assay(data_filt) %&gt;% 
  as.data.frame() %&gt;% 
  vis_miss(., sort_miss = TRUE)</code></pre>
<p><img src="figure/Turbo_vs_WT.Rmd/missing-data-filt-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="not-filtered-2" class="section level4">
<h4>Not-Filtered</h4>
<pre class="r"><code>assay(data_se) %&gt;% 
  as.data.frame() %&gt;% 
  vis_miss(., sort_miss = TRUE)</code></pre>
<p><img src="figure/Turbo_vs_WT.Rmd/missing-data-NOfilt-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="sample-cvs" class="section level3 tabset">
<h3>Sample CVs</h3>
<p>The coefficient of variation (CV) is a statistical measure of the dispersion of data points in a data series around the mean. The coefficient of variation represents the ratio of the standard deviation to the mean, and it is a useful statistic for comparing the degree of variation from one data series to another, even if the means are drastically different from one another.</p>
<div id="filtered-3" class="section level4">
<h4>Filtered</h4>
<pre class="r"><code>plot_cvs(data_filt)</code></pre>
<p><img src="figure/Turbo_vs_WT.Rmd/CVs-filt-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="not-filtered-3" class="section level4">
<h4>Not-Filtered</h4>
<pre class="r"><code>plot_cvs(data_se)</code></pre>
<p><img src="figure/Turbo_vs_WT.Rmd/CVs-NOfilt-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
</div>
</div>
<div id="upset-plot" class="section level2 tabset tabset-fade">
<h2><strong>2. Upset plot</strong></h2>
<p>With this plot we can identify the numper of prteins in common within the 2 conditions or in common between all condition.</p>
<div id="turbo." class="section level3">
<h3>TURBO.</h3>
<pre class="r"><code>define_set &lt;- assay(data_filt) %&gt;%  as.data.frame() %&gt;% select(starts_with(cond_1)) %&gt;% colnames()
n_sets &lt;- length(define_set)

assay(data_filt) %&gt;% 
  as.data.frame() %&gt;% 
  select(starts_with(cond_1)) %&gt;% 
  rownames_to_column() %&gt;% 
  pivot_longer(!rowname, names_to = &quot;samples&quot;, values_to = &quot;intensity&quot;) %&gt;% 
  mutate(intensity = if_else(is.na(intensity), 0, 1)) %&gt;% 
  pivot_wider(names_from = samples, values_from = intensity) %&gt;% 
  as.data.frame() %&gt;% 
  upset(nsets = n_sets,
        sets = define_set,
        order.by = &quot;freq&quot;,
        keep.order = T,
        text.scale = 2.5,
        point.size = 4,
        line.size = 0.5, 
        sets.bar.color = &quot;coral2&quot;,
        main.bar.color  = &quot;gray44&quot;)</code></pre>
<p><img src="figure/Turbo_vs_WT.Rmd/upset-cond1-1.png" width="1440" style="display: block; margin: auto;" /></p>
</div>
<div id="wt." class="section level3">
<h3>WT.</h3>
<pre class="r"><code>define_set &lt;- assay(data_filt) %&gt;%  as.data.frame() %&gt;% select(starts_with(cond_2)) %&gt;% colnames()
n_sets &lt;- length(define_set)

assay(data_filt) %&gt;% 
  as.data.frame() %&gt;% 
  select(starts_with(cond_2)) %&gt;% 
  rownames_to_column() %&gt;% 
  pivot_longer(!rowname, names_to = &quot;samples&quot;, values_to = &quot;intensity&quot;) %&gt;% 
  mutate(intensity = if_else(is.na(intensity), 0, 1)) %&gt;% 
  pivot_wider(names_from = samples, values_from = intensity) %&gt;% 
  as.data.frame() %&gt;% 
  upset(nsets = n_sets,
        sets = define_set,
        order.by = &quot;freq&quot;,
        keep.order = T,
        text.scale = 2.5,
        point.size = 4,
        line.size = 0.5, 
        sets.bar.color = &quot;turquoise3&quot;,
        main.bar.color  = &quot;gray44&quot;)</code></pre>
<p><img src="figure/Turbo_vs_WT.Rmd/upset-cond2-1.png" width="1440" style="display: block; margin: auto;" /></p>
</div>
<div id="both-conditions" class="section level3">
<h3>Both conditions</h3>
<pre class="r"><code>define_set &lt;- assay(data_filt) %&gt;% colnames()
n_sets &lt;- length(define_set)
n_each &lt;- n_sets / 2

assay(data_filt) %&gt;% 
  as.data.frame() %&gt;% 
  rownames_to_column() %&gt;% 
  pivot_longer(!rowname, names_to = &quot;samples&quot;, values_to = &quot;intensity&quot;) %&gt;% 
  mutate(intensity = if_else(is.na(intensity), 0, 1)) %&gt;% 
  pivot_wider(names_from = samples, values_from = intensity) %&gt;% 
  as.data.frame() %&gt;% 
  upset(nsets = n_sets,
        sets = define_set,
        order.by = &quot;freq&quot;,
        keep.order = T,
        text.scale = 2.5,
        point.size = 4,
        line.size = 0.5, 
        sets.bar.color = rep(c(&quot;red3&quot;, &quot;royalblue1&quot;), each = n_each),
        main.bar.color  = &quot;gray44&quot;)</code></pre>
<p><img src="figure/Turbo_vs_WT.Rmd/upset-both-1.png" width="1440" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="imputation" class="section level2">
<h2><strong>3. Imputation</strong></h2>
<p>An idealized version of a label-free discovery mass spectrometry proteomics experiment would provide absolute abundance measurements for a whole proteome, across varying conditions. Unfortunately, this ideal is not realized. Measurements are made on peptides requiring an inferential step to obtain protein level estimates. The inference is complicated by experimental factors that necessitate relative abundance estimation and result in widespread non-ignorable missing data. Relative abundance on the log scale takes the form of parameter contrasts. In a complete-case analysis, contrast estimates may be biased by missing data and a substantial amount of useful information will often go unused.</p>
<pre class="r"><code>data_imputed &lt;- mix_imputation_mean(data_filt, cond1 = cond_1, cond2 = cond_2)

plot_imputation(data_filt, data_imputed)</code></pre>
<p><img src="figure/Turbo_vs_WT.Rmd/imputation-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>data_filt &lt;- data_imputed</code></pre>
</div>
<div id="differential-enrichment-analysis" class="section level2">
<h2><strong>4. Differential enrichment analysis</strong></h2>
<p>Protein-wise linear models combined with empirical Bayes statistics are used for the differential enrichment analysis (or differential expression analysis) and False discovery rate (FDR) is ajusted with <em>Benjamini-Hochberg</em> procedure. Significant proteins are are define with <strong>FDR = 0.05</strong> and <strong>Fold change = 1</strong></p>
<pre class="r"><code>data_diff &lt;- test_diff_BH(data_filt, type = &quot;manual&quot;, test = test)

dep &lt;- add_rejections(data_diff, alpha = params$FDR, lfc = params$Fold_change)

results&lt;-get_results(dep)</code></pre>
<p>There are <strong>75</strong> significant proteins in <strong>TURBO.</strong> vs <strong>WT.</strong> comparison.</p>
<p>The results from the previous analysis can be easily visualized by a number of functions. These visualizations assist in the determination of the optimal cutoffs to be used, highlight the most interesting samples and contrasts, and pinpoint differentially enriched/expressed proteins.</p>
<div id="visualization-of-the-results" class="section level3 tabset tabset-fade">
<h3><strong>4.1. Visualization of the results</strong></h3>
<div id="pca" class="section level4">
<h4>PCA</h4>
<p>The PCA plot can be used to get a high-level overview of the data. This can be very useful to observe batch effects, such as clear differences between replicates.</p>
<pre class="r"><code>if(params$Imputation){
  if(nrow(dep) &gt;= 500){
    n_value &lt;- 500
  }else {
    n_value &lt;- nrow(dep)
  }
  plot_pca(dep, x = 1, y = 2, n = n_value, point_size = 4, indicate = &quot;condition&quot;, label = T, label_size = 3)
} else {
  var &lt;- apply(assay(dep), 1, sd)
  df &lt;- assay(dep)[order(var, decreasing = TRUE)[seq_len(nrow(dep))],]
  n_value &lt;- df %&gt;% as.data.frame() %&gt;% drop_na() %&gt;% nrow()
  if(n_value &gt;= 500){
    n_value &lt;- 500
  }
  plot_pca(dep, x = 1, y = 2, n = n_value, point_size = 4, indicate = &quot;condition&quot;, label = T, label_size = 3)
}</code></pre>
<p><img src="figure/Turbo_vs_WT.Rmd/PCA-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="volcano" class="section level4">
<h4>Volcano</h4>
<p>Volcano plots allows to inspect the enrichment of proteins between the two samples (x axis) and their corresponding adjusted p value (y axis).</p>
<pre class="r"><code>plot_volcano(dep, contrast=test, add_names=T, label_size=5, adjusted = F)</code></pre>
<p><img src="figure/Turbo_vs_WT.Rmd/volcano-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="volcano-and-unique" class="section level4">
<h4>Volcano and unique</h4>
<pre class="r"><code>plot_volcano_2(dep, contrast=test, add_names=T, label_size=5, adjusted = F, unique_vec = uni_name)</code></pre>
<p><img src="figure/Turbo_vs_WT.Rmd/volcano-unique-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="correlation" class="section level4">
<h4>Correlation</h4>
<pre class="r"><code>plot_cor(dep, significant = TRUE, lower = 0, upper = 1, pal = &quot;Reds&quot;)</code></pre>
<p><img src="figure/Turbo_vs_WT.Rmd/correlation-matrix-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="cluster-heatmap" class="section level4">
<h4>Cluster heatmap</h4>
<p>The heatmap representation gives an overview of all significant proteins (rows) in all samples (columns). This allows to see general trends, for example if one sample or replicate is really different compared to the others. Additionally, the clustering of samples (columns) can indicate closer related samples and clustering of proteins (rows) indicates similarly behaving proteins.</p>
<pre class="r"><code>if(results %&gt;% filter(significant) %&gt;% nrow() &lt;= 100){
  plot_heatmap(dep, type = &quot;centered&quot;, kmeans = TRUE, k = 5, show_row_names = T, indicate = &quot;condition&quot;, col_limit = 5)
}else {
  plot_heatmap(dep, type = &quot;centered&quot;, kmeans = TRUE, k = 5, show_row_names = F, indicate = &quot;condition&quot;, col_limit = 5)
}</code></pre>
<p><img src="figure/Turbo_vs_WT.Rmd/cluster-heatmap-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="result-table" class="section level3">
<h3><strong>4.2. Result table</strong></h3>
<p>In this table are summarized the results of DEP analysis, sorted by p.value. Proteins that are UP regulated are colored with a red box in Fold change column, in the orhter hand proteins that are DOWN regulated are colored with blue.</p>
<pre class="r"><code>results %&gt;% 
  select(starts_with(&quot;name&quot;) | starts_with(test)) %&gt;% 
  filter(across(ends_with(&quot;significant&quot;))) %&gt;% 
  arrange(across(ends_with(&quot;p.adj&quot;))) %&gt;%
  mutate(across(2:3, format, scientific = T, digits = 2)) %&gt;%
  dplyr::rename_with(~ tolower(gsub(paste0(test,&quot;_&quot;), &quot;&quot;, .x)), .cols = starts_with(test)) %&gt;% 
  mutate(significant = str_to_title(significant)) %&gt;%
  dplyr::rename(Fold.Change = ratio, Gene.name = name) %&gt;% 
  DT::datatable(options = list(
  columnDefs = list(list(className = &#39;dt-center&#39;, targets = 1:5)),
  pageLength = 10)) %&gt;%
  formatStyle(&#39;Fold.Change&#39;, backgroundColor = styleInterval(c(-1, 1), c(&#39;lightblue&#39;, &#39;white&#39;, &#39;tomato&#39;))) %&gt;% 
  formatStyle(&#39;significant&#39;, color = styleEqual(c(&quot;True&quot;, &quot;False&quot;), c(&#39;green&#39;, &#39;red&#39;)))</code></pre>
<div id="htmlwidget-0dd1ba32e0541469b5ed" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-0dd1ba32e0541469b5ed">{"x":{"filter":"none","data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33","34","35","36","37","38","39","40","41","42","43","44","45","46","47","48","49","50","51","52","53","54","55","56","57","58","59","60","61","62","63","64","65","66","67","68","69","70","71","72","73","74","75"],["ZC3H4","DENND5B","SSR4","WDR82","SMN1","ANXA1","CSTF2","DKC1","LACTB","SART3","CCDC12","CSNK2A2","GAR1","KPNA3","SUGP2","SLC2A1","NHP2","UTP3","SRSF1","FBL","HMGB1","GLYR1","NOSIP","POLR2L","SYMPK","SRSF5","PPIE","MED23","C17orf49","CACTIN","CPNE7","KIF22","AHNAK","SRSF4","NOP10","PRPF3","TJP1","TJP3","TRA2B","EPS15L1","CLIP2","AKAP12","CPSF1","SMARCA2","SRSF3","CUL3","FGFRL1","ACACB","PRSS3","CDK9","PRPF4","NDUFA4","SFN","GULP1","PARP2","SKIV2L2","CCNT1","PRKRA","HCFC1","HIST1H4A","ITSN1","ARHGEF2","SYF2","SFPQ","HIST1H2AD","AQR","EMD","MT-ATP6","TXN","YBX3","CREM","CPSF6","KLHL7","PPP1CA","MRPS9"],["1.4e-12","1.2e-09","1.7e-09","1.0e-08","1.9e-07","5.4e-07","1.2e-06","2.5e-06","4.9e-06","4.7e-06","3.4e-05","3.1e-05","3.0e-05","3.4e-05","4.5e-05","5.2e-05","5.6e-05","5.9e-05","8.6e-05","1.2e-04","1.3e-04","1.6e-04","1.6e-04","1.6e-04","1.5e-04","1.7e-04","2.2e-04","2.4e-04","2.8e-04","2.9e-04","3.7e-04","4.3e-04","5.3e-04","5.3e-04","6.9e-04","6.7e-04","6.7e-04","7.1e-04","7.1e-04","8.2e-04","9.6e-04","1.1e-03","1.2e-03","1.3e-03","1.3e-03","1.5e-03","1.5e-03","1.8e-03","1.8e-03","2.0e-03","2.3e-03","2.8e-03","2.8e-03","2.9e-03","3.0e-03","3.7e-03","4.8e-03","4.8e-03","5.0e-03","5.0e-03","5.6e-03","6.0e-03","6.0e-03","6.4e-03","6.5e-03","6.7e-03","7.5e-03","7.4e-03","7.6e-03","7.8e-03","8.0e-03","8.4e-03","8.7e-03","8.9e-03","9.8e-03"],["1.2e-09","4.8e-07","4.8e-07","2.1e-06","3.1e-05","7.5e-05","1.4e-04","2.6e-04","4.1e-04","4.1e-04","1.6e-03","1.6e-03","1.6e-03","1.6e-03","2.1e-03","2.3e-03","2.3e-03","2.3e-03","3.1e-03","4.0e-03","4.0e-03","4.1e-03","4.1e-03","4.1e-03","4.1e-03","4.1e-03","5.0e-03","5.2e-03","5.9e-03","6.0e-03","7.1e-03","7.4e-03","8.6e-03","8.6e-03","9.5e-03","9.5e-03","9.5e-03","9.5e-03","9.5e-03","1.0e-02","1.2e-02","1.3e-02","1.3e-02","1.3e-02","1.4e-02","1.5e-02","1.5e-02","1.7e-02","1.7e-02","1.9e-02","2.0e-02","2.2e-02","2.3e-02","2.3e-02","2.3e-02","2.7e-02","3.2e-02","3.2e-02","3.2e-02","3.2e-02","3.5e-02","3.6e-02","3.6e-02","3.7e-02","3.7e-02","3.8e-02","4.0e-02","4.0e-02","4.0e-02","4.1e-02","4.2e-02","4.3e-02","4.5e-02","4.5e-02","4.8e-02"],["True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True"],[9.1,-5.59,-4.17,6.02,4.02,4.24,2.26,1.09,-2.26,1.93,-1.1,1.23,1.02,-1.79,-1.93,1.05,1.01,2.06,-1.1,1.08,3.04,-2.11,2.02,1.25,2.23,-3.67,-2.68,3.19,-2.02,-1.14,-1.18,1.95,1.5,-1.14,1.29,1.43,-1.21,-1.04,-1.03,-1.86,-1.67,-1.56,1.05,2.37,-1.03,1.25,2.07,1.13,1.5,-1.96,2.38,-1.53,-1.19,-1.76,-2.15,1.45,-2.01,-2.07,1.66,-1.13,-1.46,1.73,-1.03,-2.49,3.4,-1.21,-1.65,-2.41,3.3,-1.56,1.66,1.75,-1.49,2.23,1.3]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>Gene.name<\/th>\n      <th>p.val<\/th>\n      <th>p.adj<\/th>\n      <th>significant<\/th>\n      <th>Fold.Change<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-center","targets":[1,2,3,4,5]},{"orderable":false,"targets":0}],"pageLength":10,"order":[],"autoWidth":false,"orderClasses":false,"rowCallback":"function(row, data, displayNum, displayIndex, dataIndex) {\nvar value=data[5]; $(this.api().cell(row, 5).node()).css({'background-color':isNaN(parseFloat(value)) ? '' : value <= -1 ? \"lightblue\" : value <= 1 ? \"white\" : \"tomato\"});\nvar value=data[4]; $(this.api().cell(row, 4).node()).css({'color':value == \"True\" ? \"green\" : value == \"False\" ? \"red\" : null});\n}"}},"evals":["options.rowCallback"],"jsHooks":[]}</script>
</div>
</div>
<div id="gene-ontology" class="section level2">
<h2><strong>5. Gene Ontology</strong></h2>
<p>Now I perfom a gene onology analysis (GO or by branch <a href="GO:MF" class="uri">GO:MF</a>, <a href="GO:BP" class="uri">GO:BP</a>, <a href="GO:CC" class="uri">GO:CC</a>) and a KEGG ontology.<br />
The enrichment results are visualized with an interactive <strong>Manhattan-like-plot</strong> named “Gostplot”.</p>
<div id="turbo.-1" class="section level3 tabset tabset-fade">
<h3><strong>5.1. TURBO.</strong></h3>
<div id="gostplot" class="section level4">
<h4>Gostplot</h4>
<p>UP regulated proteins in TURBO. were used to generate these gene ontologies.</p>
<pre class="r"><code>Gname &lt;- results %&gt;% filter(significant &amp; get(paste0(test, &quot;_ratio&quot;)) &gt; 0) %&gt;% pull(name)

if(!params$Imputation){
  Uni &lt;- unique_pair(data_filt_unique, conditions = conditions, table = T) %&gt;% 
  filter(!!sym(paste0(&quot;unique_&quot;, cond_1)) == &quot;+&quot;) %&gt;% 
  pull(name) %&gt;% unique()

  Gname &lt;- c(Gname, Uni)
}

if(length(Gname) == 0){
  print(&quot;NOT enough data.&quot;)
  GO &lt;- NULL
}else{
  GO &lt;- gost(query = Gname, organism = &quot;hsapiens&quot;, sources = c(&quot;GO:BP&quot;, &quot;GO:MF&quot;, &quot;GO:CC&quot;, &quot;KEGG&quot;), ordered_query = T, evcodes = TRUE)
  if(is.null(GO)){
    print(&quot;NOT enough data.&quot;)
  }else{
    gostplot(GO, capped = TRUE, interactive = TRUE)
  }
}</code></pre>
<div id="htmlwidget-4906484ed42468d86de4" style="width:960px;height:768px;" class="plotly html-widget"></div>
<script type="application/json" data-for="htmlwidget-4906484ed42468d86de4">{"x":{"data":[{"x":[107.014152960457,75.4238394576609,107.021947264307,76.8501970621543,107.033638720081,161.367730856274,161.363833704349,136.500004423836,119.933211591319,84.414568948279,124.640971116532,183.526936700945,90.5214060145116,93.2805895773021,86.5735911146433,183.519142397095,84.0988996423665,159.263268816858,159.228194449534,159.224297297609,159.220400145684,84.4106717963542,123.740729021893,84.3678031251809,85.9344581989687,84.4964091387008,94.5237810413277,122.688498002185,75.9304692078908,114.157632438698,130.397064509529,97.2712731483436,107.037535872006,171.293776808855,120.53727013967,160.182996671121,151.936623198149,124.637073964607,76.1604011714567,83.5922698921366,158.885245080148,158.304569443346,114.208295413721,85.6577604123047,97.2634788444939,108.619779553493,107.045330175856],"y":[6.54573576918285,6.52210394149301,5.79878477252923,5.3844820180958,4.99519922183321,4.54292077349571,4.43827840305086,4.32621749118703,4.25583245466844,3.94303286222448,3.1690971933375,3.14412049053615,2.99511155861455,2.94844535816671,2.89760149242223,2.89127847646892,2.87305915723806,2.81692377142504,2.81692377142504,2.81692377142504,2.81692377142504,2.73440249592978,2.700769030764,2.64568840052502,2.63697337263128,2.62980249956809,2.60331613410669,2.34828128788694,2.32135341820203,2.29023836436459,2.28746446117847,2.25305589635291,2.21110706682635,1.94717484937064,1.90825474938971,1.8224790842989,1.81887039817532,1.81421545063591,1.75618755460074,1.73236286837744,1.633841269145,1.55674042811375,1.52435021789812,1.49099421921847,1.4800884623788,1.44811416461695,1.43926961826753],"text":["GO:0031118 (10) <br> rRNA pseudouridine synthesis <br> 2.846e-07","GO:0000154 (34) <br> rRNA modification <br> 3.005e-07","GO:0031120 (3) <br> snRNA pseudouridine synthesis <br> 1.589e-06","GO:0001522 (18) <br> pseudouridine synthesis <br> 4.126e-06","GO:0031123 (126) <br> RNA 3'-end processing <br> 1.011e-05","GO:0098789 (13) <br> pre-mRNA cleavage required for polyadenylation <br> 2.865e-05","GO:0098787 (14) <br> mRNA cleavage involved in mRNA processing <br> 3.645e-05","GO:0051276 (1034) <br> chromosome organization <br> 4.718e-05","GO:0040031 (7) <br> snRNA modification <br> 5.548e-05","GO:0006379 (20) <br> mRNA cleavage <br> 1.140e-04","GO:0044238 (12572) <br> primary metabolic process <br> 6.775e-04","GO:1904874 (15) <br> positive regulation of telomerase RNA localization to Cajal body <br> 7.176e-04","GO:0009451 (167) <br> RNA modification <br> 1.011e-03","GO:0010467 (8909) <br> gene expression <br> 1.126e-03","GO:0007004 (71) <br> telomere maintenance via telomerase <br> 1.266e-03","GO:1904872 (18) <br> regulation of telomerase RNA localization to Cajal body <br> 1.284e-03","GO:0006278 (72) <br> RNA-dependent DNA biosynthetic process <br> 1.339e-03","GO:0090685 (19) <br> RNA localization to nucleus <br> 1.524e-03","GO:0090672 (19) <br> telomerase RNA localization <br> 1.524e-03","GO:0090671 (19) <br> telomerase RNA localization to Cajal body <br> 1.524e-03","GO:0090670 (19) <br> RNA localization to Cajal body <br> 1.524e-03","GO:0006378 (52) <br> mRNA polyadenylation <br> 1.843e-03","GO:0043631 (53) <br> RNA polyadenylation <br> 1.992e-03","GO:0006364 (225) <br> rRNA processing <br> 2.261e-03","GO:0006807 (12062) <br> nitrogen compound metabolic process <br> 2.307e-03","GO:0006403 (198) <br> RNA localization <br> 2.345e-03","GO:0010833 (84) <br> telomere maintenance via telomere lengthening <br> 2.493e-03","GO:0043170 (12328) <br> macromolecule metabolic process <br> 4.485e-03","GO:0000454 (3) <br> snoRNA guided rRNA pseudouridine synthesis <br> 4.771e-03","GO:0034641 (8767) <br> cellular nitrogen compound metabolic process <br> 5.126e-03","GO:0046483 (8099) <br> heterocycle metabolic process <br> 5.159e-03","GO:0016072 (263) <br> rRNA metabolic process <br> 5.584e-03","GO:0031124 (73) <br> mRNA 3'-end processing <br> 6.150e-03","GO:1901360 (8395) <br> organic cyclic compound metabolic process <br> 1.129e-02","GO:0042254 (302) <br> ribosome biogenesis <br> 1.235e-02","GO:0097350 (6) <br> neutrophil clearance <br> 1.505e-02","GO:0071704 (13835) <br> organic substance metabolic process <br> 1.518e-02","GO:0044237 (12839) <br> cellular metabolic process <br> 1.534e-02","GO:0000723 (137) <br> telomere maintenance <br> 1.753e-02","GO:0006139 (7940) <br> nucleobase-containing compound metabolic process <br> 1.852e-02","GO:0090501 (155) <br> RNA phosphodiester bond hydrolysis <br> 2.324e-02","GO:0090304 (7476) <br> nucleic acid metabolic process <br> 2.775e-02","GO:0034660 (536) <br> ncRNA metabolic process <br> 2.990e-02","GO:0006725 (8159) <br> cellular aromatic compound metabolic process <br> 3.229e-02","GO:0016070 (6948) <br> RNA metabolic process <br> 3.311e-02","GO:0032200 (164) <br> telomere organization <br> 3.564e-02","GO:0031126 (12) <br> sno(s)RNA 3'-end processing <br> 3.637e-02"],"key":["GO:0031118","GO:0000154","GO:0031120","GO:0001522","GO:0031123","GO:0098789","GO:0098787","GO:0051276","GO:0040031","GO:0006379","GO:0044238","GO:1904874","GO:0009451","GO:0010467","GO:0007004","GO:1904872","GO:0006278","GO:0090685","GO:0090672","GO:0090671","GO:0090670","GO:0006378","GO:0043631","GO:0006364","GO:0006807","GO:0006403","GO:0010833","GO:0043170","GO:0000454","GO:0034641","GO:0046483","GO:0016072","GO:0031124","GO:1901360","GO:0042254","GO:0097350","GO:0071704","GO:0044237","GO:0000723","GO:0006139","GO:0090501","GO:0090304","GO:0034660","GO:0006725","GO:0016070","GO:0032200","GO:0031126"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(255,153,0,1)","opacity":0.8,"size":[10.8408627476859,13.8067422172616,3.77952755905512,12.3937948397294,16.2212051730393,11.572361025388,11.7668569146846,19.3349116626413,9.70327512794354,12.6434487918721,22.365167907438,11.9437645398337,16.6815592564197,21.977392949464,15.2267827514816,12.3937948397294,15.2520552113937,12.5227987547864,12.5227987547864,12.5227987547864,12.5227987547864,14.6488211600568,14.6850501567534,17.1514566182497,22.3189705818375,16.9520093326524,15.5269966915192,22.3433185241269,3.77952755905512,21.9591006394746,21.8686009200492,17.3909726551505,15.2769246672795,21.9096459491601,17.5997303147263,9.13738580423005,22.4715220454936,22.3885676796153,16.3597394500354,21.8458893350244,16.5613178988887,21.7767387756911,18.4341530369776,21.8770483649144,21.6922664456138,16.6524323391359,11.356683354654],"symbol":"circle","line":{"width":1.88976377952756,"color":"rgba(255,153,0,1)"}},"hoveron":"points","set":"SharedData0dd1ba32","name":"GO:BP","legendgroup":"GO:BP","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","_isNestedKey":false,"frame":null},{"x":[55.058253379197,57.092955246969,66.1321767629908,66.3660505408957,59.0263118109823,66.1282788666924,59.4979572630904,55.3233103274892,64.4326939768824,59.4745698852999,62.1485267460118,57.3463185063659,55.2102713348351,69.2855748684076,54.991989142124,55.7286915425242,55.736487335121,59.9579090263032,62.140730953415,65.220069029162,55.9352800463401,62.1251393682214,62.1329351608182,62.1290372645198,62.1446288497134,55.9859526982195,57.8452492325629,69.4843675796267,66.0620146296194,54.4891605196286,62.1212414719229,61.7236560494847,62.7955775315485,54.2474909491269,64.608099310311,63.6881957838853,57.3580121952612,67.9096174750675],"y":[10.3743545286925,9.94011351966609,9.84516517127012,9.1464678390459,9.1464678390459,8.66961925978919,8.49722887450339,7.96422014855402,7.39996199801746,7.39820465694737,7.39820465694737,6.61021041031608,5.90315794742044,5.55864306638156,5.30575771995058,5.10669446358291,4.82357283273186,4.15012071378277,3.75876859871412,2.97544725775125,2.95301114324255,2.70271747568672,2.39960176961595,2.34596569258111,2.34596569258111,2.20421598511413,2.17969636850854,2.01171304054491,1.96066904508786,1.69915288363862,1.64761506495067,1.5979636038999,1.55926707905093,1.51948485684101,1.48854090646195,1.46548186460708,1.42894676615896,1.30225605309349],"text":["GO:0005654 (4161) <br> nucleoplasm <br> 4.223e-11","GO:0015030 (102) <br> Cajal body <br> 1.148e-10","GO:0072589 (4) <br> box H/ACA scaRNP complex <br> 1.428e-10","GO:0090661 (5) <br> box H/ACA telomerase RNP complex <br> 7.137e-10","GO:0031429 (5) <br> box H/ACA snoRNP complex <br> 7.137e-10","GO:0072588 (6) <br> box H/ACA RNP complex <br> 2.140e-09","GO:0031981 (5442) <br> nuclear lumen <br> 3.183e-09","GO:0005732 (28) <br> sno(s)RNA-containing ribonucleoprotein complex <br> 1.086e-08","GO:0070013 (6569) <br> intracellular organelle lumen <br> 3.981e-08","GO:0031974 (6570) <br> membrane-enclosed lumen <br> 3.998e-08","GO:0043233 (6570) <br> organelle lumen <br> 3.998e-08","GO:0016604 (869) <br> nuclear body <br> 2.454e-07","GO:0005697 (23) <br> telomerase holoenzyme complex <br> 1.250e-06","GO:0140513 (2671) <br> nuclear protein-containing complex <br> 2.763e-06","GO:0005634 (10053) <br> nucleus <br> 4.946e-06","GO:0005847 (24) <br> mRNA cleavage and polyadenylation specificity factor complex <br> 7.822e-06","GO:0005849 (28) <br> mRNA cleavage factor complex <br> 1.501e-05","GO:0032991 (7228) <br> protein-containing complex <br> 7.077e-05","GO:0043231 (14514) <br> intracellular membrane-bounded organelle <br> 1.743e-04","GO:0071001 (3) <br> U4/U6 snRNP <br> 1.058e-03","GO:0005911 (495) <br> cell-cell junction <br> 1.114e-03","GO:0043227 (15557) <br> membrane-bounded organelle <br> 1.983e-03","GO:0043229 (15870) <br> intracellular organelle <br> 3.985e-03","GO:0043228 (6362) <br> non-membrane-bounded organelle <br> 4.509e-03","GO:0043232 (6362) <br> intracellular non-membrane-bounded organelle <br> 4.509e-03","GO:0005925 (420) <br> focal adhesion <br> 6.249e-03","GO:0030055 (428) <br> cell-substrate junction <br> 6.612e-03","GO:1902494 (1424) <br> catalytic complex <br> 9.734e-03","GO:0072357 (7) <br> PTW/PP1 phosphatase complex <br> 1.095e-02","GO:0001650 (140) <br> fibrillar center <br> 1.999e-02","GO:0043226 (16674) <br> organelle <br> 2.251e-02","GO:0042383 (135) <br> sarcolemma <br> 2.524e-02","GO:0044291 (73) <br> cell-cell contact zone <br> 2.759e-02","GO:0000781 (169) <br> chromosome, telomeric region <br> 3.024e-02","GO:0070161 (839) <br> anchoring junction <br> 3.247e-02","GO:0048188 (12) <br> Set1C/COMPASS complex <br> 3.424e-02","GO:0016607 (414) <br> nuclear speck <br> 3.724e-02","GO:0098687 (359) <br> chromosomal region <br> 4.986e-02"],"key":["GO:0005654","GO:0015030","GO:0072589","GO:0090661","GO:0031429","GO:0072588","GO:0031981","GO:0005732","GO:0070013","GO:0031974","GO:0043233","GO:0016604","GO:0005697","GO:0140513","GO:0005634","GO:0005847","GO:0005849","GO:0032991","GO:0043231","GO:0071001","GO:0005911","GO:0043227","GO:0043229","GO:0043228","GO:0043232","GO:0005925","GO:0030055","GO:1902494","GO:0072357","GO:0001650","GO:0043226","GO:0042383","GO:0044291","GO:0000781","GO:0070161","GO:0048188","GO:0016607","GO:0098687"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(16,150,24,1)","opacity":0.8,"size":[21.0894334825812,15.8643936985395,7.23124222618377,8.37907206593902,8.37907206593902,9.13738580423005,21.407580413457,13.3974314476729,21.6273049615825,21.6274815687707,21.6274815687707,19.1017374331549,12.9641522813395,20.5507379900014,22.1143470436776,13.0596102594499,13.3974314476729,21.7378809735574,22.5245251275829,3.77952755905512,18.3212742415784,22.6010313759405,22.6229344868671,21.5901171298092,21.5901171298092,18.0853933663563,18.1126791153317,19.7552655592872,9.70327512794354,16.3953446427403,22.6771653543307,16.3355092846442,15.2769246672795,16.700652226649,19.0541833197588,11.356683354654,18.0645507609824,17.856404153362],"symbol":"circle","line":{"width":1.88976377952756,"color":"rgba(16,150,24,1)"}},"hoveron":"points","set":"SharedData0dd1ba32","name":"GO:CC","legendgroup":"GO:CC","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","_isNestedKey":false,"frame":null},{"x":[25.2203565991266,40.1588639580478,20.9293843574955,3.69144591541137,3.53944962347349,9.30361669465648,9.74401620719447,48.6628616249317],"y":[8.78069876483324,5.62043681502559,4.93176279069512,3.48094987442552,2.44249071424507,1.9918599215218,1.52909739323321,1.32542342780197],"text":["GO:0034513 (5) <br> box H/ACA snoRNA binding <br> 1.657e-09","GO:0070034 (22) <br> telomerase RNA binding <br> 2.396e-06","GO:0030515 (32) <br> snoRNA binding <br> 1.170e-05","GO:0003723 (3142) <br> RNA binding <br> 3.304e-04","GO:0003676 (5447) <br> nucleic acid binding <br> 3.610e-03","GO:0005515 (14799) <br> protein binding <br> 1.019e-02","GO:0008134 (602) <br> transcription factor binding <br> 2.957e-02","GO:1901363 (7416) <br> heterocyclic compound binding <br> 4.727e-02"],"key":["GO:0034513","GO:0070034","GO:0030515","GO:0003723","GO:0003676","GO:0005515","GO:0008134","GO:1901363"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(220,57,18,1)","opacity":0.8,"size":[8.37907206593902,12.8633794105958,13.6807531937651,20.7500772918076,21.4086591648813,22.5459945929087,18.5973305706692,21.7674648364439],"symbol":"circle","line":{"width":1.88976377952756,"color":"rgba(220,57,18,1)"}},"hoveron":"points","set":"SharedData0dd1ba32","name":"GO:MF","legendgroup":"GO:MF","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","_isNestedKey":false,"frame":null},{"x":[198.653066158901,198.664778565247],"y":[5.044023506423,3.73172356661621],"text":["KEGG:03008 (98) <br> Ribosome biogenesis in eukaryotes <br> 9.036e-06","KEGG:03015 (97) <br> mRNA surveillance pathway <br> 1.855e-04"],"key":["KEGG:03008","KEGG:03015"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(221,68,119,1)","opacity":0.8,"size":[15.7956488676438,15.7779607493375],"symbol":"circle","line":{"width":1.88976377952756,"color":"rgba(221,68,119,1)"}},"hoveron":"points","set":"SharedData0dd1ba32","name":"KEGG","legendgroup":"KEGG","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","_isNestedKey":false,"frame":null},{"x":[2,49.9528814359943],"y":[-1,-1],"text":"","type":"scatter","mode":"lines","line":{"width":11.3385826771654,"color":"rgba(220,57,18,1)","dash":"solid"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[53.8499055266887,71.2657061880019],"y":[-1,-1],"text":"","type":"scatter","mode":"lines","line":{"width":11.3385826771654,"color":"rgba(16,150,24,1)","dash":"solid"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[75.1627302786963,193.967406707605],"y":[-1,-1],"text":"","type":"scatter","mode":"lines","line":{"width":11.3385826771654,"color":"rgba(255,153,0,1)","dash":"solid"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[197.864430798299,200.003897024091],"y":[-1,-1],"text":"","type":"scatter","mode":"lines","line":{"width":11.3385826771654,"color":"rgba(221,68,119,1)","dash":"solid"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[180],"y":[16.2],"text":"values above this threshold are capped","hovertext":"","textfont":{"size":7.55905511811024,"color":"rgba(190,190,190,1)"},"type":"scatter","mode":"text","hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[0,210],"y":[16,16],"text":"","type":"scatter","mode":"lines","line":{"width":0.755905511811024,"color":"rgba(190,190,190,1)","dash":"dash"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null}],"layout":{"margin":{"t":29.2835201328352,"r":6.6417600664176,"b":51.8931100989696,"l":61.0377750103778},"plot_bgcolor":"rgba(255,255,255,1)","paper_bgcolor":"rgba(255,255,255,1)","font":{"color":"rgba(0,0,0,1)","family":"","size":14.6118721461187},"xaxis":{"domain":[0,1],"automargin":true,"type":"linear","autorange":false,"range":[0,210],"tickmode":"array","ticktext":["GO:MF","GO:CC","GO:BP","KEGG"],"tickvals":[25.9764407179972,62.5578058573453,134.565068493151,198.934163911195],"categoryorder":"array","categoryarray":["GO:MF","GO:CC","GO:BP","KEGG"],"nticks":null,"ticks":"","tickcolor":null,"ticklen":3.65296803652968,"tickwidth":0,"showticklabels":true,"tickfont":{"color":"rgba(77,77,77,1)","family":"","size":10.6268161062682},"tickangle":-45,"showline":true,"linecolor":"rgba(190,190,190,1)","linewidth":0.132835201328352,"showgrid":false,"gridcolor":null,"gridwidth":0,"zeroline":false,"anchor":"y","title":"","hoverformat":".2f"},"yaxis":{"domain":[0,1],"automargin":true,"type":"linear","autorange":false,"range":[-1,18.5],"tickmode":"array","ticktext":["0","2","4","6","8","10","12","14",">16"],"tickvals":[0,2,4,6,8,10,12,14,16],"categoryorder":"array","categoryarray":["0","2","4","6","8","10","12","14",">16"],"nticks":null,"ticks":"outside","tickcolor":"rgba(190,190,190,1)","ticklen":3.65296803652968,"tickwidth":0.66417600664176,"showticklabels":true,"tickfont":{"color":"rgba(77,77,77,1)","family":"","size":11.689497716895},"tickangle":-0,"showline":true,"linecolor":"rgba(190,190,190,1)","linewidth":0.66417600664176,"showgrid":false,"gridcolor":null,"gridwidth":0,"zeroline":false,"anchor":"x","title":"","hoverformat":".2f"},"annotations":[{"text":"-log10(p-adj)","x":-0.0108706932337069,"y":0.5,"showarrow":false,"ax":0,"ay":0,"font":{"color":"rgba(0,0,0,1)","family":"","size":13.2835201328352},"xref":"paper","yref":"paper","textangle":-90,"xanchor":"right","yanchor":"center","annotationType":"axis"},{"text":"query_1","x":0.5,"y":1,"showarrow":false,"ax":0,"ay":0,"font":{"color":"rgba(169,169,169,1)","family":"","size":13.2835201328352},"xref":"paper","yref":"paper","textangle":-0,"xanchor":"center","yanchor":"bottom"}],"shapes":[{"type":"rect","fillcolor":null,"line":{"color":null,"width":0,"linetype":[]},"yref":"paper","xref":"paper","x0":0,"x1":1,"y0":0,"y1":1},{"type":"rect","fillcolor":null,"line":{"color":null,"width":0,"linetype":[]},"yref":"paper","xref":"paper","x0":0,"x1":1,"y0":0,"y1":24.9730178497302,"yanchor":1,"ysizemode":"pixel"}],"showlegend":false,"legend":{"bgcolor":"rgba(255,255,255,1)","bordercolor":"transparent","borderwidth":1.88976377952756,"font":{"color":"rgba(0,0,0,1)","family":"","size":11.689497716895}},"hovermode":"closest","barmode":"relative","dragmode":"zoom"},"config":{"doubleClick":"reset","showSendToCloud":false},"source":"A","attrs":{"1f4c71034692":{"x":{},"y":{},"text":{},"colour":{},"size":{},"alpha":{},"type":"scatter"},"1f4c2b934358":{"x":{},"xend":{},"y":{},"yend":{}},"1f4c55de18a3":{"x":{},"xend":{},"y":{},"yend":{}},"1f4c50096fda":{"x":{},"xend":{},"y":{},"yend":{}},"1f4c51ac7b84":{"x":{},"xend":{},"y":{},"yend":{}},"1f4c78194f59":{"x":{},"y":{}},"1f4c5bcb71f2":{"yintercept":{}}},"cur_data":"1f4c71034692","visdat":{"1f4c71034692":["function (y) ","x"],"1f4c2b934358":["function (y) ","x"],"1f4c55de18a3":["function (y) ","x"],"1f4c50096fda":["function (y) ","x"],"1f4c51ac7b84":["function (y) ","x"],"1f4c78194f59":["function (y) ","x"],"1f4c5bcb71f2":["function (y) ","x"]},"highlight":{"on":"plotly_click","off":"plotly_doubleclick","persistent":false,"dynamic":false,"color":null,"selectize":false,"defaultValues":null,"opacityDim":0.2,"selected":{"opacity":1},"debounce":0,"ctGroups":["SharedData0dd1ba32"]},"shinyEvents":["plotly_hover","plotly_click","plotly_selected","plotly_relayout","plotly_brushed","plotly_brushing","plotly_clickannotation","plotly_doubleclick","plotly_deselect","plotly_afterplot","plotly_sunburstclick"],"base_url":"https://plot.ly"},"evals":[],"jsHooks":[]}</script>
</div>
<div id="table-results" class="section level4">
<h4>Table results</h4>
<p>The results can also be visualized with a table:</p>
<pre class="r"><code>if(is.null(GO)){
  print(&quot;NOT enough data.&quot;)
}else{
  GO$result %&gt;% 
    as.data.frame() %&gt;% 
    select(starts_with(c(&quot;source&quot;, &quot;term_name&quot;, &quot;p_value&quot;, &quot;term_size&quot;, &quot;intersection_size&quot;))) %&gt;%
    arrange(p_value) %&gt;%  
    mutate(p_value = format(p_value, scientific = T, digits = 2)) %&gt;% 
    DT::datatable(options = list(
    columnDefs = list(list(className = &#39;dt-center&#39;, targets = 1:6)),
    pageLength = 10)) %&gt;%
    formatStyle(&#39;source&#39;, backgroundColor = styleEqual(c(&quot;GO:BP&quot;, &quot;GO:MF&quot;, &quot;GO:CC&quot;, &quot;KEGG&quot;), c(&#39;orange&#39;, &#39;tomato&#39;, &#39;lightgreen&#39;, &#39;violet&#39;)))
}</code></pre>
<div id="htmlwidget-a421697d58abce2053a2" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-a421697d58abce2053a2">{"x":{"filter":"none","data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33","34","35","36","37","38","39","40","41","42","43","44","45","46","47","48","49","50","51","52","53","54","55","56","57","58","59","60","61","62","63","64","65","66","67","68","69","70","71","72","73","74","75","76","77","78","79","80","81","82","83","84","85","86","87","88","89","90","91","92","93","94","95"],["GO:CC","GO:CC","GO:CC","GO:CC","GO:CC","GO:MF","GO:CC","GO:CC","GO:CC","GO:CC","GO:CC","GO:CC","GO:CC","GO:BP","GO:BP","GO:CC","GO:BP","GO:MF","GO:CC","GO:BP","GO:CC","GO:CC","KEGG","GO:BP","GO:MF","GO:CC","GO:BP","GO:BP","GO:BP","GO:BP","GO:CC","GO:BP","GO:CC","KEGG","GO:MF","GO:BP","GO:BP","GO:BP","GO:CC","GO:CC","GO:BP","GO:BP","GO:BP","GO:BP","GO:BP","GO:BP","GO:BP","GO:BP","GO:BP","GO:CC","GO:BP","GO:BP","GO:BP","GO:BP","GO:BP","GO:MF","GO:CC","GO:BP","GO:CC","GO:CC","GO:BP","GO:BP","GO:BP","GO:BP","GO:BP","GO:CC","GO:CC","GO:CC","GO:MF","GO:CC","GO:BP","GO:BP","GO:BP","GO:BP","GO:BP","GO:BP","GO:BP","GO:CC","GO:CC","GO:BP","GO:CC","GO:CC","GO:BP","GO:MF","GO:BP","GO:CC","GO:BP","GO:CC","GO:BP","GO:CC","GO:BP","GO:BP","GO:CC","GO:MF","GO:CC"],[311,833,3152,3212,1329,5959,3151,1450,379,2716,1444,2130,898,8174,68,350,8176,9792,3961,434,294,483,203,8179,4858,485,22121,22120,15740,11489,1568,2375,2128,206,435,12697,27807,3942,2918,536,4650,2929,27805,2294,21581,21572,21571,21570,2374,2124,12466,2363,2765,2396,4969,396,2126,12196,2125,2129,198,10007,14174,5674,8180,549,1026,4012,1875,3134,24668,11644,21817,19701,12696,257,2164,165,2123,21484,2021,2296,21335,1988,10020,103,2694,2761,5672,2525,8586,8182,901,11974,3608],["nucleoplasm","Cajal body","box H/ACA scaRNP complex","box H/ACA telomerase RNP complex","box H/ACA snoRNP complex","box H/ACA snoRNA binding","box H/ACA RNP complex","nuclear lumen","sno(s)RNA-containing ribonucleoprotein complex","intracellular organelle lumen","membrane-enclosed lumen","organelle lumen","nuclear body","rRNA pseudouridine synthesis","rRNA modification","telomerase holoenzyme complex","snRNA pseudouridine synthesis","telomerase RNA binding","nuclear protein-containing complex","pseudouridine synthesis","nucleus","mRNA cleavage and polyadenylation specificity factor complex","Ribosome biogenesis in eukaryotes","RNA 3'-end processing","snoRNA binding","mRNA cleavage factor complex","pre-mRNA cleavage required for polyadenylation","mRNA cleavage involved in mRNA processing","chromosome organization","snRNA modification","protein-containing complex","mRNA cleavage","intracellular membrane-bounded organelle","mRNA surveillance pathway","RNA binding","primary metabolic process","positive regulation of telomerase RNA localization to Cajal body","RNA modification","U4/U6 snRNP","cell-cell junction","gene expression","telomere maintenance via telomerase","regulation of telomerase RNA localization to Cajal body","RNA-dependent DNA biosynthetic process","RNA localization to nucleus","telomerase RNA localization","telomerase RNA localization to Cajal body","RNA localization to Cajal body","mRNA polyadenylation","membrane-bounded organelle","RNA polyadenylation","rRNA processing","nitrogen compound metabolic process","RNA localization","telomere maintenance via telomere lengthening","nucleic acid binding","intracellular organelle","macromolecule metabolic process","non-membrane-bounded organelle","intracellular non-membrane-bounded organelle","snoRNA guided rRNA pseudouridine synthesis","cellular nitrogen compound metabolic process","heterocycle metabolic process","rRNA metabolic process","mRNA 3'-end processing","focal adhesion","cell-substrate junction","catalytic complex","protein binding","PTW/PP1 phosphatase complex","organic cyclic compound metabolic process","ribosome biogenesis","neutrophil clearance","organic substance metabolic process","cellular metabolic process","telomere maintenance","nucleobase-containing compound metabolic process","fibrillar center","organelle","RNA phosphodiester bond hydrolysis","sarcolemma","cell-cell contact zone","nucleic acid metabolic process","transcription factor binding","ncRNA metabolic process","chromosome, telomeric region","cellular aromatic compound metabolic process","anchoring junction","RNA metabolic process","Set1C/COMPASS complex","telomere organization","sno(s)RNA 3'-end processing","nuclear speck","heterocyclic compound binding","chromosomal region"],["4.2e-11","1.1e-10","1.4e-10","7.1e-10","7.1e-10","1.7e-09","2.1e-09","3.2e-09","1.1e-08","4.0e-08","4.0e-08","4.0e-08","2.5e-07","2.8e-07","3.0e-07","1.2e-06","1.6e-06","2.4e-06","2.8e-06","4.1e-06","4.9e-06","7.8e-06","9.0e-06","1.0e-05","1.2e-05","1.5e-05","2.9e-05","3.6e-05","4.7e-05","5.5e-05","7.1e-05","1.1e-04","1.7e-04","1.9e-04","3.3e-04","6.8e-04","7.2e-04","1.0e-03","1.1e-03","1.1e-03","1.1e-03","1.3e-03","1.3e-03","1.3e-03","1.5e-03","1.5e-03","1.5e-03","1.5e-03","1.8e-03","2.0e-03","2.0e-03","2.3e-03","2.3e-03","2.3e-03","2.5e-03","3.6e-03","4.0e-03","4.5e-03","4.5e-03","4.5e-03","4.8e-03","5.1e-03","5.2e-03","5.6e-03","6.2e-03","6.2e-03","6.6e-03","9.7e-03","1.0e-02","1.1e-02","1.1e-02","1.2e-02","1.5e-02","1.5e-02","1.5e-02","1.8e-02","1.9e-02","2.0e-02","2.3e-02","2.3e-02","2.5e-02","2.8e-02","2.8e-02","3.0e-02","3.0e-02","3.0e-02","3.2e-02","3.2e-02","3.3e-02","3.4e-02","3.6e-02","3.6e-02","3.7e-02","4.7e-02","5.0e-02"],[4161,102,4,5,5,5,6,5442,28,6569,6570,6570,869,10,34,23,3,22,2671,18,10053,24,98,126,32,28,13,14,1034,7,7228,20,14514,97,3142,12572,15,167,3,495,8909,71,18,72,19,19,19,19,52,15557,53,225,12062,198,84,5447,15870,12328,6362,6362,3,8767,8099,263,73,420,428,1424,14799,7,8395,302,6,13835,12839,137,7940,140,16674,155,135,73,7476,602,536,169,8159,839,6948,12,164,12,414,7416,359],[27,8,4,4,4,4,4,28,5,29,29,29,12,4,5,4,3,4,18,4,32,4,6,5,4,4,3,3,12,3,26,3,35,6,18,34,3,5,2,7,28,4,3,4,3,3,3,3,4,35,4,6,33,5,4,22,35,33,22,22,2,27,26,6,3,3,3,10,34,2,26,6,2,34,33,4,25,3,35,4,2,3,24,5,7,4,25,7,23,2,4,2,5,23,5]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>source<\/th>\n      <th>source_order<\/th>\n      <th>term_name<\/th>\n      <th>p_value<\/th>\n      <th>term_size<\/th>\n      <th>intersection_size<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-center","targets":[1,2,3,4,5,6]},{"orderable":false,"targets":0}],"pageLength":10,"order":[],"autoWidth":false,"orderClasses":false,"rowCallback":"function(row, data, displayNum, displayIndex, dataIndex) {\nvar value=data[1]; $(this.api().cell(row, 1).node()).css({'background-color':value == \"GO:BP\" ? \"orange\" : value == \"GO:MF\" ? \"tomato\" : value == \"GO:CC\" ? \"lightgreen\" : value == \"KEGG\" ? \"violet\" : null});\n}"}},"evals":["options.rowCallback"],"jsHooks":[]}</script>
</div>
<div id="treemap" class="section level4 tabset tabset-fade">
<h4>Treemap</h4>
<div id="bp" class="section level5">
<h5>BP</h5>
<pre class="r"><code>if(sum(GO$result$source == &quot;GO:BP&quot;) &gt;= 5 &amp; !is.null(GO)){
  simMatrix &lt;- GO$result %&gt;% 
  as.data.frame() %&gt;%
  filter(source == &quot;GO:BP&quot;) %&gt;% 
  pull(term_id) %&gt;% 
  calculateSimMatrix(orgdb=&quot;org.Hs.eg.db&quot;, ont=&quot;BP&quot;, method=&quot;Rel&quot;)

scores &lt;- setNames(-log10(GO$result$p_value), GO$result$term_id)

reducedTerms &lt;- reduceSimMatrix(simMatrix,
                                scores,
                                threshold=0.7,
                                orgdb=&quot;org.Hs.eg.db&quot;)

treemapPlot(reducedTerms)
BP &lt;- T
}else{
  print(&quot;NOT enough data.&quot;)
  BP &lt;- F
}</code></pre>
<p><img src="figure/Turbo_vs_WT.Rmd/BP-cond1-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="mf" class="section level5">
<h5>MF</h5>
<pre class="r"><code>if(sum(GO$result$source == &quot;GO:MF&quot;) &gt;= 5 &amp; !is.null(GO)){
simMatrix &lt;- GO$result %&gt;% 
  as.data.frame() %&gt;%
  filter(source == &quot;GO:MF&quot;) %&gt;% 
  pull(term_id) %&gt;% 
  calculateSimMatrix(orgdb=&quot;org.Hs.eg.db&quot;, ont=&quot;MF&quot;, method=&quot;Rel&quot;)

scores &lt;- setNames(-log10(GO$result$p_value), GO$result$term_id)

reducedTerms &lt;- reduceSimMatrix(simMatrix,
                                scores,
                                threshold=0.7,
                                orgdb=&quot;org.Hs.eg.db&quot;)

treemapPlot(reducedTerms)
MF &lt;- T
}else{
  print(&quot;NOT enough data.&quot;)
  MF &lt;- F
}</code></pre>
<p><img src="figure/Turbo_vs_WT.Rmd/MF-cond1-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="cc" class="section level5">
<h5>CC</h5>
<pre class="r"><code>if(sum(GO$result$source == &quot;GO:CC&quot;) &gt;= 5 &amp; !is.null(GO)){
simMatrix &lt;- GO$result %&gt;% 
  as.data.frame() %&gt;%
  filter(source == &quot;GO:CC&quot;) %&gt;% 
  pull(term_id) %&gt;% 
  calculateSimMatrix(orgdb=&quot;org.Hs.eg.db&quot;, ont=&quot;CC&quot;, method=&quot;Rel&quot;)

scores &lt;- setNames(-log10(GO$result$p_value), GO$result$term_id)

reducedTerms &lt;- reduceSimMatrix(simMatrix,
                                scores,
                                threshold=0.7,
                                orgdb=&quot;org.Hs.eg.db&quot;)

treemapPlot(reducedTerms)
CC &lt;- T
}else{
  print(&quot;NOT enough data.&quot;)
  CC &lt;- F
}</code></pre>
<p><img src="figure/Turbo_vs_WT.Rmd/CC-cond1-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="scatterplot" class="section level4">
<h4>ScatterPlot</h4>
<pre class="r"><code>if(BP | MF | CC){
scatterPlot(simMatrix, reducedTerms, size = &quot;score&quot;, labelSize = 5)
}else{
  print(&quot;NOT enough data.&quot;)
}</code></pre>
<p><img src="figure/Turbo_vs_WT.Rmd/scatter-cond1-1.png" width="1440" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="wt.-1" class="section level3 tabset tabset-fade">
<h3><strong>5.2. WT.</strong></h3>
<div id="gostplot-1" class="section level4">
<h4>Gostplot</h4>
<p>UP regulated proteins in WT. were used to generate these gene ontologies.</p>
<pre class="r"><code>Gname &lt;- results %&gt;% filter(significant &amp; get(paste0(test, &quot;_ratio&quot;)) &lt; 0) %&gt;% pull(name)

if(!params$Imputation){
  Uni &lt;- unique_pair(data_filt_unique, conditions = conditions, table = T) %&gt;% 
  filter(!!sym(paste0(&quot;unique_&quot;, cond_2)) == &quot;+&quot;) %&gt;% 
  pull(name) %&gt;% unique()

  Gname &lt;- c(Gname, Uni)
}

if(length(Gname) == 0){
  print(&quot;NOT enough data.&quot;)
  GO &lt;- NULL
}else{
  GO &lt;- gost(query = Gname, organism = &quot;hsapiens&quot;, sources = c(&quot;GO:BP&quot;, &quot;GO:MF&quot;, &quot;GO:CC&quot;, &quot;KEGG&quot;), ordered_query = T, evcodes = TRUE)
  if(is.null(GO)){
    print(&quot;NOT enough data.&quot;)
  }else{
    gostplot(GO, capped = TRUE, interactive = TRUE)
  }
}</code></pre>
<div id="htmlwidget-454929a56e939b13f92b" style="width:960px;height:768px;" class="plotly html-widget"></div>
<script type="application/json" data-for="htmlwidget-454929a56e939b13f92b">{"x":{"data":[{"x":[167.540819505229,107.587034293409,134.711211690332,97.2673759964188,84.4730262271517,134.719005994182,167.536922353304,177.97349520804,177.977392359965,186.177000009839],"y":[2.29162574304765,2.18255013274089,2.17384229096605,1.95884322383856,1.91582839137438,1.79283444530855,1.55478098382817,1.53533620370232,1.43851420516674,1.31294949085288],"text":["GO:1900364 (8) <br> negative regulation of mRNA polyadenylation <br> 5.109e-03","GO:0031441 (9) <br> negative regulation of mRNA 3'-end processing <br> 6.568e-03","GO:0050684 (141) <br> regulation of mRNA processing <br> 6.701e-03","GO:0016071 (2153) <br> mRNA metabolic process <br> 1.099e-02","GO:0006397 (1919) <br> mRNA processing <br> 1.214e-02","GO:0050686 (30) <br> negative regulation of mRNA processing <br> 1.611e-02","GO:1900363 (18) <br> regulation of mRNA polyadenylation <br> 2.788e-02","GO:1903311 (318) <br> regulation of mRNA metabolic process <br> 2.915e-02","GO:1903312 (93) <br> negative regulation of mRNA metabolic process <br> 3.643e-02","GO:1905605 (5) <br> positive regulation of blood-brain barrier permeability <br> 4.865e-02"],"key":["GO:1900364","GO:0031441","GO:0050684","GO:0016071","GO:0006397","GO:0050686","GO:1900363","GO:1903311","GO:1903312","GO:1905605"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(255,153,0,1)","opacity":0.8,"size":[11.4012349975872,11.7184148862085,17.1333499133653,20.883539608369,20.7420128906853,14.432060880028,13.3748985332554,18.3536222748863,16.46361375956,9.97595249194087],"symbol":"circle","line":{"width":1.88976377952756,"color":"rgba(255,153,0,1)"}},"hoveron":"points","set":"SharedData9b13f92b","name":"GO:BP","legendgroup":"GO:BP","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","_isNestedKey":false,"frame":null},{"x":[55.1479049940605,64.9900931475556,57.3580121952612,55.058253379197,65.2668437847429,57.3463185063659,56.1769496168418,69.4843675796267,57.685435484328,55.1595986829558,65.2434564069525,59.4979572630904,59.9111342707223,69.2855748684076,54.991989142124,55.6741209943464,54.0720856156983,56.1730517205434,65.2707416810414],"y":[5.28218267088363,4.6978635124742,4.58230854698548,4.5736709292689,4.46565827910246,4.17989857719364,3.25110635292241,3.18426512751112,2.8789877171257,2.63140587987979,2.58658381033455,2.55158875988717,2.42031263779921,2.31011399646546,2.20963074509573,2.06804140932021,1.74491578908365,1.51395196734347,1.43691585782027],"text":["GO:0005681 (195) <br> spliceosomal complex <br> 5.222e-06","GO:0070691 (2) <br> P-TEFb complex <br> 2.005e-05","GO:0016607 (414) <br> nuclear speck <br> 2.616e-05","GO:0005654 (4161) <br> nucleoplasm <br> 2.669e-05","GO:0071013 (85) <br> catalytic step 2 spliceosome <br> 3.422e-05","GO:0016604 (869) <br> nuclear body <br> 6.608e-05","GO:0008024 (8) <br> cyclin/CDK positive transcription elongation factor complex <br> 5.609e-04","GO:1902494 (1424) <br> catalytic complex <br> 6.542e-04","GO:0019908 (12) <br> nuclear cyclin-dependent protein kinase holoenzyme complex <br> 1.321e-03","GO:0005684 (93) <br> U2-type spliceosomal complex <br> 2.337e-03","GO:0071007 (30) <br> U2-type catalytic step 2 spliceosome <br> 2.591e-03","GO:0031981 (5442) <br> nuclear lumen <br> 2.808e-03","GO:0032806 (20) <br> carboxy-terminal domain protein kinase complex <br> 3.799e-03","GO:0140513 (2671) <br> nuclear protein-containing complex <br> 4.897e-03","GO:0005634 (10053) <br> nucleus <br> 6.171e-03","GO:0005829 (5373) <br> cytosol <br> 8.550e-03","GO:0000307 (43) <br> cyclin-dependent protein kinase holoenzyme complex <br> 1.799e-02","GO:0008023 (56) <br> transcription elongation factor complex <br> 3.062e-02","GO:0071014 (13) <br> post-mRNA release spliceosomal complex <br> 3.657e-02"],"key":["GO:0005681","GO:0070691","GO:0016607","GO:0005654","GO:0071013","GO:0016604","GO:0008024","GO:1902494","GO:0019908","GO:0005684","GO:0071007","GO:0031981","GO:0032806","GO:0140513","GO:0005634","GO:0005829","GO:0000307","GO:0008023","GO:0071014"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(16,150,24,1)","opacity":0.8,"size":[17.6327333482292,3.77952755905512,18.7280713266404,21.6724614471827,16.3141558855936,19.7335039353619,11.4012349975872,20.3694247159926,12.4444444199113,16.46361375956,14.432060880028,21.9840210412881,13.6022617113909,21.1456212080949,22.6771653543307,21.9693292674658,15.1180371077117,15.5960686578714,12.6358724305451],"symbol":"circle","line":{"width":1.88976377952756,"color":"rgba(16,150,24,1)"}},"hoveron":"points","set":"SharedData9b13f92b","name":"GO:CC","legendgroup":"GO:CC","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","_isNestedKey":false,"frame":null},{"x":[3.71093261950597,42.1620971389728],"y":[3.01089359765576,2.82462050125524],"text":["GO:0003729 (350) <br> mRNA binding <br> 9.752e-04","GO:0097322 (9) <br> 7SK snRNA binding <br> 1.498e-03"],"key":["GO:0003729","GO:0097322"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(220,57,18,1)","opacity":0.8,"size":[18.4908159947357,11.7184148862085],"symbol":"circle","line":{"width":1.88976377952756,"color":"rgba(220,57,18,1)"}},"hoveron":"points","set":"SharedData9b13f92b","name":"GO:MF","legendgroup":"GO:MF","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","_isNestedKey":false,"frame":null},{"x":[198.68429924249],"y":[6.29426831512614],"text":"KEGG:03040 (146) <br> Spliceosome <br> 5.078e-07","key":["KEGG:03040"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(221,68,119,1)","opacity":0.8,"size":17.1879117800433,"symbol":"circle","line":{"width":1.88976377952756,"color":"rgba(221,68,119,1)"}},"hoveron":"points","set":"SharedData9b13f92b","name":"KEGG","legendgroup":"KEGG","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","_isSimpleKey":true,"_isNestedKey":false,"frame":null},{"x":[2,49.9528814359943],"y":[-1,-1],"text":"","type":"scatter","mode":"lines","line":{"width":11.3385826771654,"color":"rgba(220,57,18,1)","dash":"solid"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[53.8499055266887,71.2657061880019],"y":[-1,-1],"text":"","type":"scatter","mode":"lines","line":{"width":11.3385826771654,"color":"rgba(16,150,24,1)","dash":"solid"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[75.1627302786963,193.967406707605],"y":[-1,-1],"text":"","type":"scatter","mode":"lines","line":{"width":11.3385826771654,"color":"rgba(255,153,0,1)","dash":"solid"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[197.864430798299,200.003897024091],"y":[-1,-1],"text":"","type":"scatter","mode":"lines","line":{"width":11.3385826771654,"color":"rgba(221,68,119,1)","dash":"solid"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[180],"y":[16.2],"text":"values above this threshold are capped","hovertext":"","textfont":{"size":7.55905511811024,"color":"rgba(190,190,190,1)"},"type":"scatter","mode":"text","hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[0,210],"y":[16,16],"text":"","type":"scatter","mode":"lines","line":{"width":0.755905511811024,"color":"rgba(190,190,190,1)","dash":"dash"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null}],"layout":{"margin":{"t":29.2835201328352,"r":6.6417600664176,"b":51.8931100989696,"l":61.0377750103778},"plot_bgcolor":"rgba(255,255,255,1)","paper_bgcolor":"rgba(255,255,255,1)","font":{"color":"rgba(0,0,0,1)","family":"","size":14.6118721461187},"xaxis":{"domain":[0,1],"automargin":true,"type":"linear","autorange":false,"range":[0,210],"tickmode":"array","ticktext":["GO:MF","GO:CC","GO:BP","KEGG"],"tickvals":[25.9764407179972,62.5578058573453,134.565068493151,198.934163911195],"categoryorder":"array","categoryarray":["GO:MF","GO:CC","GO:BP","KEGG"],"nticks":null,"ticks":"","tickcolor":null,"ticklen":3.65296803652968,"tickwidth":0,"showticklabels":true,"tickfont":{"color":"rgba(77,77,77,1)","family":"","size":10.6268161062682},"tickangle":-45,"showline":true,"linecolor":"rgba(190,190,190,1)","linewidth":0.132835201328352,"showgrid":false,"gridcolor":null,"gridwidth":0,"zeroline":false,"anchor":"y","title":"","hoverformat":".2f"},"yaxis":{"domain":[0,1],"automargin":true,"type":"linear","autorange":false,"range":[-1,18.5],"tickmode":"array","ticktext":["0","2","4","6","8","10","12","14",">16"],"tickvals":[0,2,4,6,8,10,12,14,16],"categoryorder":"array","categoryarray":["0","2","4","6","8","10","12","14",">16"],"nticks":null,"ticks":"outside","tickcolor":"rgba(190,190,190,1)","ticklen":3.65296803652968,"tickwidth":0.66417600664176,"showticklabels":true,"tickfont":{"color":"rgba(77,77,77,1)","family":"","size":11.689497716895},"tickangle":-0,"showline":true,"linecolor":"rgba(190,190,190,1)","linewidth":0.66417600664176,"showgrid":false,"gridcolor":null,"gridwidth":0,"zeroline":false,"anchor":"x","title":"","hoverformat":".2f"},"annotations":[{"text":"-log10(p-adj)","x":-0.0108706932337069,"y":0.5,"showarrow":false,"ax":0,"ay":0,"font":{"color":"rgba(0,0,0,1)","family":"","size":13.2835201328352},"xref":"paper","yref":"paper","textangle":-90,"xanchor":"right","yanchor":"center","annotationType":"axis"},{"text":"query_1","x":0.5,"y":1,"showarrow":false,"ax":0,"ay":0,"font":{"color":"rgba(169,169,169,1)","family":"","size":13.2835201328352},"xref":"paper","yref":"paper","textangle":-0,"xanchor":"center","yanchor":"bottom"}],"shapes":[{"type":"rect","fillcolor":null,"line":{"color":null,"width":0,"linetype":[]},"yref":"paper","xref":"paper","x0":0,"x1":1,"y0":0,"y1":1},{"type":"rect","fillcolor":null,"line":{"color":null,"width":0,"linetype":[]},"yref":"paper","xref":"paper","x0":0,"x1":1,"y0":0,"y1":24.9730178497302,"yanchor":1,"ysizemode":"pixel"}],"showlegend":false,"legend":{"bgcolor":"rgba(255,255,255,1)","bordercolor":"transparent","borderwidth":1.88976377952756,"font":{"color":"rgba(0,0,0,1)","family":"","size":11.689497716895}},"hovermode":"closest","barmode":"relative","dragmode":"zoom"},"config":{"doubleClick":"reset","showSendToCloud":false},"source":"A","attrs":{"1f4c79b7510a":{"x":{},"y":{},"text":{},"colour":{},"size":{},"alpha":{},"type":"scatter"},"1f4c433e23a2":{"x":{},"xend":{},"y":{},"yend":{}},"1f4c55e930c1":{"x":{},"xend":{},"y":{},"yend":{}},"1f4c22bf654c":{"x":{},"xend":{},"y":{},"yend":{}},"1f4c660a2f04":{"x":{},"xend":{},"y":{},"yend":{}},"1f4c3a1c706b":{"x":{},"y":{}},"1f4c52a7cfe":{"yintercept":{}}},"cur_data":"1f4c79b7510a","visdat":{"1f4c79b7510a":["function (y) ","x"],"1f4c433e23a2":["function (y) ","x"],"1f4c55e930c1":["function (y) ","x"],"1f4c22bf654c":["function (y) ","x"],"1f4c660a2f04":["function (y) ","x"],"1f4c3a1c706b":["function (y) ","x"],"1f4c52a7cfe":["function (y) ","x"]},"highlight":{"on":"plotly_click","off":"plotly_doubleclick","persistent":false,"dynamic":false,"color":null,"selectize":false,"defaultValues":null,"opacityDim":0.2,"selected":{"opacity":1},"debounce":0,"ctGroups":["SharedData9b13f92b"]},"shinyEvents":["plotly_hover","plotly_click","plotly_selected","plotly_relayout","plotly_brushed","plotly_brushing","plotly_clickannotation","plotly_doubleclick","plotly_deselect","plotly_afterplot","plotly_sunburstclick"],"base_url":"https://plot.ly"},"evals":[],"jsHooks":[]}</script>
</div>
<div id="table-results-1" class="section level4">
<h4>Table results</h4>
<p>The results can also be visualized with a table:</p>
<pre class="r"><code>if(is.null(GO)){
  print(&quot;NOT enough data.&quot;)
}else{
  GO$result %&gt;% 
    as.data.frame() %&gt;% 
    select(starts_with(c(&quot;source&quot;, &quot;term_name&quot;, &quot;p_value&quot;, &quot;term_size&quot;, &quot;intersection_size&quot;))) %&gt;% 
    arrange(p_value) %&gt;%  
    mutate(p_value = format(p_value, scientific = T, digits = 2)) %&gt;% 
    DT::datatable(options = list(
    columnDefs = list(list(className = &#39;dt-center&#39;, targets = 1:6)),
    pageLength = 10)) %&gt;%
    formatStyle(&#39;source&#39;, backgroundColor = styleEqual(c(&quot;GO:BP&quot;, &quot;GO:MF&quot;, &quot;GO:CC&quot;, &quot;KEGG&quot;), c(&#39;orange&#39;, &#39;tomato&#39;, &#39;lightgreen&#39;, &#39;violet&#39;)))
}</code></pre>
<div id="htmlwidget-7856457827ab1876f7c5" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-7856457827ab1876f7c5">{"x":{"filter":"none","data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32"],["KEGG","GO:CC","GO:CC","GO:CC","GO:CC","GO:CC","GO:CC","GO:CC","GO:CC","GO:MF","GO:CC","GO:MF","GO:CC","GO:CC","GO:CC","GO:CC","GO:CC","GO:BP","GO:CC","GO:BP","GO:BP","GO:CC","GO:BP","GO:BP","GO:BP","GO:CC","GO:BP","GO:BP","GO:CC","GO:BP","GO:CC","GO:BP"],[211,334,2859,901,311,2930,898,598,4012,440,985,10306,337,2924,1450,1556,3961,23705,294,8321,15281,469,5673,2390,15283,58,23704,26382,597,26383,2931,28487],["Spliceosome","spliceosomal complex","P-TEFb complex","nuclear speck","nucleoplasm","catalytic step 2 spliceosome","nuclear body","cyclin/CDK positive transcription elongation factor complex","catalytic complex","mRNA binding","nuclear cyclin-dependent protein kinase holoenzyme complex","7SK snRNA binding","U2-type spliceosomal complex","U2-type catalytic step 2 spliceosome","nuclear lumen","carboxy-terminal domain protein kinase complex","nuclear protein-containing complex","negative regulation of mRNA polyadenylation","nucleus","negative regulation of mRNA 3'-end processing","regulation of mRNA processing","cytosol","mRNA metabolic process","mRNA processing","negative regulation of mRNA processing","cyclin-dependent protein kinase holoenzyme complex","regulation of mRNA polyadenylation","regulation of mRNA metabolic process","transcription elongation factor complex","negative regulation of mRNA metabolic process","post-mRNA release spliceosomal complex","positive regulation of blood-brain barrier permeability"],["5.1e-07","5.2e-06","2.0e-05","2.6e-05","2.7e-05","3.4e-05","6.6e-05","5.6e-04","6.5e-04","9.8e-04","1.3e-03","1.5e-03","2.3e-03","2.6e-03","2.8e-03","3.8e-03","4.9e-03","5.1e-03","6.2e-03","6.6e-03","6.7e-03","8.5e-03","1.1e-02","1.2e-02","1.6e-02","1.8e-02","2.8e-02","2.9e-02","3.1e-02","3.6e-02","3.7e-02","4.9e-02"],[146,195,2,414,4161,85,869,8,1424,350,12,9,93,30,5442,20,2671,8,10053,9,141,5373,2153,1919,30,43,18,318,56,93,13,5],[9,7,2,8,21,5,10,2,10,7,2,2,4,3,21,2,6,2,29,2,5,16,14,13,3,2,2,6,2,4,2,2]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>source<\/th>\n      <th>source_order<\/th>\n      <th>term_name<\/th>\n      <th>p_value<\/th>\n      <th>term_size<\/th>\n      <th>intersection_size<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-center","targets":[1,2,3,4,5,6]},{"orderable":false,"targets":0}],"pageLength":10,"order":[],"autoWidth":false,"orderClasses":false,"rowCallback":"function(row, data, displayNum, displayIndex, dataIndex) {\nvar value=data[1]; $(this.api().cell(row, 1).node()).css({'background-color':value == \"GO:BP\" ? \"orange\" : value == \"GO:MF\" ? \"tomato\" : value == \"GO:CC\" ? \"lightgreen\" : value == \"KEGG\" ? \"violet\" : null});\n}"}},"evals":["options.rowCallback"],"jsHooks":[]}</script>
</div>
<div id="treemap-1" class="section level4 tabset tabset-fade">
<h4>Treemap</h4>
<div id="bp-1" class="section level5">
<h5>BP</h5>
<pre class="r"><code>if(sum(GO$result$source == &quot;GO:BP&quot;) &gt;= 5 &amp; !is.null(GO)){
  simMatrix &lt;- GO$result %&gt;% 
  as.data.frame() %&gt;%
  filter(source == &quot;GO:BP&quot;) %&gt;% 
  pull(term_id) %&gt;% 
  calculateSimMatrix(orgdb=&quot;org.Hs.eg.db&quot;, ont=&quot;BP&quot;, method=&quot;Rel&quot;)

scores &lt;- setNames(-log10(GO$result$p_value), GO$result$term_id)

reducedTerms &lt;- reduceSimMatrix(simMatrix,
                                scores,
                                threshold=0.7,
                                orgdb=&quot;org.Hs.eg.db&quot;)

treemapPlot(reducedTerms)
BP &lt;- T
}else{
  print(&quot;NOT enough data.&quot;)
  BP &lt;- F
}</code></pre>
<p><img src="figure/Turbo_vs_WT.Rmd/BP-cond2-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="mf-1" class="section level5">
<h5>MF</h5>
<pre class="r"><code>if(sum(GO$result$source == &quot;GO:MF&quot;) &gt;= 5 &amp; !is.null(GO)){
simMatrix &lt;- GO$result %&gt;% 
  as.data.frame() %&gt;%
  filter(source == &quot;GO:MF&quot;) %&gt;% 
  pull(term_id) %&gt;% 
  calculateSimMatrix(orgdb=&quot;org.Hs.eg.db&quot;, ont=&quot;MF&quot;, method=&quot;Rel&quot;)

scores &lt;- setNames(-log10(GO$result$p_value), GO$result$term_id)

reducedTerms &lt;- reduceSimMatrix(simMatrix,
                                scores,
                                threshold=0.7,
                                orgdb=&quot;org.Hs.eg.db&quot;)

treemapPlot(reducedTerms)
MF &lt;- T
}else{
  print(&quot;NOT enough data.&quot;)
  MF &lt;- F
}</code></pre>
<pre><code>[1] &quot;NOT enough data.&quot;</code></pre>
</div>
<div id="cc-1" class="section level5">
<h5>CC</h5>
<pre class="r"><code>if(sum(GO$result$source == &quot;GO:CC&quot;) &gt;= 5 &amp; !is.null(GO)){
simMatrix &lt;- GO$result %&gt;% 
  as.data.frame() %&gt;%
  filter(source == &quot;GO:CC&quot;) %&gt;% 
  pull(term_id) %&gt;% 
  calculateSimMatrix(orgdb=&quot;org.Hs.eg.db&quot;, ont=&quot;CC&quot;, method=&quot;Rel&quot;)

scores &lt;- setNames(-log10(GO$result$p_value), GO$result$term_id)

reducedTerms &lt;- reduceSimMatrix(simMatrix,
                                scores,
                                threshold=0.7,
                                orgdb=&quot;org.Hs.eg.db&quot;)

treemapPlot(reducedTerms)
CC &lt;- T
}else{
  print(&quot;NOT enough data.&quot;)
  CC &lt;- F
}</code></pre>
<p><img src="figure/Turbo_vs_WT.Rmd/CC-cond2-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="scatterplot-1" class="section level4">
<h4>ScatterPlot</h4>
<pre class="r"><code>if(BP | MF | CC){
scatterPlot(simMatrix, reducedTerms, size = &quot;score&quot;, labelSize = 5)
}else{
  print(&quot;NOT enough data.&quot;)
}</code></pre>
<p><img src="figure/Turbo_vs_WT.Rmd/scatter-cond2-1.png" width="1440" style="display: block; margin: auto;" /></p>
</div>
</div>
</div>
<div id="network-analysis" class="section level2">
<h2><strong>6. Network analysis</strong></h2>
<p>Overview from <strong>STRING Database</strong>: In STRING, each protein-protein interaction is annotated with one or more ‘scores’. Importantly, these scores do <strong>not</strong> indicate the strength or the specificity of the interaction. Instead, they are indicators of <strong>confidence</strong>, i.e. how likely STRING judges an interaction to be true, given the available evidence. All scores rank from 0 to 1, with 1 being the highest possible confidence. A score of 0.5 would indicate that roughly every second interaction might be erroneous (i.e., a false positive).</p>
<pre class="r"><code># if(leng_dep != 0){
#   
#   string_db &lt;- STRINGdb$new(version=&quot;11&quot;, species=9606, score_threshold=400)
# 
#   string_mapped &lt;- string_db$map(string_res, &quot;Gene.name&quot;, removeUnmappedRows = TRUE, quiet = T)
# 
#   hits &lt;- string_mapped$STRING_id[1:leng_dep]
# 
#   string_db$plot_network(hits, add_link = T)
# }else{
#   print(&quot;NOT enough data.&quot;)
# }</code></pre>
<div id="cytoscape-network" class="section level3">
<h3><strong>6.1. Cytoscape Network</strong></h3>
<p>Cytoscape is an open source software platform for visualizing complex networks and integrating these with any type of attribute data. In this presentation all proteins showing at least one confidence level from the STRING database are shown. The <strong>color</strong> code is relative to <strong>fold change</strong> and the <strong>size</strong> is directly proportional to the <strong>significance (p-value)</strong>.</p>
<pre class="r"><code>print(&quot;Enable Cytoscape option to see this results.&quot;)</code></pre>
<pre><code>[1] &quot;Enable Cytoscape option to see this results.&quot;</code></pre>
<pre class="r"><code>if(n_edge != 0){
  knitr::include_graphics(&quot;assets/network_FC_pvalue.png&quot;,  error = F)
}else{
  print(&quot;NOT enough data.&quot;)
}</code></pre>
</div>
<div id="cytoscape-enrichmentmap" class="section level3">
<h3><strong>6.2. Cytoscape EnrichmentMap</strong></h3>
<p>An <strong>enrichment map</strong> is a different sort of network. Instead of nodes representing genes, nodes represent pathways or functions. Edges between these pathways or functions represent shared genes or pathway crosstalk. An enrichment map is a way to visualize your enrichment results to help reduce redundancy and uncover main themes.</p>
<pre class="r"><code>print(&quot;Enable Cytoscape option to see this results.&quot;)</code></pre>
<pre><code>[1] &quot;Enable Cytoscape option to see this results.&quot;</code></pre>
<pre class="r"><code>Gname &lt;- results %&gt;% filter(significant) %&gt;% pull(name)

if(!params$Imputation){
  
  Uni &lt;- unique_pair(data_filt_unique, conditions = conditions, table = T) %&gt;% 
  pull(name) %&gt;% 
  unique()

  Gname &lt;- c(Gname, Uni)
}

if(length(Gname) == 0){
  
  print(&quot;NOT enough data.&quot;)
  
}else{
  
  GO &lt;- gost(query = Gname, organism = &quot;hsapiens&quot;, sources = c(&quot;GO:BP&quot;, &quot;GO:MF&quot;, &quot;GO:CC&quot;, &quot;KEGG&quot;), ordered_query = T, evcodes = TRUE)
  
  if(is.null(GO)){
    
    print(&quot;NOT enough data.&quot;)

    enrich_map = F
    
  }else{
    
    enrich_map = T
    
    gem &lt;- GO$result[,c(&quot;term_id&quot;, &quot;term_name&quot;, &quot;p_value&quot;, &quot;intersection&quot;)]
    colnames(gem) &lt;- c(&quot;GO.ID&quot;, &quot;Description&quot;, &quot;p.Val&quot;, &quot;Genes&quot;)
    gem$FDR &lt;- gem$p.Val
    gem$Phenotype = &quot;+1&quot;
    gem &lt;- gem[,c(&quot;GO.ID&quot;, &quot;Description&quot;, &quot;p.Val&quot;, &quot;FDR&quot;, &quot;Phenotype&quot;, &quot;Genes&quot;)]
    head(gem)
  
    write.table(gem, here::here(&quot;output/enrichmentmap.txt&quot;),col.name=TRUE,sep=&quot;\t&quot;,row.names=FALSE,quote=FALSE)
    
  }
}</code></pre>
<pre class="r"><code>if(enrich_map){
  knitr::include_graphics(&quot;assets/network_enrichmentmap.png&quot;,  error = F)
}else{
  print(&quot;NOT enough data.&quot;)
}</code></pre>
</div>
</div>
<div id="section" class="section level2 unnumbered">
<h2></h2>
<pre class="r"><code>saveWorkbook(excel, here::here(paste0(&quot;output/results_table_&quot;, test, &quot;.xlsx&quot;)), overwrite = T)</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.0.3 (2020-10-10)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 19042)

Matrix products: default

locale:
[1] LC_COLLATE=Italian_Italy.1252  LC_CTYPE=Italian_Italy.1252   
[3] LC_MONETARY=Italian_Italy.1252 LC_NUMERIC=C                  
[5] LC_TIME=Italian_Italy.1252    

attached base packages:
[1] parallel  stats4    stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] forcats_0.5.1               stringr_1.4.0              
 [3] dplyr_1.0.7                 purrr_0.3.4                
 [5] readr_1.4.0                 tidyr_1.1.3                
 [7] tibble_3.1.2                tidyverse_1.3.1            
 [9] openxlsx_4.2.4              here_1.0.1                 
[11] RColorBrewer_1.1-2          RCy3_2.10.2                
[13] STRINGdb_2.2.2              naniar_0.6.1               
[15] visdat_0.5.3                ggrepel_0.9.1              
[17] ggplot2_3.3.5               UpSetR_1.4.0               
[19] ggstatsplot_0.8.0           datapasta_3.1.0            
[21] patchwork_1.1.1             DT_0.18                    
[23] rrvgo_1.2.0                 org.Hs.eg.db_3.12.0        
[25] AnnotationDbi_1.52.0        gprofiler2_0.2.0           
[27] enrichR_3.0                 preprocessCore_1.52.1      
[29] SummarizedExperiment_1.20.0 Biobase_2.50.0             
[31] GenomicRanges_1.42.0        GenomeInfoDb_1.26.2        
[33] IRanges_2.24.1              S4Vectors_0.28.1           
[35] BiocGenerics_0.36.0         MatrixGenerics_1.2.1       
[37] matrixStats_0.59.0          DEP_1.12.0                 
[39] MBQN_2.2.0                  workflowr_1.6.2            

loaded via a namespace (and not attached):
  [1] rappdirs_0.3.3            R.methodsS3_1.8.1        
  [3] coda_0.19-4               bit64_4.0.5              
  [5] knitr_1.33                DelayedArray_0.16.2      
  [7] R.utils_2.10.1            data.table_1.14.0        
  [9] RCurl_1.98-1.3            doParallel_1.0.16        
 [11] generics_0.1.0            RSQLite_2.2.7            
 [13] correlation_0.6.1         chron_2.3-56             
 [15] bit_4.0.4                 lubridate_1.7.10         
 [17] xml2_1.3.2                httpuv_1.6.1             
 [19] assertthat_0.2.1          WRS2_1.1-2               
 [21] xfun_0.24                 hms_1.1.0                
 [23] jquerylib_0.1.4           evaluate_0.14            
 [25] promises_1.2.0.1          fansi_0.5.0              
 [27] readxl_1.3.1              caTools_1.18.2           
 [29] dbplyr_2.1.1              igraph_1.2.6             
 [31] DBI_1.1.1                 htmlwidgets_1.5.3        
 [33] reshape_0.8.8             kSamples_1.2-9           
 [35] Rmpfr_0.8-4               paletteer_1.3.0          
 [37] hash_2.2.6.1              ellipsis_0.3.2           
 [39] crosstalk_1.1.1           backports_1.2.1          
 [41] insight_0.14.2            gridBase_0.4-7           
 [43] vctrs_0.3.8               imputeLCMD_2.0           
 [45] Cairo_1.5-12.2            cachem_1.0.5             
 [47] withr_2.4.2               treemap_2.4-2            
 [49] cluster_2.1.0             lazyeval_0.2.2           
 [51] crayon_1.4.1              labeling_0.4.2           
 [53] pkgconfig_2.0.3           SuppDists_1.1-9.5        
 [55] slam_0.1-48               statsExpressions_1.1.0   
 [57] ProtGenerics_1.22.0       wordcloud_2.6            
 [59] rlang_0.4.11              RJSONIO_1.3-1.4          
 [61] lifecycle_1.0.0           MatrixModels_0.5-0       
 [63] sandwich_3.0-1            affyio_1.60.0            
 [65] BiocFileCache_1.14.0      modelr_0.1.8             
 [67] cellranger_1.1.0          rprojroot_2.0.2          
 [69] graph_1.68.0              Matrix_1.2-18            
 [71] mc2d_0.1-21               zoo_1.8-9                
 [73] reprex_2.0.0              whisker_0.4              
 [75] GlobalOptions_0.1.2       pheatmap_1.0.12          
 [77] png_0.1-7                 viridisLite_0.4.0        
 [79] rjson_0.2.20              mzR_2.24.1               
 [81] PMCMRplus_1.9.0           parameters_0.14.0        
 [83] bitops_1.0-7              shinydashboard_0.7.1     
 [85] R.oo_1.24.0               KernSmooth_2.23-17       
 [87] blob_1.2.1                shape_1.4.6              
 [89] multcompView_0.1-8        tmvtnorm_1.4-10          
 [91] ggsignif_0.6.2            scales_1.1.1             
 [93] memoise_2.0.0             magrittr_2.0.1           
 [95] plyr_1.8.6                gplots_3.1.1             
 [97] zlibbioc_1.36.0           compiler_4.0.3           
 [99] plotrix_3.8-1             pcaMethods_1.82.0        
[101] clue_0.3-59               cli_3.0.0                
[103] affy_1.68.0               XVector_0.30.0           
[105] pbapply_1.4-3             MASS_7.3-53              
[107] tidyselect_1.1.1          vsn_3.58.0               
[109] stringi_1.6.2             highr_0.9                
[111] yaml_2.2.1                GOSemSim_2.16.1          
[113] norm_1.0-9.5              MALDIquant_1.19.3        
[115] grid_4.0.3                sass_0.4.0               
[117] tools_4.0.3               rstudioapi_0.13          
[119] circlize_0.4.13           foreach_1.5.1            
[121] git2r_0.28.0              gridExtra_2.3            
[123] ipmisc_6.0.2              pairwiseComparisons_3.1.6
[125] farver_2.1.0              mzID_1.28.0              
[127] digest_0.6.27             BiocManager_1.30.16      
[129] shiny_1.6.0               proto_1.0.0              
[131] BWStest_0.2.2             Rcpp_1.0.6               
[133] broom_0.7.8               BayesFactor_0.9.12-4.2   
[135] performance_0.7.2         later_1.2.0              
[137] ncdf4_1.17                httr_1.4.2               
[139] MSnbase_2.15.7            ComplexHeatmap_2.6.2     
[141] effectsize_0.4.5          colorspace_2.0-2         
[143] rvest_1.0.0               XML_3.99-0.6             
[145] fs_1.5.0                  rematch2_2.1.2           
[147] gmm_1.6-6                 plotly_4.9.4.1           
[149] xtable_1.8-4              gmp_0.6-2                
[151] jsonlite_1.7.2            zeallot_0.1.0            
[153] NLP_0.2-1                 R6_2.5.0                 
[155] tm_0.7-8                  gsubfn_0.7               
[157] pillar_1.6.1              htmltools_0.5.1.1        
[159] mime_0.11                 glue_1.4.2               
[161] fastmap_1.1.0             BiocParallel_1.24.1      
[163] codetools_0.2-16          mvtnorm_1.1-2            
[165] utf8_1.2.1                lattice_0.20-41          
[167] bslib_0.2.5.1             sqldf_0.4-11             
[169] curl_4.3.2                gtools_3.9.2             
[171] zip_2.2.0                 GO.db_3.12.1             
[173] limma_3.46.0              rmarkdown_2.9            
[175] munsell_0.5.0             GetoptLong_1.0.5         
[177] GenomeInfoDbData_1.2.4    iterators_1.0.13         
[179] impute_1.64.0             haven_2.4.1              
[181] gtable_0.3.0              bayestestR_0.10.0        </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
